Index: src/extra/graphapp/ga.hide
===================================================================
--- src/extra/graphapp/ga.hide	(revision 84174)
+++ src/extra/graphapp/ga.hide	(working copy)
@@ -27,6 +27,8 @@
  GAI_del_context
  GAI_del_string
  GAI_deletion_traversal
+ GAI_edit_control_proc
+ GAI_edit_control_procedure
  GAI_find_next_valid_sibling
  GAI_find_object
  GAI_find_prev_valid_sibling
@@ -40,6 +42,7 @@
  GAI_get_image_pixel
  GAI_get_modeless
  GAI_get_monochrome_pixel
+ GAI_gmaskimage
  GAI_hAccel
  GAI_handle_control
  GAI_handle_findreplace
Index: src/extra/graphapp/graphapp.h
===================================================================
--- src/extra/graphapp/graphapp.h	(revision 84174)
+++ src/extra/graphapp/graphapp.h	(working copy)
@@ -628,7 +628,7 @@
 #define edit_control_proc	GAI_edit_control_proc
 #define edit_control_procedure	GAI_edit_control_procedure
 #define hAccel			GAI_hAccel
-#define hwndClient		GAI_hwndClient
+#define hwndClient		GA_hwndClient
 #define hwndFrame		GAI_hwndFrame
 #define hwndMain		GAI_hwndMain
 #define is_NT			GA_isNT
@@ -665,6 +665,16 @@
 #define print_image		GA_print_image
 #define save_image		GA_save_image
 #define stop_image		GA_stop_image
+
+#define askfilenameW		GA_askfilenameW
+#define askfilenamesW		GA_askfilenamesW
+#define askfilesaveW		GA_askfilesaveW
+#define gcopyalpha2		GA_gcopyalpha2
+#define gdrawimage		GA_gdrawimage
+#define gfillpolypolygon	GA_gfillpolypolygon
+#define gmaskimage		GAI_gmaskimage
+#define gstrwidth1		GA_gstrwidth1
+#define setuserfilterW		GA_setuserfilterW
 #endif
 
 /*
Index: src/gnuwin32/Rdll.hide
===================================================================
--- src/gnuwin32/Rdll.hide	(revision 84174)
+++ src/gnuwin32/Rdll.hide	(working copy)
@@ -1,6 +1,24 @@
+ ALTCOMPLEX_ELT
+ ALTINTEGER_ELT
+ ALTLIST_SET_ELT
+ ALTLOGICAL_ELT
+ ALTRAW_ELT
+ ALTREAL_ELT
+ ALTREP_COERCE
+ ALTREP_DUPLICATE_EX
+ ALTREP_INSPECT
+ ALTREP_SERIALIZED_CLASS
+ ALTREP_SERIALIZED_STATE
+ ALTREP_UNSERIALIZE_EX
+ ALTSTRING_SET_ELT
+ ALTVEC_EXTRACT_SUBSET
  BASE_SYM_CACHED
  BINDING_IS_LOCKED
  BM_norm_keep
+ BNDCELL_DVAL
+ BNDCELL_IVAL
+ BNDCELL_LVAL
+ BNDCELL_TAG
  BZ2_blockSort
  BZ2_bsInitWrite
  BZ2_bzBuffToBuffCompress@28
@@ -36,13 +54,17 @@
  BZ2_hbMakeCodeLengths
  BZ2_indexIntoF
  BZ2_rNums
+ CONS_NR
  CPFun
  ConsoleAcceptCmd
  DebugMenuitem
  DefaultPalette
  EhiWakeUp
+ EncodeEnvironment
+ EncodeExtptr
  EncodeReal2
  HASHASH
+ INIT_BNDCELL
  IS_ACTIVE_BINDING
  IS_BYTES
  IS_LATIN1
@@ -60,11 +82,18 @@
  PrintUsage
  ProfileEvent
  RExitContext
+ R_BCIntActive
  R_BCNodeStackBase
+ R_BCProtReset
+ R_BCProtTop
+ R_BCVersionOK
+ R_BCbody
+ R_BCpc
  R_BrowseLevel
  R_BrowseLines
  R_BrowserLastCommand
  R_CBoundsCheck
+ R_CheckTimeLimits
  R_ChooseFile
  R_CollectWarnings
  R_Collected
@@ -81,10 +110,12 @@
  R_ErrorCon
  R_EvalDepth
  R_ExactSymbol
+ R_ExitContext
  R_Expressions
  R_Expressions_keep
  R_FileMtime
  R_FinalizeSrcRefState
+ R_FixupExitingHandlerResult
  R_FreeSEXP
  R_FreeStringBufferL
  R_GCEnabled
@@ -94,11 +125,13 @@
  R_GetMaxNSize
  R_GetMaxVSize
  R_GetTraceback
+ R_GetTracebackOnly
  R_GetTracebackParsed
  R_GetVarLocSymbol
  R_GetVarLocValue
  R_HandlerStack
  R_HiddenFile
+ R_InitConditions
  R_InitReadItemDepth
  R_InitSrcRefState
  R_InitialData
@@ -121,9 +154,11 @@
  R_LoadRconsole
  R_Log10
  R_LookupMethod
+ R_Matprod
  R_NHeap
  R_NShowCalls
  R_NSize
+ R_NaN_is_R_NA
  R_Newhashpjw
  R_OpenInitFile
  R_OpenLibraryFile
@@ -130,6 +165,7 @@
  R_OpenSiteFile
  R_OpenSysInitFile
  R_OutputCon
+ R_PCRE_limit_recursion
  R_PCRE_study
  R_PCRE_use_JIT
  R_PPStackSize
@@ -141,6 +177,7 @@
  R_ParseErrorFile
  R_ParseFile
  R_PendingPromises
+ R_PlaceholderToken
  R_Primitive
  R_Quiet
  R_ReadItemDepth
@@ -172,6 +209,7 @@
  R_Toplevel
  R_ToplevelContext
  R_UTF8fixslash
+ R_UnwindHandlerStack
  R_UseNamesSymbol
  R_VSize
  R_Verbose
@@ -182,9 +220,14 @@
  R_XDRDecodeInteger
  R_XDREncodeDouble
  R_XDREncodeInteger
+ R_args_enable_refcnt
  R_binary
+ R_checkConstants
+ R_check_constants
  R_check_locale
  R_cmpfun
+ R_cmpfun1
+ R_compact_intrange
  R_compile_pkgs
  R_compress1
  R_compress2
@@ -194,6 +237,8 @@
  R_decompress1
  R_decompress2
  R_decompress3
+ R_deferred_coerceToString
+ R_disable_bytecode
  R_dlsym
  R_doDotCall
  R_do_data_class
@@ -205,13 +250,24 @@
  R_dot_GenericDefEnv
  R_dot_Group
  R_dot_Methods
+ R_expand_binding_value
  R_fgetc
+ R_findBCInterpreterSrcref
+ R_findExecContext
+ R_findParentContext
  R_fixslash
+ R_flexiblas_info
+ R_getBCInterpreterExpression
  R_getClockIncrement
+ R_getCurrentSrcref
  R_getDllTable
+ R_getExpressionStackOverflowError
  R_getFullPathName
  R_getFullPathNameW
+ R_getNSValue
+ R_getNodeStackOverflowError
  R_getProcTime
+ R_getProtectStackOverflowError
  R_getRegisteredRoutines
  R_getS4DataSlot
  R_getSymbolInfo
@@ -218,7 +274,10 @@
  R_gmtime
  R_gmtime_r
  R_has_methods
+ R_in_gc
  R_initAsignSymbols
+ R_initAssignSymbols
+ R_init_altrep
  R_init_base
  R_init_jit_enabled
  R_initialize_bcode
@@ -225,15 +284,27 @@
  R_inspect
  R_inspect3
  R_isMissing
+ R_isWriteableDir
  R_is_running
+ R_isatty
  R_jit_enabled
+ R_jumpctxt
  R_lazyLoadDBinsertValue
  R_localtime
  R_localtime_r
+ R_makeCStackOverflowError
+ R_makeErrorCondition
+ R_makeMissingSubscriptError
+ R_makeMissingSubscriptError1
+ R_makeNotSubsettableError
+ R_makeOutOfBoundsError
  R_max_num_math_threads
+ R_mkEVPROMISE_NR
  R_mktime
  R_moduleCdynload
+ R_nativeEncoding
  R_newCurlUrl
+ R_newservsock
  R_newsock
  R_newunz
  R_newurl
@@ -242,9 +313,14 @@
  R_osDynSymbol
  R_possible_dispatch
  R_pretty
+ R_registerBC
+ R_reinit_altrep_classes
  R_restore_globals
  R_run_onexits
  R_seemsOldStyleS4Object
+ R_setConditionField
+ R_signalErrorCondition
+ R_signalErrorConditionEx
  R_strftime
  R_subset3_dflt
  R_syscall
@@ -252,11 +328,14 @@
  R_sysfunction
  R_sysparent
  R_timegm
+ R_tryUnwrap
+ R_try_clear_args_refcnt
  R_tzname
  R_tzset
  R_tzsetwall
  R_unary
  R_unserialize
+ R_vmakeErrorCondition
  R_warn_partial_match_args
  R_warn_partial_match_attr
  R_warn_partial_match_dollar
@@ -288,6 +367,7 @@
  Rf_EncodeString
  Rf_EnsureString
  Rf_ErrorMessage
+ Rf_ExtractSubset
  Rf_GInit
  Rf_GetOptionCutoff
  Rf_GetOptionDeviceAsk
@@ -350,14 +430,21 @@
  Rf_createS3Vars
  Rf_d1mach
  Rf_ddfindVar
+ Rf_deleteCachedSymbols
  Rf_deparse1s
  Rf_deparse1w
+ Rf_errorcall_cpy
  Rf_evalList
  Rf_evalListKeepMissing
+ Rf_findFun3
+ Rf_findFunctionForBody
  Rf_findVar1
  Rf_findcontext
+ Rf_fixSubset3Args
  Rf_formatRaw
+ Rf_formatRawS
  Rf_formatString
+ Rf_formatStringS
  Rf_framedepth
  Rf_gamma_cody
  Rf_gammalims
@@ -365,7 +452,9 @@
  Rf_i1mach
  Rf_initStack
  Rf_installDDVAL
+ Rf_installNoTrChar
  Rf_installS3Signature
+ Rf_isUnmodifiedSpecSym
  Rf_isValidName
  Rf_lfastchoose
  Rf_lgammacor
@@ -380,6 +469,7 @@
  Rf_matchPar
  Rf_mbtoucs
  Rf_mkCLOSXP
+ Rf_mkCharWUTF8
  Rf_mkPROMISE
  Rf_mkSYMSXP
  Rf_mkTrue
@@ -396,6 +486,7 @@
  Rf_printNamedVector
  Rf_printRealVector
  Rf_printVector
+ Rf_printwhere
  Rf_promiseArgs
  Rf_qchisq_appr
  Rf_readS3VarsFromFrame
@@ -403,6 +494,7 @@
  Rf_regerror
  Rf_regexec
  Rf_regfree
+ Rf_registerRoutines
  Rf_setIVector
  Rf_setRVector
  Rf_sexptype2char
@@ -417,9 +509,12 @@
  Rf_tspgets
  Rf_ucstomb
  Rf_unbindVar
+ Rf_unpromiseArgs
  Rf_usemethod
  Rf_utf8toucs
+ Rf_utf8toucs32
  Rf_vectorIndex
+ Rf_wcs4toutf8
  Rf_wtransChar2
  Rf_yychar
  Rf_yylval
@@ -429,14 +524,22 @@
  RguiPackageMenu
  Rgui_Edit
  Rgui_configure
+ Ri18n_wcswidth
+ Rsnprintf_mbcs
  Rsockselect
  Rstrlen
  Rstrwid
+ Rvsnprintf_mbcs
  Rwin_rename
  Rwin_wrename
  SET_ACTIVE_BINDING_BIT
  SET_ASCII
  SET_BASE_SYM_CACHED
+ SET_BNDCELL
+ SET_BNDCELL_DVAL
+ SET_BNDCELL_IVAL
+ SET_BNDCELL_LVAL
+ SET_BNDCELL_TAG
  SET_BYTES
  SET_CACHED
  SET_CXTAIL
@@ -562,6 +665,9 @@
  csroot_
  dblep0_
  deleditordata
+ dispatch_length
+ dispatch_subset2
+ dispatch_xlength
  do_AT
  do_Cstack_info
  do_D2POSIXlt
@@ -576,7 +682,9 @@
  do_S4on
  do_abbrev
  do_abs
+ do_activeBndFun
  do_addCondHands
+ do_addGlobHands
  do_addRestart
  do_addTryHandlers
  do_addhistory
@@ -584,6 +692,7 @@
  do_adist
  do_agrep
  do_allnames
+ do_altrep_class
  do_anyNA
  do_aperm
  do_aregexec
@@ -591,6 +700,7 @@
  do_arith
  do_arrangeWindows
  do_array
+ do_asCharacterFactor
  do_asPOSIXct
  do_asPOSIXlt
  do_as_environment
@@ -605,6 +715,7 @@
  do_attributes
  do_attributesgets
  do_backsolve
+ do_balancePOSIXlt
  do_baseenv
  do_basename
  do_bcclose
@@ -682,6 +793,9 @@
  do_dotCode
  do_dotcall
  do_dotcallgr
+ do_dotsElt
+ do_dotsLength
+ do_dotsNames
  do_dput
  do_drop
  do_dump
@@ -689,7 +803,6 @@
  do_dynload
  do_dynunload
  do_eSoftVersion
- do_eSoftVersion
  do_eapply
  do_emptyenv
  do_enablejit
@@ -780,6 +893,7 @@
  do_intToUtf8
  do_interactive
  do_internal
+ do_internalsID
  do_interruptsSuspended
  do_invisible
  do_invokeRestart
@@ -841,6 +955,8 @@
  do_math4
  do_matprod
  do_matrix
+ do_maxNSize
+ do_maxVSize
  do_maxcol
  do_memCompress
  do_memDecompress
@@ -853,9 +969,12 @@
  do_mkUnbound
  do_mkcode
  do_mkjunction
+ do_mmap_file
  do_modelframe
  do_modelmatrix
+ do_munmap_file
  do_mvfft
+ do_named
  do_names
  do_namesgets
  do_nargs
@@ -865,6 +984,8 @@
  do_nextn
  do_ngettext
  do_normalizepath
+ do_numToBits
+ do_numToInts
  do_nzchar
  do_onexit
  do_open
@@ -954,8 +1075,10 @@
  do_seq
  do_seq_along
  do_seq_len
+ do_sequence
  do_serialize
  do_serializeToConn
+ do_serversocket
  do_set
  do_setFileTime
  do_setInternet2
@@ -984,11 +1107,14 @@
  do_sock
  do_sockconn
  do_sockselect
+ do_socktimeout
  do_sort
+ do_sorted_fpass
  do_split
  do_sprintf
  do_standardGeneric
  do_startHTTPD
+ do_startsWith
  do_stderr
  do_stdin
  do_stdout
@@ -995,6 +1121,7 @@
  do_stop
  do_stopHTTPD
  do_storage_mode
+ do_str2lang
  do_strptime
  do_strrep
  do_strsplit
@@ -1043,6 +1170,7 @@
  do_trunc
  do_truncate
  do_tryCatchHelper
+ do_tryWrap
  do_typeof
  do_tzone_name
  do_unclass
@@ -1062,6 +1190,7 @@
  do_validUTF8
  do_vapply
  do_version
+ do_vhash
  do_warning
  do_which
  do_while
@@ -1074,6 +1203,7 @@
  do_winprogressbar
  do_winver
  do_withVisible
+ do_wrap_meta
  do_writeClipboard
  do_writebin
  do_writechar
@@ -1102,6 +1232,8 @@
  getDefaults
  getGraphMenu
  getIdentification
+ getLexicalCall
+ getLexicalContext
  getParseContext
  getTZinfo
  getWindowTitle
@@ -1121,6 +1253,7 @@
  hqr_
  htribk_
  htridi_
+ inherits2
  int_arraySubscript
  internal_shellexec
  intpr0_
@@ -1328,13 +1461,14 @@
  lzma_vli_encode
  lzma_vli_size
  mb_char_len
+ mbcsTruncateToValid
  mbcsValid
  menuclear
  menuconfig
  menueditornew
  menueditoropen
+ mkCharWUTF8
  mkPRIMSXP
- mkCharWUTF8
  mkdtemp
  nCPFun
  newWpipe
@@ -1383,6 +1517,10 @@
  pnbeta_raw
  pnchisq_raw
  pointsize
+ printComplexVectorS
+ printIntegerVectorS
+ printRawVector
+ printRealVectorS
  process_site_Renviron
  process_system_Renviron
  process_user_Renviron
Index: src/modules/lapack/Makefile.win
===================================================================
--- src/modules/lapack/Makefile.win	(revision 84174)
+++ src/modules/lapack/Makefile.win	(working copy)
@@ -44,7 +44,7 @@
 
 $(MODULE).dll: $(OBJS)
 
-../../../$(BINDIR)/Rlapack.dll: $(LIBOBJECTS)
+../../../$(BINDIR)/Rlapack.dll: $(LIBOBJECTS) Rlapack.def
 	$(DLL) -shared -s -o $@ $^ -L../../../$(IMPDIR) -lR -lRblas $(FLIBS)
 
 clean:
