Index: src/gnuwin32/installer/JRins.R
===================================================================
--- src/gnuwin32/installer/JRins.R	(revision 79595)
+++ src/gnuwin32/installer/JRins.R	(working copy)
@@ -27,7 +27,7 @@
     srcdir = gsub("/", "\\", srcdir, fixed = TRUE)
 
     Rver <- readLines("../../../VERSION")[1L]
-    Rver <- sub("Under .*$", "Pre-release", Rver)
+    Rver <- sub("Under .*$", "Pre-release-UCRT3", Rver)
     ## This is now over 2^16, so truncate
     SVN <- sub("Revision: ", "", readLines("../../../SVN-REVISION"))[1L]
     SVN <- as.character(as.numeric(SVN) - 50000L)
