Index: src/gnuwin32/Makefile
===================================================================
--- src/gnuwin32/Makefile	(revision 86034)
+++ src/gnuwin32/Makefile	(working copy)
@@ -90,7 +90,7 @@
    ../extra/tre/libtre.a
 
 R-DLLLIBS = $(OPENMP) -L. $(FLIBS) -lRblas -L../../$(IMPDIR) -lRgraphapp \
-  -lRiconv -lcomctl32 -lole32 -luuid -lwinmm -lversion $(LIBEXTRAS)
+  -lRiconv -lcomctl32 -lole32 -luuid -lwinmm -lversion -ldeflate $(LIBEXTRAS)
 
 
 .PHONY:  clean veryclean rlibs fixfiles front-ends rmodules
Index: src/gnuwin32/fixed/h/config.h
===================================================================
--- src/gnuwin32/fixed/h/config.h	(revision 86034)
+++ src/gnuwin32/fixed/h/config.h	(working copy)
@@ -469,9 +469,11 @@
 
 /* Define to 1 if you have libdeflate headers and library. */
 /* #undef HAVE_LIBDEFLATE */
+#define HAVE_LIBDEFLATE 1
 
 /* Define to 1 if you have the <libdeflate.h> header file. */
 /* #undef HAVE_LIBDEFLATE_H */
+#define HAVE_LIBDEFLATE_H 1
 
 /* Define to 1 if you have the `dl' library (-ldl). */
 /* #undef HAVE_LIBDL */
@@ -1242,6 +1244,7 @@
 
 /* Define to use libdefault rather than libz for lazy-loaded R objects */
 /* #undef USE_LIBDEFLATE */
+#define USE_LIBDEFLATE 1
 
 /* Define if the POSIX multithreading library can be used. (For intl) */
 /* #undef USE_POSIX_THREADS */
