Index: src/gnuwin32/fixed/etc/Makeconf
===================================================================
--- src/gnuwin32/fixed/etc/Makeconf	(revision 85957)
+++ src/gnuwin32/fixed/etc/Makeconf	(working copy)
@@ -44,13 +44,13 @@
   ifneq ($(strip $(R_CUSTOM_TOOLS_SOFT)),)
     LOCAL_SOFT = $(R_CUSTOM_TOOLS_SOFT)
   else
-    ## The rtools43 installer sets RTOOLS43_HOME (or RTOOLS43_AARCH64_HOME)
+    ## The rtools44 installer sets RTOOLS44_HOME (or RTOOLS44_AARCH64_HOME)
 
-    # INSTALLER-BUILD-x86_64: RTOOLS43_HOME ?= c:/rtools43
-    # INSTALLER-BUILD-x86_64: LOCAL_SOFT = $(subst \,/,$(RTOOLS43_HOME))/x86_64-w64-mingw32.static.posix
+    # INSTALLER-BUILD-x86_64: RTOOLS44_HOME ?= c:/rtools44
+    # INSTALLER-BUILD-x86_64: LOCAL_SOFT = $(subst \,/,$(RTOOLS44_HOME))/x86_64-w64-mingw32.static.posix
 
-    # INSTALLER-BUILD-aarch64: RTOOLS43_HOME ?= c:/rtools43-aarch64
-    # INSTALLER-BUILD-aarch64: LOCAL_SOFT = $(subst \,/,$(RTOOLS43_AARCH64_HOME))/aarch64-w64-mingw32.static.posix
+    # INSTALLER-BUILD-aarch64: RTOOLS44_HOME ?= c:/rtools44-aarch64
+    # INSTALLER-BUILD-aarch64: LOCAL_SOFT = $(subst \,/,$(RTOOLS44_AARCH64_HOME))/aarch64-w64-mingw32.static.posix
   endif
 endif
 R_TOOLS_SOFT ?= $(LOCAL_SOFT)
Index: src/gnuwin32/fixed/etc/Rcmd_environ
===================================================================
--- src/gnuwin32/fixed/etc/Rcmd_environ	(revision 85957)
+++ src/gnuwin32/fixed/etc/Rcmd_environ	(working copy)
@@ -36,8 +36,8 @@
 SED=${SED-sed}
 
 ## keep in step with Rprofile.windows
-# INSTALLER-BUILD-x86_64:R_RTOOLS43_PATH="${RTOOLS43_HOME:-c:/rtools43}/x86_64-w64-mingw32.static.posix/bin;${RTOOLS43_HOME:-c:/rtools43}/usr/bin"
-# INSTALLER-BUILD-aarch64:R_RTOOLS43_PATH="${RTOOLS43_AARCH64_HOME:-c:/rtools43-aarch64}/aarch64-w64-mingw32.static.posix/bin;${RTOOLS43_AARCH64_HOME:-c:/rtools43-aarch64}/usr/bin"
+# INSTALLER-BUILD-x86_64:R_RTOOLS44_PATH="${RTOOLS44_HOME:-c:/rtools44}/x86_64-w64-mingw32.static.posix/bin;${RTOOLS44_HOME:-c:/rtools44}/usr/bin"
+# INSTALLER-BUILD-aarch64:R_RTOOLS44_PATH="${RTOOLS44_AARCH64_HOME:-c:/rtools44-aarch64}/aarch64-w64-mingw32.static.posix/bin;${RTOOLS44_AARCH64_HOME:-c:/rtools44-aarch64}/usr/bin"
 #   trailing slash to protect against trailing backslash in PATH
-# INSTALLER-BUILD:PATH="${R_CUSTOM_TOOLS_PATH:-${R_RTOOLS43_PATH}};${PATH}/"
+# INSTALLER-BUILD:PATH="${R_CUSTOM_TOOLS_PATH:-${R_RTOOLS44_PATH}};${PATH}/"
 
Index: src/library/profile/Rprofile.windows
===================================================================
--- src/library/profile/Rprofile.windows	(revision 85957)
+++ src/library/profile/Rprofile.windows	(working copy)
@@ -63,25 +63,25 @@
 
 local({
     # keep in step with Rcmd_environ
-    setRtools43Path <- 0
-    # INSTALLER-BUILD: setRtools43Path <- 1
+    setRtools44Path <- 0
+    # INSTALLER-BUILD: setRtools44Path <- 1
 
-    if (setRtools43Path) {
+    if (setRtools44Path) {
         aarch64 <- FALSE
         # INSTALLER-BUILD-aarch64: aarch64 <- TRUE
 
         if (aarch64) {
-            rthome <- Sys.getenv("RTOOLS43_AARCH64_HOME",
-                                 "c:/rtools43-aarch64")
+            rthome <- Sys.getenv("RTOOLS44_AARCH64_HOME",
+                                 "c:/rtools44-aarch64")
             rtpath <- paste0(rthome, "/aarch64-w64-mingw32.static.posix/bin;",
                              rthome, "/usr/bin")
         } else {
-            rthome <- Sys.getenv("RTOOLS43_HOME", "c:/rtools43")
+            rthome <- Sys.getenv("RTOOLS44_HOME", "c:/rtools44")
             rtpath <- paste0(rthome, "/x86_64-w64-mingw32.static.posix/bin;",
                              rthome, "/usr/bin")
         }
         path <- Sys.getenv("R_CUSTOM_TOOLS_PATH", rtpath)
-        Sys.setenv(R_RTOOLS43_PATH = rtpath)
+        Sys.setenv(R_RTOOLS44_PATH = rtpath)
         Sys.setenv(PATH = paste0(path, ";", Sys.getenv("PATH")))
     }
 })
