diff -r -u src/library/grDevices/src/Makefile.win src/library/grDevices/src/Makefile.win
--- src/library/grDevices/src/Makefile.win	2021-09-21 09:58:27.803700200 +0200
+++ src/library/grDevices/src/Makefile.win	2021-09-21 09:58:45.995812500 +0200
@@ -36,7 +36,7 @@
 
 winbitmap-CPPFLAGS = -DHAVE_PNG -DHAVE_JPEG -DHAVE_TIFF -I"$(EXT_LIBS)/include"
 
-PKG_LIBS = -L"$(EXT_LIBS)"/lib$(R_ARCH) -lRgraphapp -lpng -ltiff -ljpeg -lz
+PKG_LIBS = -L"$(EXT_LIBS)"/lib$(R_ARCH) -lRgraphapp -lpng -ltiff -ljpeg -lzstd -lz -lwebp -llzma
 
 
 all: makeMakedeps shlib
