Index: src/library/grDevices/src/Makefile.win
===================================================================
--- src/library/grDevices/src/Makefile.win	(revision 83997)
+++ src/library/grDevices/src/Makefile.win	(working copy)
@@ -36,7 +36,7 @@
 
 winbitmap-CPPFLAGS = -DHAVE_PNG -DHAVE_JPEG -DHAVE_TIFF -I"$(EXT_LIBS)/include"
 
-PKG_LIBS = -L"$(EXT_LIBS)"/lib$(R_ARCH) -lRgraphapp -lpng -ltiff -ljpeg -lzstd -lz -lwebp -llzma
+PKG_LIBS = -L"$(EXT_LIBS)"/lib$(R_ARCH) -lRgraphapp -lpng -ltiff -ljpeg -lzstd -lz -lwebp -lsharpyuv -llzma
 
 
 all: makeMakedeps shlib
