diff -r -u src/gnuwin32/front-ends/graphappmain.c src/gnuwin32/front-ends/graphappmain.c
--- src/gnuwin32/front-ends/graphappmain.c	2021-11-18 14:45:37.730088300 +0100
+++ src/gnuwin32/front-ends/graphappmain.c	2021-11-18 14:45:51.636697300 +0100
@@ -17,13 +17,13 @@
 
 #ifdef _W64
     extern int __argc;
-    extern char **__argv;
+    __declspec(dllimport) extern char **__argv;
 
     GA_startgraphapp(Instance, PrevInstance, CmdShow);
     AppMain(__argc, __argv);
 #else
     extern int _argc;
-    extern char **_argv;
+    __declspec(dllimport) extern char **_argv;
 
     GA_startgraphapp(Instance, PrevInstance, CmdShow);
     AppMain(_argc, _argv);
