diff -r -u src/library/grDevices/src/Makefile.win src/library/grDevices/src/Makefile.win
--- src/library/grDevices/src/Makefile.win	2021-11-18 14:45:37.948901100 +0100
+++ src/library/grDevices/src/Makefile.win	2021-11-18 14:45:54.656001000 +0100
@@ -36,7 +36,7 @@
 
 winbitmap-CPPFLAGS = -DHAVE_PNG -DHAVE_JPEG -DHAVE_TIFF -I"$(EXT_LIBS)/include"
 
-PKG_LIBS = -L"$(EXT_LIBS)"/lib$(R_ARCH) -lRgraphapp -lpng -ltiff -ljpeg -lz
+PKG_LIBS = -L"$(EXT_LIBS)"/lib$(R_ARCH) -lRgraphapp -lpng -ltiff -ljpeg -lzstd -lz -lwebp -llzma
 
 
 all: makeMakedeps shlib
