diff -r -u src/library/tools/R/QC.R src/library/tools/R/QC.R
--- src/library/tools/R/QC.R	2021-04-07 07:42:37.428564000 -0700
+++ src/library/tools/R/QC.R	2021-04-07 07:45:02.809385800 -0700
@@ -4881,7 +4881,7 @@
     if(dir.exists(code_dir)) {
         all_files <- mydir(code_dir)
         ## Under Windows, need a Makefile.win for methods.
-        R_files <- c("sysdata.rda", "Makefile.win",
+        R_files <- c("sysdata.rda", "Makefile.win", "Makefile.ucrt",
                      list_files_with_type(code_dir, "code",
                                           full.names = FALSE,
                                           OS_subdirs = OS_subdirs))
diff -r -u src/library/tools/R/build.R src/library/tools/R/build.R
--- src/library/tools/R/build.R	2021-04-07 07:42:37.731821700 -0700
+++ src/library/tools/R/build.R	2021-04-07 07:45:02.943090700 -0700
@@ -449,17 +449,17 @@
             messageLog(Log, "cleaning src")
             if (WINDOWS) {
                 have_make <- nzchar(Sys.which(Sys.getenv("MAKE", "make")))
-                if (file.exists("Makefile.win")) {
+                if (file.exists(fn <- "Makevars.ucrt") || file.exists(fn <- "Makefile.win")) {
                     if (have_make)
-                        Ssystem(Sys.getenv("MAKE", "make"), "-f Makefile.win clean")
+                        Ssystem(Sys.getenv("MAKE", "make"), paste0("-f ", fn, " clean"))
                     else warning("unable to run 'make clean' in 'src'",
                                  domain = NA)
                 } else {
-                    if (file.exists("Makevars.win")) {
+                    if (file.exists(fn <- "Makevars.ucrt") || file.exists(fn <- "Makevars.win")) {
                         if (have_make) {
                             makefiles <- paste("-f",
                                                shQuote(file.path(R.home("share"), "make", "clean.mk")),
-                                           "-f Makevars.win")
+                                           "-f", fn)
                             Ssystem(Sys.getenv("MAKE", "make"),
                                     c(makefiles, "clean"))
                         } else warning("unable to run 'make clean' in 'src'",
@@ -643,8 +643,8 @@
     fix_nonLF_in_make_files <- function(pkgname, Log) {
         fix_nonLF_in_files(pkgname,
                            paste0("^(",
-                                  paste(c("Makefile", "Makefile.in", "Makefile.win",
-                                          "Makevars", "Makevars.in", "Makevars.win"),
+                                  paste(c("Makefile", "Makefile.in", "Makefile.win", "Makefile.ucrt",
+                                          "Makevars", "Makevars.in", "Makevars.win", "Makevars.ucrt"),
                                         collapse = "|"), ")$"), Log)
         ## Other Makefiles
         makes <- dir(pkgname, pattern = "^Makefile$",
diff -r -u src/library/tools/R/check.R src/library/tools/R/check.R
--- src/library/tools/R/check.R	2021-04-07 07:42:37.880089400 -0700
+++ src/library/tools/R/check.R	2021-04-07 07:45:03.016450500 -0700
@@ -1671,7 +1671,8 @@
             ## that source files have the predefined extensions.
             ## </NOTE>
             if (!any(file.exists(file.path("src",
-                                           c("Makefile", "Makefile.win",
+                                           c("Makefile", "Makefile.win", 
+					     "Makefile.ucrt",
                                              "install.libs.R"))))) {
                 if (!length(dir("src", pattern = "\\.([cfmM]|cc|cpp|f90|f95|mm)"))) {
                     if (!any) warningLog(Log)
@@ -3053,7 +3054,8 @@
         ## statically linked).
         makefiles <- Sys.glob(file.path("src",
                                         c("Makevars", "Makevars.in",
-                                          "Makefile", "Makefile.win")))
+                                          "Makefile", "Makefile.win",
+					  "Makefile.ucrt")))
         if(length(makefiles)) {
             checkingLog(Log, "for portable use of $(BLAS_LIBS) and $(LAPACK_LIBS)")
             any <- FALSE
@@ -3099,8 +3101,9 @@
 
         makefiles <- Sys.glob(file.path("src",
                                         c("Makevars", "Makevars.in",
-                                          "Makevars.win",
-                                          "Makefile", "Makefile.win")))
+                                          "Makevars.win", "Makevars.ucrt",
+                                          "Makefile", "Makefile.win",
+					  "Makefile.ucrt")))
 
         if(length(makefiles)) {
             checkingLog(Log, "use of PKG_*FLAGS in Makefiles")
@@ -3336,7 +3339,7 @@
         ## spaces for which there is no portable way to quote/escape.
         all_files <-
             dir(".",
-                pattern = "^(Makefile|Makefile.in|Makefile.win|makefile|GNUmakefile)$",
+                pattern = "^(Makefile|Makefile.in|Makefile.win|Makefile.ucrt|makefile|GNUmakefile)$",
                 recursive = TRUE)
         all_files <- unique(sort(all_files))
         if(length(all_files)) {
@@ -5708,6 +5711,7 @@
                 setwd(srcd)
                 if (!file.exists("Makefile") &&
                     !file.exists("Makefile.win") &&
+		    !file.exists("Makefile.ucrt") &&
                     !(file.exists("Makefile.in") && spec_install)) {
                     ## Recognized extensions for sources or headers.
                     srcfiles <- dir(".", all.files = TRUE)
diff -r -u src/library/tools/R/install.R src/library/tools/R/install.R
--- src/library/tools/R/install.R	2021-04-07 07:43:25.883064500 -0700
+++ src/library/tools/R/install.R	2021-04-07 07:45:03.079959800 -0700
@@ -507,7 +507,9 @@
 
             owd <- setwd("src")
             if (WINDOWS) {
-                if (file.exists("Makefile.win"))
+                if (file.exists("Makefile.ucrt"))
+                    system(paste(MAKE, "-f Makefile.ucrt clean"))
+                else if (file.exists("Makefile.win"))
                     system(paste(MAKE, "-f Makefile.win clean"))
                 else
                     unlink(c("Makedeps",
@@ -1202,13 +1204,15 @@
                 if (!is.na(f <- Sys.getenv("R_MAKEVARS_USER",
                                            NA_character_))) {
                     if (file.exists(f))  makefiles <- f
-                } else if (file.exists(f <- path.expand("~/.R/Makevars.win")))
+                } else if (file.exists(f <- path.expand("~/.R/Makevars.ucrt")))
+                    makefiles <- f
+                else if (file.exists(f <- path.expand("~/.R/Makevars.win")))
                     makefiles <- f
                 else if (file.exists(f <- path.expand("~/.R/Makevars")))
                     makefiles <- f
-                if (file.exists("Makefile.win")) {
-                    makefiles <- c("Makefile.win", makefiles)
-                    message("  running 'src/Makefile.win' ...", domain = NA)
+                if (file.exists(f <- "Makefile.ucrt") || file.exists(f <- "Makefile.win")) {
+                    makefiles <- c(f, makefiles)
+                    message(paste0("  running 'src/", f, "' ..."), domain = NA)
                     res <- system(paste("make --no-print-directory",
                                         paste("-f", shQuote(makefiles), collapse = " ")))
                     if (res == 0L) shlib_install(instdir, rarch)
@@ -2502,6 +2506,9 @@
         if (!is.na(f <- Sys.getenv("R_MAKEVARS_USER", NA_character_))) {
             if (file.exists(f))  makefiles <- c(makefiles, f)
         } else if (rarch == "/x64" &&
+                   file.exists(f <- path.expand("~/.R/Makevars.ucrt")))
+            makefiles <- c(makefiles, f)
+        else if (rarch == "/x64" &&
                    file.exists(f <- path.expand("~/.R/Makevars.win64")))
             makefiles <- c(makefiles, f)
         else if (file.exists(f <- path.expand("~/.R/Makevars.win")))
@@ -2513,9 +2520,9 @@
     }
 
     makeobjs <- paste0("OBJECTS=", shQuote(objs))
-    if (WINDOWS && file.exists("Makevars.win")) {
-        makefiles <- c("Makevars.win", makefiles)
-        lines <- readLines("Makevars.win", warn = FALSE)
+    if (WINDOWS && (file.exists(fn <- "Makevars.ucrt") || file.exists(fn <- "Makevars.win"))) {
+        makefiles <- c(fn, makefiles)
+        lines <- readLines(fn, warn = FALSE)
         if (length(grep("^OBJECTS *=", lines, perl=TRUE, useBytes = TRUE)))
             makeobjs <- ""
         if (length(ll <- grep("^CXX_STD *=", lines, perl = TRUE,
@@ -3043,6 +3050,9 @@
             if(file.exists(f)) m <- f
         }
         else if((Sys.getenv("R_ARCH") == "/x64") &&
+                file.exists(f <- path.expand("~/.R/Makevars.ucrt")))
+            m <- f
+        else if((Sys.getenv("R_ARCH") == "/x64") &&
                 file.exists(f <- path.expand("~/.R/Makevars.win64")))
             m <- f
         else if(file.exists(f <- path.expand("~/.R/Makevars.win")))
diff -r -u src/scripts/config src/scripts/config
--- src/scripts/config	2021-04-07 07:42:38.289508600 -0700
+++ src/scripts/config	2021-04-07 07:45:03.139563900 -0700
@@ -256,6 +256,8 @@
   if test "${R_OSTYPE}" = "windows"; then
     if test -f "${R_MAKEVARS_USER}"; then
       makefiles="${makefiles} -f \"${R_MAKEVARS_USER}\""
+    elif test ${R_ARCH} = "/x64" && test -f "${HOME}/.R/Makevars.ucrt"; then
+      makefiles="${makefiles} -f \"${HOME}\"/.R/Makevars.ucrt"
     elif test ${R_ARCH} = "/x64" && test -f "${HOME}/.R/Makevars.win64"; then
       makefiles="${makefiles} -f \"${HOME}\"/.R/Makevars.win64"
     elif test -f "${HOME}/.R/Makevars.win"; then
