diff -r -u src/library/profile/Rprofile.windows src/library/profile/Rprofile.windows
--- src/library/profile/Rprofile.windows	2021-10-26 10:31:34.932369500 +0200
+++ src/library/profile/Rprofile.windows	2021-10-26 10:32:46.203680200 +0200
@@ -56,7 +56,7 @@
     tests_startup <- Sys.getenv("R_TESTS")
     if(nzchar(tests_startup)) source(tests_startup)
     ca_cert <- Sys.getenv("CURL_CA_BUNDLE", NA_character_)
-    if(!is.na(ca_cert) &&
+    if(is.na(ca_cert) &&
        file.exists(ca_path <- file.path(R.home("etc"), "curl-ca-bundle.crt")))
         Sys.setenv(CURL_CA_BUNDLE = ca_path)
 })
