diff -r -u src/gnuwin32/front-ends/RGui32.manifest src/gnuwin32/front-ends/RGui32.manifest
--- src/gnuwin32/front-ends/RGui32.manifest	2021-09-21 09:58:27.042780200 +0200
+++ src/gnuwin32/front-ends/RGui32.manifest	2021-09-21 09:59:06.980397900 +0200
@@ -6,6 +6,11 @@
     name="Rgui.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>RGui console</description>
 <dependency>
     <dependentAssembly>
diff -r -u src/gnuwin32/front-ends/RGui64.manifest src/gnuwin32/front-ends/RGui64.manifest
--- src/gnuwin32/front-ends/RGui64.manifest	2021-09-21 09:58:27.120877500 +0200
+++ src/gnuwin32/front-ends/RGui64.manifest	2021-09-21 09:59:06.996018300 +0200
@@ -6,6 +6,11 @@
     name="Rgui.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>RGui console</description>
 <dependency>
     <dependentAssembly>
diff -r -u src/gnuwin32/front-ends/Rcmd.exe.manifest src/gnuwin32/front-ends/Rcmd.exe.manifest
--- src/gnuwin32/front-ends/Rcmd.exe.manifest	2021-09-21 09:58:27.214746800 +0200
+++ src/gnuwin32/front-ends/Rcmd.exe.manifest	2021-09-21 09:59:06.996018300 +0200
@@ -6,6 +6,11 @@
     name="Rcmd.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>R utility</description>
 <ms_asmv2:trustInfo xmlns:ms_asmv2="urn:schemas-microsoft-com:asm.v2">
    <ms_asmv2:security>
diff -r -u src/gnuwin32/front-ends/Rcmd64.exe.manifest src/gnuwin32/front-ends/Rcmd64.exe.manifest
--- src/gnuwin32/front-ends/Rcmd64.exe.manifest	2021-09-21 09:58:27.278329700 +0200
+++ src/gnuwin32/front-ends/Rcmd64.exe.manifest	2021-09-21 09:59:06.996018300 +0200
@@ -6,6 +6,11 @@
     name="Rcmd.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>R utility</description>
 <ms_asmv2:trustInfo xmlns:ms_asmv2="urn:schemas-microsoft-com:asm.v2">
    <ms_asmv2:security>
diff -r -u src/gnuwin32/front-ends/Rscript.exe.manifest src/gnuwin32/front-ends/Rscript.exe.manifest
--- src/gnuwin32/front-ends/Rscript.exe.manifest	2021-09-21 09:58:27.373725500 +0200
+++ src/gnuwin32/front-ends/Rscript.exe.manifest	2021-09-21 09:59:07.011642200 +0200
@@ -6,6 +6,11 @@
     name="Rscript.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>R utility</description>
 <dependency>
     <dependentAssembly>
diff -r -u src/gnuwin32/front-ends/Rterm32.manifest src/gnuwin32/front-ends/Rterm32.manifest
--- src/gnuwin32/front-ends/Rterm32.manifest	2021-09-21 09:58:27.437426700 +0200
+++ src/gnuwin32/front-ends/Rterm32.manifest	2021-09-21 09:59:07.011642200 +0200
@@ -6,6 +6,11 @@
     name="Rterm.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>R terminal</description>
 <dependency>
     <dependentAssembly>
diff -r -u src/gnuwin32/front-ends/Rterm64.manifest src/gnuwin32/front-ends/Rterm64.manifest
--- src/gnuwin32/front-ends/Rterm64.manifest	2021-09-21 09:58:27.526960100 +0200
+++ src/gnuwin32/front-ends/Rterm64.manifest	2021-09-21 09:59:07.027267500 +0200
@@ -6,6 +6,11 @@
     name="Rterm.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>R terminal</description>
 <dependency>
     <dependentAssembly>
