diff -r -u src/library/tools/R/check.R src/library/tools/R/check.R
--- src/library/tools/R/check.R	2021-10-27 00:31:56.809124300 +0200
+++ src/library/tools/R/check.R	2021-10-28 14:27:35.171774800 +0200
@@ -4821,7 +4821,7 @@
                 Sys.setenv(R_RD4PDF = "times")
                 args <- c( "Rd2pdf ", Rd2pdf_opts,
                           paste0("--build-dir=", shQuote(build_dir)),
-                          "--no-clean", "--no-index",
+                          "--no-clean", "--no-index", "--force",
                           "-o ", man_file, topdir)
                 if (run_Rcmd(args, "Rdlatex.log", timeout = tlim)) {
                     ## FIXME: the info is almost certainly in Rdlatex.log
