diff -r -u src/gnuwin32/front-ends/Renviron.c src/gnuwin32/front-ends/Renviron.c
--- src/gnuwin32/front-ends/Renviron.c	2021-11-03 14:48:20.164786700 +0100
+++ src/gnuwin32/front-ends/Renviron.c	2021-11-03 14:51:01.869601100 +0100
@@ -81,7 +81,7 @@
 static char *findterm(char *s)
 {
     char *p, *q, *r, *r2, *ss=s;
-    static char ans[1000];
+    static char ans[10000];
     int nans;
 
     if(!strlen(s)) return "";
@@ -99,11 +99,11 @@
 	strncpy(r, p, q - p + 1);
 	r[q - p + 1] = '\0';
 	r2 = subterm(r);
-	if(strlen(ans) + strlen(r2) < 1000) strcat(ans, r2); else return ss;
+	if(strlen(ans) + strlen(r2) < 10000) strcat(ans, r2); else return ss;
 	/* now repeat on the tail */
 	s = q+1;
     }
-    if(strlen(ans) + strlen(s) < 1000) strcat(ans, s); else return ss;
+    if(strlen(ans) + strlen(s) < 10000) strcat(ans, s); else return ss;
     return ans;
 }
 
@@ -149,8 +149,8 @@
 }
 
 
-#define BUF_SIZE 255
-#define MSG_SIZE 2000
+#define BUF_SIZE 1024 
+#define MSG_SIZE 8000
 int process_Renviron(const char *filename)
 {
     FILE *fp;
