Index: src/library/tools/R/install.R
===================================================================
--- src/library/tools/R/install.R	(revision 79644)
+++ src/library/tools/R/install.R	(working copy)
@@ -1068,6 +1068,45 @@
 
         if (preclean) run_clean()
 
+        if (WINDOWS) {
+            it_patches_base <- Sys.getenv("_R_INSTALL_TIME_PATCHES",
+                "https://www.r-project.org/nosvn/winutf8/ucrt3/")
+        
+            patches_idx <- tryCatch({
+                    idxfile <- file(paste0(it_patches_base, "/",
+                                           "patches_idx.rds"))
+                    patches_idx <- readRDS(idxfile)
+                    close(idxfile)
+                    patches_idx
+                },
+                error = function(e) NULL)
+
+            if (is.null(patches_idx))
+                message("WARNING: installation-time patches will not be applied, could not get the patches index")
+            else {
+                patches_msg <- FALSE
+                for(p in patches_idx[[pkg_name]]) {
+                    if (!patches_msg) {
+                        patches_msg <- TRUE
+                        starsmsg(stars, "applying installation-time patches")
+                    }
+                    purl <- paste0(it_patches_base, "/", p)
+                    have_patch <- nzchar(Sys.which("patch"))
+                    if (!have_patch)
+                        stop("patch utility is needed for installation-time patching")
+                    dir.create("install_time_patches", recursive=TRUE)
+                    fname <- paste0("install_time_patches/", basename(p))
+                    download.file(purl, destfile = fname)
+                    if (system2("patch", args = "-p2", stdin = fname) != 0)
+                        message("WARNING: failed to apply patch ", p, "\n")
+                    else
+                        message("Applied installation-time patch ", purl,
+                                " and saved it as ", fname,
+                                " in package installation\n")
+                 }
+            }
+        }
+
         if (use_configure) {
             if (WINDOWS) {
                 if (file.exists("configure.win")) {
@@ -1329,9 +1368,12 @@
             ## Windows, even if it is installed.
             if (!grepl(" x64 ", utils::win.version())) test_archs <- "i386"
         }
-
+ 
         if (have_cross) Sys.unsetenv("R_ARCH")
 
+        if (WINDOWS && dir.exists("install_time_patches"))
+            file.copy("install_time_patches", instdir, recursive = TRUE)
+
         ## R files must start with a letter
 	if (install_R && dir.exists("R") && length(dir("R"))) {
 	    starsmsg(stars, "R")
