Index: src/gnuwin32/installer/Makefile
===================================================================
--- src/gnuwin32/installer/Makefile	(revision 79595)
+++ src/gnuwin32/installer/Makefile	(working copy)
@@ -102,11 +102,6 @@
 # This is odd -- better to clean the copy
 	(cd ..; $(MAKE) --no-print-directory PKGDIR=. pkgclean-windlgs)
 	$(CP) -pR ../windlgs $(RPREFIX)/src/library
-ifeq "$(WIN)" "64" 	
-# Edit Makeconf
-	$(SED) -e 's!^BINPREF [?]= .*!BINPREF ?= "$$(RTOOLS40_ROOT)/mingw64/bin/"!' \
-	  $(R_HOME)/etc/x64/Makeconf > $(RPREFIX)/etc/x64/Makeconf
-endif	  
 # test dir
 	mkdir $(RPREFIX)/tests
 	$(MAKE) --no-print-directory -C ../../../tests -f Makefile.win \
