Index: src/library/utils/R/zzz.R
===================================================================
--- src/library/utils/R/zzz.R	(revision 79502)
+++ src/library/utils/R/zzz.R	(working copy)
@@ -38,7 +38,9 @@
         if(.Platform$OS.type == "windows") {
             list(unzip = "internal",
                  editor = if(length(grep("Rgui", commandArgs(), TRUE))) "internal" else "notepad",
-                 repos = c(CRAN = "@CRAN@"),
+                 repos = c(CRAN_UCRT = "https://www.r-project.org/nosvn/winutf8/ucrt3/CRAN",
+                           BIOC_UCRT = "https://www.r-project.org/nosvn/winutf8/ucrt3/BIOC",
+                           CRAN = "@CRAN@"),
                  askYesNo = if (.Platform$GUI == "Rgui") askYesNoWinDialog
                  )
         } else
