Index: src/gnuwin32/front-ends/RGui32.manifest
===================================================================
--- src/gnuwin32/front-ends/RGui32.manifest	(revision 79449)
+++ src/gnuwin32/front-ends/RGui32.manifest	(working copy)
@@ -6,6 +6,11 @@
     name="Rgui.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>RGui console</description>
 <dependency>
     <dependentAssembly>
Index: src/gnuwin32/front-ends/RGui64.manifest
===================================================================
--- src/gnuwin32/front-ends/RGui64.manifest	(revision 79449)
+++ src/gnuwin32/front-ends/RGui64.manifest	(working copy)
@@ -6,6 +6,11 @@
     name="Rgui.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>RGui console</description>
 <dependency>
     <dependentAssembly>
Index: src/gnuwin32/front-ends/Rcmd.exe.manifest
===================================================================
--- src/gnuwin32/front-ends/Rcmd.exe.manifest	(revision 79449)
+++ src/gnuwin32/front-ends/Rcmd.exe.manifest	(working copy)
@@ -6,6 +6,11 @@
     name="Rcmd.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>R utility</description>
 <ms_asmv2:trustInfo xmlns:ms_asmv2="urn:schemas-microsoft-com:asm.v2">
    <ms_asmv2:security>
Index: src/gnuwin32/front-ends/Rcmd64.exe.manifest
===================================================================
--- src/gnuwin32/front-ends/Rcmd64.exe.manifest	(revision 79449)
+++ src/gnuwin32/front-ends/Rcmd64.exe.manifest	(working copy)
@@ -6,6 +6,11 @@
     name="Rcmd.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>R utility</description>
 <ms_asmv2:trustInfo xmlns:ms_asmv2="urn:schemas-microsoft-com:asm.v2">
    <ms_asmv2:security>
Index: src/gnuwin32/front-ends/Rscript.exe.manifest
===================================================================
--- src/gnuwin32/front-ends/Rscript.exe.manifest	(revision 79449)
+++ src/gnuwin32/front-ends/Rscript.exe.manifest	(working copy)
@@ -6,6 +6,11 @@
     name="Rscript.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>R utility</description>
 <dependency>
     <dependentAssembly>
Index: src/gnuwin32/front-ends/Rterm32.manifest
===================================================================
--- src/gnuwin32/front-ends/Rterm32.manifest	(revision 79449)
+++ src/gnuwin32/front-ends/Rterm32.manifest	(working copy)
@@ -6,6 +6,11 @@
     name="Rterm.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>R terminal</description>
 <dependency>
     <dependentAssembly>
Index: src/gnuwin32/front-ends/Rterm64.manifest
===================================================================
--- src/gnuwin32/front-ends/Rterm64.manifest	(revision 79449)
+++ src/gnuwin32/front-ends/Rterm64.manifest	(working copy)
@@ -6,6 +6,11 @@
     name="Rterm.exe"
     type="win32"
 />
+<application>
+  <windowsSettings>
+    <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+  </windowsSettings>
+</application>
 <description>R terminal</description>
 <dependency>
     <dependentAssembly>
