diff -ru orig/libbacktrace/configure new/libbacktrace/configure
--- orig/libbacktrace/configure	2020-03-12 11:08:30.000000000 +0000
+++ new/libbacktrace/configure	2020-04-16 20:51:37.714796973 +0000
@@ -12987,6 +12987,7 @@
 
   fi
 fi
+have_mmap=no
 if test "$have_mmap" = "no"; then
   VIEW_FILE=read.lo
   ALLOC_FILE=alloc.lo
diff -ru orig/libbacktrace/configure.ac new/libbacktrace/configure.ac
--- orig/libbacktrace/configure.ac	2020-03-12 11:07:23.000000000 +0000
+++ new/libbacktrace/configure.ac	2020-04-16 19:54:01.620106462 +0000
@@ -303,6 +303,8 @@
     AC_CHECK_FUNC(mmap, [have_mmap=yes], [have_mmap=no])
   fi
 fi
+# with mingw, mmap.h may be present via win32-mmap
+have_mmap=no
 if test "$have_mmap" = "no"; then
   VIEW_FILE=read.lo
   ALLOC_FILE=alloc.lo
