diff -ru orig/libsanitizer/configure new/libsanitizer/configure
--- orig/libsanitizer/configure	2020-03-12 11:08:30.000000000 +0000
+++ new/libsanitizer/configure	2020-04-16 20:52:06.774750716 +0000
@@ -16595,6 +16595,7 @@
 
   fi
 fi
+have_mmap=no
 if test "$have_mmap" = "no"; then
   VIEW_FILE=read.lo
   ALLOC_FILE=alloc.lo
diff -ru orig/libsanitizer/configure.ac new/libsanitizer/configure.ac
--- orig/libsanitizer/configure.ac	2020-03-12 11:07:24.000000000 +0000
+++ new/libsanitizer/configure.ac	2020-04-16 20:31:39.924660269 +0000
@@ -247,6 +247,8 @@
     AC_CHECK_FUNC(mmap, [have_mmap=yes], [have_mmap=no])
   fi
 fi
+# with mingw, mmap.h may be present via win32-mmap
+have_mmap=no
 if test "$have_mmap" = "no"; then
   VIEW_FILE=read.lo
   ALLOC_FILE=alloc.lo
