diff -r -u src/gnuwin32/installer/Makefile src/gnuwin32/installer/Makefile
--- src/gnuwin32/installer/Makefile	2021-11-18 14:45:37.870788000 +0100
+++ src/gnuwin32/installer/Makefile	2021-11-18 14:46:00.699248300 +0100
@@ -104,7 +104,7 @@
 	$(CP) -pR ../windlgs $(RPREFIX)/src/library
 ifeq "$(WIN)" "64" 	
 # Edit Makeconf
-	$(SED) -e 's!^BINPREF [?]= .*!BINPREF ?= "$$(RTOOLS40_ROOT)/mingw64/bin/"!' \
+	$(SED) -e 's!^LOCAL_SOFT [?]= .*!LOCAL_SOFT ?= $$(patsubst %/bin/gcc,%,$$(shell which gcc))!' \
 	  $(R_HOME)/etc/x64/Makeconf > $(RPREFIX)/etc/x64/Makeconf
 endif	  
 # test dir
