diff -Nru orig/mfa/src/gibbs.cpp patched/mfa/src/gibbs.cpp
--- orig/mfa/src/gibbs.cpp	2021-10-26 21:19:36.000000000 +0200
+++ patched/mfa/src/gibbs.cpp	2022-03-14 18:29:55.358005700 +0100
@@ -1,5 +1,6 @@
 #include <Rcpp.h>
 #include <cmath>
+#include <R_ext/Constants.h>
 
 using namespace Rcpp;
 
@@ -10,7 +11,7 @@
 
 // [[Rcpp::export]]
 double log_d_norm(double x, double mu, double sigma) {
-  double ldn = -0.5 * log(2 * PI);
+  double ldn = -0.5 * log(2 * M_PI);
   ldn -= log(sigma) + 1 / (2 * sigma * sigma) * (x - mu) * (x - mu);
   return ldn;
 }
