diff -Nru orig/Rhtslib/src/htslib-1.7/config.h patched/Rhtslib/src/htslib-1.7/config.h
--- orig/Rhtslib/src/htslib-1.7/config.h	2022-03-09 21:04:35.000000000 +0100
+++ patched/Rhtslib/src/htslib-1.7/config.h	2022-03-14 19:17:42.045468200 +0100
@@ -3,5 +3,5 @@
 #define HAVE_LIBLZMA 1
 #define HAVE_LZMA_H 1
 #define HAVE_FSEEKO 1
-#define HAVE_DRAND48 1
+/* #define HAVE_DRAND48 1 */
 #define HAVE_LIBCURL 1
