diff -Nru orig/adimpro/src/awsimage.f patched/adimpro/src/awsimage.f
--- orig/adimpro/src/awsimage.f	2023-03-22 18:13:35.000000000 +0100
+++ patched/adimpro/src/awsimage.f	2024-02-01 18:22:52.984524400 +0100
@@ -189,6 +189,7 @@
 C   kern     specifies the location kernel
 C   wght     scaling factor for second and third dimension (larger values shrink)
 C
+      use omp_lib
       implicit none
       external kldistd,lkern
       double precision kldistd,lkern
@@ -197,8 +198,6 @@
       integer ih,ih1,ii,i1,i2,j1,j2,k,n,thrednr,
      1        jind,jind2,jwind2,dlw,clw,jw1,jw2
       double precision swj,swj0,z1,z2,wj,hakt2
-!$      integer omp_get_thread_num
-!$      external omp_get_thread_num
       hakt2=hakt*hakt
       ih=FLOOR(hakt)
       dlw=2*ih+1
diff -Nru orig/adimpro/src/shrink.f patched/adimpro/src/shrink.f
--- orig/adimpro/src/shrink.f	2023-03-22 18:13:35.000000000 +0100
+++ patched/adimpro/src/shrink.f	2024-02-01 18:24:56.491090900 +0100
@@ -124,13 +124,12 @@
       RETURN
       END
       subroutine shrinkg(x,nx1,nx2,y,ny1,ny2,tol,z,nz,method,nc)
+      use omp_lib
       implicit none
       integer nx1,ny1,nx2,ny2,nz,x(nx1,nx2),y(ny1,ny2),nc
       double precision z(nz,nc),tol
       integer iy1,iy2,ja1,ja2,je1,je2,jx1,jx2,k,method,thrednr
       double precision yy,d1,d2
-!$      integer omp_get_thread_num
-!$      external omp_get_thread_num
 C
 C   x - original image
 C   y - new image
@@ -199,13 +198,12 @@
       RETURN
       END
       subroutine shrinkc(x,nx1,nx2,y,ny1,ny2,tol,z,nz,method,nc)
+      use omp_lib
       implicit none
       integer nx1,ny1,nx2,ny2,nz,x(3,nx1,nx2),y(3,ny1,ny2),nc
       double precision z(3,nz,nc),tol
       integer iy1,iy2,ja1,ja2,je1,je2,jx1,jx2,k,method,thrednr
       double precision yy(3),d1,d2
-!$      integer omp_get_thread_num
-!$      external omp_get_thread_num
 C
 C   x - original image
 C   y - new image
