diff -Nru orig/archive/src/Makevars.ucrt patched/archive/src/Makevars.ucrt
--- orig/archive/src/Makevars.ucrt	2021-07-27 20:57:14.000000000 +0200
+++ patched/archive/src/Makevars.ucrt	2021-11-22 13:09:53.386317000 +0100
@@ -1,2 +1,21 @@
-CRT=-ucrt
-include Makevars.win
+PKG_CPPFLAGS= \
+  -I.
+
+PKG_LIBS = \
+  -larchive -lxml2 -lcrypto -lnettle -lregex -lexpat -llzo2 -llzma -llz4 -lbz2 -lz -lzstd -liconv -lws2_32
+
+
+# Originally from https://github.com/cran/curl/blob/3897ba5203dee940e2ce40ac23a0d1106da93df6/src/Makevars.win
+
+LIB_CON = ../inst/lib$(R_ARCH)/libconnection.dll
+
+all: $(OBJECTS) $(SHLIB) $(LIB_CON)
+
+$(LIB_CON): connection/connection.o
+	mkdir -p $(dir $(LIB_CON))
+	$(SHLIB_LINK) $(LIBR) -o $@ $^
+
+clean:
+	rm -f $(OBJECTS) $(SHLIB) $(LIB_CON) connection/connection.o
+
+.PHONY: all clean
diff -Nru orig/archive/src/archive/src/Makevars.ucrt patched/archive/src/archive/src/Makevars.ucrt
--- orig/archive/src/archive/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/archive/src/archive/src/Makevars.ucrt	2021-11-22 13:08:10.200598400 +0100
@@ -0,0 +1,21 @@
+PKG_CPPFLAGS= \
+  -I.
+
+PKG_LIBS = \
+  -larchive -lxml2 -lcrypto -lnettle -lregex -lexpat -llzo2 -llzma -llz4 -lbz2 -lz -lzstd -liconv -lws2_32
+
+
+# Originally from https://github.com/cran/curl/blob/3897ba5203dee940e2ce40ac23a0d1106da93df6/src/Makevars.win
+
+LIB_CON = ../inst/lib$(R_ARCH)/libconnection.dll
+
+all: $(OBJECTS) $(SHLIB) $(LIB_CON)
+
+$(LIB_CON): connection/connection.o
+	mkdir -p $(dir $(LIB_CON))
+	$(SHLIB_LINK) $(LIBR) -o $@ $^
+
+clean:
+	rm -f $(OBJECTS) $(SHLIB) $(LIB_CON) connection/connection.o
+
+.PHONY: all clean
