diff -Nru orig/BMTME/R/BME.R patched/BMTME/R/BME.R
--- orig/BMTME/R/BME.R	2020-08-25 17:55:36.000000000 +0200
+++ patched/BMTME/R/BME.R	2023-01-17 09:29:08.242650700 +0100
@@ -73,7 +73,7 @@
     class(out) <- 'BMECV'
 
   } else if (parallelCores > 1 && inherits(testingSet, 'CrossValidation')) {
-    cl <- snow::makeCluster(parallelCores)
+    cl <- snow::makeCluster(parallelCores, type="SOCK")
     doSNOW::registerDoSNOW(cl)
     nCV <- length(testingSet$CrossValidation_list)
 
diff -Nru orig/BMTME/R/BMORS.R patched/BMTME/R/BMORS.R
--- orig/BMTME/R/BMORS.R	2020-08-25 17:55:36.000000000 +0200
+++ patched/BMTME/R/BMORS.R	2023-01-17 09:29:29.350006500 +0100
@@ -99,7 +99,7 @@
     out <- list(results = results, nIter = nIter, burnIn = burnIn, thin = thin, executionTime = proc.time()[3] - time.init)
     class(out) <- 'BMORSCV'
   } else if (parallelCores > 1 && inherits(testingSet, 'CrossValidation')) {
-    cl <- snow::makeCluster(parallelCores)
+    cl <- snow::makeCluster(parallelCores, type="SOCK")
     doSNOW::registerDoSNOW(cl)
     nCV <- length(testingSet$CrossValidation_list)
 
diff -Nru orig/BMTME/R/BMTME.R patched/BMTME/R/BMTME.R
--- orig/BMTME/R/BMTME.R	2020-08-25 17:55:36.000000000 +0200
+++ patched/BMTME/R/BMTME.R	2023-01-17 09:29:53.997206700 +0100
@@ -83,7 +83,7 @@
     out <- list(results = results, n_cores = parallelCores, nIter = nIter, burnIn = burnIn, thin = thin, executionTime = proc.time()[3] - time.init)
     class(out) <- 'BMTMECV'
   } else if (parallelCores > 1 && inherits(testingSet, 'CrossValidation')) {
-    cl <- snow::makeCluster(parallelCores)
+    cl <- snow::makeCluster(parallelCores, type="SOCK")
     doSNOW::registerDoSNOW(cl)
     nCV <- length(testingSet$CrossValidation_list)
 
