diff -Nru orig/btb/src/Makevars.win patched/btb/src/Makevars.win
--- orig/btb/src/Makevars.win	2022-10-24 12:48:56.000000000 +0200
+++ patched/btb/src/Makevars.win	2025-01-29 12:08:19.999500000 +0100
@@ -9,4 +9,4 @@
 SHLIB_CXX14LDFLAGS = -shared
 
 PKG_CXXFLAGS = -DRCPP_PARALLEL_USE_TBB=1
-PKG_LIBS = $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS) `$(R_HOME)/bin/Rscript -e "RcppParallel::RcppParallelLibs()"`
+PKG_LIBS = $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS) $(shell "$(R_HOME)/bin$(R_ARCH_BIN)/Rscript" -e "RcppParallel::RcppParallelLibs()")
