--- orig/copula/R/AllClass.R	2023-12-07 13:06:30.000000000 +0100
+++ patched/copula/R/AllClass.R	2025-03-05 12:46:30.000000000 +0100
@@ -1,4 +1,4 @@
-## Copyright (C) 2012--2023 Marius Hofert, Ivan Kojadinovic, Martin Maechler, and Jun Yan
+## Copyright (C) 2012--2025 Marius Hofert, Ivan Kojadinovic, Martin Maechler, and Jun Yan
 ##
 ## This program is free software; you can redistribute it and/or modify it under
 ## the terms of the GNU General Public License as published by the Free Software
@@ -114,8 +114,8 @@
              ## ... TODO
 
              ## Check more :
-	     if (object@psi(0, theta= 1/2) != 1)
-		 return("psi(0, theta=1/2) != 1 -- seems an invalid generator")
+	     if (abs(object@psi(0, theta= 1/2) - 1) > 4 * .Machine$double.eps)
+		 return("psi(0, theta=1/2) != 1 {+/- eps} -- psi() too inaccurate as generator")
 
              ## ....
              ## ....
