diff -Nru orig/Cyclops/src/cyclops/Types.h patched/Cyclops/src/cyclops/Types.h
--- orig/Cyclops/src/cyclops/Types.h	2023-09-25 15:56:29.000000000 +0200
+++ patched/Cyclops/src/cyclops/Types.h	2024-02-02 11:59:03.252333700 +0100
@@ -30,17 +30,10 @@
 //    }
 // #endif
 
-#ifdef WIN_BUILD
-    #include <tr1/unordered_map>
-    namespace bsccs {
-        using std::tr1::unordered_map;
-    }
-#else
-    #include <unordered_map>
-    namespace bsccs {
-        using std::unordered_map;
-    }
-#endif
+#include <unordered_map>
+namespace bsccs {
+    using std::unordered_map;
+}
 
 namespace bsccs {
 
