diff -Nru orig/devEMF/src/devEMF.cpp patched/devEMF/src/devEMF.cpp
--- orig/devEMF/src/devEMF.cpp	2023-06-28 22:51:12.000000000 +0200
+++ patched/devEMF/src/devEMF.cpp	2024-02-02 16:13:16.118413800 +0100
@@ -314,7 +314,7 @@
                   const pGEcontext gc, pDevDesc dd) {
             static_cast<CDevEMF*>(dd->deviceSpecific)->Line(x1,y1,x2,y2,gc);
         }
-        void Polyline(int n, double *x, double *y, 
+        void EMFcb_Polyline(int n, double *x, double *y, 
                       const pGEcontext gc, pDevDesc dd) {
             static_cast<CDevEMF*>(dd->deviceSpecific)->Polyline(n,x,y, gc);
         }
@@ -331,7 +331,7 @@
                   const pGEcontext gc, pDevDesc dd) {
             static_cast<CDevEMF*>(dd->deviceSpecific)->Rect(x0,y0,x1,y1,gc);
         }
-        void Polygon(int n, double *x, double *y, 
+        void EMFcb_Polygon(int n, double *x, double *y, 
                      const pGEcontext gc, pDevDesc dd) {
             static_cast<CDevEMF*>(dd->deviceSpecific)->Polygon(n,x,y,gc);
         }
@@ -957,8 +957,8 @@
 #if R_GE_version >= 8
     dd->path = EMFcb::Path;
 #endif
-    dd->polygon = EMFcb::Polygon;
-    dd->polyline = EMFcb::Polyline;
+    dd->polygon = EMFcb::EMFcb_Polygon;
+    dd->polyline = EMFcb::EMFcb_Polyline;
     dd->locator = EMFcb::Locator;
     dd->mode = EMFcb::Mode;
     dd->metricInfo = EMFcb::MetricInfo;
