diff -Nru orig/dtts/src/align.cpp patched/dtts/src/align.cpp
--- orig/dtts/src/align.cpp	2024-01-31 14:39:22.000000000 +0100
+++ patched/dtts/src/align.cpp	2024-02-02 16:43:26.116677300 +0100
@@ -24,13 +24,13 @@
 std::ostream &operator<<(std::ostream &stream,
                          const std::chrono::time_point<Clock, Duration> &time_point) {
   const time_t time = Clock::to_time_t(time_point);
-#if (__GNUC__ > 4 || ((__GNUC__ == 4) && __GNUC_MINOR__ > 8 && __GNUC_REVISION__ > 1))
+#if (__GNUC__ > 4 || defined(__WIN32) || ((__GNUC__ == 4) && __GNUC_MINOR__ > 8 && __GNUC_REVISION__ > 1))
   // Maybe the put_time will be implemented later?
   struct tm tm;
   // thanks to https://stackoverflow.com/a/38034148/143305 for the next test
   #if defined(__unix__)
   localtime_r(&time, &tm);
-  #elif defined(_MSC_VER)
+  #elif defined(_MSC_VER) || defined(__WIN32)
   localtime_s(&tm, &time);
   #endif
   return stream << std::put_time(&tm, "%c"); // Print standard date&time
