diff -Nru orig/flan/src/Makevars.win patched/flan/src/Makevars.win
--- orig/flan/src/Makevars.win	2019-12-12 17:28:55.000000000 +0100
+++ patched/flan/src/Makevars.win	2024-02-05 17:38:09.818002800 +0100
@@ -1,5 +1,5 @@
 ## This assumes that the LIB_GSL variable points to working GSL libraries
-PKG_CPPFLAGS=-std=c++0x -Wall -pedantic -I$(LIB_GSL)/include -I. -I../inst/include
+PKG_CXXFLAGS=-std=c++0x -Wall -pedantic -I$(LIB_GSL)/include -I. -I../inst/include
 ## 32 or 64bits?
 ifeq "${R_ARCH}" "/x64"
         PKG_LIBS=-L$(LIB_GSL)/lib/x64 -lgsl -lgslcblas  $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)
