diff -Nru orig/geno2proteo/exec/DNAseqTranslation_CDS.perl patched/geno2proteo/exec/DNAseqTranslation_CDS.perl
--- orig/geno2proteo/exec/DNAseqTranslation_CDS.perl	2017-09-27 12:33:36.000000000 +0200
+++ patched/geno2proteo/exec/DNAseqTranslation_CDS.perl	2021-11-23 10:46:54.924824400 +0100
@@ -18,7 +18,7 @@
  open DOC, $geneticCodesFile or die "Cannot open the file$geneticCodesFile\n";
  my $num=1;
  while(<DOC>) {
-   chomp;
+   s/[\r\n]+//;
    my @items = split/[\t ]+/;
    #$items[1] = 9 if($items[1] eq '*');
    $geneticCodes{$items[0]} = $items[1];
@@ -40,8 +40,7 @@
  my $numExons=0;
  
  while(<DOC>) {
-   chomp;
-   
+   s/[\r\n]+//; 
    my @items = split/[\t ]+/;
    my $lastItem = scalar @items;
    $lastItem -= 1;
diff -Nru orig/geno2proteo/exec/getDNAseqsFromGenome_loci.perl patched/geno2proteo/exec/getDNAseqsFromGenome_loci.perl
--- orig/geno2proteo/exec/getDNAseqsFromGenome_loci.perl	2017-04-19 13:25:18.000000000 +0200
+++ patched/geno2proteo/exec/getDNAseqsFromGenome_loci.perl	2021-11-23 10:47:27.597232100 +0100
@@ -30,7 +30,7 @@
 my $num=0;
 while(<DOC>) {
  ++$num;
- chomp;
+ s/[\r\n]+//;
  if(/^>([^ ]+) /) {
    #print "$num $_\n" ;
    if($dnaSeq ne '') {
@@ -69,7 +69,7 @@
 my @lociNames;
 my @bedLines;
 while(<DOC>) {
-  chomp;
+  s/[\r\n]+//;
   $bedLines[$numLoci] = $_;
   my @terms = split/[\t ]+/;
   
