diff -Nru orig/httpgd/src/Makevars.ucrt patched/httpgd/src/Makevars.ucrt
--- orig/httpgd/src/Makevars.ucrt	2021-08-05 15:17:24.000000000 +0200
+++ patched/httpgd/src/Makevars.ucrt	2021-10-20 21:36:02.599443700 +0200
@@ -1,2 +1,13 @@
-CRT=-ucrt
-include Makevars.win
+CXX_STD = CXX17
+
+PKG_CPPFLAGS = -Ilib \
+	-DBOOST_NO_AUTO_PTR \
+	-DFMT_HEADER_ONLY \
+	-I$(R_TOOLS_SOFT)/include/cairo
+
+PKG_LIBS = -lmswsock -lcairo -lpixman-1 -lfontconfig -lncrypt -lksecdd -lbcrypt -lexpat -lharfbuzz_too -lfreetype_too -lharfbuzz -lfreetype -lpng16 -lpng -lbz2 -lgdi32 -lintl -liconv -lwsock32 -lws2_32 -lz -lcfitsio
+
+all: clean 
+
+clean:
+	rm -f $(OBJECTS)
