diff -Nru orig/jmvconnect/src/column.h patched/jmvconnect/src/column.h
--- orig/jmvconnect/src/column.h	2021-07-06 06:42:00.000000000 +0200
+++ patched/jmvconnect/src/column.h	2024-02-13 09:31:29.907310600 +0100
@@ -14,12 +14,6 @@
 #include <utility>
 #include <cmath>
 
-#ifdef _WIN32
-#define ALIGN_8 alignas(8)
-#else
-#define ALIGN_8 __attribute__ ((aligned (8)))
-#endif
-
 class DataSet;
 
 typedef struct
@@ -28,7 +22,7 @@
     int length;
     int capacity;
 
-    char values[8] ALIGN_8;
+    alignas(8) char values[8];
 
 } Block;
 
