diff -Nru orig/mpmi/src/Makevars.ucrt patched/mpmi/src/Makevars.ucrt
--- orig/mpmi/src/Makevars.ucrt	1969-12-31 16:00:00.000000000 -0800
+++ patched/mpmi/src/Makevars.ucrt	2021-02-18 02:03:06.577815200 -0800
@@ -0,0 +1,16 @@
+PKG_FFLAGS=$(SHLIB_OPENMP_FFLAGS)
+PKG_LIBS=$(SHLIB_OPENMP_CFLAGS)
+
+all: $(SHLIB)
+
+cmi.o: 10iface.o
+
+cminjk.o: 10iface.o  
+
+dmi.o: 10iface.o  
+
+mmi.o: 10iface.o  
+
+mminjk.o: 10iface.o  
+
+
