diff -Nru orig/nanotime/src/Makevars.ucrt patched/nanotime/src/Makevars.ucrt
--- orig/nanotime/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/nanotime/src/Makevars.ucrt	2021-02-17 19:45:41.665006300 +0100
@@ -0,0 +1,5 @@
+## We want C++11 
+CXX_STD = CXX11
+
+## We need headers from our package, the directory is not automatically included
+PKG_CXXFLAGS = -I../inst/include -mno-ms-bitfields -D_POSIX_THREAD_SAFE_FUNCTIONS
