diff -Nru orig/ncdf4/src/Makevars.ucrt patched/ncdf4/src/Makevars.ucrt
--- orig/ncdf4/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/ncdf4/src/Makevars.ucrt	2021-03-19 09:22:53.679077000 +0100
@@ -0,0 +1,10 @@
+
+PKG_LIBS = \
+-lnetcdf -lmfhdf -lhdf5_hl -lportablexdr -ldf -lhdf5 -lcurl -lrtmp -lssl -lssh2 -lidn2 -lunistring -liconv -lgcrypt -lcrypto -lgpg-error -lwsock32 -lws2_32 -ljpeg -lz -lcfitsio -lzstd -lsbml-static -lcrypt32 -lwldap32
+
+all: clean 
+
+clean:
+	rm -Rf $(SHLIB) $(OBJECTS)
+
+.PHONY: all clean
