diff -Nru orig/parallelpam/src/dissimmat_full.cpp patched/parallelpam/src/dissimmat_full.cpp
--- orig/parallelpam/src/dissimmat_full.cpp	2023-07-13 16:44:19.000000000 +0200
+++ patched/parallelpam/src/dissimmat_full.cpp	2024-02-07 17:32:07.933147800 +0100
@@ -17,6 +17,7 @@
  */
 
 #include <dissimmat.h>
+#include <pthread.h>
 
 extern unsigned char DEB;
 
diff -Nru orig/parallelpam/src/dissimmat_sparse.cpp patched/parallelpam/src/dissimmat_sparse.cpp
--- orig/parallelpam/src/dissimmat_sparse.cpp	2023-07-13 16:44:30.000000000 +0200
+++ patched/parallelpam/src/dissimmat_sparse.cpp	2024-02-07 17:32:39.054066600 +0100
@@ -17,6 +17,7 @@
  */
 
 #include <dissimmat.h>
+#include <pthread.h>
 
 extern unsigned char DEB;
 
diff -Nru orig/parallelpam/src/fastpam.cpp patched/parallelpam/src/fastpam.cpp
--- orig/parallelpam/src/fastpam.cpp	2023-06-19 15:26:21.000000000 +0200
+++ patched/parallelpam/src/fastpam.cpp	2024-02-07 17:32:54.201077100 +0100
@@ -17,6 +17,7 @@
  */
 
 #include <fastpam.h>
+#include <pthread.h>
 
 extern unsigned char DEB;
 
diff -Nru orig/parallelpam/src/silhouette.cpp patched/parallelpam/src/silhouette.cpp
--- orig/parallelpam/src/silhouette.cpp	2023-06-19 15:26:21.000000000 +0200
+++ patched/parallelpam/src/silhouette.cpp	2024-02-07 17:33:11.835021900 +0100
@@ -17,6 +17,7 @@
  */
 
 #include<silhouette.h>
+#include <pthread.h>
 
 extern unsigned char DEB;
 
diff -Nru orig/parallelpam/src/threadhelper.cpp patched/parallelpam/src/threadhelper.cpp
--- orig/parallelpam/src/threadhelper.cpp	2023-06-19 15:26:21.000000000 +0200
+++ patched/parallelpam/src/threadhelper.cpp	2024-02-07 17:33:28.494626300 +0100
@@ -17,6 +17,7 @@
  */
 
 #include <threadhelper.h>
+#include <pthread.h>
 
 // Auxiliary function to decide the number of threads
 unsigned int ChooseNumThreads(int nthreads)
