diff -Nru orig/pexm/src/Makevars.ucrt patched/pexm/src/Makevars.ucrt
--- orig/pexm/src/Makevars.ucrt	1969-12-31 16:00:00.000000000 -0800
+++ patched/pexm/src/Makevars.ucrt	2021-02-22 00:03:15.163006700 -0800
@@ -0,0 +1,15 @@
+##############################################################################################################
+# This Makevars.win (Windows) is based on similar files taken as examples from the packages runjags and rjags.
+#
+# Remark: The user must adapt the JAGS_ROOT variable below when:
+#         - the JAGS version is not 4.3.0;
+#         - JAGS is not installed in the standard location "c:\Program Files\JAGS". 
+##############################################################################################################
+
+JAGS_ROOT ?= c:\progra~1\JAGS\JAGS-4.3.0
+JAGS_ROOT ?= "/c/progra~1/JAGS/JAGS-4.3.0"
+
+PKG_CPPFLAGS=-I"$(JAGS_ROOT)/include"
+PKG_LIBS=-L"$(JAGS_ROOT)/${R_ARCH}/bin" -ljrmath-0 -ljags-4
+OBJECTS = functions/DPexFun.o functions/HCPexFun.o functions/HPexFun.o functions/PPexFun.o functions/QPexFun.o distributions/DPex.o init.o pexm.o
+
