diff -Nru orig/pgenlibr/src/Makevars.win patched/pgenlibr/src/Makevars.win
--- orig/pgenlibr/src/Makevars.win	2023-06-10 00:08:21.000000000 +0200
+++ patched/pgenlibr/src/Makevars.win	2024-04-30 08:52:52.156320500 +0200
@@ -1,5 +1,5 @@
 PKG_CPPFLAGS = -DLIBDEFLATE_STATIC -DPGENLIB_NOPRINT -I. -Ilibdeflate
-PKG_LIBS = -L. -lPGDEFLATE -lPLINK2 -lz -lpthread -lzstd
+PKG_LIBS = -L. -lPLINK2 -lPGDEFLATE -lz -lpthread -lzstd
 
 LIBDEFLATE_SOURCES = libdeflate/lib/zlib_decompress.c libdeflate/lib/zlib_compress.c libdeflate/lib/utils.c libdeflate/lib/gzip_compress.c libdeflate/lib/crc32.c libdeflate/lib/adler32.c libdeflate/lib/deflate_decompress.c libdeflate/lib/deflate_compress.c libdeflate/lib/gzip_decompress.c libdeflate/lib/arm/arm_cpu_features.c libdeflate/lib/x86/x86_cpu_features.c
 LIBDEFLATE = $(LIBDEFLATE_SOURCES:.c=.o)
