diff -Nru orig/proj4/src/Makevars.ucrt patched/proj4/src/Makevars.ucrt
--- orig/proj4/src/Makevars.ucrt	2022-01-30 20:38:42.000000000 -0800
+++ patched/proj4/src/Makevars.ucrt	2022-09-18 10:55:29.443751700 -0700
@@ -1,3 +1,3 @@
-## from Tomas, uses P4 API so verify on toolchain updates
+## from Tomas, uses P6 API so verify on toolchain updates
 PKG_LIBS = -lproj -lsqlite3 -lcurl -ltiff -ljpeg -lrtmp -lssl -lssh2 -lgcrypt -lcrypto -lgdi32 -lz -lzstd -lwebp -llzma -lgdi32 -lcrypt32 -lidn2 -lunistring -liconv -lgpg-error -lws2_32 -lwldap32 -lwinmm -lstdc++
-PKG_CPPFLAGS = -DACCEPT_USE_OF_DEPRECATED_PROJ_API_H=1
+PKG_CPPFLAGS = -DUSE_PROJ6_API=1
