diff -Nru orig/prophet/NAMESPACE patched/prophet/NAMESPACE
--- orig/prophet/NAMESPACE	2020-04-29 02:36:35.000000000 +0200
+++ patched/prophet/NAMESPACE	2021-03-19 14:55:25.543002000 +0100
@@ -19,4 +19,3 @@
 import(Rcpp)
 import(rlang)
 importFrom(dplyr,"%>%")
-useDynLib(prophet, .registration = TRUE)
diff -Nru orig/prophet/R/zzz.R patched/prophet/R/zzz.R
--- orig/prophet/R/zzz.R	2020-03-02 22:20:12.000000000 +0100
+++ patched/prophet/R/zzz.R	2021-03-19 15:12:28.435681200 +0100
@@ -4,6 +4,12 @@
 # LICENSE file in the root directory of this source tree.
 
 .onLoad <- function(libname, pkgname) {
+  if (.Platform$OS.type == "windows") {
+     tbbpath <- system.file(paste0("lib/", .Platform$r_arch),
+                            package="RcppParallel")
+     library.dynam("prophet", pkgname, libname, DLLpath = tbbpath)
+  }
+
   # Create environment for storing stan model
   assign("prophet_model_env", new.env(), parent.env(environment()))
   tryCatch({
