diff -Nru orig/PSPManalysis/inst/C/PSPMdemo.c patched/PSPManalysis/inst/C/PSPMdemo.c
--- orig/PSPManalysis/inst/C/PSPMdemo.c	2020-10-13 07:59:18.000000000 +0200
+++ patched/PSPManalysis/inst/C/PSPMdemo.c	2021-11-25 12:19:07.284114800 +0100
@@ -943,7 +943,7 @@
 void InitialiseVars(void)
 
 {
-#if (defined(_MSC_VER) && (_MSC_VER < 1500)) || (defined(R_PACKAGE) && defined(_WIN32))
+#if (defined(_MSC_VER) && (_MSC_VER < 1500)) || (defined(R_PACKAGE) && defined(_WIN32) && !(defined(_UCRT) || ((__MSVCRT_VERSION__ >= 0x1400) || (__MSVCRT_VERSION__ >= 0xE00 && __MSVCRT_VERSION__ < 0x1000))))
   (void)_set_output_format(_TWO_DIGIT_EXPONENT);
 #endif
 
diff -Nru orig/PSPManalysis/inst/C/PSPMequi.c patched/PSPManalysis/inst/C/PSPMequi.c
--- orig/PSPManalysis/inst/C/PSPMequi.c	2020-10-13 08:09:13.000000000 +0200
+++ patched/PSPManalysis/inst/C/PSPMequi.c	2021-11-25 12:19:37.660481300 +0100
@@ -1279,7 +1279,7 @@
 {
   int         i;
 
-#if (defined(_MSC_VER) && (_MSC_VER < 1500)) || (defined(R_PACKAGE) && defined(_WIN32))
+#if (defined(_MSC_VER) && (_MSC_VER < 1500)) || (defined(R_PACKAGE) && defined(_WIN32) && !(defined(_UCRT) || ((__MSVCRT_VERSION__ >= 0x1400) || (__MSVCRT_VERSION__ >= 0xE00 && __MSVCRT_VERSION__ < 0x1000))))
   (void)_set_output_format(_TWO_DIGIT_EXPONENT);
 #endif
 
diff -Nru orig/PSPManalysis/inst/C/PSPMevodyn.c patched/PSPManalysis/inst/C/PSPMevodyn.c
--- orig/PSPManalysis/inst/C/PSPMevodyn.c	2020-10-13 08:15:25.000000000 +0200
+++ patched/PSPManalysis/inst/C/PSPMevodyn.c	2021-11-25 12:20:28.891713900 +0100
@@ -946,7 +946,7 @@
 {
   int         i;
 
-#if (defined(_MSC_VER) && (_MSC_VER < 1500)) || (defined(R_PACKAGE) && defined(_WIN32))
+#if (defined(_MSC_VER) && (_MSC_VER < 1500)) || (defined(R_PACKAGE) && defined(_WIN32) && !(defined(_UCRT) || ((__MSVCRT_VERSION__ >= 0x1400) || (__MSVCRT_VERSION__ >= 0xE00 && __MSVCRT_VERSION__ < 0x1000))))
   (void)_set_output_format(_TWO_DIGIT_EXPONENT);
 #endif
 
diff -Nru orig/PSPManalysis/inst/C/PSPMind.c patched/PSPManalysis/inst/C/PSPMind.c
--- orig/PSPManalysis/inst/C/PSPMind.c	2020-10-13 08:17:07.000000000 +0200
+++ patched/PSPManalysis/inst/C/PSPMind.c	2021-11-25 12:20:55.990110400 +0100
@@ -398,7 +398,7 @@
 
 {
   int ii;
-#if (defined(_MSC_VER) && (_MSC_VER < 1500)) || (defined(R_PACKAGE) && defined(_WIN32))
+#if (defined(_MSC_VER) && (_MSC_VER < 1500)) || (defined(R_PACKAGE) && defined(_WIN32) && !(defined(_UCRT) || ((__MSVCRT_VERSION__ >= 0x1400) || (__MSVCRT_VERSION__ >= 0xE00 && __MSVCRT_VERSION__ < 0x1000))))
   (void)_set_output_format(_TWO_DIGIT_EXPONENT);
 #endif
 
