diff -Nru orig/QF/src/Makevars.ucrt patched/QF/src/Makevars.ucrt
--- orig/QF/src/Makevars.ucrt	2021-10-06 12:36:34.000000000 +0200
+++ patched/QF/src/Makevars.ucrt	2025-03-26 17:24:15.170679000 +0100
@@ -1,2 +1,9 @@
-CRT=-ucrt
-include Makevars.win
+ifeq (,$(shell pkg-config --version 2>/dev/null))
+  PKG_LIBS = -lgsl -lgslcblas -lm
+else
+  PKG_CPPFLAGS = $(shell pkg-config --cflags gsl)
+  PKG_LIBS = $(shell pkg-config --libs gsl)
+endif
+
+PKG_CPPFLAGS += -I../inst/include
+
