diff -Nru orig/RcppThread/R/LdFlags.R patched/RcppThread/R/LdFlags.R
--- orig/RcppThread/R/LdFlags.R	2023-10-18 14:19:53.000000000 +0200
+++ patched/RcppThread/R/LdFlags.R	2024-02-08 17:56:36.113030700 +0100
@@ -25,9 +25,6 @@
 }
 
 checkForLibAtomic <- function() {
-    if (.Platform$OS.type == "windows")
-        return(TRUE)
-
     tmp <- createTestFiles()
     writeLibAtomicTest(tmp["src"])
     failed <- runCmd(getCompiler(), tmp["src"], "-o", tmp["out"], "-latomic")
