diff -Nru orig/RcppCWB/src/Makevars.ucrt patched/RcppCWB/src/Makevars.ucrt
--- orig/RcppCWB/src/Makevars.ucrt	2022-03-18 08:24:19.000000000 +0100
+++ patched/RcppCWB/src/Makevars.ucrt	2022-04-27 11:40:56.783253200 +0200
@@ -5,7 +5,7 @@
 
 PKG_CPPFLAGS=-I$(R_PACKAGE_SOURCE)/cqp -I$(R_PACKAGE_SOURCE)/cl -I$(R_PACKAGE_SOURCE)/CQi -DPCRE_STATIC
 
-PKG_LIBS=-L$(R_PACKAGE_SOURCE)/cl -L$(R_PACKAGE_SOURCE)/cqp -L$(R_PACKAGE_SOURCE)/utils -lcwb -lcqp -lcl -lglib-2.0 -lintl -liconv -lws2_32 -lpcre -lole32 
+PKG_LIBS=-L$(R_PACKAGE_SOURCE)/cl -L$(R_PACKAGE_SOURCE)/cqp -L$(R_PACKAGE_SOURCE)/utils -lcwb -lcqp -lcl -lglib-2.0 -luuid -lintl -liconv -lws2_32 -lpcre -lole32 
 
 ${SHLIB}: libcl.a libcqp.a libcwb.a
 
