diff -Nru orig/redux/src/Makevars.ucrt patched/redux/src/Makevars.ucrt
--- orig/redux/src/Makevars.ucrt	1969-12-31 16:00:00.000000000 -0800
+++ patched/redux/src/Makevars.ucrt	2021-03-19 09:09:05.348201900 -0700
@@ -0,0 +1,10 @@
+# -*- makefile -*-
+PKG_CPPFLAGS=-I$(R_TOOLS_SOFT)/include/hiredis -DSTRICT_R_HEADERS
+PKG_LIBS=-lhiredis -lws2_32
+
+all: clean
+
+clean:
+	rm -f $(OBJECTS) $(SHLIB)
+
+.PHONY: all clean
diff -Nru orig/redux/src/registration.c patched/redux/src/registration.c
--- orig/redux/src/registration.c	2018-05-29 08:23:52.000000000 -0700
+++ patched/redux/src/registration.c	2021-03-19 09:11:28.108117400 -0700
@@ -4,6 +4,10 @@
 #include <R_ext/Rdynload.h>
 #include <Rversion.h>
 
+#ifdef _WIN32
+# include<winsock2.h>
+#endif
+
 static const R_CallMethodDef callMethods[] = {
   {"Credux_redis_connect",       (DL_FUNC) &redux_redis_connect,        2},
   {"Credux_redis_connect_unix",  (DL_FUNC) &redux_redis_connect_unix,   1},
