diff -Nru orig/remaCor/src/Makevars.win patched/remaCor/src/Makevars.win
--- orig/remaCor/src/Makevars.win	2023-06-21 05:24:23.000000000 +0200
+++ patched/remaCor/src/Makevars.win	2024-02-08 11:47:39.401459400 +0100
@@ -1,6 +1,6 @@
 ## With Rcpp 0.11.0 and later, we no longer need to set PKG_LIBS as there is
 ## no user-facing library. The include path to headers is already set by R.
-PKG_LIBS = -lblas -lgfortran -llapack -lquadmath
+PKG_LIBS = -lblas -llapack $(FLIBS)
 
 ## With R 3.1.0 or later, you can uncomment the following line to tell R to 
 ## enable compilation with C++11 (or even C++14) where available
