diff -ru orig/rgdal/src/Makevars.ucrt patched/rgdal/src/Makevars.ucrt
--- orig/rgdal/src/Makevars.ucrt	2022-05-07 05:06:06.000000000 -0700
+++ patched/rgdal/src/Makevars.ucrt	2022-09-20 14:36:20.114222700 -0700
@@ -3,11 +3,11 @@
 	-DPROJ_H_API
 
 PKG_LIBS = \
-        -lgdal -lgta -lsqlite3 -lmysqlclient -lspatialite -lproj -lgeos_c -lgeos  \
+        -lgdal -lblosc -lkea -lhdf5_cpp -lhdf5 -lgta -lsqlite3 -lmysqlclient -lspatialite -lminizip -lproj -lgeos_c -lgeos  \
         -ljson-c -lnetcdf -lpq \
         -lpoppler -llcms2 -lfreetype -lharfbuzz -lfreetype -llz4 -lpcre2-8 \
         -lwebp -lcurl -lidn2 -lunistring -lssh2 -lgcrypt -lgpg-error -lssl \
-        -lhdf5_hl -lhdf5 -lexpat -lfreexl \
+        -lhdf5_hl -lhdf5 -lsz -lexpat -lfreexl \
         -lmfhdf -ldf -lportablexdr  -lpcre \
         -lopenjp2 -ljasper -lpng -ljpeg -ltiff -lwebp -lgeotiff -lgif -lxml2 -llzma \
         -lodbc32 -lodbccp32 -liconv -lpsapi -lwldap32 -lsecur32 -lgdi32 -lnormaliz \
