diff -Nru orig/rpg/src/Makevars.win patched/rpg/src/Makevars.win
--- orig/rpg/src/Makevars.win	2017-10-29 23:50:30.000000000 +0100
+++ patched/rpg/src/Makevars.win	2020-07-02 18:05:47.996631600 +0200
@@ -1,10 +1,6 @@
-PKG_CPPFLAGS= -I../windows/libpq-9.5.2/include
-PKG_LIBS= -L../windows/libpq-9.5.2/lib${R_ARCH} -lpq -lssl -lcrypto -lwsock32 -lsecur32 -lws2_32 -lgdi32 -lcrypt32 -lwldap32
+PKG_LIBS= -lpq -lssl -lcrypto -lz -lwsock32 -lsecur32 -lws2_32 -lgdi32 -lcrypt32 -lwldap32
 
-all: clean winlibs
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R"
+all: clean
 
 clean:
 	rm -f $(OBJECTS) $(SHLIB)
diff -Nru orig/rpg/tools/winlibs.R patched/rpg/tools/winlibs.R
--- orig/rpg/tools/winlibs.R	2017-10-25 20:38:59.000000000 +0200
+++ patched/rpg/tools/winlibs.R	1970-01-01 01:00:00.000000000 +0100
@@ -1,8 +0,0 @@
-# Build against static libpq
-if(!file.exists("../windows/libpq-9.5.2/include/libpq-fe.h")){
-  if(getRversion() < "3.3.0") setInternet2()
-  download.file("https://github.com/rwinlib/libpq/archive/v9.5.2.zip", "lib.zip", quiet = TRUE)
-  dir.create("../windows", showWarnings = FALSE)
-  unzip("lib.zip", exdir = "../windows")
-  unlink("lib.zip")
-}
