diff -Nru orig/scellpam/src/dissimmat_full.cpp patched/scellpam/src/dissimmat_full.cpp
--- orig/scellpam/src/dissimmat_full.cpp	2023-10-13 01:41:50.000000000 +0200
+++ patched/scellpam/src/dissimmat_full.cpp	2024-02-08 12:21:08.151417100 +0100
@@ -17,6 +17,7 @@
  */
 
 #include <dissimmat.h>
+#include <pthread.h>
 
 extern unsigned char DEB;
 
diff -Nru orig/scellpam/src/dissimmat_sparse.cpp patched/scellpam/src/dissimmat_sparse.cpp
--- orig/scellpam/src/dissimmat_sparse.cpp	2023-10-13 01:41:50.000000000 +0200
+++ patched/scellpam/src/dissimmat_sparse.cpp	2024-02-08 12:21:24.653169600 +0100
@@ -17,6 +17,7 @@
  */
 
 #include <dissimmat.h>
+#include <pthread.h>
 
 extern unsigned char DEB;
 
diff -Nru orig/scellpam/src/fastpam.cpp patched/scellpam/src/fastpam.cpp
--- orig/scellpam/src/fastpam.cpp	2023-10-13 01:41:50.000000000 +0200
+++ patched/scellpam/src/fastpam.cpp	2024-02-08 12:21:38.590137300 +0100
@@ -17,6 +17,7 @@
  */
 
 #include <fastpam.h>
+#include <pthread.h>
 
 extern unsigned char DEB;
 
diff -Nru orig/scellpam/src/silhouette.cpp patched/scellpam/src/silhouette.cpp
--- orig/scellpam/src/silhouette.cpp	2023-10-13 01:41:50.000000000 +0200
+++ patched/scellpam/src/silhouette.cpp	2024-02-08 12:21:50.846433700 +0100
@@ -17,6 +17,7 @@
  */
 
 #include<silhouette.h>
+#include <pthread.h>
 
 extern unsigned char DEB;
 
diff -Nru orig/scellpam/src/threadhelper.cpp patched/scellpam/src/threadhelper.cpp
--- orig/scellpam/src/threadhelper.cpp	2023-10-13 01:41:50.000000000 +0200
+++ patched/scellpam/src/threadhelper.cpp	2024-02-08 12:22:27.910040100 +0100
@@ -17,6 +17,7 @@
  */
 
 #include <threadhelper.h>
+#include <pthread.h>
 
 // Auxiliary function to decide the number of threads
 unsigned int ChooseNumThreads(int nthreads)
