diff -Nru orig/stringi/src/Makevars.win patched/stringi/src/Makevars.win
--- orig/stringi/src/Makevars.win	2023-11-21 05:57:49.000000000 +0100
+++ patched/stringi/src/Makevars.win	2024-05-06 13:21:42.572383700 +0200
@@ -44,7 +44,7 @@
 
 $(SHLIB): $(OBJECTS) libicu_common.a libicu_i18n.a libicu_stubdata.a
 
-PKG_LIBS=-L. -licu_stubdata -licu_common -licu_i18n
+PKG_LIBS=-L. -licu_i18n -licu_common -licu_stubdata
 
 libicu_common.a: $(ICU_COMMON_OBJECTS)
 
