diff -Nru orig/stringi/src/icu69/common/uassert.h patched/stringi/src/icu69/common/uassert.h
--- orig/stringi/src/icu69/common/uassert.h	2021-08-11 17:37:55.000000000 -0700
+++ patched/stringi/src/icu69/common/uassert.h	2022-12-30 13:37:42.343617700 -0800
@@ -47,7 +47,7 @@
 #else
 #ifdef U_STRINGI_PATCHES
 // #include <R.h>
-extern "C" void Rf_error(const char *str, ...);
+extern "C" [[noreturn]] void Rf_error(const char *str, ...);
 #   define UPRV_UNREACHABLE (Rf_error("Internal error in ICU: UPRV_UNREACHABLE"))
 #else /* !U_STRINGI_PATCHES */
 #   define UPRV_UNREACHABLE abort()
