diff -ruN orig/sybilSBML/src/Makevars.win patched/sybilSBML/src/Makevars.win
--- orig/sybilSBML/src/Makevars.win	2020-03-19 04:42:28.000000000 -0700
+++ patched/sybilSBML/src/Makevars.win	2021-01-11 09:38:41.830129600 -0800
@@ -1,7 +1,7 @@
-PKG_CPPFLAGS= -D_R_=1 -DUSE_R=1 -I"$(LIB_SBML)/include"
+PKG_CPPFLAGS= -D_R_=1 -DUSE_R=1
 
-PKG_LIBS = -L"$(LIB_SBML)/bin" -lsbml
+PKG_LIBS = -lsbml
 
-all: $(SHLIB)
-	mkdir -p "$(R_PACKAGE_DIR)/libs$(R_ARCH)"
-	cp "$(LIB_SBML)/bin/libsbml.dll" "$(R_PACKAGE_DIR)\libs$(R_ARCH)"
+#all: $(SHLIB)
+#	mkdir -p "$(R_PACKAGE_DIR)/libs$(R_ARCH)"
+#	cp "$(LIB_SBML)/bin/libsbml.dll" "$(R_PACKAGE_DIR)\libs$(R_ARCH)"
