diff -Nru orig/sysfonts/src/Makevars.ucrt patched/sysfonts/src/Makevars.ucrt
--- orig/sysfonts/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/sysfonts/src/Makevars.ucrt	2021-02-17 19:45:43.791503300 +0100
@@ -0,0 +1,13 @@
+PKG_CPPFLAGS = -I$(R_TOOLS_SOFT)/include/freetype2
+PKG_LIBS = -lfreetype -lharfbuzz -lfreetype -lpng -lz -lbz2
+
+
+.PHONY: all deps clean
+
+all: $(SHLIB)
+
+clean:
+	$(RM) *.o
+	$(RM) *.dll
+	$(RM) *.so
+	$(RM) *.dylib
