diff -Nru orig/tth/src/Makefile.win patched/tth/src/Makefile.win
--- orig/tth/src/Makefile.win	2020-03-19 14:57:07.000000000 +0100
+++ patched/tth/src/Makefile.win	2021-02-17 19:45:44.166506600 +0100
@@ -1,8 +1,4 @@
-## WARNING: requires GNU make (allowed on Windows)
-sp =
-sp +=
-sq = $(subst $(sp),\ ,$1)
-include $(call sq,${R_HOME}/etc${R_ARCH}/Makeconf)
+include $(R_HOME)/etc$(R_ARCH)/Makeconf
 
 CFLAGS = -w
 
