diff -Nru orig/udunits2/src/Makevars.ucrt patched/udunits2/src/Makevars.ucrt
--- orig/udunits2/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/udunits2/src/Makevars.ucrt	2022-03-11 17:42:21.470441200 +0100
@@ -0,0 +1,14 @@
+PKG_LIBS = -ludunits2 -lexpat
+
+all: clean winlibs
+
+winlibs:
+	mkdir -p ../inst
+	rm -Rf ../inst/share
+	mkdir -p ../inst/share/udunits
+	cp -r $(R_TOOLS_SOFT)/share/udunits/* ../inst/share/udunits
+
+clean:
+	rm -Rf $(SHLIB) $(OBJECTS)
+
+.PHONY: all winlibs clean
