diff -Nru orig/unrtf/src/Makevars patched/unrtf/src/Makevars
--- orig/unrtf/src/Makevars	2020-01-13 00:47:37.000000000 +0100
+++ patched/unrtf/src/Makevars	2021-11-22 21:53:20.641269000 +0100
@@ -1,5 +1,5 @@
 INSTDIR=../inst
-PKG_CPPFLAGS = -Ilibunrtf -DHAVE_CONFIG_H -DPKGDATADIR=\"../share\"
+PKG_CPPFLAGS = -Ilibunrtf -DHAVE_CONFIG_H -DPKGDATADIR=\"../share\" -D__USE_MINGW_ACCESS
 
 LIBUNRTF = \
 	libunrtf/attr.o libunrtf/convert.o libunrtf/error.o libunrtf/hash.o libunrtf/main.o \
