diff -Nru orig/xml2/src/Makevars.ucrt patched/xml2/src/Makevars.ucrt
--- orig/xml2/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/xml2/src/Makevars.ucrt	2025-01-29 17:05:19.741873200 +0100
@@ -0,0 +1,10 @@
+PKG_CPPFLAGS=-I../inst/include
+
+ifeq (,$(shell pkg-config --version 2>/dev/null))
+  PKG_CPPFLAGS += -I${R_TOOLS_SOFT}/include/libxml2 -DLIBXML_STATIC
+  PKG_LIBS = -lxml2 -lz -liconv -lm -lws2_32 -lbcrypt -llzma
+else
+  PKG_CPPFLAGS += $(shell pkg-config --cflags libxml-2.0)
+  PKG_LIBS = $(shell pkg-config --libs libxml-2.0)
+endif
+
