diff -Nru tk8.6.12-orig/generic/tkInt.h tk8.6.12-patched/generic/tkInt.h
--- tk8.6.12-orig/generic/tkInt.h	2021-10-29 13:08:23.000000000 -0400
+++ tk8.6.12-patched/generic/tkInt.h	2022-01-13 09:13:00.148799036 -0500
@@ -120,9 +120,9 @@
 #endif
 
 #ifndef TCL_Z_MODIFIER
-#   if defined(_WIN64)
+#   if defined(_WIN64) && !defined(_UCRT)
 #	define TCL_Z_MODIFIER	"I"
-#   elif defined(__GNUC__) && !defined(_WIN32)
+#   elif defined(__GNUC__) && !defined(_WIN32) || defined(_UCRT) && defined(_WIN64)
 #	define TCL_Z_MODIFIER	"z"
 #   else
 #	define TCL_Z_MODIFIER	""
