diff -Nru tk8.6.16-orig/generic/tkInt.h tk8.6.16-patched/generic/tkInt.h
--- tk8.6.16-orig/generic/tkInt.h	2024-11-27 11:20:04.000000000 -0500
+++ tk8.6.16-patched/generic/tkInt.h	2025-02-19 11:43:14.109589458 -0500
@@ -124,9 +124,9 @@
 #endif
 
 #ifndef TCL_Z_MODIFIER
-#   if defined(_WIN64)
+#   if defined(_WIN64) && !defined(_UCRT)
 #	define TCL_Z_MODIFIER	"I"
-#   elif defined(__GNUC__) && !defined(_WIN32)
+#   elif defined(__GNUC__) && !defined(_WIN32) || defined(_UCRT) && defined(_WIN64)
 #	define TCL_Z_MODIFIER	"z"
 #   else
 #	define TCL_Z_MODIFIER	""
