diff -Nru orig/Rhtslib/R/zzz.R patched/Rhtslib/R/zzz.R
--- orig/Rhtslib/R/zzz.R	2025-04-15 20:49:11.000000000 +0200
+++ patched/Rhtslib/R/zzz.R	2025-10-19 19:07:19.851456000 +0200
@@ -26,7 +26,7 @@
                 if (file.exists(paste0(syslibs, "libbrotlidec.a")))
                     libs <- append(libs, c("brotlidec", "brotlicommon"))
             }
-            libs <- append(libs, c("bcrypt", "idn2", "unistring",
+            libs <- append(libs, c("secur32", "bcrypt", "idn2", "unistring",
                            "iconv", "ssl", "crypto", "z", "crypt32", "wsock32",
                            "wldap32", "ssh2", "gcrypt", "gpg-error", "ws2_32",
                            "zstd", "regex"))
diff -Nru orig/Rhtslib/src/Makevars.win patched/Rhtslib/src/Makevars.win
--- orig/Rhtslib/src/Makevars.win	2025-04-15 20:49:11.000000000 +0200
+++ patched/Rhtslib/src/Makevars.win	2025-10-19 19:07:46.180684500 +0200
@@ -7,7 +7,7 @@
 LIBNGHTTP2 = $(or $(and $(wildcard $(R_TOOLS_SOFT)/lib/libnghttp2.a),-lnghttp2),)
 LIBPSL = $(or $(and $(wildcard $(R_TOOLS_SOFT)/lib/libpsl.a),-lpsl),)
 LIBBROTLI = $(or $(and $(wildcard $(R_TOOLS_SOFT)/lib/libbrotlidec.a),-lbrotlidec -lbrotlicommon),)
-PKG_LIBS+=-lm -lbz2 -llzma -lcurl $(LIBNGHTTP2) $(LIBPSL) $(LIBBROTLI) -lbcrypt -lidn2 -lunistring -liconv -lssl -lcrypto -lz -lcrypt32 -lwsock32 -lwldap32 -lssh2 -lgcrypt -lgpg-error -lws2_32 -lzstd -lregex
+PKG_LIBS+=-lm -lbz2 -llzma -lcurl -lsecur32 $(LIBNGHTTP2) $(LIBPSL) $(LIBBROTLI) -lbcrypt -lidn2 -lunistring -liconv -lssl -lcrypto -lz -lcrypt32 -lwsock32 -lwldap32 -lssh2 -lgcrypt -lgpg-error -lws2_32 -lzstd -lregex
 
 populate-usrlib-dir: htslib mk-usrlib-dir
 	cd "${HTSLIB_SOURCE_DIR}" && cp libhts.a "${USRLIB_DIR}"
