diff -Nru orig/RProtoBufLib/configure.win patched/RProtoBufLib/configure.win
--- orig/RProtoBufLib/configure.win	1970-01-01 01:00:00.000000000 +0100
+++ patched/RProtoBufLib/configure.win	2024-03-13 11:16:16.930108900 +0100
@@ -0,0 +1,18 @@
+#! /bin/sh
+
+ARCH=`$R_HOME/bin/Rscript --no-echo -e 'cat(R.version$arch)'`
+
+if [ X$ARCH != Xx86_64 ] ; then
+  if which lld >/dev/null 2>/dev/null ; then
+    ./configure LD=lld $*
+  else
+    ./configure $*
+  fi
+  cp src/Makevars src/Makevars.ucrt
+  sed -i -e 's!GS_LIB =.*!GS_LIB = GatingSet.pb.o!g' \
+         -e 's!libprotobuf\* !libprotobuf.a !g' \
+         -e 's!.*GatingSet.pb.cc.*!	${CXX} ${CXXPICFLAGS} ${PKG_CPPFLAGS} -o ${GS_LIB_DIR}/${GS_LIB} -c ${GS_LIB_DIR}/GatingSet.pb.cc!g' \
+         -e 's!libGatingSet.pb${DYLIB_EXT}!GatingSet.pb.o!g' \
+         src/Makevars.ucrt
+fi
+
