diff -Nru orig/rtracklayer/src/ucsc/bbiRead.c patched/rtracklayer/src/ucsc/bbiRead.c
--- orig/rtracklayer/src/ucsc/bbiRead.c	2023-10-20 20:14:45.000000000 +0200
+++ patched/rtracklayer/src/ucsc/bbiRead.c	2024-03-14 14:51:39.718884500 +0100
@@ -632,7 +632,7 @@
 /* Protect from bad input. */
 if (start >= end)
     return result;
-bzero(summary, summarySize * sizeof(summary[0]));
+memset(summary, 0, summarySize * sizeof(summary[0]));
 
 /* Figure out what size of data we want.  We actually want to get 2 data points per summary
  * value if possible to minimize the effect of a data point being split between summary pixels. */
