diff -Nru orig/affyPLM/src/Makevars.win patched/affyPLM/src/Makevars.win
--- orig/affyPLM/src/Makevars.win	2024-12-15 19:40:17.000000000 +0100
+++ patched/affyPLM/src/Makevars.win	2025-01-29 13:35:21.555542600 +0100
@@ -1,5 +1,12 @@
 PKG_CPPFLAGS += -DHAVE_ZLIB
 PKG_LIBS += $(FLIBS) $(LAPACK_LIBS) $(BLAS_LIBS)
 
+ifeq (,$(shell pkg-config --version 2>/dev/null))
+  PKG_LIBS += -lz
+else
+  PKG_CPPFLAGS = $(shell pkg-config --cflags zlib)
+  PKG_LIBS += $(shell pkg-config --libs zlib)
+endif
+
 %.o: %.c
 	$(CC) $(ALL_CPPFLAGS) $(ALL_CFLAGS) -c $< -o $@
