diff -Nru orig/chromstaR/src/Makevars patched/chromstaR/src/Makevars
--- orig/chromstaR/src/Makevars	2023-10-20 20:31:19.000000000 +0200
+++ patched/chromstaR/src/Makevars	2024-03-16 11:25:55.437492900 +0100
@@ -1,2 +1,3 @@
 PKG_LIBS = $(SHLIB_OPENMP_CXXFLAGS)
+PKG_CPPFLAGS = -DR_NO_REMAP
 PKG_CXXFLAGS = $(SHLIB_OPENMP_CXXFLAGS)
diff -Nru orig/chromstaR/src/Makevars.win patched/chromstaR/src/Makevars.win
--- orig/chromstaR/src/Makevars.win	2023-10-20 20:31:19.000000000 +0200
+++ patched/chromstaR/src/Makevars.win	1970-01-01 01:00:00.000000000 +0100
@@ -1,2 +0,0 @@
-PKG_LIBS = $(SHLIB_OPENMP_CXXFLAGS)
-PKG_CXXFLAGS = $(SHLIB_OPENMP_CXXFLAGS)
