diff -Nru orig/DiffBind/src/croi_main.cpp patched/DiffBind/src/croi_main.cpp
--- orig/DiffBind/src/croi_main.cpp	2023-10-20 20:17:28.000000000 +0200
+++ patched/DiffBind/src/croi_main.cpp	2024-03-16 11:30:21.769681300 +0100
@@ -1,12 +1,13 @@
 #include <unistd.h>
-#include <R.h>
-#include <Rinternals.h>
 
 #include "nodeGroup.h"
 #include "croi_func.h"
 #include "iBucket.h"
 #include "densitySet.h"
 
+#include <R.h>
+#include <Rinternals.h>
+
 extern "C" {
 
 SEXP croi_count_reads(SEXP filename_r,SEXP insertLength_r,SEXP filetype_r,
