diff -Nru orig/flowWorkspace/src/Makevars.ucrt patched/flowWorkspace/src/Makevars.ucrt
--- orig/flowWorkspace/src/Makevars.ucrt	2024-01-13 15:56:13.000000000 +0100
+++ patched/flowWorkspace/src/Makevars.ucrt	2024-03-15 09:12:08.931415600 +0100
@@ -1,7 +1,8 @@
 
 CXX_STD = CXX17
 
-PKG_CPPFLAGS =-DROUT -I../inst/include -DBOOST_NO_AUTO_PTR -DRCPP_PARALLEL_USE_TBB=1 -fpermissive 
+PKG_CPPFLAGS =-DROUT -I../inst/include -DBOOST_NO_AUTO_PTR -DRCPP_PARALLEL_USE_TBB=1 -fpermissive \
+              -D_LIBCPP_ENABLE_CXX17_REMOVED_UNARY_BINARY_FUNCTION 
 
 #needs to wrap in $(shell) to strip the quotes returned by rhdf5lib::pkgconfig
 RHDF5_LIBS= $(shell "${R_HOME}/bin/Rscript" -e "Rhdf5lib::pkgconfig('PKG_CXX_LIBS')")
