diff -Nru orig/lpsymphony/configure patched/lpsymphony/configure
--- orig/lpsymphony/configure	2023-10-25 01:56:12.000000000 +0200
+++ patched/lpsymphony/configure	2024-03-14 15:29:30.270125500 +0100
@@ -9,8 +9,8 @@
 
 R="${R_HOME}/bin/R"
 
-SYMPHONY_CPPFLAGS=`pkg-config --cflags SYMPHONY >/dev/null 2>&1`
-SYMPHONY_LIBS=`pkg-config --libs SYMPHONY >/dev/null 2>&1`
+SYMPHONY_CPPFLAGS=`pkg-config --cflags SYMPHONY 2>/dev/null`
+SYMPHONY_LIBS=`pkg-config --libs SYMPHONY 2>/dev/null`
 
 test -z "${SYMPHONY_LIBS}" && SYMPHONY_LIBS="-lSym -lCgl -lOsiClp -lClp -lOsi -lCoinUtils"
 
diff -Nru orig/lpsymphony/configure.win patched/lpsymphony/configure.win
--- orig/lpsymphony/configure.win	2023-10-20 20:27:51.000000000 +0200
+++ patched/lpsymphony/configure.win	2024-03-14 15:31:14.218330600 +0100
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+ARCH=`$R_HOME/bin/Rscript --no-echo -e 'cat(R.version$arch)'`
+
+if [ X$ARCH != Xx86_64 ] ; then
+  ./configure $* 
+  cp src/Makevars src/Makevars.ucrt
+fi
