diff -Nru orig/msa/src/ClustalOmega/src/argtable2/arg_int.c patched/msa/src/ClustalOmega/src/argtable2/arg_int.c
--- orig/msa/src/ClustalOmega/src/argtable2/arg_int.c	2021-10-26 21:08:47.000000000 +0200
+++ patched/msa/src/ClustalOmega/src/argtable2/arg_int.c	2021-11-29 19:12:57.342311100 +0100
@@ -32,7 +32,8 @@
 #include <limits.h>
 
 /* local error codes */
-enum {EMINCOUNT=1,EMAXCOUNT,EBADINT,EOVERFLOW};
+// EOVERFLOW conflicts in MinGW 9
+enum {EMINCOUNT=1,EMAXCOUNT,EBADINT,CO_EOVERFLOW};
 
 static void resetfn(struct arg_int *parent)
     {
diff -Nru orig/msa/src/gc-7.6.12/msaMakefile.win patched/msa/src/gc-7.6.12/msaMakefile.win
--- orig/msa/src/gc-7.6.12/msaMakefile.win	2021-10-26 21:08:47.000000000 +0200
+++ patched/msa/src/gc-7.6.12/msaMakefile.win	2021-11-29 19:45:38.031937000 +0100
@@ -9,8 +9,8 @@
 
 gc72:
 	export PKG_LIBS="$(PKG_LIBS) -I$(AO_INCLUDE_DIR) -I$(INCLUDE_DIR) -I$(PRIVATE_INCLUDE_DIR)"; \
-	export PKG_CFLAGS="$(PKG_CFLAGS) -DALL_INTERIOR_POINTERS -DGC_DLL -DGC_THREADS -D_CRT_SECURE_NO_DEPRECATE -I$(AO_INCLUDE_DIR) -I$(INCLUDE_DIR) -I$(PRIVATE_INCLUDE_DIR)"; \
-	export PKG_CXXFLAGS="$(PKG_CXXFLAGS) -DALL_INTERIOR_POINTERS -DALL_INTERIOR_POINTERS -DGC_DLL -DGC_THREADS -D_CRT_SECURE_NO_DEPRECATE -I$(AO_INCLUDE_DIR) -I$(INCLUDE_DIR) -I$(PRIVATE_INCLUDE_DIR)"; \
+	export PKG_CFLAGS="-fcommon $(PKG_CFLAGS) -DALL_INTERIOR_POINTERS -DGC_DLL -DGC_THREADS -D_CRT_SECURE_NO_DEPRECATE -I$(AO_INCLUDE_DIR) -I$(INCLUDE_DIR) -I$(PRIVATE_INCLUDE_DIR)"; \
+	export PKG_CXXFLAGS="-fcommon $(PKG_CXXFLAGS) -DALL_INTERIOR_POINTERS -DALL_INTERIOR_POINTERS -DGC_DLL -DGC_THREADS -D_CRT_SECURE_NO_DEPRECATE -I$(AO_INCLUDE_DIR) -I$(INCLUDE_DIR) -I$(PRIVATE_INCLUDE_DIR)"; \
 	${R_HOME}/bin${R_ARCH_BIN}/R.exe CMD SHLIB -o gc7612.dll *.c; \
 	${R_HOME}/bin${R_ARCH_BIN}/R.exe CMD SHLIB -o gccpp7612.dll -L. -lgc72 *.cpp; \
 	$(AR) rcs libGC.a $(OBJNamesGC); \
