diff -Nru orig/Rhtslib/R/zzz.R patched/Rhtslib/R/zzz.R
--- orig/Rhtslib/R/zzz.R	2022-11-01 11:53:58.000000000 -0700
+++ patched/Rhtslib/R/zzz.R	2022-12-01 03:07:48.503874400 -0800
@@ -15,7 +15,7 @@
         if (platform == "Windows") {
             ## See how PKG_LIBS is defined in Rhtslib/src/Makevars.win
             ## and make sure to produce the same value here.
-            libs <- c("z", "m", "bz2", "lzma", "curl", "idn2", "unistring",
+            libs <- c("z", "m", "bz2", "lzma", "curl", "bcrypt", "idn2", "unistring",
                       "iconv", "ssl", "crypto", "crypt32", "wsock32",
                       "wldap32", "ssh2", "gcrypt", "gpg-error", "ws2_32",
                       "zstd", "regex")
diff -Nru orig/Rhtslib/src/Makevars.win patched/Rhtslib/src/Makevars.win
--- orig/Rhtslib/src/Makevars.win	2022-11-01 11:53:58.000000000 -0700
+++ patched/Rhtslib/src/Makevars.win	2022-12-01 03:08:08.691581200 -0800
@@ -9,7 +9,7 @@
 
 ## Linker options. Make sure the Rhtslib::pkgconfig() function (defined in
 ## R/zzz.R) produces the same PKG_LIBS value.
-PKG_LIBS+=-lz -lm -lbz2 -llzma -lcurl -lidn2 -lunistring -liconv -lssl -lcrypto -lcrypt32 -lwsock32 -lwldap32 -lssh2 -lgcrypt -lgpg-error -lws2_32 -lzstd -lregex
+PKG_LIBS+=-lz -lm -lbz2 -llzma -lcurl -lbcrypt -lidn2 -lunistring -liconv -lssl -lcrypto -lcrypt32 -lwsock32 -lwldap32 -lssh2 -lgcrypt -lgpg-error -lws2_32 -lzstd -lregex
 
 populate-usrlib-dir: htslib mk-usrlib-dir
 	cd "${HTSLIB_SOURCE_DIR}" && cp libhts.a "${USRLIB_DIR}"
diff -Nru orig/Rhtslib/src/htslib-1.15.1/Makefile.Rhtslib.win patched/Rhtslib/src/htslib-1.15.1/Makefile.Rhtslib.win
--- orig/Rhtslib/src/htslib-1.15.1/Makefile.Rhtslib.win	2022-11-01 11:53:58.000000000 -0700
+++ patched/Rhtslib/src/htslib-1.15.1/Makefile.Rhtslib.win	2022-12-01 03:24:52.691298000 -0800
@@ -38,7 +38,7 @@
 LDFLAGS  := $(shell ${R_HOME}/bin/R CMD config LDFLAGS)
 
 # Default libraries to link if configure is not used
-htslib_default_libs = -lz -lm -lbz2 -llzma -lcurl -lidn2 -lunistring -liconv -lssl -lcrypto -lcrypt32 -lwsock32 -lwldap32 -lssh2 -lgcrypt -lgpg-error -lws2_32 -lzstd -lregex
+htslib_default_libs = -lz -lm -lbz2 -llzma -lcurl -lbcrypt -lidn2 -lunistring -liconv -lssl -lcrypto -lcrypt32 -lwsock32 -lwldap32 -lssh2 -lgcrypt -lgpg-error -lws2_32 -lzstd -lregex
 
 CPPFLAGS += -D_FILE_OFFSET_BITS=64 -DCURL_STATICLIB
 # TODO: make the 64-bit support for VCF optional via configure, for now add -DVCF_ALLOW_INT64
@@ -75,7 +75,7 @@
 # Set by config.mk if plugins are enabled
 plugindir =
 
-all: lib-static lib-shared $(BUILT_PROGRAMS) plugins $(BUILT_TEST_PROGRAMS) \
+all: lib-static $(BUILT_PROGRAMS) plugins $(BUILT_TEST_PROGRAMS) \
      htslib_static.mk htslib-uninstalled.pc
 
 ALL_CPPFLAGS = -I. $(CPPFLAGS)
