diff -Nru orig/RProtoBufLib/configure.win patched/RProtoBufLib/configure.win
--- orig/RProtoBufLib/configure.win	1969-12-31 16:00:00.000000000 -0800
+++ patched/RProtoBufLib/configure.win	2021-06-28 13:15:46.169708900 -0700
@@ -0,0 +1,4 @@
+#! /bin/bash
+
+./configure $*
+cp src/Makevars src/Makevars.ucrt
Binary files orig/RProtoBufLib/src/.Makevars.in.swp and patched/RProtoBufLib/src/.Makevars.in.swp differ
diff -Nru orig/RProtoBufLib/src/Makevars.in patched/RProtoBufLib/src/Makevars.in
--- orig/RProtoBufLib/src/Makevars.in	2021-05-19 12:15:25.000000000 -0700
+++ patched/RProtoBufLib/src/Makevars.in	2021-06-28 15:00:00.416607700 -0700
@@ -6,13 +6,13 @@
 
 
 #expose compiled object of GatingSet pb API for cytolib to link to
-GS_LIB=libGatingSet.pb${DYLIB_EXT}
+GS_LIB=GatingSet.pb.o
 GS_LIB_DIR = GatingSet_pb_lib
 
 all: copying  
 
 $(GS_LIB):
-	${SHLIB_CXXLD} ${SHLIB_CXXLDFLAGS} ${CXXPICFLAGS} ${PKG_CPPFLAGS} -o ${GS_LIB_DIR}/${GS_LIB} ${GS_LIB_DIR}/GatingSet.pb.cc
+	${CXX} ${CXXPICFLAGS} ${PKG_CPPFLAGS} -o ${GS_LIB_DIR}/${GS_LIB} -c ${GS_LIB_DIR}/GatingSet.pb.cc
 	 	
 	
 #copy hdf5 library headers to package include
@@ -20,6 +20,6 @@
 	mkdir -p "${USER_INCLUDE}"
 	cp -r @PBBUILD@/include/google ${USER_INCLUDE}
 	mkdir -p "${USER_LIB_DIR}"
-	cp @PBBUILD@/lib/libprotobuf* ${USER_LIB_DIR}
+	cp @PBBUILD@/lib/libprotobuf.a ${USER_LIB_DIR}
 	cp ${GS_LIB_DIR}/$(GS_LIB) ${USER_LIB_DIR}/${GS_LIB}
 
