diff -Nru orig/Rsamtools/src/bam_mate_iter.cpp patched/Rsamtools/src/bam_mate_iter.cpp
--- orig/Rsamtools/src/bam_mate_iter.cpp	2023-10-20 20:15:58.000000000 +0200
+++ patched/Rsamtools/src/bam_mate_iter.cpp	2024-03-14 14:00:12.244768200 +0100
@@ -1,8 +1,9 @@
-#include <Rdefines.h>
 #include "BamRangeIterator.h"
 #include "BamFileIterator.h"
 #include "bam_mate_iter.h"
 
+#include <Rdefines.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
