diff -Nru orig/rtracklayer/src/Makevars.win patched/rtracklayer/src/Makevars.win
--- orig/rtracklayer/src/Makevars.win	2025-01-14 19:40:57.000000000 +0100
+++ patched/rtracklayer/src/Makevars.win	2025-03-06 10:29:02.478428900 +0100
@@ -1,7 +1,8 @@
+LIBNGHTTP2 = $(or $(and $(wildcard $(R_TOOLS_SOFT)/lib/libnghttp2.a),-lnghttp2),)
 LIBPSL = $(or $(and $(wildcard $(R_TOOLS_SOFT)/lib/libpsl.a),-lpsl),)
 LIBBROTLI = $(or $(and $(wildcard $(R_TOOLS_SOFT)/lib/libbrotlidec.a),-lbrotlidec -lbrotlicommon),)
 
-PKG_LIBS+=-lcurl $(LIBPSL) $(LIBBROTLI) -lssh2 -lssl -lwldap32\
+PKG_LIBS+=-lcurl $(LIBNGHTTP2) $(LIBPSL) $(LIBBROTLI) -lssh2 -lssl -lwldap32\
     -lbcrypt -lcrypto -lgcrypt -lcrypt32 -lgpg-error -lidn2\
     -lunistring -liconv -lzstd -lws2_32 -lz
 include Makevars.common
