diff -Nru orig/slalom/src/Makevars patched/slalom/src/Makevars
--- orig/slalom/src/Makevars	2023-10-20 20:35:33.000000000 +0200
+++ patched/slalom/src/Makevars	2024-03-16 12:04:33.249351200 +0100
@@ -1 +1,2 @@
+CXX_STD = CXX14
 PKG_LIBS = $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)
diff -Nru orig/slalom/src/Makevars.win patched/slalom/src/Makevars.win
--- orig/slalom/src/Makevars.win	2023-10-20 20:35:33.000000000 +0200
+++ patched/slalom/src/Makevars.win	1970-01-01 01:00:00.000000000 +0100
@@ -1,2 +0,0 @@
-
-PKG_LIBS = $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)
