diff -Nru orig/zlibbioc/src/zlib-1.2.5/win32/Makefile.gcc patched/zlibbioc/src/zlib-1.2.5/win32/Makefile.gcc
--- orig/zlibbioc/src/zlib-1.2.5/win32/Makefile.gcc	2023-10-20 20:17:00.000000000 +0200
+++ patched/zlibbioc/src/zlib-1.2.5/win32/Makefile.gcc	2024-03-12 14:33:48.965476500 +0100
@@ -37,10 +37,14 @@
 
 PREFIX = $(BINPREF)
 
+CC = $(PREFIX)gcc
+
 ifeq "$(WIN)" "64"
-CC = $(PREFIX)gcc -m64
-else
-CC = $(PREFIX)gcc -m32
+CC += -m64
+endif
+
+ifeq "$(WIN)" "32"
+CC += -m32
 endif
 
 CFLAGS = $(LOC) -O3 -Wall
@@ -56,10 +60,12 @@
 ARFLAGS = rcs
 
 RC = $(PREFIX)windres
+RCFLAGS = --define GCC_WINDRES
 ifeq "$(WIN)" "64"
-RCFLAGS = --define GCC_WINDRES --target=pe-x86-64
-else
-RCFLAGS = --define GCC_WINDRES --target=pe-i386
+RCFLAGS += --target=pe-x86-64
+endif
+ifeq "$(WIN)" "32"
+RCFLAGS += --target=pe-i386
 endif
 
 STRIP = $(PREFIX)strip
