diff -Nru c-blosc-1.21.1-orig/blosc/blosc.c c-blosc-1.21.1-patched/blosc/blosc.c
--- c-blosc-1.21.1-orig/blosc/blosc.c	2021-10-06 04:16:29.000000000 -0400
+++ c-blosc-1.21.1-patched/blosc/blosc.c	2022-09-14 14:31:51.938339057 -0400
@@ -57,7 +57,7 @@
 #endif  /* _WIN32 */
 
 /* Include the win32/pthread.h library for all the Windows builds. See #224. */
-#if defined(_WIN32)
+#if 0 && defined(_WIN32)
   #include "win32/pthread.h"
   #include "win32/pthread.c"
 #else
diff -Nru c-blosc-1.21.1-orig/blosc/blosc-export.h c-blosc-1.21.1-patched/blosc/blosc-export.h
--- c-blosc-1.21.1-orig/blosc/blosc-export.h	2021-10-06 04:16:29.000000000 -0400
+++ c-blosc-1.21.1-patched/blosc/blosc-export.h	2022-09-14 14:31:44.642384296 -0400
@@ -30,7 +30,7 @@
   #define BLOSC_EXPORT
 #endif  /* defined(BLOSC_SHARED_LIBRARY) */
 
-#if defined(__GNUC__) || defined(__clang__)
+#if !defined(__MINGW32__) && (defined(__GNUC__) || defined(__clang__))
   #define BLOSC_NO_EXPORT __attribute__((visibility("hidden")))
 #else
   #define BLOSC_NO_EXPORT
diff -Nru c-blosc-1.21.1-orig/blosc/shuffle.c c-blosc-1.21.1-patched/blosc/shuffle.c
--- c-blosc-1.21.1-orig/blosc/shuffle.c	2021-10-06 04:16:29.000000000 -0400
+++ c-blosc-1.21.1-patched/blosc/shuffle.c	2022-09-14 14:31:44.642384296 -0400
@@ -14,7 +14,7 @@
 #include "blosc-comp-features.h"
 #include <stdio.h>
 
-#if defined(_WIN32)
+#if !defined(__MINGW32__)
 #include "win32/pthread.h"
 #else
 #include <pthread.h>
