diff -Nru brotli-1.0.9-orig/scripts/libbrotlicommon.pc.in brotli-1.0.9-patched/scripts/libbrotlicommon.pc.in
--- brotli-1.0.9-orig/scripts/libbrotlicommon.pc.in	2020-08-27 10:12:55.000000000 -0400
+++ brotli-1.0.9-patched/scripts/libbrotlicommon.pc.in	2023-09-18 11:34:17.367503501 -0400
@@ -7,5 +7,5 @@
 URL: https://github.com/google/brotli
 Description: Brotli common dictionary library
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -R${libdir} -lbrotlicommon
+Libs: -L${libdir} -lbrotlicommon
 Cflags: -I${includedir}
diff -Nru brotli-1.0.9-orig/scripts/libbrotlidec.pc.in brotli-1.0.9-patched/scripts/libbrotlidec.pc.in
--- brotli-1.0.9-orig/scripts/libbrotlidec.pc.in	2020-08-27 10:12:55.000000000 -0400
+++ brotli-1.0.9-patched/scripts/libbrotlidec.pc.in	2023-09-18 11:33:50.079494943 -0400
@@ -7,6 +7,6 @@
 URL: https://github.com/google/brotli
 Description: Brotli decoder library
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -R${libdir} -lbrotlidec
+Libs: -L${libdir} -lbrotlidec
 Requires.private: libbrotlicommon >= 1.0.2
 Cflags: -I${includedir}
diff -Nru brotli-1.0.9-orig/scripts/libbrotlienc.pc.in brotli-1.0.9-patched/scripts/libbrotlienc.pc.in
--- brotli-1.0.9-orig/scripts/libbrotlienc.pc.in	2020-08-27 10:12:55.000000000 -0400
+++ brotli-1.0.9-patched/scripts/libbrotlienc.pc.in	2023-09-18 11:34:03.447499163 -0400
@@ -7,6 +7,6 @@
 URL: https://github.com/google/brotli
 Description: Brotli encoder library
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -R${libdir} -lbrotlienc
+Libs: -L${libdir} -lbrotlienc
 Requires.private: libbrotlicommon >= 1.0.2
 Cflags: -I${includedir}
