diff -Nru SYMPHONY-5.6.17-orig/Clp/src/Clp_C_Interface.cpp SYMPHONY-5.6.17-patched/Clp/src/Clp_C_Interface.cpp
--- SYMPHONY-5.6.17-orig/Clp/src/Clp_C_Interface.cpp	2019-01-06 14:43:06.000000000 -0500
+++ SYMPHONY-5.6.17-patched/Clp/src/Clp_C_Interface.cpp	2023-10-23 11:22:19.938097794 -0400
@@ -1232,7 +1232,7 @@
 #endif
 
 COINLIBAPI Clp_Solve *COINLINKAGE
-ClpSolve_new()
+ClpSolve_new(void)
 {
   return new Clp_Solve();
 }
diff -Nru SYMPHONY-5.6.17-orig/Clp/src/Clp_C_Interface.h SYMPHONY-5.6.17-patched/Clp/src/Clp_C_Interface.h
--- SYMPHONY-5.6.17-orig/Clp/src/Clp_C_Interface.h	2019-01-06 14:43:06.000000000 -0500
+++ SYMPHONY-5.6.17-patched/Clp/src/Clp_C_Interface.h	2023-10-23 11:21:37.474154597 -0400
@@ -65,7 +65,7 @@
 /** Destructor */
 COINLIBAPI void COINLINKAGE Clp_deleteModel(Clp_Simplex *model);
 /** Default constructor */
-COINLIBAPI Clp_Solve *COINLINKAGE ClpSolve_new();
+COINLIBAPI Clp_Solve *COINLINKAGE ClpSolve_new(void);
 /** Destructor */
 COINLIBAPI void COINLINKAGE ClpSolve_delete(Clp_Solve *solve);
 /*@}*/
