--- orig/configure	2020-04-16 23:17:05.664959591 +0000
+++ new/configure	2020-04-16 23:17:43.500856740 +0000
@@ -4247,7 +4247,7 @@
 set_default host_cc
 
 pkg_config_fail_message=""
-if ! $pkg_config --version >/dev/null 2>&1; then
+if ! $pkg_config --about >/dev/null 2>&1; then
     warn "$pkg_config not found, library detection may fail."
     pkg_config=false
 elif is_in -static $cc $LDFLAGS && ! is_in --static $pkg_config $pkg_config_flags; then
