diff -Nru fluidsynth-2.4.6-orig/CMakeLists.txt fluidsynth-2.4.6-patched/CMakeLists.txt
--- fluidsynth-2.4.6-orig/CMakeLists.txt	2025-05-04 07:47:50.000000000 -0400
+++ fluidsynth-2.4.6-patched/CMakeLists.txt	2025-05-25 13:26:34.672030861 -0400
@@ -872,7 +872,7 @@
 endif ()
 
 if (WIN32)
-  set ( implibname "fluidsynth-${LIB_VERSION_CURRENT}" )
+  set ( implibname "fluidsynth" )
 else ()
   set ( implibname "fluidsynth" )
 endif ()
diff -Nru fluidsynth-2.4.6-orig/src/CMakeLists.txt fluidsynth-2.4.6-patched/src/CMakeLists.txt
--- fluidsynth-2.4.6-orig/src/CMakeLists.txt	2025-05-04 07:47:50.000000000 -0400
+++ fluidsynth-2.4.6-patched/src/CMakeLists.txt	2025-05-25 13:19:23.512274156 -0400
@@ -326,7 +326,7 @@
       PREFIX "lib"
       OUTPUT_NAME "fluidsynth-${LIB_VERSION_CURRENT}"
       IMPORT_PREFIX "lib"
-      ARCHIVE_OUTPUT_NAME "fluidsynth-${LIB_VERSION_CURRENT}"
+      ARCHIVE_OUTPUT_NAME "fluidsynth"
       VERSION ${LIB_VERSION_INFO}
       SOVERSION ${LIB_VERSION_CURRENT}
     )
