diff -Nru fluidsynth-2.4.3-orig/src/CMakeLists.txt fluidsynth-2.4.3-patched/src/CMakeLists.txt
--- fluidsynth-2.4.3-orig/src/CMakeLists.txt	2025-01-25 14:38:47.000000000 -0500
+++ fluidsynth-2.4.3-patched/src/CMakeLists.txt	2025-02-27 10:22:45.230713777 -0500
@@ -311,7 +311,7 @@
   set_target_properties ( libfluidsynth
     PROPERTIES
       PUBLIC_HEADER "${public_HEADERS}"
-      OUTPUT_NAME "fluidsynth-${LIB_VERSION_CURRENT}"
+      OUTPUT_NAME "fluidsynth"
       VERSION ${LIB_VERSION_INFO}
       SOVERSION ${LIB_VERSION_CURRENT}
     )
