diff -Nru freexl-2.0.0-orig/config.h.in freexl-2.0.0-patched/config.h.in
--- freexl-2.0.0-orig/config.h.in	2023-07-25 01:12:29.000000000 -0400
+++ freexl-2.0.0-patched/config.h.in	2023-09-14 08:46:50.832020602 -0400
@@ -25,7 +25,7 @@
 /* Define to 1 if you have the <memory.h> header file. */
 #undef HAVE_MEMORY_H
 
-/* Define to 1 if you have the <minizip/unzip.h> header file. */
+/* Define to 1 if you have the <unzip.h> header file. */
 #undef HAVE_MINIZIP_UNZIP_H
 
 /* Define to 1 if you have the `sqrt' function. */
diff -Nru freexl-2.0.0-orig/config-msvc.h freexl-2.0.0-patched/config-msvc.h
--- freexl-2.0.0-orig/config-msvc.h	2023-07-25 01:12:29.000000000 -0400
+++ freexl-2.0.0-patched/config-msvc.h	2023-09-14 08:46:50.876020610 -0400
@@ -25,7 +25,7 @@
 /* Define to 1 if you have the <memory.h> header file. */
 #define HAVE_MEMORY_H 1
 
-/* Define to 1 if you have the <minizip/unzip.h> header file. */
+/* Define to 1 if you have the <unzip.h> header file. */
 /* #undef HAVE_MINIZIP_UNZIP_H */
 
 /* Define to 1 if you have the `sqrt' function. */
diff -Nru freexl-2.0.0-orig/configure freexl-2.0.0-patched/configure
--- freexl-2.0.0-orig/configure	2023-07-25 01:12:29.000000000 -0400
+++ freexl-2.0.0-patched/configure	2023-09-14 08:46:50.856020606 -0400
@@ -17926,16 +17926,16 @@
   $as_echo "#define OMIT_XMLDOC 1" >>confdefs.h
 
 else
-for ac_header in minizip/unzip.h
+for ac_header in unzip.h
 do :
-  ac_fn_c_check_header_mongrel "$LINENO" "minizip/unzip.h" "ac_cv_header_minizip_unzip_h" "$ac_includes_default"
+  ac_fn_c_check_header_mongrel "$LINENO" "unzip.h" "ac_cv_header_minizip_unzip_h" "$ac_includes_default"
 if test "x$ac_cv_header_minizip_unzip_h" = xyes; then :
   cat >>confdefs.h <<_ACEOF
 #define HAVE_MINIZIP_UNZIP_H 1
 _ACEOF
 
 else
-  as_fn_error $? "cannot find minizip/unzip.h, bailing out" "$LINENO" 5
+  as_fn_error $? "cannot find unzip.h, bailing out" "$LINENO" 5
 fi
 
 done
diff -Nru freexl-2.0.0-orig/configure.ac freexl-2.0.0-patched/configure.ac
--- freexl-2.0.0-orig/configure.ac	2023-07-25 01:12:29.000000000 -0400
+++ freexl-2.0.0-patched/configure.ac	2023-09-14 08:46:50.964020622 -0400
@@ -80,7 +80,7 @@
 if test x"$enable_xmldocs" != "xyes"; then
   AC_DEFINE(OMIT_XMLDOC)
 else
-AC_CHECK_HEADERS(minizip/unzip.h,, [AC_MSG_ERROR([cannot find minizip/unzip.h, bailing out])])
+AC_CHECK_HEADERS(unzip.h,, [AC_MSG_ERROR([cannot find unzip.h, bailing out])])
 AC_SEARCH_LIBS(unzLocateFile, minizip,,AC_MSG_ERROR(['libminizip' is required but it doesn't seem to be installed on this system.]))
 
 AC_CHECK_HEADERS(expat.h,, [AC_MSG_ERROR([cannot find expat.h, bailing out])])
diff -Nru freexl-2.0.0-orig/src/freexl_ods.c freexl-2.0.0-patched/src/freexl_ods.c
--- freexl-2.0.0-orig/src/freexl_ods.c	2023-07-25 01:12:29.000000000 -0400
+++ freexl-2.0.0-patched/src/freexl_ods.c	2023-09-14 08:46:50.844020604 -0400
@@ -83,7 +83,7 @@
 #include "freexl.h"
 #include "freexl_internals.h"
 
-#include <minizip/unzip.h>
+#include <unzip.h>
 
 #ifdef _WIN32
 #define strcasecmp	_stricmp
diff -Nru freexl-2.0.0-orig/src/freexl_xlsx.c freexl-2.0.0-patched/src/freexl_xlsx.c
--- freexl-2.0.0-orig/src/freexl_xlsx.c	2023-07-25 01:12:29.000000000 -0400
+++ freexl-2.0.0-patched/src/freexl_xlsx.c	2023-09-14 08:46:50.848020605 -0400
@@ -84,7 +84,7 @@
 #include "freexl.h"
 #include "freexl_internals.h"
 
-#include <minizip/unzip.h>
+#include <unzip.h>
 
 #ifdef _WIN32
 #define strcasecmp	_stricmp
