diff -Nru gdal-3.9.3-orig/frmts/gtiff/libtiff/gdal_libtiff_symbol_rename.h gdal-3.9.3-patched/frmts/gtiff/libtiff/gdal_libtiff_symbol_rename.h
--- gdal-3.9.3-orig/frmts/gtiff/libtiff/gdal_libtiff_symbol_rename.h	2024-10-07 12:02:06.000000000 -0400
+++ gdal-3.9.3-patched/frmts/gtiff/libtiff/gdal_libtiff_symbol_rename.h	2024-10-16 07:40:55.089012569 -0400
@@ -1,6 +1,8 @@
 /* This is a generated file by dump_symbols.h. *DO NOT EDIT MANUALLY !* */
 #ifndef GDAL_LIBTIFF_SYMBOL_RENAME_H_INCLUDED
 #define GDAL_LIBTIFF_SYMBOL_RENAME_H_INCLUDED
+#define _TIFFCheckDirNumberAndOffset gdal__TIFFCheckDirNumberAndOffset
+#define _TIFFGetDirNumberFromOffset gdal__TIFFGetDirNumberFromOffset
 #define add_ms gdal_add_ms
 #define allocChoppedUpStripArrays gdal_allocChoppedUpStripArrays
 #define buildMap gdal_buildMap
