diff -ru gdal-3.3.2-orig/apps/commonutils.h gdal-3.3.2-new/apps/commonutils.h
--- gdal-3.3.2-orig/apps/commonutils.h	2021-09-24 11:23:42.669007602 +0000
+++ gdal-3.3.2-new/apps/commonutils.h	2021-09-24 20:35:39.599706347 +0000
@@ -75,7 +75,44 @@
 #define MAIN_END
 
 #endif // defined(WIN32)
-#endif // defined(__cplusplus)
+#else // defined(__cplusplus) ----------------------------------
+
+#if defined(WIN32) && (defined(_MSC_VER) || defined(SUPPORTS_WMAIN))
+
+#include <wchar.h>
+#include <stdlib.h>
+#include "cpl_conv.h"
+#include "cpl_string.h"
+
+int wmain( int argc, wchar_t ** argv_w, wchar_t ** envp );
+
+#define MAIN_START(argc, argv) \
+  int wmain( int argc, wchar_t ** argv_w, wchar_t **envp ) \
+  { \
+    char **argv = (char**)(CPLCalloc(argc + 1, sizeof(char*))); \
+    { \
+      int i; \
+      for( i = 0; i < argc; i++ ) \
+      { \
+        argv[i] = CPLRecodeFromWChar( argv_w[i], CPL_ENC_UCS2, CPL_ENC_UTF8 ); \
+      } \
+    }
+
+// NOTE: memory will not be explicitly freed on exit/return from wmain()
+
+#define MAIN_END CSLDestroy(argv); \
+    return 0; \
+  }
+
+#else // defined(WIN32)
+
+#define MAIN_START(argc, argv) \
+    int main( int argc, char ** argv )
+
+#define MAIN_END
+
+#endif // defined(WIN32)
+#endif // (!__cplusplus__)
 
 
 CPL_C_START
diff -ru gdal-3.3.2-orig/apps/dumpoverviews.cpp gdal-3.3.2-new/apps/dumpoverviews.cpp
--- gdal-3.3.2-orig/apps/dumpoverviews.cpp	2021-09-24 11:23:42.669007602 +0000
+++ gdal-3.3.2-new/apps/dumpoverviews.cpp	2021-09-24 11:24:29.197007787 +0000
@@ -31,6 +31,7 @@
 #include "cpl_string.h"
 #include "gdal_priv.h"
 #include "ogr_spatialref.h"
+#include "commonutils.h"
 
 CPL_CVSID("$Id: dumpoverviews.cpp 355b41831cd2685c85d1aabe5b95665a2c6e99b7 2019-06-19 17:07:04 +0200 Even Rouault $")
 
@@ -51,7 +52,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main( int argc, char ** argv )
+MAIN_START(argc, argv)
 
 {
     GDALAllRegister();
@@ -180,6 +181,7 @@
 
     return 0;
 }
+MAIN_END
 
 /************************************************************************/
 /*                              DumpBand()                              */
diff -ru gdal-3.3.2-orig/apps/gdal2ogr.c gdal-3.3.2-new/apps/gdal2ogr.c
--- gdal-3.3.2-orig/apps/gdal2ogr.c	2021-09-24 11:23:42.669007602 +0000
+++ gdal-3.3.2-new/apps/gdal2ogr.c	2021-09-24 11:24:29.197007787 +0000
@@ -31,6 +31,7 @@
 #include "ogr_api.h"
 #include "ogr_srs_api.h"
 #include "cpl_string.h"
+#include "commonutils.h"
 
 CPL_CVSID("$Id: gdal2ogr.c 355b41831cd2685c85d1aabe5b95665a2c6e99b7 2019-06-19 17:07:04 +0200 Even Rouault $")
 
@@ -73,7 +74,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main(int argc, char* argv[])
+MAIN_START(argc, argv)
 {
     const char     *pszFormat = "ESRI Shapefile";
     char           *pszLayerName = NULL;
@@ -337,3 +338,4 @@
 
     return 0;
 }
+MAIN_END
\ No newline at end of file
diff -ru gdal-3.3.2-orig/apps/gdalasyncread.cpp gdal-3.3.2-new/apps/gdalasyncread.cpp
--- gdal-3.3.2-orig/apps/gdalasyncread.cpp	2021-09-24 11:23:42.669007602 +0000
+++ gdal-3.3.2-new/apps/gdalasyncread.cpp	2021-09-24 20:35:43.335706362 +0000
@@ -32,6 +32,7 @@
 #include "gdal_version.h"
 #include "gdal_priv.h"
 #include "ogr_spatialref.h"
+#include "commonutils.h"
 
 CPL_CVSID("$Id: gdalasyncread.cpp 3933b3a21990c6b7072b1a4e50e8520c7bfaf1f8 2021-06-09 17:21:21 +0800 C41eb $")
 
@@ -73,7 +74,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main( int argc, char ** argv )
+MAIN_START(argc, argv)
 
 {
     GDALDatasetH        hSrcDS, hDstDS;
@@ -558,3 +559,4 @@
     GDALDumpOpenDatasets( stderr );
     GDALDestroyDriverManager();
 }
+MAIN_END
diff -ru gdal-3.3.2-orig/apps/gdalflattenmask.c gdal-3.3.2-new/apps/gdalflattenmask.c
--- gdal-3.3.2-orig/apps/gdalflattenmask.c	2021-09-24 11:23:42.669007602 +0000
+++ gdal-3.3.2-new/apps/gdalflattenmask.c	2021-09-24 11:24:29.201007787 +0000
@@ -29,6 +29,7 @@
 #include "gdal.h"
 #include "cpl_conv.h"
 #include "cpl_string.h"
+#include "commonutils.h"
 
 CPL_CVSID("$Id: gdalflattenmask.c 355b41831cd2685c85d1aabe5b95665a2c6e99b7 2019-06-19 17:07:04 +0200 Even Rouault $")
 
@@ -55,7 +56,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main(int argc, char* argv[])
+MAIN_START(argc, argv)
 {
     const char* pszFormat = "GTiff";
     const char* pszSrcFilename = NULL;
@@ -338,3 +339,4 @@
 
     return 0;
 }
+MAIN_END
diff -ru gdal-3.3.2-orig/apps/gdaltorture.cpp gdal-3.3.2-new/apps/gdaltorture.cpp
--- gdal-3.3.2-orig/apps/gdaltorture.cpp	2021-09-24 11:23:42.669007602 +0000
+++ gdal-3.3.2-new/apps/gdaltorture.cpp	2021-09-24 11:24:29.201007787 +0000
@@ -29,6 +29,7 @@
 #include "gdal.h"
 #include "cpl_string.h"
 #include "cpl_conv.h"
+#include "commonutils.h"
 
 #include <cassert>
 
@@ -237,7 +238,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main( int argc, char **argv )
+MAIN_START(argc, argv)
 {
     GDALAllRegister();
 
@@ -292,3 +293,4 @@
 
     return 0;
 }
+MAIN_END
\ No newline at end of file
diff -ru gdal-3.3.2-orig/apps/gdalwarpsimple.c gdal-3.3.2-new/apps/gdalwarpsimple.c
--- gdal-3.3.2-orig/apps/gdalwarpsimple.c	2021-09-24 11:23:42.669007602 +0000
+++ gdal-3.3.2-new/apps/gdalwarpsimple.c	2021-09-24 11:24:29.201007787 +0000
@@ -31,6 +31,7 @@
 #include "gdal_alg.h"
 #include "cpl_string.h"
 #include "ogr_srs_api.h"
+#include "commonutils.h"
 
 CPL_CVSID("$Id: gdalwarpsimple.c ff8146d84de7cba8e09d212d5481ea7d2ede3e98 2017-06-27 20:47:31Z Even Rouault $")
 
@@ -91,7 +92,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main( int argc, char ** argv )
+MAIN_START(argc, argv)
 
 {
     GDALDatasetH        hSrcDS, hDstDS;
@@ -313,6 +314,7 @@
 
     exit( 0 );
 }
+MAIN_END
 
 /************************************************************************/
 /*                        GDALWarpCreateOutput()                        */
diff -ru gdal-3.3.2-orig/apps/multireadtest.cpp gdal-3.3.2-new/apps/multireadtest.cpp
--- gdal-3.3.2-orig/apps/multireadtest.cpp	2021-09-24 11:23:42.669007602 +0000
+++ gdal-3.3.2-new/apps/multireadtest.cpp	2021-09-24 11:24:29.205007787 +0000
@@ -30,6 +30,7 @@
 #include "gdal_alg.h"
 #include "cpl_multiproc.h"
 #include "cpl_string.h"
+#include "commonutils.h"
 #include <vector>
 
 CPL_CVSID("$Id: multireadtest.cpp cf1507fcfdb6ef16ec9afab68cc0d7d969c7bcbd 2020-02-27 21:22:11 +0100 Even Rouault $")
@@ -121,8 +122,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main( int argc, char ** argv )
-
+MAIN_START(argc, argv)
 {
 /* -------------------------------------------------------------------- */
 /*      Process arguments.                                              */
@@ -254,3 +254,4 @@
 
     return 0;
 }
+MAIN_END
diff -ru gdal-3.3.2-orig/apps/testreprojmulti.cpp gdal-3.3.2-new/apps/testreprojmulti.cpp
--- gdal-3.3.2-orig/apps/testreprojmulti.cpp	2021-09-24 11:23:42.669007602 +0000
+++ gdal-3.3.2-new/apps/testreprojmulti.cpp	2021-09-24 11:24:29.205007787 +0000
@@ -32,6 +32,7 @@
 #include "cpl_atomic_ops.h"
 #include "cpl_multiproc.h"
 #include "ogr_spatialref.h"
+#include "commonutils.h"
 
 CPL_CVSID("$Id: testreprojmulti.cpp 355b41831cd2685c85d1aabe5b95665a2c6e99b7 2019-06-19 17:07:04 +0200 Even Rouault $")
 
@@ -63,7 +64,7 @@
         CPLAtomicInc(&nIter);
         memcpy(padfResultX, padfRefX, 1024 * sizeof(double));
         memcpy(padfResultY, padfRefY, 1024 * sizeof(double));
-        poCT->TransformEx( 1024, padfResultX, padfResultY, nullptr, nullptr );
+        poCT->Transform( 1024, padfResultX, padfResultY, nullptr);
 
         /* Check that the results are consistent with the reference results */
         assert(memcmp(padfResultX, padfRefResultX, 1024 * sizeof(double)) == 0);
@@ -74,7 +75,7 @@
     }
 }
 
-int main(int argc, char* argv[])
+MAIN_START(argc, argv)
 {
     int nThreads = 2;
 
@@ -109,7 +110,7 @@
     memcpy(padfRefResultX, padfRefX, 1024 * sizeof(double));
     memcpy(padfRefResultY, padfRefY, 1024 * sizeof(double));
 
-    poCT->TransformEx( 1024, padfRefResultX, padfRefResultY, nullptr, nullptr );
+    poCT->Transform( 1024, padfRefResultX, padfRefResultY, nullptr);
 
     for(int i=0;i<nThreads;i++)
         CPLCreateThread(ReprojFunc, nullptr);
@@ -119,3 +120,4 @@
 
     return 0;
 }
+MAIN_END
diff -ru gdal-3.3.2-orig/configure gdal-3.3.2-new/configure
--- gdal-3.3.2-orig/configure	2021-09-24 11:23:42.661007602 +0000
+++ gdal-3.3.2-new/configure	2021-09-24 11:34:26.061010161 +0000
@@ -33461,7 +33461,7 @@
     NETCDF_VERSION=`$NETCDF_NCCONFIG --version`
     NETCDF_PREFIX=`$NETCDF_NCCONFIG --prefix`
     NETCDF_INCLUDEDIR=`$NETCDF_NCCONFIG --includedir`
-    NETCDF_LIBS=`$NETCDF_NCCONFIG --libs`
+    NETCDF_LIBS=`$NETCDF_NCCONFIG --static --libs`
 
     if echo "x" | sed -E "s/x/y/" >/dev/null 2>/dev/null; then
         STRIPPED_LIBRARY_NAME="$(echo "${NETCDF_LIBS}" | sed -E 's:-L/(usr/)?(lib(64)?|lib/x86_64-linux-gnu)/*[[:space:]]+::')"
@@ -33525,6 +33525,8 @@
       EXTRA_INCLUDES="-I$NETCDF_INCLUDEDIR $EXTRA_INCLUDES"
       NETCDF_ROOT=$NETCDF_PREFIX
       LIBS="$NETCDF_LIBS $LIBS"
+    else
+      as_fn_error $? "\"Netcdf via nc-config not working\"" "$LINENO" 5
     fi
 
 
@@ -38506,7 +38508,7 @@
 
       HAVE_GEOS="no"
 
-      GEOS_LIBS="`${GEOS_CONFIG} --ldflags` -lgeos_c"
+      GEOS_LIBS="`${GEOS_CONFIG} --static-clibs`"
       GEOS_CFLAGS="`${GEOS_CONFIG} --cflags`"
       GEOS_VERSION="`${GEOS_CONFIG} --version`"
 
@@ -42601,7 +42603,7 @@
         if test $RETVAL -eq 0; then
             { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-            CXXFLAGS="$CXXFLAGS -DSUPPORTS_WMAIN"
+            CPPFLAGS="$CPPFLAGS -DSUPPORTS_WMAIN"
             LDFLAGS="$LDFLAGS -municode"
         else
             { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
diff -ru gdal-3.3.2-orig/configure.ac gdal-3.3.2-new/configure.ac
--- gdal-3.3.2-orig/configure.ac	2021-09-24 11:23:42.517007601 +0000
+++ gdal-3.3.2-new/configure.ac	2021-09-24 11:31:51.581009547 +0000
@@ -3163,7 +3163,7 @@
     NETCDF_VERSION=`$NETCDF_NCCONFIG --version`
     NETCDF_PREFIX=`$NETCDF_NCCONFIG --prefix`
     NETCDF_INCLUDEDIR=`$NETCDF_NCCONFIG --includedir`
-    NETCDF_LIBS=`$NETCDF_NCCONFIG --libs`
+    NETCDF_LIBS=`$NETCDF_NCCONFIG --static --libs`
     STRIP_SYSTEM_LIBRARY_PATHS("${NETCDF_LIBS}")
     NETCDF_LIBS="$STRIPPED_LIBRARY_NAME"
     echo ""
@@ -3176,6 +3176,8 @@
       EXTRA_INCLUDES="-I$NETCDF_INCLUDEDIR $EXTRA_INCLUDES"
       NETCDF_ROOT=$NETCDF_PREFIX
       LIBS="$NETCDF_LIBS $LIBS"
+    else
+      AC_MSG_ERROR("Netcdf via nc-config not working")
     fi
 
 dnl previous behavior without nc-config
@@ -6032,7 +6034,7 @@
         rm -f testwmain*
         if test $RETVAL -eq 0; then
             AC_MSG_RESULT([yes])
-            CXXFLAGS="$CXXFLAGS -DSUPPORTS_WMAIN"
+            CPPFLAGS="$CPPFLAGS -DSUPPORTS_WMAIN"
             LDFLAGS="$LDFLAGS -municode"
         else
             AC_MSG_RESULT([no])
diff -ru gdal-3.3.2-orig/m4/geos.m4 gdal-3.3.2-new/m4/geos.m4
--- gdal-3.3.2-orig/m4/geos.m4	2021-09-24 11:23:42.605007601 +0000
+++ gdal-3.3.2-new/m4/geos.m4	2021-09-24 11:24:29.225007787 +0000
@@ -124,7 +124,7 @@
 
       HAVE_GEOS="no"
 
-      GEOS_LIBS="`${GEOS_CONFIG} --ldflags` -lgeos_c"
+      GEOS_LIBS="`${GEOS_CONFIG} --static-clibs`"
       GEOS_CFLAGS="`${GEOS_CONFIG} --cflags`"
       GEOS_VERSION="`${GEOS_CONFIG} --version`"
 
diff -ru gdal-3.3.2-orig/port/cpl_port.h gdal-3.3.2-new/port/cpl_port.h
--- gdal-3.3.2-orig/port/cpl_port.h	2021-09-24 11:23:42.661007602 +0000
+++ gdal-3.3.2-new/port/cpl_port.h	2021-09-24 11:24:29.225007787 +0000
@@ -113,15 +113,6 @@
 /*      MinGW stuff                                                     */
 /* ==================================================================== */
 
-/* We need __MSVCRT_VERSION__ >= 0x0700 to have "_aligned_malloc" */
-/* Latest versions of mingw32 define it, but with older ones, */
-/* we need to define it manually */
-#if defined(__MINGW32__)
-#ifndef __MSVCRT_VERSION__
-#define __MSVCRT_VERSION__ 0x0700
-#endif
-#endif
-
 /* Needed for std=c11 on Solaris to have strcasecmp() */
 #if defined(GDAL_COMPILATION) && defined(__sun__) && (__STDC_VERSION__ + 0) >= 201112L && (_XOPEN_SOURCE + 0) < 600
 #ifdef _XOPEN_SOURCE
