diff -Nru gdal-3.6.2-orig/frmts/gtiff/libtiff/gdal_libtiff_symbol_rename.h gdal-3.6.2-patched/frmts/gtiff/libtiff/gdal_libtiff_symbol_rename.h
--- gdal-3.6.2-orig/frmts/gtiff/libtiff/gdal_libtiff_symbol_rename.h	2023-01-02 09:38:17.000000000 -0500
+++ gdal-3.6.2-patched/frmts/gtiff/libtiff/gdal_libtiff_symbol_rename.h	2023-09-04 12:40:19.340485465 -0400
@@ -1,4 +1,6 @@
 /* This is a generated file by dump_symbols.h. *DO NOT EDIT MANUALLY !* */
+#define _TIFFCheckDirNumberAndOffset gdal__TIFFCheckDirNumberAndOffset
+#define _TIFFGetDirNumberFromOffset gdal__TIFFGetDirNumberFromOffset
 #define add_ms gdal_add_ms
 #define allocChoppedUpStripArrays gdal_allocChoppedUpStripArrays
 #define buildMap gdal_buildMap
