diff -Nru gdal-3.5.0-orig/apps/commonutils.h gdal-3.5.0-patched/apps/commonutils.h
--- gdal-3.5.0-orig/apps/commonutils.h	2022-05-10 10:03:37.000000000 -0400
+++ gdal-3.5.0-patched/apps/commonutils.h	2022-09-13 14:32:47.631089215 -0400
@@ -75,7 +75,44 @@
 #define MAIN_END
 
 #endif // defined(WIN32)
-#endif // defined(__cplusplus)
+#else // defined(__cplusplus) ----------------------------------
+
+#if defined(WIN32) && (defined(_MSC_VER) || defined(SUPPORTS_WMAIN))
+
+#include <wchar.h>
+#include <stdlib.h>
+#include "cpl_conv.h"
+#include "cpl_string.h"
+
+int wmain( int argc, wchar_t ** argv_w, wchar_t ** envp );
+
+#define MAIN_START(argc, argv) \
+  int wmain( int argc, wchar_t ** argv_w, wchar_t **envp ) \
+  { \
+    char **argv = (char**)(CPLCalloc(argc + 1, sizeof(char*))); \
+    { \
+      int i; \
+      for( i = 0; i < argc; i++ ) \
+      { \
+        argv[i] = CPLRecodeFromWChar( argv_w[i], CPL_ENC_UCS2, CPL_ENC_UTF8 ); \
+      } \
+    }
+
+// NOTE: memory will not be explicitly freed on exit/return from wmain()
+
+#define MAIN_END CSLDestroy(argv); \
+    return 0; \
+  }
+
+#else // defined(WIN32)
+
+#define MAIN_START(argc, argv) \
+    int main( int argc, char ** argv )
+
+#define MAIN_END
+
+#endif // defined(WIN32)
+#endif // (!__cplusplus__)
 
 
 CPL_C_START
diff -Nru gdal-3.5.0-orig/apps/dumpoverviews.cpp gdal-3.5.0-patched/apps/dumpoverviews.cpp
--- gdal-3.5.0-orig/apps/dumpoverviews.cpp	2022-05-10 10:03:37.000000000 -0400
+++ gdal-3.5.0-patched/apps/dumpoverviews.cpp	2022-09-13 14:32:47.631089215 -0400
@@ -31,6 +31,7 @@
 #include "cpl_string.h"
 #include "gdal_priv.h"
 #include "ogr_spatialref.h"
+#include "commonutils.h"
 
 CPL_CVSID("$Id$")
 
@@ -51,7 +52,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main( int argc, char ** argv )
+MAIN_START(argc, argv)
 
 {
     GDALAllRegister();
@@ -180,6 +181,7 @@
 
     return 0;
 }
+MAIN_END
 
 /************************************************************************/
 /*                              DumpBand()                              */
diff -Nru gdal-3.5.0-orig/apps/gdal2ogr.c gdal-3.5.0-patched/apps/gdal2ogr.c
--- gdal-3.5.0-orig/apps/gdal2ogr.c	2022-05-10 10:03:37.000000000 -0400
+++ gdal-3.5.0-patched/apps/gdal2ogr.c	2022-09-13 14:32:47.631089215 -0400
@@ -31,6 +31,7 @@
 #include "ogr_api.h"
 #include "ogr_srs_api.h"
 #include "cpl_string.h"
+#include "commonutils.h"
 
 CPL_CVSID("$Id$")
 
@@ -73,7 +74,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main(int argc, char* argv[])
+MAIN_START(argc, argv)
 {
     const char     *pszFormat = "ESRI Shapefile";
     char           *pszLayerName = NULL;
@@ -337,3 +338,4 @@
 
     return 0;
 }
+MAIN_END
\ No newline at end of file
diff -Nru gdal-3.5.0-orig/apps/gdalasyncread.cpp gdal-3.5.0-patched/apps/gdalasyncread.cpp
--- gdal-3.5.0-orig/apps/gdalasyncread.cpp	2022-05-10 10:03:37.000000000 -0400
+++ gdal-3.5.0-patched/apps/gdalasyncread.cpp	2022-09-13 14:32:47.631089215 -0400
@@ -32,6 +32,7 @@
 #include "gdal_version.h"
 #include "gdal_priv.h"
 #include "ogr_spatialref.h"
+#include "commonutils.h"
 
 CPL_CVSID("$Id$")
 
@@ -73,7 +74,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main( int argc, char ** argv )
+MAIN_START(argc, argv)
 
 {
     GDALDatasetH        hSrcDS, hDstDS;
@@ -558,3 +559,4 @@
     GDALDumpOpenDatasets( stderr );
     GDALDestroyDriverManager();
 }
+MAIN_END
diff -Nru gdal-3.5.0-orig/apps/gdalflattenmask.c gdal-3.5.0-patched/apps/gdalflattenmask.c
--- gdal-3.5.0-orig/apps/gdalflattenmask.c	2022-05-10 10:03:37.000000000 -0400
+++ gdal-3.5.0-patched/apps/gdalflattenmask.c	2022-09-13 14:32:47.631089215 -0400
@@ -29,6 +29,7 @@
 #include "gdal.h"
 #include "cpl_conv.h"
 #include "cpl_string.h"
+#include "commonutils.h"
 
 CPL_CVSID("$Id$")
 
@@ -55,7 +56,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main(int argc, char* argv[])
+MAIN_START(argc, argv)
 {
     const char* pszFormat = "GTiff";
     const char* pszSrcFilename = NULL;
@@ -338,3 +339,4 @@
 
     return 0;
 }
+MAIN_END
diff -Nru gdal-3.5.0-orig/apps/gdaltorture.cpp gdal-3.5.0-patched/apps/gdaltorture.cpp
--- gdal-3.5.0-orig/apps/gdaltorture.cpp	2022-05-10 10:03:37.000000000 -0400
+++ gdal-3.5.0-patched/apps/gdaltorture.cpp	2022-09-13 14:32:47.631089215 -0400
@@ -29,6 +29,7 @@
 #include "gdal.h"
 #include "cpl_string.h"
 #include "cpl_conv.h"
+#include "commonutils.h"
 
 #include <cassert>
 
@@ -237,7 +238,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main( int argc, char **argv )
+MAIN_START(argc, argv)
 {
     GDALAllRegister();
 
@@ -292,3 +293,4 @@
 
     return 0;
 }
+MAIN_END
\ No newline at end of file
diff -Nru gdal-3.5.0-orig/apps/gdalwarpsimple.c gdal-3.5.0-patched/apps/gdalwarpsimple.c
--- gdal-3.5.0-orig/apps/gdalwarpsimple.c	2022-05-10 10:03:37.000000000 -0400
+++ gdal-3.5.0-patched/apps/gdalwarpsimple.c	2022-09-13 14:32:47.631089215 -0400
@@ -31,6 +31,7 @@
 #include "gdal_alg.h"
 #include "cpl_string.h"
 #include "ogr_srs_api.h"
+#include "commonutils.h"
 
 CPL_CVSID("$Id$")
 
@@ -91,7 +92,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main( int argc, char ** argv )
+MAIN_START(argc, argv)
 
 {
     GDALDatasetH        hSrcDS, hDstDS;
@@ -313,6 +314,7 @@
 
     exit( 0 );
 }
+MAIN_END
 
 /************************************************************************/
 /*                        GDALWarpCreateOutput()                        */
diff -Nru gdal-3.5.0-orig/apps/multireadtest.cpp gdal-3.5.0-patched/apps/multireadtest.cpp
--- gdal-3.5.0-orig/apps/multireadtest.cpp	2022-05-10 10:03:37.000000000 -0400
+++ gdal-3.5.0-patched/apps/multireadtest.cpp	2022-09-13 14:32:47.631089215 -0400
@@ -30,6 +30,7 @@
 #include "gdal_alg.h"
 #include "cpl_multiproc.h"
 #include "cpl_string.h"
+#include "commonutils.h"
 #include <vector>
 
 CPL_CVSID("$Id$")
@@ -121,8 +122,7 @@
 /*                                main()                                */
 /************************************************************************/
 
-int main( int argc, char ** argv )
-
+MAIN_START(argc, argv)
 {
 /* -------------------------------------------------------------------- */
 /*      Process arguments.                                              */
@@ -254,3 +254,4 @@
 
     return 0;
 }
+MAIN_END
diff -Nru gdal-3.5.0-orig/apps/testreprojmulti.cpp gdal-3.5.0-patched/apps/testreprojmulti.cpp
--- gdal-3.5.0-orig/apps/testreprojmulti.cpp	2022-05-10 10:03:37.000000000 -0400
+++ gdal-3.5.0-patched/apps/testreprojmulti.cpp	2022-09-13 14:32:47.631089215 -0400
@@ -32,6 +32,7 @@
 #include "cpl_atomic_ops.h"
 #include "cpl_multiproc.h"
 #include "ogr_spatialref.h"
+#include "commonutils.h"
 
 CPL_CVSID("$Id$")
 
@@ -74,7 +75,7 @@
     }
 }
 
-int main(int argc, char* argv[])
+MAIN_START(argc, argv)
 {
     int nThreads = 2;
 
@@ -119,3 +120,4 @@
 
     return 0;
 }
+MAIN_END
diff -Nru gdal-3.5.0-orig/configure gdal-3.5.0-patched/configure
--- gdal-3.5.0-orig/configure	2022-05-10 10:03:41.000000000 -0400
+++ gdal-3.5.0-patched/configure	2022-09-13 14:32:47.639089169 -0400
@@ -32284,78 +32284,20 @@
     NETCDF_VERSION=`$NETCDF_NCCONFIG --version`
     NETCDF_PREFIX=`$NETCDF_NCCONFIG --prefix`
     NETCDF_INCLUDEDIR=`$NETCDF_NCCONFIG --includedir`
-    NETCDF_LIBS=`$NETCDF_NCCONFIG --libs`
-    NETCDF_STATIC_LIBS=`$NETCDF_NCCONFIG --static --libs`
-
-    if echo "x" | sed -E "s/x/y/" >/dev/null 2>/dev/null; then
-        STRIPPED_LIBRARY_NAME="$(echo "${NETCDF_LIBS}" | sed -E 's:-L/(usr/)?(lib(64)?|lib/x86_64-linux-gnu)/*[[:space:]]+::')"
-    elif echo "x" | sed -r "s/x/y/" >/dev/null 2>/dev/null; then
-                STRIPPED_LIBRARY_NAME="$(echo "${NETCDF_LIBS}" | sed -r 's:-L/(usr/)?(lib(64)?|lib/x86_64-linux-gnu)/*[[:space:]]+::')"
-    else
-        STRIPPED_LIBRARY_NAME="$(echo "${NETCDF_LIBS}")"
-    fi
-
-    NETCDF_LIBS="$STRIPPED_LIBRARY_NAME"
+    NETCDF_LIBS=`$NETCDF_NCCONFIG --static --libs`
     echo ""
     { $as_echo "$as_me:${as_lineno-$LINENO}: result:  got version=\"$NETCDF_VERSION\", prefix=\"$NETCDF_PREFIX\",
  libs=\"$NETCDF_LIBS\", includedir=\"$NETCDF_INCLUDEDIR\"" >&5
 $as_echo " got version=\"$NETCDF_VERSION\", prefix=\"$NETCDF_PREFIX\",
  libs=\"$NETCDF_LIBS\", includedir=\"$NETCDF_INCLUDEDIR\"" >&6; }
 
-    SAVED_LIBS="${LIBS}"
-    LIBS="$NETCDF_LIBS"
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for nc_open in -lnetcdf" >&5
 $as_echo_n "checking for nc_open in -lnetcdf... " >&6; }
 if ${ac_cv_lib_netcdf_nc_open+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lnetcdf  $LIBS"
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char nc_open ();
-int
-main ()
-{
-return nc_open ();
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_netcdf_nc_open=yes
-else
-  ac_cv_lib_netcdf_nc_open=no
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-LIBS=$ac_check_lib_save_LIBS
-fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_netcdf_nc_open" >&5
-$as_echo "$ac_cv_lib_netcdf_nc_open" >&6; }
-if test "x$ac_cv_lib_netcdf_nc_open" = xyes; then :
-  NETCDF_SETTING=yes
-else
-  NETCDF_SETTING=no
-fi
-
-    if test "$NETCDF_SETTING" = "no" ; then
-      unset ac_cv_lib_netcdf_nc_open
-      LIBS="$NETCDF_STATIC_LIBS"
-      { $as_echo "$as_me:${as_lineno-$LINENO}: checking for nc_open in -lnetcdf" >&5
-$as_echo_n "checking for nc_open in -lnetcdf... " >&6; }
-if ${ac_cv_lib_netcdf_nc_open+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  ac_check_lib_save_LIBS=$LIBS
-LIBS="-lnetcdf  $LIBS"
+LIBS="-lnetcdf $NETCDF_LIBS $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -32391,13 +32333,13 @@
   NETCDF_SETTING=no
 fi
 
-    fi
-    LIBS="${SAVED_LIBS}"
 
     if test "$NETCDF_SETTING" = "yes" ; then
       EXTRA_INCLUDES="-I$NETCDF_INCLUDEDIR $EXTRA_INCLUDES"
       NETCDF_ROOT=$NETCDF_PREFIX
       LIBS="$NETCDF_LIBS $LIBS"
+    else
+      as_fn_error $? "\"Netcdf via nc-config not working\"" "$LINENO" 5
     fi
 
 
@@ -37142,7 +37084,7 @@
 
       HAVE_GEOS="no"
 
-      GEOS_LIBS="`${GEOS_CONFIG} --ldflags` -lgeos_c"
+      GEOS_LIBS="`${GEOS_CONFIG} --static-clibs`"
       GEOS_CFLAGS="`${GEOS_CONFIG} --cflags`"
       GEOS_VERSION="`${GEOS_CONFIG} --version`"
 
@@ -41477,7 +41419,7 @@
         if test $RETVAL -eq 0; then
             { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-            CXXFLAGS="$CXXFLAGS -DSUPPORTS_WMAIN"
+            CPPFLAGS="$CPPFLAGS -DSUPPORTS_WMAIN"
             LDFLAGS="$LDFLAGS -municode"
         else
             { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
diff -Nru gdal-3.5.0-orig/configure.ac gdal-3.5.0-patched/configure.ac
--- gdal-3.5.0-orig/configure.ac	2022-05-10 10:03:37.000000000 -0400
+++ gdal-3.5.0-patched/configure.ac	2022-09-14 04:39:20.903982131 -0400
@@ -1537,16 +1537,21 @@
     ,,)
 
 HAVE_SPATIALITE=no
+HAVE_SPAT_PKG=no
 
 if test -z "$with_spatialite" -o "$with_spatialite" = "no"; then
     AC_MSG_RESULT(disabled)
 elif test "$with_spatialite" = "yes"; then
     AC_CHECK_HEADERS(sqlite3.h)
     if test "$ac_cv_header_sqlite3_h" = "yes"; then
-        AC_MSG_CHECKING([for spatialite.h in /usr/include or /usr/local/include])
-        if test -f "/usr/include/spatialite.h" -o -f "/usr/local/include/spatialite.h"; then
-            AC_MSG_RESULT(found)
+        #AC_MSG_CHECKING([for spatialite.h in /usr/include or /usr/local/include])
+        #if test -f "/usr/include/spatialite.h" -o -f "/usr/local/include/spatialite.h"; then
+        PKG_CHECK_MODULES(SPATIALITE, spatialite, [HAVE_SPAT_PKG=yes], [HAVE_SPAT_PKG=no])        
+        if test "$HAVE_SPAT_PKG" = "yes"; then
+            OLD_LIBS="$LIBS"
+            LIBS="$LIBS -lsqlite3 -lspatialite -lgeos_c -lgeos -lxml2 -liconv -llzma -lproj -lstdc++ -lws2_32 -lm -lz"
             AC_CHECK_LIB(spatialite,spatialite_init,SPATIALITE_INIT_FOUND=yes,SPATIALITE_INIT_FOUND=no,-lsqlite3)
+            LIBS="$OLD_LIBS"
             if test "$SPATIALITE_INIT_FOUND" = "yes"; then
                 HAVE_SPATIALITE=yes
                 SPATIALITE_LIBS="-lspatialite -lsqlite3"
@@ -2936,28 +2941,19 @@
     NETCDF_VERSION=`$NETCDF_NCCONFIG --version`
     NETCDF_PREFIX=`$NETCDF_NCCONFIG --prefix`
     NETCDF_INCLUDEDIR=`$NETCDF_NCCONFIG --includedir`
-    NETCDF_LIBS=`$NETCDF_NCCONFIG --libs`
-    NETCDF_STATIC_LIBS=`$NETCDF_NCCONFIG --static --libs`
-    STRIP_SYSTEM_LIBRARY_PATHS("${NETCDF_LIBS}")
-    NETCDF_LIBS="$STRIPPED_LIBRARY_NAME"
+    NETCDF_LIBS=`$NETCDF_NCCONFIG --static --libs`
     echo ""
     AC_MSG_RESULT([ got version="$NETCDF_VERSION", prefix="$NETCDF_PREFIX",
  libs="$NETCDF_LIBS", includedir="$NETCDF_INCLUDEDIR"])
 
-    SAVED_LIBS="${LIBS}"
-    LIBS="$NETCDF_LIBS"
-    AC_CHECK_LIB([netcdf], [nc_open], [NETCDF_SETTING=yes], [NETCDF_SETTING=no],)
-    if test "$NETCDF_SETTING" = "no" ; then
-      unset ac_cv_lib_netcdf_nc_open
-      LIBS="$NETCDF_STATIC_LIBS"
-      AC_CHECK_LIB([netcdf], [nc_open], [NETCDF_SETTING=yes], [NETCDF_SETTING=no],)
-    fi
-    LIBS="${SAVED_LIBS}"
+    AC_CHECK_LIB([netcdf], [nc_open], [NETCDF_SETTING=yes], [NETCDF_SETTING=no], $NETCDF_LIBS)
 
     if test "$NETCDF_SETTING" = "yes" ; then
       EXTRA_INCLUDES="-I$NETCDF_INCLUDEDIR $EXTRA_INCLUDES"
       NETCDF_ROOT=$NETCDF_PREFIX
       LIBS="$NETCDF_LIBS $LIBS"
+    else
+      AC_MSG_ERROR("Netcdf via nc-config not working")
     fi
 
 dnl previous behavior without nc-config
@@ -5629,7 +5625,7 @@
         rm -f testwmain*
         if test $RETVAL -eq 0; then
             AC_MSG_RESULT([yes])
-            CXXFLAGS="$CXXFLAGS -DSUPPORTS_WMAIN"
+            CPPFLAGS="$CPPFLAGS -DSUPPORTS_WMAIN"
             LDFLAGS="$LDFLAGS -municode"
         else
             AC_MSG_RESULT([no])
diff -Nru gdal-3.5.0-orig/m4/geos.m4 gdal-3.5.0-patched/m4/geos.m4
--- gdal-3.5.0-orig/m4/geos.m4	2022-05-10 10:03:38.000000000 -0400
+++ gdal-3.5.0-patched/m4/geos.m4	2022-09-13 14:32:47.643089147 -0400
@@ -124,7 +124,7 @@
 
       HAVE_GEOS="no"
 
-      GEOS_LIBS="`${GEOS_CONFIG} --ldflags` -lgeos_c"
+      GEOS_LIBS="`${GEOS_CONFIG} --static-clibs`"
       GEOS_CFLAGS="`${GEOS_CONFIG} --cflags`"
       GEOS_VERSION="`${GEOS_CONFIG} --version`"
 
diff -Nru gdal-3.5.0-orig/ogr/ogrsf_frmts/sqlite/ogrsqliteutility.cpp gdal-3.5.0-patched/ogr/ogrsf_frmts/sqlite/ogrsqliteutility.cpp
--- gdal-3.5.0-orig/ogr/ogrsf_frmts/sqlite/ogrsqliteutility.cpp	2022-05-10 10:03:38.000000000 -0400
+++ gdal-3.5.0-patched/ogr/ogrsf_frmts/sqlite/ogrsqliteutility.cpp	2022-09-13 14:35:48.582070744 -0400
@@ -349,7 +349,8 @@
     std::set<std::string> uniqueFieldsUC;
 
     // std::regex in gcc < 4.9 is broken
-#if !defined(__GNUC__) || defined(__clang__) || __GNUC__ >= 5
+    // seems still broken in newer versions, including gcc 10.3
+#if 0
 
     try
     {
diff -Nru gdal-3.5.0-orig/port/cpl_port.h gdal-3.5.0-patched/port/cpl_port.h
--- gdal-3.5.0-orig/port/cpl_port.h	2022-05-10 10:03:38.000000000 -0400
+++ gdal-3.5.0-patched/port/cpl_port.h	2022-09-13 14:32:47.643089147 -0400
@@ -113,15 +113,6 @@
 /*      MinGW stuff                                                     */
 /* ==================================================================== */
 
-/* We need __MSVCRT_VERSION__ >= 0x0700 to have "_aligned_malloc" */
-/* Latest versions of mingw32 define it, but with older ones, */
-/* we need to define it manually */
-#if defined(__MINGW32__)
-#ifndef __MSVCRT_VERSION__
-#define __MSVCRT_VERSION__ 0x0700
-#endif
-#endif
-
 /* Needed for std=c11 on Solaris to have strcasecmp() */
 #if defined(GDAL_COMPILATION) && defined(__sun__) && (__STDC_VERSION__ + 0) >= 201112L && (_XOPEN_SOURCE + 0) < 600
 #ifdef _XOPEN_SOURCE
