diff -Nru gdk-pixbuf-2.42.10-orig/gdk-pixbuf/meson.build gdk-pixbuf-2.42.10-patched/gdk-pixbuf/meson.build
--- gdk-pixbuf-2.42.10-orig/gdk-pixbuf/meson.build	2022-10-25 12:53:28.000000000 -0400
+++ gdk-pixbuf-2.42.10-patched/gdk-pixbuf/meson.build	2024-10-14 15:41:17.250908890 -0400
@@ -338,7 +338,7 @@
 
   bin = executable(bin_name, bin_source,
                    gdkpixbuf_enum_h,
-                   dependencies: gdk_pixbuf_deps + [ gdkpixbuf_dep ],
+                   dependencies: loaders_deps + gdk_pixbuf_deps + [ gdkpixbuf_dep ],
                    include_directories: [ root_inc, gdk_pixbuf_inc ],
                    c_args: common_cflags + gdk_pixbuf_cflags,
                    install: true)
