diff -Nru geos-3.11.2-orig/include/geos/shape/fractal/HilbertEncoder.h geos-3.11.2-patched/include/geos/shape/fractal/HilbertEncoder.h
--- geos-3.11.2-orig/include/geos/shape/fractal/HilbertEncoder.h	2023-03-16 19:57:01.000000000 -0400
+++ geos-3.11.2-patched/include/geos/shape/fractal/HilbertEncoder.h	2024-02-13 16:39:04.293222846 -0500
@@ -18,6 +18,7 @@
 #include <geos/export.h>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 // Forward declarations
 namespace geos {
diff -Nru geos-3.11.2-orig/tests/unit/capi/GEOSMakeValidTest.cpp geos-3.11.2-patched/tests/unit/capi/GEOSMakeValidTest.cpp
--- geos-3.11.2-orig/tests/unit/capi/GEOSMakeValidTest.cpp	2023-03-16 19:57:01.000000000 -0400
+++ geos-3.11.2-patched/tests/unit/capi/GEOSMakeValidTest.cpp	2024-02-13 16:48:27.595871787 -0500
@@ -5,6 +5,7 @@
 #include <geos_c.h>
 // std
 #include <cstdarg>
+#include <cstdint>
 #include <cstdio>
 #include <cstdlib>
 #include <cmath>
