diff -Nru geos-3.9.1-orig/tools/geos-config.in geos-3.9.1-patched/tools/geos-config.in
--- geos-3.9.1-orig/tools/geos-config.in	2021-02-03 18:19:53.000000000 -0500
+++ geos-3.9.1-patched/tools/geos-config.in	2021-09-09 12:16:51.943891208 -0400
@@ -46,22 +46,11 @@
     --version)
       echo @VERSION@
       ;;
-    --libs)
-      # TODO: make an alias for --clibs
-      # see http://trac.osgeo.org/geos/ticket/497
-      echo -L${libdir} -lgeos-@VERSION_RELEASE@
+    --clibs|--static-clibs)
+      echo -L${libdir} -lgeos_c -lgeos -lm -lstdc++
       ;;
-    --clibs)
-      echo -L${libdir} -lgeos_c
-      ;;
-    --cclibs)
-      echo -L${libdir} -lgeos
-      ;;
-    --static-clibs)
-      echo -L${libdir} -lgeos_c -lgeos -lm
-      ;;
-    --static-cclibs)
-      echo -L${libdir} -lgeos -lm
+    --libs|--cclibs|--static-cclibs)
+      echo -L${libdir} -lgeos -lm -lstdc++
       ;;
     --cflags)
       echo -I${prefix}/include
