diff -Nru geos-3.14.0-orig/tools/geos-config.in geos-3.14.0-patched/tools/geos-config.in
--- geos-3.14.0-orig/tools/geos-config.in	2025-08-21 11:05:34.000000000 -0400
+++ geos-3.14.0-patched/tools/geos-config.in	2025-10-11 08:13:08.654313567 -0400
@@ -41,21 +41,10 @@
     --version)
       echo @VERSION@
       ;;
-    --libs)
-      # TODO: make an alias for --clibs
-      # see http://trac.osgeo.org/geos/ticket/497
-      echo -L${libdir} -lgeos
-      ;;
-    --clibs)
-      echo -L${libdir} -lgeos_c
-      ;;
-    --cclibs)
-      echo -L${libdir} -lgeos
-      ;;
-    --static-clibs)
+    --clibs|--static-clibs)
       echo -L${libdir} -lgeos_c -lgeos -lstdc++ -lm
       ;;
-    --static-cclibs)
+    --libs|--cclibs|--static-cclibs)
       echo -L${libdir} -lgeos -lstdc++ -lm
       ;;
     --cflags)
