diff -Nru geos-3.11.2-orig/tools/geos-config.in geos-3.11.2-patched/tools/geos-config.in
--- geos-3.11.2-orig/tools/geos-config.in	2023-03-16 19:57:01.000000000 -0400
+++ geos-3.11.2-patched/tools/geos-config.in	2023-03-20 05:55:26.260230086 -0400
@@ -41,19 +41,10 @@
     --version)
       echo @VERSION@
       ;;
-    --libs)
-      echo -L${libdir} -lgeos
-      ;;
-    --clibs)
-      echo -L${libdir} -lgeos_c
-      ;;
-    --cclibs)
-      echo -L${libdir} -lgeos
-      ;;
-    --static-clibs)
+    --clibs|--static-clibs)
       echo -L${libdir} -lgeos_c -lgeos -lstdc++ -lm
       ;;
-    --static-cclibs)
+    --libs|--cclibs|--static-cclibs)
       echo -L${libdir} -lgeos -lstdc++ -lm
       ;;
     --cflags)
