diff -Nru geos-3.11.1-orig/tools/geos-config.in geos-3.11.1-patched/tools/geos-config.in
--- geos-3.11.1-orig/tools/geos-config.in	2022-11-13 14:24:41.000000000 -0500
+++ geos-3.11.1-patched/tools/geos-config.in	2023-01-03 10:19:55.400608367 -0500
@@ -41,20 +41,11 @@
     --version)
       echo @VERSION@
       ;;
-    --libs)
-      echo -L${libdir} -lgeos
+    --clibs|--static-clibs)
+      echo -L${libdir} -lgeos_c -lgeos -lm -lstdc++
       ;;
-    --clibs)
-      echo -L${libdir} -lgeos_c
-      ;;
-    --cclibs)
-      echo -L${libdir} -lgeos
-      ;;
-    --static-clibs)
-      echo -L${libdir} -lgeos_c -lgeos -lm
-      ;;
-    --static-cclibs)
-      echo -L${libdir} -lgeos -lm
+    --libs|--cclibs|--static-cclibs)
+      echo -L${libdir} -lgeos -lm -lstdc++
       ;;
     --cflags)
       echo -I${includedir}
