diff -Nru giflib-5.1.9-orig/Makefile giflib-5.1.9-patched/Makefile
--- giflib-5.1.9-orig/Makefile	2019-03-28 14:57:23.000000000 -0400
+++ giflib-5.1.9-patched/Makefile	2023-06-29 07:55:04.681671458 -0400
@@ -68,13 +68,13 @@
 $(UTILS):: libgif.a libutil.a
 
 libgif.so: $(OBJECTS) $(HEADERS)
-	$(CC) $(CFLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,libgif.so.$(LIBMAJOR) -o libgif.so $(OBJECTS)
+	$(CC) $(CFLAGS) -shared $(LDFLAGS) -o libgif.so $(OBJECTS)
 
 libgif.a: $(OBJECTS) $(HEADERS)
 	$(AR) rcs libgif.a $(OBJECTS)
 
 libutil.so: $(UOBJECTS) $(UHEADERS)
-	$(CC) $(CFLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,libutil.so.$(LIBMAJOR) -o libutil.so $(UOBJECTS)
+	$(CC) $(CFLAGS) -shared $(LDFLAGS) -o libutil.so $(UOBJECTS)
 
 libutil.a: $(UOBJECTS) $(UHEADERS)
 	$(AR) rcs libutil.a $(UOBJECTS)
