diff -Nru hdf-4.2.15-orig/configure.ac hdf-4.2.15-patched/configure.ac
--- hdf-4.2.15-orig/configure.ac	2020-03-03 12:40:50.000000000 -0500
+++ hdf-4.2.15-patched/configure.ac	2025-02-24 18:07:51.702521631 -0500
@@ -913,6 +913,7 @@
   *-*-solaris2*)        BAR="solaris"     ;;
   *-apple*)             BAR="apple"       ;;
   *-pc-cygwin*)         BAR="linux"       ;;
+  *-*-mingw*)	         BAR="linux"       ;;
   *)                    echo "*** unknown host $host!"; exit 1 ;;
 esac
 src_files=""
diff -Nru hdf-4.2.15-orig/mfhdf/ncgen/Makefile.am hdf-4.2.15-patched/mfhdf/ncgen/Makefile.am
--- hdf-4.2.15-orig/mfhdf/ncgen/Makefile.am	2020-03-03 12:40:50.000000000 -0500
+++ hdf-4.2.15-patched/mfhdf/ncgen/Makefile.am	2025-02-24 18:07:51.703521629 -0500
@@ -26,16 +26,6 @@
 
 bin_PROGRAMS = ncgen
 
-if HDF_BUILD_NETCDF
-if HDF_BUILD_FORTRAN
-noinst_PROGRAMS = ctest0 ftest0
-else 
-noinst_PROGRAMS = ctest0
-endif
-else
-noinst_PROGRAMS = ctest0
-endif
-
 ## Information for building the "ncgen" program
 ncgen_SOURCES = close.c escapes.c generate.c genlib.c getfill.c init.c      \
                 load.c main.c ncgentab.c
