diff -Nru ImageMagick-7.1.2-5-orig/MagickCore/utility-private.h ImageMagick-7.1.2-5-patched/MagickCore/utility-private.h
--- ImageMagick-7.1.2-5-orig/MagickCore/utility-private.h	2025-10-06 08:42:40.000000000 -0400
+++ ImageMagick-7.1.2-5-patched/MagickCore/utility-private.h	2025-10-11 08:01:23.774487947 -0400
@@ -399,7 +399,7 @@
   path_wide=create_wchar_path(path);
   if (path_wide == (WCHAR *) NULL)
     return(-1);
-  status=_wstati64(path_wide,attributes);
+  status=wstat(path_wide,attributes);
   path_wide=(WCHAR *) RelinquishMagickMemory(path_wide);
   return(status);
 #endif
