diff -Nru ImageMagick-7.1.1-44-orig/MagickCore/utility-private.h ImageMagick-7.1.1-44-patched/MagickCore/utility-private.h
--- ImageMagick-7.1.1-44-orig/MagickCore/utility-private.h	2025-02-22 08:55:05.000000000 -0500
+++ ImageMagick-7.1.1-44-patched/MagickCore/utility-private.h	2025-02-25 05:25:33.242300669 -0500
@@ -387,7 +387,7 @@
    path_wide=create_wchar_path(path);
    if (path_wide == (WCHAR *) NULL)
      return(-1);
-   status=_wstati64(path_wide,attributes);
+   status=wstat(path_wide,attributes);
    path_wide=(WCHAR *) RelinquishMagickMemory(path_wide);
    return(status);
 #endif
