diff -Nru wjakob-tbb-9e219e2-orig/src/tbb/tbb_misc.cpp wjakob-tbb-9e219e2-patched/src/tbb/tbb_misc.cpp
--- wjakob-tbb-9e219e2-orig/src/tbb/tbb_misc.cpp	2021-06-14 04:50:28.000000000 -0400
+++ wjakob-tbb-9e219e2-patched/src/tbb/tbb_misc.cpp	2022-12-01 07:14:27.662544824 -0500
@@ -17,6 +17,10 @@
 // Source file for miscellaneous entities that are infrequently referenced by
 // an executing program.
 
+#if _WIN32||_WIN64
+#include "tbb/machine/windows_api.h"
+#endif
+
 #include "tbb/tbb_stddef.h"
 #include "tbb_assert_impl.h" // Out-of-line TBB assertion handling routines are instantiated here.
 #include "tbb/tbb_exception.h"
diff -Nru wjakob-tbb-9e219e2-orig/src/tbbmalloc/tbbmalloc.cpp wjakob-tbb-9e219e2-patched/src/tbbmalloc/tbbmalloc.cpp
--- wjakob-tbb-9e219e2-orig/src/tbbmalloc/tbbmalloc.cpp	2021-06-14 04:50:28.000000000 -0400
+++ wjakob-tbb-9e219e2-patched/src/tbbmalloc/tbbmalloc.cpp	2022-12-01 07:14:27.662544824 -0500
@@ -14,6 +14,12 @@
     limitations under the License.
 */
 
+#if USE_PTHREAD
+#include <dlfcn.h> // dlopen
+#elif USE_WINTHREAD
+#include "tbb/machine/windows_api.h"
+#endif
+
 #include "TypeDefinitions.h" // Customize.h and proxy.h get included
 #include "tbbmalloc_internal_api.h"
 
@@ -21,12 +27,6 @@
 
 #undef UNICODE
 
-#if USE_PTHREAD
-#include <dlfcn.h> // dlopen
-#elif USE_WINTHREAD
-#include "tbb/machine/windows_api.h"
-#endif
-
 namespace rml {
 namespace internal {
 
