diff -Nru libgpg-error-1.56-orig/src/gpg-error.def.in libgpg-error-1.56-patched/src/gpg-error.def.in
--- libgpg-error-1.56-orig/src/gpg-error.def.in	2025-04-06 10:38:17.000000000 -0400
+++ libgpg-error-1.56-patched/src/gpg-error.def.in	2026-01-17 11:15:46.801054858 -0500
@@ -24,7 +24,6 @@
  */
 
 #include <config.h>
-
 EXPORTS
  gpg_strerror                 @1
  gpg_strerror_r               @2
diff -Nru libgpg-error-1.56-orig/src/mkheader.c libgpg-error-1.56-patched/src/mkheader.c
--- libgpg-error-1.56-orig/src/mkheader.c	2025-08-14 11:04:08.000000000 -0400
+++ libgpg-error-1.56-patched/src/mkheader.c	2026-01-17 11:15:46.801054858 -0500
@@ -606,7 +606,7 @@
     }
   else if (!strcmp (tag, "include:os-add"))
     {
-      if (!strcmp (host_os, "mingw32"))
+      if (strstr (host_os, "mingw"))
         {
           include_file (fname, lnr, "w32-add.h", write_line);
         }
