diff -Nru libgit2-1.7.1-orig/src/libgit2/CMakeLists.txt libgit2-1.7.1-patched/src/libgit2/CMakeLists.txt
--- libgit2-1.7.1-orig/src/libgit2/CMakeLists.txt	2023-08-14 16:49:40.000000000 -0400
+++ libgit2-1.7.1-patched/src/libgit2/CMakeLists.txt	2023-09-18 12:43:27.592618543 -0400
@@ -28,7 +28,7 @@
 list(SORT SRC_GIT2)
 target_sources(libgit2 PRIVATE ${SRC_GIT2})
 
-if(WIN32 AND NOT CYGWIN)
+if(WIN32 AND NOT CYGWIN AND BUILD_SHARED_LIBS)
 	# Add resource information on Windows
 	set(SRC_RC "git2.rc")
 endif()
@@ -56,7 +56,12 @@
 # Compile and link libgit2
 #
 
-add_library(libgit2package ${SRC_RC} ${LIBGIT2_OBJECTS})
+if(BUILD_SHARED_LIBS)
+  add_library(libgit2package ${SRC_RC} ${LIBGIT2_OBJECTS})
+else()
+  add_library(libgit2package ${LIBGIT2_OBJECTS})
+endif()
+
 target_link_libraries(libgit2package ${LIBGIT2_SYSTEM_LIBS})
 target_include_directories(libgit2package SYSTEM PRIVATE ${LIBGIT2_INCLUDES})
 
