diff -Nru libgsf-1.14.47-orig/gsf/Makefile.in libgsf-1.14.47-patched/gsf/Makefile.in
--- libgsf-1.14.47-orig/gsf/Makefile.in	2022-06-01 16:37:40.881679139 +0000
+++ libgsf-1.14.47-patched/gsf/Makefile.in	2022-06-01 16:37:20.705459363 +0000
@@ -93,7 +93,7 @@
 @WITH_WIN32_TRUE@am__append_1 = lib.def stamp-lib.def
 @WITH_WIN32_FALSE@libgsf_1_la_DEPENDENCIES = $(am__DEPENDENCIES_1) \
 @WITH_WIN32_FALSE@	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
-@WITH_WIN32_TRUE@am__append_2 = -no-undefined -export-symbols lib.def
+@WITH_WIN32_TRUE@am__append_2 = -no-undefined
 @HAVE_LIBEXE_TRUE@am__append_3 = gsf-1.exp gsf-1.lib
 @HAVE_INTROSPECTION_TRUE@am__append_4 = Gsf-1.gir
 @HAVE_INTROSPECTION_TRUE@am__append_5 = $(gir_DATA) $(typelibs_DATA)
