diff -Nru l-smash-2.14.5-orig/configure l-smash-2.14.5-patched/configure
--- l-smash-2.14.5-orig/configure	2018-02-10 19:08:16.000000000 -0500
+++ l-smash-2.14.5-patched/configure	2023-09-25 09:19:50.520640892 -0400
@@ -494,7 +494,7 @@
     cat >> config.mak2 << EOF
 cli/${tool}${EXT}: cli/${tool}.o $OBJ_TOOLS $STATICLIB $SHAREDLIB
 	\$(CC) \$(CFLAGS) \$(LDFLAGS) -o \$@ \$< $OBJ_TOOLS -llsmash \$(LIBS)
-	-@ \$(if \$(STRIP), \$(STRIP) \$@)
+	-@ \$(if \$(STRIP), \$(STRIP) \$@.exe)
 
 EOF
 done
diff -Nru l-smash-2.14.5-orig/Makefile l-smash-2.14.5-patched/Makefile
--- l-smash-2.14.5-orig/Makefile	2018-02-10 19:08:16.000000000 -0500
+++ l-smash-2.14.5-patched/Makefile	2023-09-25 09:30:03.196513959 -0400
@@ -44,7 +44,7 @@
 
 install: all install-lib
 	install -d $(DESTDIR)$(bindir)
-	install -m 755 $(TOOLS) $(DESTDIR)$(bindir)
+	install -m 755 $(TOOLS:%=%.exe) $(DESTDIR)$(bindir)
 
 install-lib: liblsmash.pc lib
 	install -d $(DESTDIR)$(includedir)
