diff -Nru mirror-rtmpdump-fa8646d-orig/Makefile mirror-rtmpdump-fa8646d-patched/Makefile
--- mirror-rtmpdump-fa8646d-orig/Makefile	2015-12-23 13:09:27.000000000 -0600
+++ mirror-rtmpdump-fa8646d-patched/Makefile	2021-08-30 16:06:03.831224699 -0500
@@ -26,11 +26,9 @@
 
 bindir=$(prefix)/bin
 sbindir=$(prefix)/sbin
-mandir=$(prefix)/man
 
 BINDIR=$(DESTDIR)$(bindir)
 SBINDIR=$(DESTDIR)$(sbindir)
-MANDIR=$(DESTDIR)$(mandir)
 
 LIBS_posix=
 LIBS_darwin=
@@ -59,11 +57,9 @@
 $(PROGS): $(LIBRTMP)
 
 install:	$(PROGS)
-	-mkdir -p $(BINDIR) $(SBINDIR) $(MANDIR)/man1 $(MANDIR)/man8
+	-mkdir -p $(BINDIR) $(SBINDIR)
 	cp rtmpdump$(EXT) $(BINDIR)
 	cp rtmpgw$(EXT) rtmpsrv$(EXT) rtmpsuck$(EXT) $(SBINDIR)
-	cp rtmpdump.1 $(MANDIR)/man1
-	cp rtmpgw.8 $(MANDIR)/man8
 	@cd librtmp; $(MAKE) install
 
 clean:
