diff -Nru libSBML-5.19.0-Source-orig/CMakeLists.txt libSBML-5.19.0-Source-patched/CMakeLists.txt
--- libSBML-5.19.0-Source-orig/CMakeLists.txt	2020-11-19 05:17:49.000000000 -0600
+++ libSBML-5.19.0-Source-patched/CMakeLists.txt	2021-08-30 13:37:37.170680190 -0500
@@ -968,9 +968,10 @@
     set( MISC_PREFIX "${CMAKE_INSTALL_DATADIR}/libsbml/" )
     set(LIBSBML_LIBRARY sbml)
 else()
-    set( MISC_PREFIX ".\\" )
+   # set( MISC_PREFIX ".\\" )
     set(PATH_SEP "\\")
     set(FILE_SEP ";")
+    set(MISC_PREFIX "${CMAKE_INSTALL_DATADIR}/libsbml/")
     if(MINGW)
         set(LIBSBML_LIBRARY sbml)
     else()
