diff -Nru libsbml-5.20.2-orig/CMakeLists.txt libsbml-5.20.2-patched/CMakeLists.txt
--- libsbml-5.20.2-orig/CMakeLists.txt	2023-09-21 07:16:41.000000000 -0400
+++ libsbml-5.20.2-patched/CMakeLists.txt	2023-09-25 18:20:30.131162183 -0400
@@ -706,9 +706,10 @@
     set( MISC_PREFIX "${CMAKE_INSTALL_DATADIR}/libsbml/" )
     set(LIBSBML_LIBRARY sbml)
 else()
-    set( MISC_PREFIX ".\\" )
+    #set( MISC_PREFIX ".\\" )
     set(PATH_SEP "\\")
     set(FILE_SEP ";")
+    set(MISC_PREFIX "${CMAKE_INSTALL_DATADIR}/libsbml/")
     if(MINGW)
         set(LIBSBML_LIBRARY sbml)
     else()
diff -Nru libsbml-5.20.2-orig/CMakeModules/FindLIBXML.cmake libsbml-5.20.2-patched/CMakeModules/FindLIBXML.cmake
--- libsbml-5.20.2-orig/CMakeModules/FindLIBXML.cmake	2023-09-21 07:16:41.000000000 -0400
+++ libsbml-5.20.2-patched/CMakeModules/FindLIBXML.cmake	2023-09-26 09:28:19.078857940 -0400
@@ -4,13 +4,14 @@
 string(TOUPPER ${PROJECT_NAME} _UPPER_PROJECT_NAME)
 set(_PROJECT_DEPENDENCY_DIR ${_UPPER_PROJECT_NAME}_DEPENDENCY_DIR)
 
-find_library(LIBXML_LIBRARY
-    NAMES  libxml2s xml2s libxml2.lib xml2
-    PATHS ${${_PROJECT_DEPENDENCY_DIR}}/lib64
-          ${${_PROJECT_DEPENDENCY_DIR}}/lib
-    NO_DEFAULT_PATH
-)
-
+if (NOT LIBXML_LIBRARY)
+    find_library(LIBXML_LIBRARY
+        NAMES  libxml2s xml2s libxml2.lib xml2
+        PATHS ${${_PROJECT_DEPENDENCY_DIR}}/lib64
+              ${${_PROJECT_DEPENDENCY_DIR}}/lib
+        NO_DEFAULT_PATH
+    )
+endif()
 
 if (NOT LIBXML_LIBRARY)
     find_library(LIBXML_LIBRARY
@@ -71,7 +72,7 @@
 set(ADDITIONAL_LIBS "${ADDITIONAL_LIBS}ZLIB::ZLIB;")
 endif()
 if (WIN32)
-set(ADDITIONAL_LIBS "ws2_32.lib;${ADDITIONAL_LIBS}")
+set(ADDITIONAL_LIBS ${ADDITIONAL_LIBS} bcrypt z lzma ws2_32)
 endif()
 
 
