diff -Nru libxml2-2.12.10-orig/libxml-2.0.pc.in libxml2-2.12.10-patched/libxml-2.0.pc.in
--- libxml2-2.12.10-orig/libxml-2.0.pc.in	2025-02-18 11:18:55.000000000 -0500
+++ libxml2-2.12.10-patched/libxml-2.0.pc.in	2025-05-22 14:33:47.713115977 -0400
@@ -8,5 +8,6 @@
 Version: @VERSION@
 Description: libXML library version2.
 Requires@XML_PC_PRIVATE@: @XML_PC_REQUIRES@
+Requires.private: liblzma
 Libs: -L${libdir} @XML_LIBS@ @XML_PC_LIBS_PRIVATE@ @XML_PC_LIBS@ @LIBS@
 Cflags: @XML_INCLUDEDIR@ @XML_CFLAGS@
