diff -Nru libxml2-2.10.3-orig/libxml-2.0.pc.in libxml2-2.10.3-patched/libxml-2.0.pc.in
--- libxml2-2.10.3-orig/libxml-2.0.pc.in	2022-05-02 08:10:21.000000000 -0400
+++ libxml2-2.10.3-patched/libxml-2.0.pc.in	2022-12-13 05:11:14.911968419 -0500
@@ -8,6 +8,7 @@
 Version: @VERSION@
 Description: libXML library version2.
 Requires:
+Requires.private: liblzma
 Libs: -L${libdir} @XML_LIBS@
 Libs.private: @XML_PRIVATE_LIBS@ @LIBS@
 Cflags: @XML_INCLUDEDIR@ @XML_CFLAGS@
