diff -Nru zeromq-libzmq-de5ee18-orig/CMakeLists.txt zeromq-libzmq-de5ee18-patched/CMakeLists.txt
--- zeromq-libzmq-de5ee18-orig/CMakeLists.txt	2023-08-26 17:19:35.000000000 -0400
+++ zeromq-libzmq-de5ee18-patched/CMakeLists.txt	2023-09-29 11:09:05.852703758 -0400
@@ -1726,7 +1726,7 @@
 
 if(WIN32)
   set(ZEROMQ_CMAKECONFIG_INSTALL_DIR
-      "CMake"
+      "lib/cmake/zmq"
       CACHE STRING "install path for ZeroMQConfig.cmake")
 else()
   # CMake search path wants either "share" (AKA GNUInstallDirs DATAROOTDIR) for arch-independent, or LIBDIR for arch-
diff -Nru zeromq-libzmq-de5ee18-orig/src/clock.cpp zeromq-libzmq-de5ee18-patched/src/clock.cpp
--- zeromq-libzmq-de5ee18-orig/src/clock.cpp	2023-08-26 17:19:35.000000000 -0400
+++ zeromq-libzmq-de5ee18-patched/src/clock.cpp	2023-09-29 11:27:34.321470240 -0400
@@ -99,10 +99,10 @@
   init_compatible_get_tick_count64 ();
 #endif
 
-#ifndef ZMQ_HAVE_WINDOWS
+//#ifndef ZMQ_HAVE_WINDOWS
 const uint64_t usecs_per_msec = 1000;
 const uint64_t nsecs_per_usec = 1000;
-#endif
+//#endif
 const uint64_t usecs_per_sec = 1000000;
 
 zmq::clock_t::clock_t () :
