diff -Nru zeromq-libzmq-c89390f-orig/CMakeLists.txt zeromq-libzmq-c89390f-patched/CMakeLists.txt
--- zeromq-libzmq-c89390f-orig/CMakeLists.txt	2020-05-25 12:34:17.000000000 -0500
+++ zeromq-libzmq-c89390f-patched/CMakeLists.txt	2021-08-30 16:20:55.559799326 -0500
@@ -1615,7 +1615,7 @@
 
 if(WIN32)
   set(ZEROMQ_CMAKECONFIG_INSTALL_DIR
-      "CMake"
+      "lib/cmake/zmq"
       CACHE STRING "install path for ZeroMQConfig.cmake")
 else()
   # CMake search path wants either "share" (AKA GNUInstallDirs DATAROOTDIR) for arch-independent, or LIBDIR for arch-
