diff -Nru mesa-20.3.0-orig/meson.build mesa-20.3.0-new/meson.build
--- mesa-20.3.0-orig/meson.build	2020-12-03 12:59:53.233230800 -0500
+++ mesa-20.3.0-new/meson.build	2020-12-15 09:56:02.364514973 -0500
@@ -963,8 +963,6 @@
       '-D_ALLOW_KEYWORD_MACROS',
       '-D_HAS_EXCEPTIONS=0', # Tell C++ STL to not use exceptions
     ]
-  else
-    pre_args += ['-D__MSVCRT_VERSION__=0x0700']
   endif
 elif host_machine.system() == 'openbsd'
   pre_args += '-D_ISOC11_SOURCE'
diff -Nru mesa-20.3.0-orig/scons/gallium.py mesa-20.3.0-new/scons/gallium.py
--- mesa-20.3.0-orig/scons/gallium.py	2020-12-03 12:59:53.233230800 -0500
+++ mesa-20.3.0-new/scons/gallium.py	2020-12-15 09:55:19.556561841 -0500
@@ -403,7 +403,6 @@
             ('WINVER', '0x0A00'),
         ]
         if gcc_compat:
-            cppdefines += [('__MSVCRT_VERSION__', '0x0700')]
             cppdefines += ['_USE_MATH_DEFINES']
         if msvc:
             cppdefines += [
