diff -Nru mesa-22.0.2-orig/meson.build mesa-22.0.2-patched/meson.build
--- mesa-22.0.2-orig/meson.build	2022-04-22 00:04:00.000000000 -0400
+++ mesa-22.0.2-patched/meson.build	2022-04-25 12:08:21.744992517 -0400
@@ -1096,7 +1096,8 @@
       '-DNOMINMAX',
     ]
   else
-    pre_args += ['-D__MSVCRT_VERSION__=0x0700']
+# not with UCRT
+#    pre_args += ['-D__MSVCRT_VERSION__=0x0700']
   endif
 elif host_machine.system() == 'openbsd'
   pre_args += '-D_ISOC11_SOURCE'
