diff -Nru minizip-ng-4.0.1-orig/CMakeLists.txt minizip-ng-4.0.1-patched/CMakeLists.txt
--- minizip-ng-4.0.1-orig/CMakeLists.txt	2023-08-04 14:23:54.000000000 -0400
+++ minizip-ng-4.0.1-patched/CMakeLists.txt	2023-09-15 08:50:18.316318686 -0400
@@ -251,7 +251,7 @@
         list(APPEND MINIZIP_LIB ${BZIP2_LIBRARIES})
         list(APPEND MINIZIP_LBD ${BZIP2_LIBRARY_DIRS})
 
-        set(PC_PRIVATE_LIBS "${PC_PRIVATE_LIBS} -lbzip2")
+        set(PC_PRIVATE_LIBS "${PC_PRIVATE_LIBS} -lbz2")
     elseif(MZ_FETCH_LIBS)
         clone_repo(bzip2 https://sourceware.org/git/bzip2.git)
 
@@ -688,7 +688,7 @@
 endif()
 if(MSVC)
     # VS debugger has problems when executable and static library are named the same
-    set_target_properties(${MINIZIP_TARGET} PROPERTIES OUTPUT_NAME lib${MINIZIP_TARGET})
+    set_target_properties(${MINIZIP_TARGET} PROPERTIES OUTPUT_NAME ${MINIZIP_TARGET})
 endif()
 if(MZ_LZMA)
     set_target_properties(${MINIZIP_TARGET} PROPERTIES C_STANDARD 99)
diff -Nru minizip-ng-4.0.1-orig/minizip.c minizip-ng-4.0.1-patched/minizip.c
--- minizip-ng-4.0.1-orig/minizip.c	2023-08-04 14:23:54.000000000 -0400
+++ minizip-ng-4.0.1-patched/minizip.c	2023-09-15 06:48:32.469376839 -0400
@@ -202,7 +202,7 @@
 
     /* Print the progress of the current compress operation */
     if (options->verbose)
-        printf("%s - %" PRId64 " / %" PRId64 " (%.02f%%)\n", file_info->filename, position,
+        printf("%s - %" PRId64 " / %" PRId64 " (%0.2f%%)\n", file_info->filename, position,
             file_info->uncompressed_size, progress);
     return MZ_OK;
 }
@@ -317,7 +317,7 @@
 
     /* Print the progress of the current extraction */
     if (options->verbose)
-        printf("%s - %" PRId64 " / %" PRId64 " (%.02f%%)\n", file_info->filename, position,
+        printf("%s - %" PRId64 " / %" PRId64 " (%0.2f%%)\n", file_info->filename, position,
             file_info->uncompressed_size, progress);
 
     return MZ_OK;
diff -Nru minizip-ng-4.0.1-orig/minizip.pc.cmakein minizip-ng-4.0.1-patched/minizip.pc.cmakein
--- minizip-ng-4.0.1-orig/minizip.pc.cmakein	2023-08-04 14:23:54.000000000 -0400
+++ minizip-ng-4.0.1-patched/minizip.pc.cmakein	2023-09-15 09:10:59.775394606 -0400
@@ -8,7 +8,7 @@
 Description: Zip manipulation library
 Version: @VERSION@
 
-Requires: zlib
-Libs: -L${libdir} -L${sharedlibdir} -l@MINIZIP_TARGET@
-Libs.private:@PC_PRIVATE_LIBS@
+Requires: zlib openssl
+Libs: -L${libdir} -L${sharedlibdir} -l@MINIZIP_TARGET@ -lbz2
+Libs.private:@PC_PRIVATE_LIBS@ -lbcrypt
 Cflags: -I${includedir}
