diff -Nru minizip-ng-4.0.4-orig/minizip.c minizip-ng-4.0.4-patched/minizip.c
--- minizip-ng-4.0.4-orig/minizip.c	2023-12-29 10:05:05.000000000 -0500
+++ minizip-ng-4.0.4-patched/minizip.c	2024-01-08 05:23:25.338544459 -0500
@@ -202,7 +202,7 @@
 
     /* Print the progress of the current compress operation */
     if (options->verbose)
-        printf("%s - %" PRId64 " / %" PRId64 " (%.02f%%)\n", file_info->filename, position,
+        printf("%s - %" PRId64 " / %" PRId64 " (%0.2f%%)\n", file_info->filename, position,
             file_info->uncompressed_size, progress);
     return MZ_OK;
 }
@@ -317,7 +317,7 @@
 
     /* Print the progress of the current extraction */
     if (options->verbose)
-        printf("%s - %" PRId64 " / %" PRId64 " (%.02f%%)\n", file_info->filename, position,
+        printf("%s - %" PRId64 " / %" PRId64 " (%0.2f%%)\n", file_info->filename, position,
             file_info->uncompressed_size, progress);
 
     return MZ_OK;
diff -Nru minizip-ng-4.0.4-orig/minizip.pc.cmakein minizip-ng-4.0.4-patched/minizip.pc.cmakein
--- minizip-ng-4.0.4-orig/minizip.pc.cmakein	2023-12-29 10:05:05.000000000 -0500
+++ minizip-ng-4.0.4-patched/minizip.pc.cmakein	2024-01-08 05:28:58.735212467 -0500
@@ -8,7 +8,7 @@
 Description: Zip manipulation library
 Version: @VERSION@
 
-Requires.private: @PC_PRIVATE_DEPS@
-Libs: -L${libdir} -L${sharedlibdir} -l@MINIZIP_TARGET@
-Libs.private:@PC_PRIVATE_LIBS@
+Requires.private: @PC_PRIVATE_DEPS@ openssl
+Libs: -L${libdir} -L${sharedlibdir} -l@MINIZIP_TARGET@ -lbz2
+Libs.private:@PC_PRIVATE_LIBS@ -lbcrypt
 Cflags: -I${includedir}
