diff -Nru minizip-ng-4.0.8-orig/minizip.c minizip-ng-4.0.8-patched/minizip.c
--- minizip-ng-4.0.8-orig/minizip.c	2025-01-08 13:12:15.000000000 -0500
+++ minizip-ng-4.0.8-patched/minizip.c	2025-01-20 05:35:32.558575985 -0500
@@ -198,7 +198,7 @@
 
     /* Print the progress of the current compress operation */
     if (options->verbose) {
-        printf("%s - %" PRId64 " / %" PRId64 " (%.02f%%)\n", file_info->filename, position,
+        printf("%s - %" PRId64 " / %" PRId64 " (%0.2f%%)\n", file_info->filename, position,
                file_info->uncompressed_size, progress);
     }
     return MZ_OK;
@@ -313,7 +313,7 @@
 
     /* Print the progress of the current extraction */
     if (options->verbose) {
-        printf("%s - %" PRId64 " / %" PRId64 " (%.02f%%)\n", file_info->filename, position,
+        printf("%s - %" PRId64 " / %" PRId64 " (%0.2f%%)\n", file_info->filename, position,
                file_info->uncompressed_size, progress);
     }
 
diff -Nru minizip-ng-4.0.8-orig/minizip.pc.cmakein minizip-ng-4.0.8-patched/minizip.pc.cmakein
--- minizip-ng-4.0.8-orig/minizip.pc.cmakein	2025-01-08 13:12:15.000000000 -0500
+++ minizip-ng-4.0.8-patched/minizip.pc.cmakein	2025-01-20 05:31:05.982025777 -0500
@@ -8,7 +8,7 @@
 Description: Zip manipulation library
 Version: @VERSION@
 
-Requires.private: @PC_PRIVATE_DEPS@
-Libs: -L${libdir} -L${sharedlibdir} -l@MINIZIP_TARGET@
-Libs.private:@PC_PRIVATE_LIBS@
+Requires.private: @PC_PRIVATE_DEPS@ openssl
+Libs: -L${libdir} -L${sharedlibdir} -l@MINIZIP_TARGET@ -lbz2
+Libs.private:@PC_PRIVATE_LIBS@ -lbcrypt
 Cflags: -I${includedir}
