diff -Nru minizip-ng-3.0.3-orig/CMakeLists.txt minizip-ng-3.0.3-patched/CMakeLists.txt
--- minizip-ng-3.0.3-orig/CMakeLists.txt	2021-09-06 15:44:00.000000000 -0400
+++ minizip-ng-3.0.3-patched/CMakeLists.txt	2021-09-09 14:19:02.342572765 -0400
@@ -255,7 +255,7 @@
         list(APPEND MINIZIP_LIB ${BZIP2_LIBRARIES})
         list(APPEND MINIZIP_LBD ${BZIP2_LIBRARY_DIRS})
 
-        set(PC_PRIVATE_LIBS "${PC_PRIVATE_LIBS} -lbzip2")
+        set(PC_PRIVATE_LIBS "${PC_PRIVATE_LIBS} -lbz2")
     elseif(MZ_FETCH_LIBS)
         clone_repo(bzip2 https://sourceware.org/git/bzip2.git)
 
@@ -313,7 +313,7 @@
         list(APPEND MINIZIP_INC ${LIBLZMA_INCLUDE_DIRS})
         list(APPEND MINIZIP_LIB ${LIBLZMA_LIBRARIES})
 
-        set(PC_PRIVATE_LIBS "${PC_PRIVATE_LIBS} -lliblzma")
+        set(PC_PRIVATE_LIBS "${PC_PRIVATE_LIBS} -llzma")
     elseif(MZ_FETCH_LIBS)
         clone_repo(liblzma https://git.tukaani.org/xz.git)
 
@@ -411,6 +411,7 @@
         message(STATUS "Using OpenSSL ${OPENSSL_VERSION}")
 
         list(APPEND MINIZIP_SRC mz_crypt_openssl.c)
+        message(STATUS "Adding OPENSSL LIBRARIES ${OPENSSL_LIBRARIES}")
         list(APPEND MINIZIP_LIB ${OPENSSL_LIBRARIES})
         list(APPEND MINIZIP_LBD ${OPENSSL_LIBRARY_DIRS})
         list(APPEND MINIZIP_INC ${OPENSSL_INCLUDE_DIR})
@@ -437,7 +438,7 @@
             message(STATUS "Using CryptoAPI")
 
             list(APPEND MINIZIP_SRC mz_crypt_win32.c)
-            list(APPEND MINIZIP_LIB crypt32.lib)
+            list(APPEND MINIZIP_LIB -lcrypt32)
         endif()
     else()
         list(APPEND MINIZIP_DEF -DMZ_ZIP_NO_CRYPTO)
@@ -708,7 +709,7 @@
 endif()
 if(MSVC)
     # VS debugger has problems when executable and static library are named the same
-    set_target_properties(${PROJECT_NAME} PROPERTIES OUTPUT_NAME lib${PROJECT_NAME})
+    set_target_properties(${PROJECT_NAME} PROPERTIES OUTPUT_NAME ${PROJECT_NAME})
 endif()
 if(NOT RISCOS)
     set_target_properties(${PROJECT_NAME} PROPERTIES POSITION_INDEPENDENT_CODE 1)
diff -Nru minizip-ng-3.0.3-orig/minizip.c minizip-ng-3.0.3-patched/minizip.c
--- minizip-ng-3.0.3-orig/minizip.c	2021-09-06 15:44:00.000000000 -0400
+++ minizip-ng-3.0.3-patched/minizip.c	2021-09-09 13:48:13.237764494 -0400
@@ -202,7 +202,7 @@
 
     /* Print the progress of the current compress operation */
     if (options->verbose)
-        printf("%s - %" PRId64 " / %" PRId64 " (%.02f%%)\n", file_info->filename, position,
+        printf("%s - %" PRId64 " / %" PRId64 " (%0.2f%%)\n", file_info->filename, position,
             file_info->uncompressed_size, progress);
     return MZ_OK;
 }
@@ -314,7 +314,7 @@
 
     /* Print the progress of the current extraction */
     if (options->verbose)
-        printf("%s - %" PRId64 " / %" PRId64 " (%.02f%%)\n", file_info->filename, position,
+        printf("%s - %" PRId64 " / %" PRId64 " (%0.2f%%)\n", file_info->filename, position,
             file_info->uncompressed_size, progress);
 
     return MZ_OK;
diff -Nru minizip-ng-3.0.3-orig/minizip.pc.cmakein minizip-ng-3.0.3-patched/minizip.pc.cmakein
--- minizip-ng-3.0.3-orig/minizip.pc.cmakein	2021-09-06 15:44:00.000000000 -0400
+++ minizip-ng-3.0.3-patched/minizip.pc.cmakein	2021-09-09 14:43:11.644978924 -0400
@@ -8,7 +8,7 @@
 Description: Minizip zip file manipulation library
 Version: @VERSION@
 
-Requires: zlib
-Libs: -L${libdir} -L${sharedlibdir} -l@PROJECT_NAME@
+Requires: zlib openssl
+Libs: -L${libdir} -L${sharedlibdir} -l@PROJECT_NAME@ -lbz2
 Libs.private:@PC_PRIVATE_LIBS@
 Cflags: -I${includedir}
