diff -Nru minizip-ng-4.0.10-orig/minizip.c minizip-ng-4.0.10-patched/minizip.c
--- minizip-ng-4.0.10-orig/minizip.c	2025-05-04 21:36:06.000000000 -0400
+++ minizip-ng-4.0.10-patched/minizip.c	2025-05-23 03:59:17.264357982 -0400
@@ -215,7 +215,7 @@
 
     /* Print the progress of the current compress operation */
     if (options->verbose) {
-        printf("%s - %" PRId64 " / %" PRId64 " (%.02f%%)\n", file_info->filename, position,
+        printf("%s - %" PRId64 " / %" PRId64 " (%0.2f%%)\n", file_info->filename, position,
                file_info->uncompressed_size, progress);
     }
     return MZ_OK;
@@ -307,6 +307,7 @@
     minizip_opt *options = (minizip_opt *)userdata;
     char *utf8_string = NULL;
 
+    MZ_UNUSED(handle);
     MZ_UNUSED(path);
 
     if ((options->encoding > 0) && (file_info->flag & MZ_ZIP_FLAG_UTF8) == 0) {
@@ -338,7 +339,7 @@
 
     /* Print the progress of the current extraction */
     if (options->verbose) {
-        printf("%s - %" PRId64 " / %" PRId64 " (%.02f%%)\n", file_info->filename, position,
+        printf("%s - %" PRId64 " / %" PRId64 " (%0.2f%%)\n", file_info->filename, position,
                file_info->uncompressed_size, progress);
     }
 
diff -Nru minizip-ng-4.0.10-orig/minizip.pc.cmakein minizip-ng-4.0.10-patched/minizip.pc.cmakein
--- minizip-ng-4.0.10-orig/minizip.pc.cmakein	2025-05-04 21:36:06.000000000 -0400
+++ minizip-ng-4.0.10-patched/minizip.pc.cmakein	2025-05-22 12:07:43.701305167 -0400
@@ -8,7 +8,7 @@
 Description: Zip manipulation library
 Version: @VERSION@
 
-Requires.private: @PC_PRIVATE_DEPS@
-Libs: -L${libdir} -L${sharedlibdir} -l@MINIZIP_TARGET@
-Libs.private:@PC_PRIVATE_LIBS@
+Requires.private: @PC_PRIVATE_DEPS@ openssl
+Libs: -L${libdir} -L${sharedlibdir} -l@MINIZIP_TARGET@ -lbz2
+Libs.private:@PC_PRIVATE_LIBS@ -lbcrypt
 Cflags: -I${includedir}
