diff --git a/src/muParserDLL.cpp b/src/muParserDLL.cpp
index 5730f71..dfb2ec8 100644
--- a/src/muParserDLL.cpp
+++ b/src/muParserDLL.cpp
@@ -142,10 +142,11 @@ inline ParserTag* AsParserTag(muParserHandle_t a_hParser)
 	return static_cast<ParserTag*>(a_hParser);
 }
 
-
-#if defined(_WIN32)
-
-BOOL APIENTRY DllMain(HANDLE /*hModule*/, DWORD ul_reason_for_call, LPVOID /*lpReserved*/)
+// omit for static linking in GDAL which uses the C++ API only
+// #if defined(_WIN32)
+//
+// BOOL APIENTRY DllMain(HANDLE /*hModule*/, DWORD ul_reason_for_call, LPVOID /*lpReserved*/)
+/*
 {
 	switch (ul_reason_for_call)
 	{
@@ -160,8 +161,8 @@ BOOL APIENTRY DllMain(HANDLE /*hModule*/, DWORD ul_reason_for_call, LPVOID /*lpR
 
 	return TRUE;
 }
-
-#endif
+*/
+// #endif
 
 //---------------------------------------------------------------------------
 //
