diff -Nru opencv-4.5.1-orig/cmake/OpenCVGenSetupVars.cmake opencv-4.5.1-patched/cmake/OpenCVGenSetupVars.cmake
--- opencv-4.5.1-orig/cmake/OpenCVGenSetupVars.cmake	2020-12-21 14:01:38.000000000 -0600
+++ opencv-4.5.1-patched/cmake/OpenCVGenSetupVars.cmake	2021-08-30 16:49:21.024562031 -0500
@@ -1,5 +1,5 @@
 if(WIN32)
-  ocv_update(OPENCV_SETUPVARS_INSTALL_PATH ".")
+  ocv_update(OPENCV_SETUPVARS_INSTALL_PATH "bin")
   ocv_update(OPENCV_SCRIPT_EXTENSION ".cmd")
   ocv_update(OPENCV_SETUPVARS_TEMPLATE "setup_vars_win32.cmd.in")
 else()
diff -Nru opencv-4.5.1-orig/CMakeLists.txt opencv-4.5.1-patched/CMakeLists.txt
--- opencv-4.5.1-orig/CMakeLists.txt	2020-12-21 14:01:38.000000000 -0600
+++ opencv-4.5.1-patched/CMakeLists.txt	2021-08-30 16:46:40.725119863 -0500
@@ -965,11 +965,11 @@
 if(ANDROID OR NOT UNIX)
   install(FILES ${OPENCV_LICENSE_FILE}
         PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
-        DESTINATION ./ COMPONENT libs)
+        DESTINATION ./share/opencv4 COMPONENT libs)
   if(OPENCV_README_FILE)
     install(FILES ${OPENCV_README_FILE}
             PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
-            DESTINATION ./ COMPONENT libs)
+            DESTINATION ./share/opencv4 COMPONENT libs)
   endif()
 endif()
 
