diff -Nru pkgconf-pkgconf-1.8.0-orig/libpkgconf/personality.c pkgconf-pkgconf-1.8.0-patched/libpkgconf/personality.c
--- pkgconf-pkgconf-1.8.0-orig/libpkgconf/personality.c	2021-07-24 22:40:52.000000000 -0400
+++ pkgconf-pkgconf-1.8.0-patched/libpkgconf/personality.c	2023-10-09 12:10:41.216783693 -0400
@@ -26,7 +26,7 @@
 	.name = "default",
 #ifdef _WIN32
 	.want_default_static = true,
-	.want_default_pure = true,
+	.want_default_pure = false,
 #endif
 };
 
