diff -Nru pkgconf-pkgconf-1.8.0-orig/libpkgconf/libpkgconf.h pkgconf-pkgconf-1.8.0-patched/libpkgconf/libpkgconf.h
--- pkgconf-pkgconf-1.8.0-orig/libpkgconf/libpkgconf.h	2021-07-24 22:40:52.000000000 -0400
+++ pkgconf-pkgconf-1.8.0-patched/libpkgconf/libpkgconf.h	2023-09-19 05:31:03.584500375 -0400
@@ -264,7 +264,7 @@
 
 #if defined(__GNUC__) || defined(__INTEL_COMPILER)
 #define PRINTFLIKE(fmtarg, firstvararg) \
-        __attribute__((__format__ (__printf__, fmtarg, firstvararg)))
+        __attribute__((__format__ (gnu_printf, fmtarg, firstvararg)))
 #define DEPRECATED \
         __attribute__((deprecated))
 #else
diff -Nru pkgconf-pkgconf-1.8.0-orig/libpkgconf/stdinc.h pkgconf-pkgconf-1.8.0-patched/libpkgconf/stdinc.h
--- pkgconf-pkgconf-1.8.0-orig/libpkgconf/stdinc.h	2021-07-24 22:40:52.000000000 -0400
+++ pkgconf-pkgconf-1.8.0-patched/libpkgconf/stdinc.h	2023-09-19 05:30:57.668498734 -0400
@@ -31,11 +31,7 @@
 # include <windows.h>
 # include <malloc.h>
 # define PATH_DEV_NULL	"nul"
-# ifdef _WIN64
-#  define SIZE_FMT_SPECIFIER	"%I64u"
-# else
-#  define SIZE_FMT_SPECIFIER	"%u"
-# endif
+# define SIZE_FMT_SPECIFIER	"%zu"
 # ifndef ssize_t
 # ifndef __MINGW32__
 #  include <BaseTsd.h>
