diff -Nru poppler-data-0.4.12-orig/Makefile poppler-data-0.4.12-patched/Makefile
--- poppler-data-0.4.12-orig/Makefile	2023-02-01 14:02:31.000000000 -0500
+++ poppler-data-0.4.12-patched/Makefile	2025-04-11 15:27:56.203143968 -0400
@@ -4,6 +4,7 @@
 prefix     = /usr/local
 datadir    = $(prefix)/share
 pkgdatadir = $(datadir)/poppler
+pkgconfdir = $(prefix)/lib
 
 all : poppler-data.pc
 
@@ -14,8 +15,8 @@
 	install -m 0644 $(name-to-unicode-files) $(DESTDIR)$(pkgdatadir)/nameToUnicode
 	install -d -m 0755 $(DESTDIR)$(pkgdatadir)/unicodeMap
 	install -m 0644 $(unicode-map-files) $(DESTDIR)$(pkgdatadir)/unicodeMap
-	install -d -m 0755 $(DESTDIR)$(datadir)/pkgconfig
-	install -m 0644 poppler-data.pc $(DESTDIR)$(datadir)/pkgconfig
+	install -d -m 0755 $(DESTDIR)$(pkgconfdir)/pkgconfig
+	install -m 0644 poppler-data.pc $(DESTDIR)$(pkgconfdir)/pkgconfig
 
 	for f in $(cmap-files); do					    \
 		install -d -m 0755 $(DESTDIR)$(pkgdatadir)/`dirname $$f`; \
