diff -Nru poppler-25.01.0-orig/CMakeLists.txt poppler-25.01.0-patched/CMakeLists.txt
--- poppler-25.01.0-orig/CMakeLists.txt	2025-01-01 19:28:08.000000000 -0500
+++ poppler-25.01.0-patched/CMakeLists.txt	2025-01-20 12:32:50.541070168 -0500
@@ -87,7 +87,7 @@
 option(RUN_GPERF_IF_PRESENT "Run gperf if it is found" ON)
 option(INSTALL_GLIB_DEMO "Whether to install the poppler-glib-demo program." OFF)
 if(WIN32)
-  option(ENABLE_RELOCATABLE "Do not hardcode the poppler library location (on Windows)." ON)
+  option(ENABLE_RELOCATABLE "Do not hardcode the poppler library location (on Windows)." ${BUILD_SHARED_LIBS})
 else()
   set(ENABLE_RELOCATABLE OFF)
 endif()
@@ -322,6 +322,10 @@
 endif()
 set(ENABLE_LIBOPENJPEG "${WITH_OPENJPEG}")
 
+if(ENABLE_LIBOPENJPEG AND NOT BUILD_SHARED_LIBS)
+  add_definitions(-DOPJ_STATIC)
+endif()
+
 find_soft_mandatory_package(ENABLE_LCMS LCMS2 2.12)
 set(USE_CMS ${ENABLE_LCMS})
 
diff -Nru poppler-25.01.0-orig/poppler.pc.cmake poppler-25.01.0-patched/poppler.pc.cmake
--- poppler-25.01.0-orig/poppler.pc.cmake	2025-01-01 19:28:08.000000000 -0500
+++ poppler-25.01.0-patched/poppler.pc.cmake	2025-01-20 12:23:27.834302254 -0500
@@ -6,5 +6,6 @@
 Description: PDF rendering library
 Version: @POPPLER_VERSION@
 
+Requires.private: @POPPLER_REQUIRES@
 Libs: -L${libdir} -lpoppler
 Cflags: -I${includedir}/poppler
