diff -Nru poppler-26.01.0-orig/CMakeLists.txt poppler-26.01.0-patched/CMakeLists.txt
--- poppler-26.01.0-orig/CMakeLists.txt	2026-01-06 17:13:29.000000000 -0500
+++ poppler-26.01.0-patched/CMakeLists.txt	2026-01-21 09:43:46.052924217 -0500
@@ -88,7 +88,7 @@
 option(RUN_GPERF_IF_PRESENT "Run gperf if it is found" ON)
 option(INSTALL_GLIB_DEMO "Whether to install the poppler-glib-demo program." OFF)
 if(WIN32)
-  option(ENABLE_RELOCATABLE "Do not hardcode the poppler library location (on Windows)." ON)
+  option(ENABLE_RELOCATABLE "Do not hardcode the poppler library location (on Windows)." ${BUILD_SHARED_LIBS})
 else()
   set(ENABLE_RELOCATABLE OFF)
 endif()
@@ -326,6 +326,9 @@
   message(FATAL_ERROR "Invalid ENABLE_LIBOPENJPEG value: ${ENABLE_LIBOPENJPEG}")
 endif()
 set(ENABLE_LIBOPENJPEG "${WITH_OPENJPEG}")
+if(ENABLE_LIBOPENJPEG AND NOT BUILD_SHARED_LIBS)
+  add_definitions(-DOPJ_STATIC)
+endif()
 
 find_soft_mandatory_package(ENABLE_LCMS LCMS2 ${LCMS2_REQUIRED_VERSION})
 set(USE_CMS ${ENABLE_LCMS})
diff -Nru poppler-26.01.0-orig/goo/gfile.cc poppler-26.01.0-patched/goo/gfile.cc
--- poppler-26.01.0-orig/goo/gfile.cc	2026-01-06 17:13:29.000000000 -0500
+++ poppler-26.01.0-patched/goo/gfile.cc	2026-01-21 09:56:19.725526201 -0500
@@ -43,6 +43,9 @@
 #    include <fcntl.h>
 #    include <cstring>
 #    include <pwd.h>
+#else
+#    include <stdlib.h>
+#    include <windows.h>
 #endif // _WIN32
 #include <cstdio>
 #include <limits>
diff -Nru poppler-26.01.0-orig/poppler.pc.cmake poppler-26.01.0-patched/poppler.pc.cmake
--- poppler-26.01.0-orig/poppler.pc.cmake	2026-01-06 17:13:29.000000000 -0500
+++ poppler-26.01.0-patched/poppler.pc.cmake	2026-01-21 09:43:46.052924217 -0500
@@ -7,6 +7,7 @@
 Version: @POPPLER_VERSION@
 Requires.private: @PC_REQUIRES_PRIVATE@
 
+Requires.private: @POPPLER_REQUIRES@
 Libs: -L${libdir} -lpoppler
 Libs.private: @PC_LIBS_PRIVATE@
 Cflags: -I${includedir}/poppler
