diff -Nru postgresql-13.4-orig/configure.in postgresql-13.4-patched/configure.in
--- postgresql-13.4-orig/configure.in	2021-08-09 16:49:05.000000000 -0400
+++ postgresql-13.4-patched/configure.in	2021-09-09 06:43:37.605020383 -0400
@@ -17,6 +17,7 @@
 dnl
 m4_pattern_forbid(^PGAC_)dnl to catch undefined macros
 
+AC_PREREQ(2.63)
 AC_INIT([PostgreSQL], [13.4], [pgsql-bugs@lists.postgresql.org], [], [https://www.postgresql.org/])
 
 m4_if(m4_defn([m4_PACKAGE_VERSION]), [2.69], [], [m4_fatal([Autoconf version 2.69 is required.
