diff -Nru postgresql-13.4-orig/src/Makefile.shlib postgresql-13.4-patched/src/Makefile.shlib
--- postgresql-13.4-orig/src/Makefile.shlib	2021-08-09 16:49:05.000000000 -0400
+++ postgresql-13.4-patched/src/Makefile.shlib	2021-09-09 08:48:42.965970602 -0400
@@ -249,7 +249,7 @@
   ifdef SO_MAJOR_VERSION
     shlib		= lib$(NAME)$(DLSUFFIX)
   endif
-  haslibarule   = yes
+  haslibarule   =
 endif
 
 
@@ -354,8 +354,8 @@
 # Win32 case
 
 # See notes in src/backend/parser/Makefile about the following two rules
-$(stlib): $(shlib)
-	touch $@
+#$(stlib): $(shlib)
+#	touch $@
 
 # XXX A backend that loads a module linked with libgcc_s_dw2-1.dll will exit
 # uncleanly, hence -static-libgcc.  (Last verified with MinGW-w64 compilers
@@ -372,7 +372,7 @@
 DLL_DEFFILE = lib$(NAME)dll.def
 
 $(shlib): $(OBJS) $(DLL_DEFFILE) | $(SHLIB_PREREQS)
-	$(CC) $(CFLAGS)  -shared -static-libgcc -o $@  $(OBJS) $(DLL_DEFFILE) $(LDFLAGS) $(LDFLAGS_SL) $(SHLIB_LINK) $(LIBS) -Wl,--out-implib=$(stlib)
+	$(CC) $(CFLAGS)  -shared -static-libgcc -o $@  $(OBJS) $(DLL_DEFFILE) $(LDFLAGS) $(LDFLAGS_SL) $(SHLIB_LINK) $(LIBS)
 
 UC_NAME = $(shell echo $(NAME) | tr 'abcdefghijklmnopqrstuvwxyz' 'ABCDEFGHIJKLMNOPQRSTUVWXYZ')
 
@@ -408,7 +408,7 @@
 ##
 
 .PHONY: install-lib install-lib-static install-lib-shared installdirs-lib
-install-lib: install-lib-shared
+#install-lib: install-lib-shared
 ifdef soname
 install-lib: install-lib-static
 install-lib: install-lib-pc
