diff -ru proj-8.2.1-orig/proj.pc.in proj-8.2.1-patched/proj.pc.in
--- proj-8.2.1-orig/proj.pc.in	2021-12-26 10:31:08.000000000 -0500
+++ proj-8.2.1-patched/proj.pc.in	2022-09-16 09:53:42.388219742 -0400
@@ -10,5 +10,5 @@
 Requires:
 Version: @VERSION@
 Libs: -L${libdir} -lproj
-Libs.private: @EXTRA_LIBS@
+Libs.private: @EXTRA_LIBS@ @SQLITE3_LIBS@ @TIFF_LIBS@ @CURL_LIBS@
 Cflags: -I${includedir}
Only in proj-8.2.1-patched/: proj.pc.in~
