From bbdc4cd142850ddea772e5373a2f5c474a7f1af5 Mon Sep 17 00:00:00 2001
From: Claudio Bley <claudio.bley@tweag.io>
Date: Fri, 17 Jan 2025 13:24:24 +0100
Subject: [PATCH 1/2] Link with dbghelp library on Windows

---
 build_defs/cpp_opts.bzl | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/build_defs/cpp_opts.bzl b/build_defs/cpp_opts.bzl
index f46b7dcc0a65c..7c14ebc7bd07a 100644
--- a/build_defs/cpp_opts.bzl
+++ b/build_defs/cpp_opts.bzl
@@ -39,6 +39,9 @@ LINK_OPTS = select({
         "-lm",
         "-framework CoreFoundation",
     ],
+    "@platforms//os:windows": [
+        "-ldbghelp",
+    ],
     "//conditions:default": [
         "-lpthread",
         "-lm",

From 1d9270535fca3c777e68c5e09048fce4c22835c1 Mon Sep 17 00:00:00 2001
From: Claudio Bley <claudio.bley@tweag.io>
Date: Fri, 17 Jan 2025 13:26:59 +0100
Subject: [PATCH 2/2] Do not define `PROTOBUF_DESCRIPTOR_WEAK_MESSAGES_ALLOWED`
 on Windows

---
 src/google/protobuf/port_def.inc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/google/protobuf/port_def.inc b/src/google/protobuf/port_def.inc
index 0f64087c30841..f105721837170 100644
--- a/src/google/protobuf/port_def.inc
+++ b/src/google/protobuf/port_def.inc
@@ -571,7 +571,7 @@ static_assert(PROTOBUF_ABSL_MIN(20230125, 3),
 #error PROTOBUF_DESCRIPTOR_WEAK_MESSAGES_ALLOWED was previously defined
 #endif
 #if defined(__GNUC__) && defined(__clang__) && !defined(__APPLE__) && \
-    !defined(_MSC_VER)
+    !defined(_MSC_VER) && !defined(_WIN32)
 #define PROTOBUF_DESCRIPTOR_WEAK_MESSAGES_ALLOWED
 #endif
 
