diff -Nru rasqal-0.9.33-orig/src/win32_rasqal_config.h.in rasqal-0.9.33-new/src/win32_rasqal_config.h.in
--- rasqal-0.9.33-orig/src/win32_rasqal_config.h.in	2013-04-05 13:17:36.000000000 -0400
+++ rasqal-0.9.33-new/src/win32_rasqal_config.h.in	2020-12-16 13:07:49.331969799 -0500
@@ -42,36 +42,10 @@
 
 #define HAVE_STRICMP 1
 
-/* MS names for these functions */
-// next line breaks build on wntmsci12
-//#define vsnprintf _vsnprintf
-#define snprintf _snprintf
-#define access _access
-#define stricmp _stricmp
-#define strnicmp _strnicmp
-
-/* definition from winsock2.h */
-#ifndef _TIMEVAL_DEFINED
-#define _TIMEVAL_DEFINED
-typedef struct timeval {
-  long tv_sec;
-  long tv_usec;
-} timeval;
-#endif
-
-struct timezone
-{
-  int tz_minuteswest; /* minutes West of Greenwich */
-  int tz_dsttime;     /* type of dst correction */
-};
-
-int rasqal_gettimeofday(struct timeval *tv, struct timezone *tz);
-#undef HAVE_GETTIMEOFDAY
+#define HAVE_GETTIMEOFDAY 1
 
 #include <float.h>
 #define isnan(n) _isnan(n)
-/* no round function available */
-#define round(x) floor(x+0.5)
 
 /* These are SPARQL token definitions */
 #ifdef OPTIONAL
@@ -152,13 +126,12 @@
 /* #define RAPTOR_TRIPLES_SOURCE_REDLAND 1 */
 
 /* Use PCRE regex library */
-//#define RASQAL_REGEX_PCRE 1
+#define RASQAL_REGEX_PCRE 1
 
 #define HAVE_LIBXML_SCHEMASINTERNALS_H 1
 #define HAVE_LIBXML_XMLSCHEMASTYPES_H 1
 #define HAVE_LIBXML_XMLSCHEMAS_H 1
 
-#define RASQAL_DECIMAL_NONE 1
 #define RASQAL_UUID_INTERNAL 1
 #define RASQAL_DIGEST_INTERNAL 1
 
