diff -Nru redland-1.0.17-orig/src/win32_rdf_config.h.in redland-1.0.17-new/src/win32_rdf_config.h.in
--- redland-1.0.17-orig/src/win32_rdf_config.h.in	2013-04-05 13:17:40.000000000 -0400
+++ redland-1.0.17-new/src/win32_rdf_config.h.in	2020-12-16 13:46:43.541522071 -0500
@@ -42,17 +42,6 @@
 
 #define HAVE_STRICMP 1
 
-/* MS names for these functions */
-// next line breaks build on wntmsci12
-//#define vsnprintf _vsnprintf
-#define snprintf _snprintf
-#define access _access
-#define stricmp _stricmp
-#define strnicmp _strnicmp
-#define close _close
-#define unlink _unlink
-#define mktemp _mktemp
-
 #define HAVE_C99_VSNPRINTF 1
 
 /* Define the following if you are calling the librdf from C# or any language which uses stdcall */
@@ -76,40 +65,40 @@
  */
 
 /* BDB has close method with 2 args */
-#define HAVE_BDB_CLOSE_2_ARGS 1
+/* #undef HAVE_BDB_CLOSE_2_ARGS */
 
 /* BDB defines DBC */
-#define HAVE_BDB_CURSOR 1
+/* #undef HAVE_BDB_CURSOR */
 
 /* BDB cursor method has 4 arguments */
-#define HAVE_BDB_CURSOR_4_ARGS 1
+/* #undef HAVE_BDB_CURSOR_4_ARGS */
 
 /* BDB defines DB_TXN */
-#define HAVE_BDB_DB_TXN 1
+/* #undef HAVE_BDB_DB_TXN */
 
 /* BDB has fd method with 2 args */
-#define HAVE_BDB_FD_2_ARGS 1
+/* #undef HAVE_BDB_FD_2_ARGS */
 
 /* Have BDB hash support */
-#define HAVE_BDB_HASH 1
+/* #undef HAVE_BDB_HASH */
 
 /* BDB has open method with 6 args */
 /* #undef HAVE_BDB_OPEN_6_ARGS */
 
 /* BDB has open method with 7 args */
-#define HAVE_BDB_OPEN_7_ARGS 1
+/* #undef HAVE_BDB_OPEN_7_ARGS */
 
 /* BDB has set_flags method */
-#define HAVE_BDB_SET_FLAGS 1
+/* #undef HAVE_BDB_SET_FLAGS */
 
 /* BDB has dbopen method */
 /* #undef HAVE_DBOPEN */
 
 /* BDB has db_create method */
-#define HAVE_DB_CREATE 1
+/* #undef HAVE_DB_CREATE */
 
 /* Define to 1 if you have the <db.h> header file. */
-#define HAVE_DB_H 1
+/* #undef HAVE_DB_H */
 
 /* Define to 1 if you have the <dlfcn.h> header file. */
 /* undef HAVE_DLFCN_H */
@@ -160,7 +149,7 @@
 #define HAVE_MKTEMP 1
 
 /* MySQL libraries are present */
-#define HAVE_MYSQL 1
+/* #undef HAVE_MYSQL */
 
 /* Define to 1 if you have the <openssl/crypto.h> header file. */
 /* #undef HAVE_OPENSSL_CRYPTO_H */
@@ -257,7 +246,7 @@
 #define STORAGE_TREES 1
 
 /* Building MySQL storage */
-/* #define STORAGE_MYSQL 1 */
+/* #undef STORAGE_MYSQL */
 
 /* Building PostgreSQL storage */
 /*#undef STORAGE_POSTGRESQL*/
