diff -Nru orig/rswipl/src/swipl-devel/src/pl-alloc.c patched/rswipl/src/swipl-devel/src/pl-alloc.c
--- orig/rswipl/src/swipl-devel/src/pl-alloc.c	2024-01-25 22:36:28.000000000 +0100
+++ patched/rswipl/src/swipl-devel/src/pl-alloc.c	2024-03-02 23:27:58.848438500 +0100
@@ -55,7 +55,7 @@
 /* Provides error checking for the weak declarations below */
 #include <gperftools/malloc_extension_c.h>
 #endif
-#ifdef HAVE_SYS_MMAN_H
+#if defined(HAVE_SYS_MMAN_H) && defined(HAVE_MMAP) && defined(HAVE_SYSCONF)
 #define MMAP_STACK 1
 #include <sys/mman.h>
 #include <unistd.h>
