diff -Nru librttopo-orig/configure.ac librttopo-patched/configure.ac
--- librttopo-orig/configure.ac	2019-07-27 02:55:02.000000000 -0400
+++ librttopo-patched/configure.ac	2021-09-09 13:06:43.863199810 -0400
@@ -111,6 +111,7 @@
 fi
 # Extract the linker and include flags
 GEOS_LDFLAGS=`$GEOSCONFIG --ldflags`
+GEOS_LIBS=`$GEOSCONFIG --libs`
 GEOS_CFLAGS=-I`$GEOSCONFIG --includes`
 AC_SUBST([GEOS_LDFLAGS])
 AC_SUBST([GEOS_CFLAGS])
@@ -121,10 +122,10 @@
 CPPFLAGS="$CPPFLAGS_SAVE"
 # Ensure we can link against libgeos_c
 LIBS_SAVE="$LIBS"
-LIBS="$GEOS_LDFLAGS"
-AC_SEARCH_LIBS(GEOSContext_setErrorMessageHandler_r,geos_c,,AC_MSG_ERROR([could not find libgeos_c (or obsolete 'libgeos_c' < v.3.5.0 found) - you may need to specify the directory of a geos-config file using --with-geosconfig]))
+LIBS="$GEOS_LIBS"
+AC_SEARCH_LIBS(GEOSContext_setErrorMessageHandler_r,geos_c,,AC_MSG_ERROR([could not find libgeos_c (or obsolete 'libgeos_c' < v.3.5.0 found) - you may need to specify the directory of a geos-config file using --with-geosconfig]),`$GEOSCONFIG --libs`) 
 LIBS="$LIBS_SAVE"
-LIBS="$LIBS $GEOS_LDFLAGS -lgeos_c"
+LIBS="$LIBS $GEOS_LDFLAGS -lgeos_c $GEOS_LIBS"
 GEOS_MAJOR_VERSION=`$GEOSCONFIG --version | cut -d. -f1 | sed 's/[[^0-9]]//g'`
 GEOS_MINOR_VERSION=`$GEOSCONFIG --version | cut -d. -f2 | sed 's/[[^0-9]]//g'`
 RTGEOM_GEOS_VERSION="$GEOS_MAJOR_VERSION$GEOS_MINOR_VERSION"
