diff -Nru yaml-cpp-yaml-cpp-0.6.2-orig/CMakeLists.txt yaml-cpp-yaml-cpp-0.6.2-patched/CMakeLists.txt
--- yaml-cpp-yaml-cpp-0.6.2-orig/CMakeLists.txt	2018-03-05 22:56:23.000000000 -0600
+++ yaml-cpp-yaml-cpp-0.6.2-patched/CMakeLists.txt	2021-08-30 16:35:00.934196099 -0500
@@ -321,7 +321,7 @@
 	"${PROJECT_BINARY_DIR}/yaml-cpp-config.cmake" @ONLY)
 
 if(WIN32 AND NOT CYGWIN)
-	set(INSTALL_CMAKE_DIR CMake)
+	set(INSTALL_CMAKE_DIR ${LIB_INSTALL_DIR}/cmake/yaml-cpp)
 else()
 	set(INSTALL_CMAKE_DIR ${LIB_INSTALL_DIR}/cmake/yaml-cpp)
 endif()
