diff -Nru orig/duckdb/src/duckdb/third_party/fast_float/fast_float/fast_float.h patched/duckdb/src/duckdb/third_party/fast_float/fast_float/fast_float.h
--- orig/duckdb/src/duckdb/third_party/fast_float/fast_float/fast_float.h	2023-11-14 11:57:47.000000000 +0100
+++ patched/duckdb/src/duckdb/third_party/fast_float/fast_float/fast_float.h	2024-02-05 14:27:09.919749000 +0100
@@ -242,7 +242,7 @@
 fastfloat_really_inline value128 full_multiplication(uint64_t a,
                                                      uint64_t b) {
   value128 answer;
-#ifdef _M_ARM64
+#if defined(FASTFLOAT_VISUAL_STUDIO) && defined(_M_ARM64)
   // ARM64 has native support for 64-bit multiplications, no need to emulate
   answer.high = __umulh(a, b);
   answer.low = a * b;
