src/scripts/R.sh.in
--- src/scripts/R.sh.in	2007-03-19 12:19:19.000000000 -0400
+++ R.sh.in	2007-03-29 17:22:47.000000000 -0400
@@ -19,7 +19,8 @@
 # be overridden from the environment.
 # This script is shared by parallel installs, so nothing in it should
 # depend on the sub-architecture except the default here.
-: ${R_ARCH=@R_ARCH@}
+# CRAN local change: The default for Mac OS X is the current architecture
+: ${R_ARCH=/`arch`}
 
 usage="
 Usage: R [options] [< infile] [> outfile]
