diff -Nru gcc-14.2.0-orig/libgomp/configure gcc-14.2.0-patched/libgomp/configure
--- gcc-14.2.0-orig/libgomp/configure	2024-08-01 04:18:42.000000000 -0400
+++ gcc-14.2.0-patched/libgomp/configure	2025-02-26 20:23:06.775113346 -0500
@@ -11887,6 +11887,7 @@
       FC=no
     fi ;;
 esac
+FC="$GFORTRAN"
 ac_ext=${ac_fc_srcext-f}
 ac_compile='$FC -c $FCFLAGS $ac_fcflags_srcext conftest.$ac_ext >&5'
 ac_link='$FC -o conftest$ac_exeext $FCFLAGS $LDFLAGS $ac_fcflags_srcext conftest.$ac_ext $LIBS >&5'
diff -Nru gcc-14.2.0-orig/libgomp/configure.ac gcc-14.2.0-patched/libgomp/configure.ac
--- gcc-14.2.0-orig/libgomp/configure.ac	2024-08-01 04:18:42.000000000 -0400
+++ gcc-14.2.0-patched/libgomp/configure.ac	2025-02-26 20:23:06.775113346 -0500
@@ -170,6 +170,7 @@
       FC=no
     fi ;;
 esac
+FC="$GFORTRAN"
 AC_PROG_FC(gfortran)
 FCFLAGS="$FCFLAGS -Wall -L../libgfortran"
 
