diff -Nru orig/libopenexr/src/Makevars.win.in patched/libopenexr/src/Makevars.win.in
--- orig/libopenexr/src/Makevars.win.in	2025-06-30 05:06:55.000000000 +0200
+++ patched/libopenexr/src/Makevars.win.in	2025-10-15 16:23:28.498189400 +0200
@@ -92,8 +92,8 @@
 	src/OpenEXR/build/src/lib/OpenEXR/Makefile \
 	src/OpenEXR/build/src/lib/OpenEXRCore/Makefile \
 	src/OpenEXR/build/src/lib/OpenEXRUtil/Makefile \
-	src/OpenEXR/build/CMakeFiles/3.30.4/CompilerIdC/CMakeCCompilerId.c \
-	src/OpenEXR/build/CMakeFiles/3.30.4/CompilerIdCXX/CMakeCXXCompilerId.cpp \
+	src/OpenEXR/build/CMakeFiles/*/CompilerIdC/CMakeCCompilerId.c \
+	src/OpenEXR/build/CMakeFiles/*/CompilerIdCXX/CMakeCXXCompilerId.cpp \
 	src/OpenEXR/build/cmake/IexConfig.h \
 	src/OpenEXR/build/cmake/IexConfigInternal.h \
 	src/OpenEXR/build/cmake/IlmThreadConfig.h \
@@ -112,4 +112,4 @@
 clean: 
 	rm -rf @PACKAGE_BASE_DIR@/src/OpenEXR/build && 
 	rm -rf @PACKAGE_BASE_DIR@/src/Makevars &&
-	rm -rf @PACKAGE_BASE_DIR@/src/Makevars.win
\ No newline at end of file
+	rm -rf @PACKAGE_BASE_DIR@/src/Makevars.win
