diff -Nru orig/mpmi/src/Makevars patched/mpmi/src/Makevars
--- orig/mpmi/src/Makevars	2019-10-30 02:56:48.000000000 -0700
+++ patched/mpmi/src/Makevars	2021-01-15 06:13:53.334030800 -0800
@@ -1,5 +1,5 @@
 USE_FC_TO_LINK=
-PKG_FCFLAGS=$(SHLIB_OPENMP_FFLAGS)
+PKG_FFLAGS=$(SHLIB_OPENMP_FFLAGS)
 PKG_LIBS=$(SHLIB_OPENMP_FFLAGS)
 
 all: $(SHLIB)
diff -Nru orig/mpmi/src/Makevars.win patched/mpmi/src/Makevars.win
--- orig/mpmi/src/Makevars.win	1969-12-31 16:00:00.000000000 -0800
+++ patched/mpmi/src/Makevars.win	2021-01-15 06:14:27.209279800 -0800
@@ -0,0 +1,16 @@
+PKG_FFLAGS = $(SHLIB_OPENMP_FFLAGS)
+PKG_LIBS = $(SHLIB_OPENMP_CFLAGS)
+
+all: $(SHLIB)
+
+cmi.o: 10iface.o
+
+cminjk.o: 10iface.o  
+
+dmi.o: 10iface.o  
+
+mmi.o: 10iface.o  
+
+mminjk.o: 10iface.o  
+
+
