--- llvm-project-19.1.7.src-orig/cmake/Modules/GetClangResourceDir.cmake	2025-01-14 04:41:02.000000000 -0500
+++ llvm-project-19.1.7.src-patched/cmake/Modules/GetClangResourceDir.cmake	2025-02-27 18:03:50.338859418 -0500
@@ -13,7 +13,7 @@
     set(ret_dir bin/${CLANG_RESOURCE_DIR})
   else()
     if (NOT CLANG_VERSION_MAJOR)
-      string(REGEX MATCH "^[0-9]+" CLANG_VERSION_MAJOR ${PACKAGE_VERSION})
+      string(REGEX MATCH "^[0-9]+" CLANG_VERSION_MAJOR "${PACKAGE_VERSION}")
     endif()
     set(ret_dir lib${LLVM_LIBDIR_SUFFIX}/clang/${CLANG_VERSION_MAJOR})
   endif()
