diff -Nru orig/adbcpostgresql/src/Makevars.ucrt patched/adbcpostgresql/src/Makevars.ucrt
--- orig/adbcpostgresql/src/Makevars.ucrt	2023-10-30 14:26:47.000000000 +0100
+++ patched/adbcpostgresql/src/Makevars.ucrt	2024-01-17 13:23:56.108650400 +0100
@@ -17,7 +17,7 @@
 
 PKG_CPPFLAGS = -I../src -DADBC_EXPORT="" -D__USE_MINGW_ANSI_STDIO
 
-PKG_LIBS = -lpq -lpgcommon -lpgport -lssl -lcrypto -lz -lsecur32 -lws2_32 -lwldap32
+PKG_LIBS = -lpq -lpgcommon -lpgport -lssl -lcrypto -lz -lsecur32 -lws2_32 -lwldap32 -lcrypt32
 
 OBJECTS = init.o \
     error.o \
