diff -Nru orig/apcf/src/Makevars.ucrt patched/apcf/src/Makevars.ucrt
--- orig/apcf/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/apcf/src/Makevars.ucrt	2021-03-15 17:31:08.313651000 +0100
@@ -0,0 +1,22 @@
+PKG_LIBS = \
+       -lgdal -lgta -lsqlite3 -lmysqlclient -lspatialite -lproj -lgeos_c -lgeos  \
+       -ljson-c -lnetcdf -lpq \
+       -lwebp -lcurl -lidn2 -lunistring -lssh2 -lgcrypt -lgpg-error -lssl \
+       -lhdf5_hl -lhdf5 -lexpat -lfreexl \
+       -lmfhdf -ldf -lportablexdr  -lpcre \
+       -lopenjp2 -ljasper -lpng -ljpeg -ltiff -lwebp -lgeotiff -lgif -lxml2 -llzma \
+       -lodbc32 -lodbccp32 -liconv -lpsapi -lwldap32 -lsecur32 -lgdi32 -lnormaliz \
+       -lbz2 -lzstd -lz -lcrypto -lcrypt32 -lcfitsio -lws2_32
+
+all: clean winlibs
+
+CXX_STD = CXX11
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
+winlibs:
+	cp -r $(R_TOOLS_SOFT)/share/gdal ../inst/
+
+
+.PHONY: all winlibs clean
