diff -ruN orig/apcf/src/Makevars.win patched/apcf/src/Makevars.win
--- orig/apcf/src/Makevars.win	2019-06-12 00:08:50.000000000 -0700
+++ patched/apcf/src/Makevars.win	2021-01-08 03:14:53.953376200 -0800
@@ -1,32 +1,18 @@
-VERSION = 2.2.3
-COMPILED_BY ?= gcc-4.6.3
-RWINLIB = ../windows/gdal2-$(VERSION)
-TARGET = lib$(subst gcc,,$(COMPILED_BY))$(R_ARCH)
-
-PKG_CPPFLAGS =\
-	-I$(RWINLIB)/include/gdal \
-	-I$(RWINLIB)/include/geos
-
 PKG_LIBS = \
-	-L$(RWINLIB)/$(TARGET) \
-	-L$(RWINLIB)/lib$(R_ARCH) \
-	-lgdal -lsqlite3 -lspatialite -lproj -lgeos_c -lgeos  \
-	-ljson-c -lnetcdf -lmariadbclient -lpq -lintl -lwebp -lcurl -lssh2 -lssl -lcrypto \
-	-lkea -lhdf5_cpp -lhdf5_hl -lhdf5 -lexpat -lfreexl -lcfitsio \
-	-lmfhdf -ldf -lxdr \
-	-lopenjp2 -ljasper -lpng16 -ljpeg -ltiff -lgeotiff -lgif -lxml2 -llzma -lszip -lz \
-	-lodbc32 -lodbccp32 -liconv -lpsapi -lws2_32 -lcrypt32 -lwldap32 -lsecur32 -lgdi32
+       -lgdal -lgta -lsqlite3 -lmysqlclient -lspatialite -lproj -lgeos_c -lgeos  \
+       -ljson-c -lnetcdf -lpq \
+       -lwebp -lcurl -lidn2 -lunistring -lssh2 -lgcrypt -lgpg-error -lssl \
+       -lhdf5_hl -lhdf5 -lexpat -lfreexl \
+       -lmfhdf -ldf -lportablexdr  -lpcre \
+       -lopenjp2 -ljasper -lpng -ljpeg -ltiff -lwebp -lgeotiff -lgif -lxml2 -llzma \
+       -lodbc32 -lodbccp32 -liconv -lpsapi -lwldap32 -lsecur32 -lgdi32 -lnormaliz \
+       -lbz2 -lzstd -lz -lcrypto -lcrypt32 -lcfitsio -lws2_32
 
-all: clean winlibs
+all: clean 
 
 CXX_STD = CXX11
 
-winlibs:
-	mkdir -p ../inst
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" --vanilla "../tools/winlibs.R" $(VERSION)
-	cp -r $(RWINLIB)/share/gdal ../inst/
-
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
 
-.PHONY: all winlibs clean
+.PHONY: all clean
diff -ruN orig/apcf/tools/winlibs.R patched/apcf/tools/winlibs.R
--- orig/apcf/tools/winlibs.R	2019-06-12 00:08:50.000000000 -0700
+++ patched/apcf/tools/winlibs.R	1969-12-31 16:00:00.000000000 -0800
@@ -1,15 +0,0 @@
-# inherited from sf (https://github.com/r-spatial/sf)
-
-if(getRversion() < "3.3.0") {
-  stop("Your version of R is too old. This package requires R-3.3.0 or newer on Windows.")
-}
-
-# For details see: https://github.com/rwinlib/gdal2
-VERSION <- commandArgs(TRUE)
-if(!file.exists(sprintf("../windows/gdal2-%s/include/gdal/gdal.h", VERSION))){
-  if(getRversion() < "3.3.0") setInternet2()
-  download.file(sprintf("https://github.com/rwinlib/gdal2/archive/v%s.zip", VERSION), "lib.zip", quiet = TRUE)
-  dir.create("../windows", showWarnings = FALSE)
-  unzip("lib.zip", exdir = "../windows")
-  unlink("lib.zip")
-}
\ No newline at end of file
