diff -Nru orig/av/src/Makevars.ucrt patched/av/src/Makevars.ucrt
--- orig/av/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/av/src/Makevars.ucrt	2021-02-17 19:45:27.286171300 +0100
@@ -0,0 +1,14 @@
+PKG_CPPFLAGS = -D__USE_MINGW_ANSI_STDIO=1 -fopenmp
+PKG_LIBS = -fopenmp \
+	-lavfilter -lswscale -lpostproc -lavformat -lavcodec -lswresample \
+	-lavutil -lavresample -lvidstab -lvo-amrwbenc -lopencore-amrwb -lopencore-amrnb -lbluray \
+	-lvorbis -lvorbisenc -logg -ltheora -lspeex -lopus -lssp -lvpx -lass -lbs2b \
+	-lfribidi -lharfbuzz -lfontconfig -lfreetype -lexpat -lpng \
+	-lbz2 -lgnutls -lhogweed -lnettle -lgmp -lxml2 -lidn2 -ltasn1 -lsecur32 -lws2_32 -liconv -llzma -lz -lmp3lame  \
+	-lx264 -lxvidcore -lintl -lunistring -liconv -lcrypt32 -lole32 -lm -luser32 -lbcrypt
+
+all: clean
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
