diff -Nru orig/av/src/Makevars.win patched/av/src/Makevars.win
--- orig/av/src/Makevars.win	2020-01-18 17:18:25.000000000 +0100
+++ patched/av/src/Makevars.win	2020-07-02 18:06:00.607847400 +0200
@@ -1,16 +1,14 @@
-VERSION = 4.2.2
-RWINLIB = ../windows/ffmpeg-$(VERSION)
-PKG_CPPFLAGS = -I$(RWINLIB)/include -D__USE_MINGW_ANSI_STDIO=1
-PKG_LIBS = -L$(RWINLIB)/lib$(R_ARCH) \
-	-lavfilter -lswscale -lpostproc -lavformat -lavcodec -lswresample -lavutil \
-	-lvorbis -lvorbisenc -logg -lvpx \
-	-lbz2 -lsecur32 -lws2_32 -liconv -lz -lmp3lame -lx264 \
-	-lxvidcore -lole32 -lm -luser32 -lbcrypt
+PKG_CPPFLAGS = -D__USE_MINGW_ANSI_STDIO=1 -fopenmp
+PKG_LIBS = -fopenmp \
+	-lavfilter -lswscale -lpostproc -lavformat -lavcodec -lswresample \
+	-lavutil -lavresample -lvidstab -lvo-amrwbenc -lopencore-amrwb -lopencore-amrnb -lbluray \
+	-lvorbis -lvorbisenc -logg -ltheora -lspeex -lopus -lssp -lvpx -lass -lbs2b \
+	-lfribidi -lharfbuzz -lfontconfig -lfreetype -lexpat -lpng \
+	-lbz2 -lgnutls -lhogweed -lnettle -lgmp -lxml2 -lidn2 -ltasn1 -lsecur32 -lws2_32 -liconv -llzma -lz -lmp3lame  \
+	-lx264 -lxvidcore -lintl -lunistring -liconv -lcrypt32 -lole32 -lm -luser32 -lbcrypt
 
-all: winlibs clean
+all: clean
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
 
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
