diff -Nru orig/cairoDevice/src/Makevars.ucrt patched/cairoDevice/src/Makevars.ucrt
--- orig/cairoDevice/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/cairoDevice/src/Makevars.ucrt	2021-02-17 19:45:39.536615300 +0100
@@ -0,0 +1,2 @@
+PKG_CPPFLAGS=-D_R_=1 -DUSE_R=1 -mms-bitfields -I$(R_TOOLS_SOFT)/include/gtk-2.0 -I$(R_TOOLS_SOFT)/lib/gtk-2.0/include -I$(R_TOOLS_SOFT)/include/atk-1.0 -I$(R_TOOLS_SOFT)/include/cairo -I$(R_TOOLS_SOFT)/include/pango-1.0 -I$(R_TOOLS_SOFT)/include/glib-2.0 -I$(R_TOOLS_SOFT)/lib/glib-2.0/include -I$(R_TOOLS_SOFT)/include/cairo -I$(R_TOOLS_SOFT)/include/gdk-pixbuf-2.0 -I$(R_TOOLS_SOFT)/include -I$(R_HOME)/include
+PKG_LIBS=-lgtk-win32-2.0 -lgdk-win32-2.0 -latk-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lpangoft2-1.0 -lcairo -lpangowin32-1.0 -lgio-2.0 -lpixman-1 -lfontconfig -lpango-1.0 -lusp10 -lgdi32 -lgmodule-2.0 -liphlpapi -ldnsapi -lgobject-2.0 -lffi -lexpat -lfreetype_too -lharfbuzz_too -lfreetype -lpng16 -lpng -lharfbuzz -lglib-2.0 -lpcre -lbz2 -lintl -liconv -lws2_32 -ltiff -llzma -ljpeg -lz -lzstd -lwebpdecoder -lwebp -limm32 -lole32 -luuid
