diff -ruN orig/cblasr/src/Makevars.win patched/cblasr/src/Makevars.win
--- orig/cblasr/src/Makevars.win	2019-06-10 12:21:45.000000000 -0700
+++ patched/cblasr/src/Makevars.win	2021-01-14 09:49:35.034580100 -0800
@@ -1,2 +1,2 @@
-PKG_LIBS = $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)
-PKG_CXXFLAGS = -I. -I../inst/include/
+PKG_LIBS = $(BLAS_LIBS) $(LAPACK_LIBS) $(FLIBS)
+PKG_CPPFLAGS = -I. -I../inst/include/
