diff -Nru orig/cepreader/src/Makefile.win patched/cepreader/src/Makefile.win
--- orig/cepreader/src/Makefile.win	2019-04-30 11:15:52.000000000 +0200
+++ patched/cepreader/src/Makefile.win	2020-07-02 18:06:21.688939000 +0200
@@ -1,5 +1,5 @@
-FC:=`"${R_HOME}/bin/R" --arch i386 CMD config FC`
-FLIBS:=`"${R_HOME}/bin/R" --arch i386 CMD config FLIBS`
-FFLAGS:=`"${R_HOME}/bin/R" --arch i386 CMD config FFLAGS`
+FC:=`"${R_HOME}/bin/R" --arch x64 CMD config FC`
+FLIBS:=`"${R_HOME}/bin/R" --arch x64 CMD config FLIBS`
+FFLAGS:=`"${R_HOME}/bin/R" --arch x64 CMD config FFLAGS`
 cepread: cepin.f
 	$(FC) $(FLIBS) $(FFLAGS) cepin.f -o cepread
