diff -Nru orig/cepreader/src/Makefile.win patched/cepreader/src/Makefile.win
--- orig/cepreader/src/Makefile.win	2020-12-08 13:46:43.000000000 +0100
+++ patched/cepreader/src/Makefile.win	2021-02-17 19:45:39.692865400 +0100
@@ -1,4 +1,4 @@
-FC:=`"${R_HOME}/bin/R" --arch i386 CMD config FC`
-FFLAGS:=`"${R_HOME}/bin/R" --arch i386 CMD config FFLAGS`
+FC:=`"${R_HOME}/bin/R" --arch x64 CMD config FC`
+FFLAGS:=`"${R_HOME}/bin/R" --arch x64 CMD config FFLAGS`
 cepread: cepin.f
 	$(FC) $(FFLAGS) cepin.f -o cepread
