diff -ruN orig/cepreader/src/Makefile.win patched/cepreader/src/Makefile.win
--- orig/cepreader/src/Makefile.win	2020-12-08 04:46:43.000000000 -0800
+++ patched/cepreader/src/Makefile.win	2021-01-07 01:08:45.872335400 -0800
@@ -1,4 +1,4 @@
-FC:=`"${R_HOME}/bin/R" --arch i386 CMD config FC`
-FFLAGS:=`"${R_HOME}/bin/R" --arch i386 CMD config FFLAGS`
+FC:=`"${R_HOME}/bin/R" --arch x64 CMD config FC`
+FFLAGS:=`"${R_HOME}/bin/R" --arch x64 CMD config FFLAGS`
 cepread: cepin.f
 	$(FC) $(FFLAGS) cepin.f -o cepread
