diff -Nru orig/cld3/src/Makevars.win patched/cld3/src/Makevars.win
--- orig/cld3/src/Makevars.win	2020-01-31 11:52:50.000000000 +0100
+++ patched/cld3/src/Makevars.win	2020-07-02 18:05:42.136422700 +0200
@@ -1,12 +1,8 @@
 # ProtoBuf Version
-VERSION = 3.8.0
-RWINLIB = ../windows/protobuf-${VERSION}
 CXX_STD = CXX11
 
-PKG_CPPFLAGS = -I. -I$(RWINLIB)/include
+PKG_CPPFLAGS = -I. 
 PKG_LIBS = -Llibcld3 -lstatcld3 \
-	-L$(RWINLIB)/lib${subst gcc,,${COMPILED_BY}}${R_ARCH} \
-	-L$(RWINLIB)/lib${R_ARCH} \
 	-lprotobuf-lite
 
 STATLIB = libcld3/libstatcld3.a
@@ -47,8 +43,7 @@
 	$(AR) rcs $(STATLIB) $(LIBCLD3)
 
 winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
-	"$(RWINLIB)/bin$(WIN)/protoc" cld_3/protos/*.proto --cpp_out=.
+	protoc cld_3/protos/*.proto --cpp_out=.
 
 clean:
 	rm -f $(OBJECTS) $(SHLIB) $(LIBCLD3) $(STATLIB) cld_3/protos/*.pb.cc cld_3/protos/*.pb.h
