diff -Nru orig/cli/src/thread.c patched/cli/src/thread.c
--- orig/cli/src/thread.c	2021-07-16 21:36:21.000000000 +0200
+++ patched/cli/src/thread.c	2021-08-05 20:25:42.408899700 +0200
@@ -43,6 +43,15 @@
   cli__tick_ts.tv_nsec = (cticktime % 1000) * 1000 * 1000;
   int ret = 0;
 
+#ifndef _UCRT
+  // TODO: MinGW9 (& UCRT?)
+  //
+  // at least with MinGW9 and UCRT (but not MinGW8), the package fails
+  // during test-load (during installation) when the threads are enabled;
+  // it is killed by signal 40
+  //
+  // enabling pthread_detach below avoids this problem, but leads to a 
+  // warning that the cli thread could not be cancelled
   if (! getenv("CLI_NO_THREAD")) {
     ret = pthread_create(
       & tick_thread,
@@ -56,7 +65,9 @@
 #ifndef _WIN32
     if (!ret) pthread_detach(tick_thread);
 #endif
-  } else {
+  } else
+#endif
+  {
     cli__reset = 0;
   }
 
