diff -ruN orig/clpAPI/src/Makevars.win patched/clpAPI/src/Makevars.win
--- orig/clpAPI/src/Makevars.win	2020-08-05 06:34:58.000000000 -0700
+++ patched/clpAPI/src/Makevars.win	2021-01-14 00:30:02.361978000 -0800
@@ -1,2 +1,2 @@
-PKG_CPPFLAGS=-g -D_R_=1 -DUSE_R=1 -I${CLP_PATH}/include/coin
-PKG_LIBS=-L${CLP_PATH}/lib/coin -L${CLP_PATH}/lib -lClp -lCoinUtils -lstdc++ -lz
+PKG_CPPFLAGS=-g -D_R_=1 -DUSE_R=1 -I${LOCAL_SOFT}/include/coin
+PKG_LIBS=-lClp -lCoinUtils -lstdc++ -lz -lbz2 $(BLAS_LIBS) $(LAPACK_LIBS)
