diff -Nru orig/clustermq/src/Makevars.ucrt patched/clustermq/src/Makevars.ucrt
--- orig/clustermq/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/clustermq/src/Makevars.ucrt	2021-02-17 19:45:39.911616000 +0100
@@ -0,0 +1,12 @@
+CXX_STD = CXX11
+
+PKG_CPPFLAGS = -DSTRICT_R_HEADERS -DZMQ_STATIC \
+	-I../inst
+
+PKG_LIBS = \
+	-lzmq -lsodium -liphlpapi -lws2_32 -fstack-protector
+
+all: clean 
+
+clean:
+	rm -f $(OBJECTS) $(SHLIB)
