diff -Nru orig/curl/src/Makevars.ucrt patched/curl/src/Makevars.ucrt
--- orig/curl/src/Makevars.ucrt	1969-12-31 16:00:00.000000000 -0800
+++ patched/curl/src/Makevars.ucrt	2021-06-21 07:21:50.830858600 -0700
@@ -0,0 +1,15 @@
+PKG_LIBS= \
+	-lwinhttp -lcurl -lrtmp -lssl -lssh2 -lgcrypt -lcrypto -lgdi32 -lz -lzstd -lgdi32 -lcrypt32 -lidn2 -lunistring -liconv -lgpg-error -lws2_32 -lwldap32 -lwinmm
+
+PKG_CPPFLAGS= \
+	-DCURL_STATICLIB -DSTRICT_R_HEADERS
+
+all: clean winlibs
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
+winlibs: clean
+	echo '#include <curl/curl.h>' | $(CC) $(PKG_CPPFLAGS) -std=gnu99 -E -xc - | grep "^[ \t]*CURLOPT_.*," | sed s/,// > ../tools/option_table.txt
+
+.PHONY: all winlibs clean
