diff -Nru orig/dbarts/src/Makevars.win patched/dbarts/src/Makevars.win
--- orig/dbarts/src/Makevars.win	2023-07-20 21:27:33.000000000 +0200
+++ patched/dbarts/src/Makevars.win	2024-02-02 13:21:08.336581100 +0100
@@ -1,10 +1,14 @@
 ## pass -DMATCH_BAYES_TREE to get same results as BayesTree, including excess random calls
 MATCH_BAYES_TREE=
 THREAD_SAFE_UNLOAD=
-SSE2_FLAG=-msse2
-SSE4_1_FLAG=-msse4.1
-AVX_FLAG=-mavx
-AVX2_FLAG=-mavx2
+
+ARCH=$(shell "${R_HOME}/bin/R" --no-echo -e 'cat(R.version$$arch)')
+ifeq "$(ARCH)" "x86_64"
+  SSE2_FLAG=-msse2
+  SSE4_1_FLAG=-msse4.1
+  AVX_FLAG=-mavx
+  AVX2_FLAG=-mavx2
+endif
 
 INST_DIR=../inst
 
