diff -Nru orig/exactextractr/src/Makevars.ucrt patched/exactextractr/src/Makevars.ucrt
--- orig/exactextractr/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/exactextractr/src/Makevars.ucrt	2021-02-17 19:45:40.099121700 +0100
@@ -0,0 +1,15 @@
+
+PKG_LIBS = -lgeos_c -lgeos
+PKG_CXXFLAGS = -std=c++14
+
+EE = exactextract/src
+
+SOURCES = $(EE)/area.cpp $(EE)/box.cpp $(EE)/cell.cpp $(EE)/coordinate.cpp $(EE)/floodfill.cpp $(EE)/geos_utils.cpp $(EE)/grid.cpp $(EE)/perimeter_distance.cpp $(EE)/raster_cell_intersection.cpp $(EE)/side.cpp $(EE)/traversal.cpp $(EE)/traversal_areas.cpp $(EE)/weighted_quantiles.cpp
+OBJECTS = RcppExports.o exact_extract.o $(SOURCES:.cpp=.o)
+
+all: clean
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
+.PHONY: all clean
