diff -Nru orig/freetypeharfbuzz/configure.win patched/freetypeharfbuzz/configure.win
--- orig/freetypeharfbuzz/configure.win	2018-07-19 11:45:07.000000000 +0200
+++ patched/freetypeharfbuzz/configure.win	2020-07-02 18:06:21.563960900 +0200
@@ -1,3 +0,0 @@
-#!/usr/bin/env sh
-
-exit 0
diff -Nru orig/freetypeharfbuzz/src/Makevars.win patched/freetypeharfbuzz/src/Makevars.win
--- orig/freetypeharfbuzz/src/Makevars.win	2018-10-22 00:49:35.000000000 +0200
+++ patched/freetypeharfbuzz/src/Makevars.win	2020-07-02 18:06:21.267043000 +0200
@@ -1,24 +1,9 @@
-VERSION = 1.15.10
-RWINLIB = ../windows/cairo-${VERSION}
-
-PKG_CPPFLAGS = \
-	-I${RWINLIB}/include/freetype2 \
-	-I${RWINLIB}/include/harfbuzz
-
-PKG_LIBS = -L${RWINLIB}/lib${R_ARCH} -lharfbuzz -lfreetype -lpng \
- -lbz2 -lz
+PKG_CPPFLAGS = -I$(LOCAL_SOFT)/include/harfbuzz  -I$(LOCAL_SOFT)/include/freetype2
+PKG_LIBS = -lharfbuzz -lfreetype -lpng -lbz2 -lz
 
 .PHONY: all
-all: winlibs teardown
-
-.PHONY: winlibs
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" ${VERSION}
-
-.PHONY: teardown
-teardown: $(SHLIB)
-	rm -rf ../windows
+all: 
 
 .PHONY: clean
 clean:
-	rm -rf $(SHLIB) $(OBJECTS) ../windows
+	rm -rf $(SHLIB) $(OBJECTS)
