diff -Nru orig/gdalcubes/src/Makevars.ucrt patched/gdalcubes/src/Makevars.ucrt
--- orig/gdalcubes/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/gdalcubes/src/Makevars.ucrt	2021-03-15 17:30:37.016870300 +0100
@@ -0,0 +1,68 @@
+
+OBJECTS =   gdalcubes/src/apply_pixel.o \
+            gdalcubes/src/config.o \
+            gdalcubes/src/collection_format.o \
+            gdalcubes/src/datetime.o \
+            gdalcubes/src/filesystem.o \
+            gdalcubes/src/utils.o \
+			gdalcubes/src/cube.o \
+			gdalcubes/src/cube_factory.o \
+			gdalcubes/src/error.o \
+			gdalcubes/src/image_collection.o \
+			gdalcubes/src/image_collection_cube.o \
+			gdalcubes/src/image_collection_ops.o \
+			gdalcubes/src/join_bands.o \
+			gdalcubes/src/select_time.o \
+			gdalcubes/src/reduce_time.o \
+			gdalcubes/src/reduce_space.o \
+			gdalcubes/src/window_time.o \
+			gdalcubes/src/select_bands.o \
+			gdalcubes/src/filter_pixel.o \
+			gdalcubes/src/filter_geom.o \
+			gdalcubes/src/fill_time.o\
+			gdalcubes/src/stream.o \
+			gdalcubes/src/stream_reduce_time.o \
+			gdalcubes/src/stream_reduce_space.o \
+			gdalcubes/src/stream_apply_pixel.o \
+			gdalcubes/src/stream_apply_time.o \
+			gdalcubes/src/swarm.o \
+			gdalcubes/src/view.o \
+			gdalcubes/src/dummy.o \
+			gdalcubes/src/warp.o \
+			gdalcubes/src/vector_queries.o \
+			gdalcubes/src/external/tinyexpr/tinyexpr.o \
+			gdalcubes/src/external/tiny-process-library/process.o \
+			gdalcubes/src/external/tiny-process-library/process_win.o \
+			gdalcubes/src/external/json11/json11.o \
+			gdalcubes.o \
+			RcppExports.o
+
+PKG_CPPFLAGS = \
+               -DCURL_STATICLIB -DR_PACKAGE
+ 
+
+PKG_LIBS = \
+	-lgdal -lgta -lsqlite3 -lmysqlclient -lspatialite -lproj -lgeos_c -lgeos  \
+	-ljson-c -lnetcdf -lpq \
+	-lwebp -lcurl -lidn2 -lunistring -lssh2 -lgcrypt -lgpg-error -lssl \
+	-lhdf5_hl -lhdf5 -lexpat -lfreexl \
+	-lmfhdf -ldf -lportablexdr  -lpcre \
+	-lopenjp2 -ljasper -lpng -ljpeg -ltiff -lwebp -lgeotiff -lgif -lxml2 -llzma \
+	-lodbc32 -lodbccp32 -liconv -lpsapi -lwldap32 -lsecur32 -lgdi32 -lnormaliz \
+	-lbz2 -lzstd -lz -lcrypto -lcrypt32 -lcfitsio -lws2_32
+	
+
+all: clean  winlibs
+
+CXX_STD=CXX11
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
+winlibs:
+	cp -r $(LOCAL_SOFT)/share/gdal ../inst/
+	cp -r $(LOCAL_SOFT)/share/proj ../inst/
+
+	
+.PHONY: all winlibs clean
+	
