diff -Nru orig/gdalcubes/src/Makevars.win patched/gdalcubes/src/Makevars.win
--- orig/gdalcubes/src/Makevars.win	2020-05-16 21:32:35.000000000 +0200
+++ patched/gdalcubes/src/Makevars.win	2020-07-02 18:06:43.645237200 +0200
@@ -1,9 +1,4 @@
 
-RWINLIB_GDAL = ../windows/gdal2-2.2.3
-RWINLIB_NETCDF = ../windows/netcdf-4.4.1.1-dap
-RWINLIB_CURL = ../windows/libcurl-7.59.0
-RWINLIB_SQLITE = ../windows/sqlite-amalgamation-3260000
-
 OBJECTS =   gdalcubes/src/apply_pixel.o \
             gdalcubes/src/config.o \
 			gdalcubes/src/cube.o \
@@ -32,36 +27,29 @@
 			gdalcubes.o \
 			RcppExports.o
 
-PKG_CPPFLAGS = -I$(RWINLIB_GDAL)/include/gdal \
-               -I$(RWINLIB_NETCDF)/include \
-               -I$(RWINLIB_CURL)/include \
-               -I$(RWINLIB_SQLITE) \
+PKG_CPPFLAGS = \
                -DCURL_STATICLIB -DR_PACKAGE
  
 
-PKG_LIBS = -L$(RWINLIB_NETCDF)/lib$(R_ARCH) -L$(RWINLIB_GDAL)/lib$(R_ARCH) -L$(RWINLIB_CURL)/lib$(R_ARCH) \
-  -lgdal -lsqlite3 -lspatialite -lproj -lgeos_c -lgeos  \
-	-ljson-c -lnetcdf -lmariadbclient -lpq -lintl -lwebp -lcurl -lssh2 -lssl -lcrypto \
-	-lkea -lhdf5_cpp -lhdf5_hl -lhdf5 -lexpat -lfreexl -lcfitsio \
-	-lmfhdf -ldf -lxdr \
-	-lopenjp2 -ljasper -lpng16 -ljpeg -ltiff -lgeotiff -lgif -lxml2 -llzma -lszip -lz \
-	-lodbc32 -lodbccp32 -liconv -lpsapi -lws2_32 -lcrypt32 -lwldap32 -lsecur32 -lgdi32 
+PKG_LIBS = \
+	-lgdal -lgta -lsqlite3 -lmysqlclient -lspatialite -lproj -lgeos_c -lgeos  \
+	-ljson-c -lnetcdf -lpq \
+	-lwebp -lcurl -lidn2 -lunistring -lssh2 -lgcrypt -lgpg-error -lssl \
+	-lhdf5_hl -lhdf5 -lexpat -lfreexl \
+	-lmfhdf -ldf -lportablexdr  -lpcre \
+	-lopenjp2 -ljasper -lpng -ljpeg -ltiff -lwebp -lgeotiff -lgif -lxml2 -llzma \
+	-lodbc32 -lodbccp32 -liconv -lpsapi -lwldap32 -lsecur32 -lgdi32 -lnormaliz \
+	-lbz2 -lzstd -lz -lcrypto -lcrypt32 -lcfitsio -lws2_32
 	
 
-all: clean winlibs
+all: clean 
 
 CXX_STD=CXX11
 
-winlibs:
-	mkdir -p ../inst
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" --vanilla "../tools/winlibs.R"
-	cp -r $(RWINLIB_GDAL)/share/gdal ../inst/
-	cp -r $(RWINLIB_GDAL)/share/proj ../inst/
-
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
 	
-.PHONY: all winlibs clean
+.PHONY: all clean
 	
 	
 	
