diff -Nru orig/GENLIB/src/Makevars.win patched/GENLIB/src/Makevars.win
--- orig/GENLIB/src/Makevars.win	2022-02-24 05:05:45.000000000 +0100
+++ patched/GENLIB/src/Makevars.win	2024-02-05 19:52:21.026257800 +0100
@@ -1,6 +1,6 @@
 ## Use the R_HOME indirection to support installations of multiple R version
 PKG_LIBS = $(shell "${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" -e "Rcpp:::LdFlags()")
-PKG_CPPFLAGS = -std=c++0x
+PKG_CXXFLAGS = -std=c++0x
 ##CXX_STD = CXX11 
 
 
