diff -Nru orig/gert/src/Makevars.ucrt patched/gert/src/Makevars.ucrt
--- orig/gert/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/gert/src/Makevars.ucrt	2021-02-17 19:45:40.411618700 +0100
@@ -0,0 +1,14 @@
+
+PKG_CPPFLAGS = \
+	-DR_NO_REMAP -DSTRICT_R_HEADERS
+
+PKG_LIBS = \
+        -lgit2 -lpcre -lssh2 -lz -lssl -lcrypto -lgcrypt -lgpg-error \
+        -lwinhttp -lws2_32 -lcrypt32 -lole32 -lrpcrt4
+
+all: clean
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
+.PHONY: all clean
