diff -Nru orig/gert/src/Makevars.win patched/gert/src/Makevars.win
--- orig/gert/src/Makevars.win	2019-02-13 21:25:20.000000000 +0100
+++ patched/gert/src/Makevars.win	2020-07-02 18:06:24.017436700 +0200
@@ -1,19 +1,10 @@
-VERSION = 0.27.8
-RWINLIB = ../windows/libgit2-${VERSION}
 
-PKG_CPPFLAGS = -I${RWINLIB}/include \
+PKG_CPPFLAGS = \
 	-DR_NO_REMAP -DSTRICT_R_HEADERS
 
-PKG_LIBS = -L${RWINLIB}/lib${R_ARCH} \
-	-lgit2 -lcurl -lhttp_parser -lssh2 -lz -lssl -lcrypto \
-	-lwinhttp -lgdi32 -lole32 -lrpcrt4 -lws2_32 -lcrypt32 -lwldap32
+PKG_LIBS = \
+        -lgit2 -lpcre -lssh2 -lz -lssl -lcrypto -lgcrypt -lgpg-error \
+        -lwinhttp -lws2_32 -lcrypt32 -lole32 -lrpcrt4
+all: clean 
 
-all: clean winlibs
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
-
-clean:
-	rm -f $(SHLIB) $(OBJECTS)
-
-.PHONY: all winlibs clean
+.PHONY: all clean
