diff -Nru orig/ggiraph/src/Makevars.ucrt patched/ggiraph/src/Makevars.ucrt
--- orig/ggiraph/src/Makevars.ucrt	2021-12-14 22:19:13.000000000 +0100
+++ patched/ggiraph/src/Makevars.ucrt	2025-03-26 14:46:18.389952100 +0100
@@ -1,2 +1,8 @@
-CRT=-ucrt
-include Makevars.win
+ifeq (,$(shell pkg-config --version 2>/dev/null))
+  PKG_LIBS = -lpng -lz
+else
+  PKG_CPPFLAGS = $(shell pkg-config --cflags libpng)
+  PKG_LIBS = $(shell pkg-config --libs libpng)
+endif
+
+PKG_CPPFLAGS += -DSTRICT_R_HEADERS
