diff -Nru orig/git2r/src/Makevars.ucrt patched/git2r/src/Makevars.ucrt
--- orig/git2r/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/git2r/src/Makevars.ucrt	2022-03-10 14:04:46.205081600 +0100
@@ -0,0 +1,16 @@
+
+PKG_CPPFLAGS = \
+	-DR_NO_REMAP -DSTRICT_R_HEADERS \
+        -DGIT2R_HAVE_BUF_DISPOSE \
+        -DGIT2R_HAVE_OBJECT_ANY \
+        -DGIT2R_LIBGIT2_V0_99_0_RENAMES
+
+PKG_LIBS = \
+        -lgit2 -lpcre -lssh2 -lz -lssl -lcrypto -lgcrypt -lgpg-error \
+        -lwinhttp -lws2_32 -lcrypt32 -lole32 -lrpcrt4
+all: clean 
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
+.PHONY: all clean
