diff -Nru orig/gpboost/src/gpboost_R.cpp patched/gpboost/src/gpboost_R.cpp
--- orig/gpboost/src/gpboost_R.cpp	2024-07-16 10:26:02.000000000 +0200
+++ patched/gpboost/src/gpboost_R.cpp	2024-08-19 20:28:17.314202200 +0200
@@ -13,6 +13,7 @@
 
 #include <R_ext/Rdynload.h>
 
+#define STRICT_R_HEADERS
 #define R_NO_REMAP
 #define R_USE_C99_IN_CXX
 #include <R_ext/Error.h>
diff -Nru orig/gpboost/src/gpboost_R.h patched/gpboost/src/gpboost_R.h
--- orig/gpboost/src/gpboost_R.h	2024-07-16 10:26:02.000000000 +0200
+++ patched/gpboost/src/gpboost_R.h	2024-08-19 20:27:55.329393100 +0200
@@ -8,6 +8,7 @@
 
 #include <LightGBM/c_api.h>
 
+#define STRICT_R_HEADERS
 #define R_NO_REMAP
 #define R_USE_C99_IN_CXX
 #include <Rinternals.h>
diff -Nru orig/gpboost/src/include/LightGBM/utils/log.h patched/gpboost/src/include/LightGBM/utils/log.h
--- orig/gpboost/src/include/LightGBM/utils/log.h	2024-07-16 10:26:09.000000000 +0200
+++ patched/gpboost/src/include/LightGBM/utils/log.h	2024-08-19 20:14:15.501042600 +0200
@@ -16,6 +16,7 @@
 #include <string>
 
 #ifdef LGB_R_BUILD
+#define STRICT_R_HEADERS
 #define R_NO_REMAP
 #define R_USE_C99_IN_CXX
 #include <R_ext/Error.h>
