diff -Nru orig/gpg/src/Makevars.win patched/gpg/src/Makevars.win
--- orig/gpg/src/Makevars.win	2019-12-02 12:58:51.000000000 +0100
+++ patched/gpg/src/Makevars.win	2020-07-02 18:06:25.830193400 +0200
@@ -1,17 +1,10 @@
-VERSION = 1.11.1
-RWINLIB = ../windows/gpgme-$(VERSION)
-PKG_LIBS = -L$(RWINLIB)/lib${R_ARCH} \
+PKG_LIBS = \
 	-lgpgme -lassuan -lgpg-error -lws2_32
 
-PKG_CPPFLAGS= -I$(RWINLIB)/include \
+PKG_CPPFLAGS= \
 	-DHAVE_GPGME_IO_READWRITE
 
-all: clean winlibs
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
-	mkdir -p ../inst/bin
-	cp -f $(RWINLIB)/bin/* ../inst/bin/
+all: clean
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
