diff -ruN orig/hdf5r/src/Makevars.win patched/hdf5r/src/Makevars.win
--- orig/hdf5r/src/Makevars.win	2020-08-15 07:13:01.000000000 -0700
+++ patched/hdf5r/src/Makevars.win	2021-01-11 04:38:17.611131300 -0800
@@ -1,10 +1,6 @@
-WRAP = 1_8_16
-VERSION = 1.8.16
-RWINLIB = ../windows/hdf5-$(VERSION)
-PKG_LIBS = -L${RWINLIB}/lib-4.9.3${R_ARCH} -lhdf5 -lhdf5_hl -lz -lm 
-PKG_CPPFLAGS = -I${RWINLIB}/include -I$(WRAP) -I. -D__USE_MINGW_ANSI_STDIO
-
-all: winlibs
+WRAP = 1_12_0
+PKG_CPPFLAGS = -I$(WRAP) -I. -DH5_USE_110_API
+PKG_LIBS = -lhdf5_hl -lhdf5 -lz -lm
 
 OBJECTS = $(WRAP)/const_export.o $(WRAP)/datatype_export.o $(WRAP)/Wrapper_auto_H5A.o \
 	$(WRAP)/Wrapper_auto_H5.o $(WRAP)/Wrapper_auto_H5D.o $(WRAP)/Wrapper_auto_H5DS.o $(WRAP)/Wrapper_auto_H5E.o \
@@ -15,9 +11,5 @@
 	$(WRAP)/Wrapper_auto_H5FDlog.o $(WRAP)/Wrapper_auto_H5FDsec2.o $(WRAP)/Wrapper_auto_H5FDstdio.o \
 	convert.o hdf5r_init.o H5Error.o H5ls.o Wrapper_manual_H5T.o
 
-winlibs:
-	@echo "Linking to HDF5 $(RWINLIB)"
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
-
-.PHONY: all winlibs
+.PHONY: all
 
