diff -Nru orig/httpgd/src/Makevars.win patched/httpgd/src/Makevars.win
--- orig/httpgd/src/Makevars.win	2021-01-10 17:59:27.000000000 +0100
+++ patched/httpgd/src/Makevars.win	2021-02-09 16:21:52.093868600 +0100
@@ -1,17 +1,12 @@
 CXX_STD = CXX17
 
-RWINLIB = ../windows/baselibs-3.5.0
-
-PKG_CPPFLAGS = -Ilib -I${RWINLIB}/include \
+PKG_CPPFLAGS = -Ilib \
 	-DBOOST_NO_AUTO_PTR \
 	-DFMT_HEADER_ONLY
 
-PKG_LIBS = -L${RWINLIB}/lib${R_ARCH} -lpng -lz -lWs2_32 -lwsock32
-
-all: clean winlibs
+PKG_LIBS = -lpng -lz -lWs2_32 -lwsock32
 
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" ${VERSION}
+all: clean 
 
 clean:
 	rm -f $(OBJECTS)
