diff -Nru orig/httpgd/src/Makevars.ucrt patched/httpgd/src/Makevars.ucrt
--- orig/httpgd/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/httpgd/src/Makevars.ucrt	2021-02-17 19:45:40.708502600 +0100
@@ -0,0 +1,12 @@
+CXX_STD = CXX17
+
+PKG_CPPFLAGS = -Ilib \
+	-DBOOST_NO_AUTO_PTR \
+	-DFMT_HEADER_ONLY
+
+PKG_LIBS = -lpng -lz -lWs2_32 -lwsock32
+
+all: clean 
+
+clean:
+	rm -f $(OBJECTS)
