diff -Nru orig/igraph/src/Makevars.ucrt patched/igraph/src/Makevars.ucrt
--- orig/igraph/src/Makevars.ucrt	2024-01-02 22:50:27.000000000 +0100
+++ patched/igraph/src/Makevars.ucrt	2024-02-07 15:56:16.635216300 +0100
@@ -10,6 +10,6 @@
         -D_GNU_SOURCE=1
 
 PKG_LIBS = -L"${LIB_XML}/lib" -lxml2 -liconv -lz -lws2_32 -lstdc++ \
-  -lglpk -lgmp $(BLAS_LIBS) $(LAPACK_LIBS) -llzma
+  -lglpk -lgmp $(BLAS_LIBS) $(LAPACK_LIBS) $(FLIBS) -llzma
 
 OBJECTS=${SOURCES}
