diff -Nru orig/ijtiff/src/Makevars.win patched/ijtiff/src/Makevars.win
--- orig/ijtiff/src/Makevars.win	2018-04-12 11:24:07.000000000 +0200
+++ patched/ijtiff/src/Makevars.win	2020-07-02 18:05:41.245710400 +0200
@@ -1,15 +1,6 @@
-ifeq "$(WIN)" "64"
-PKG_CPPFLAGS= -I../windows/libtiff-4.0.9/include
-PKG_LIBS= -L../windows/libtiff-4.0.9/lib${R_ARCH} -ltiff -ljpeg -llzma -lz
-else
-PKG_LIBS= -ltiff -ljpeg -llzma -lz
-endif
+PKG_LIBS= -ltiff -ljpeg -lz -lzstd -lwebp -llzma
 
-
-all: clean winlibs
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R"
+all: clean 
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
