diff -Nru orig/imager/src/Makevars.ucrt patched/imager/src/Makevars.ucrt
--- orig/imager/src/Makevars.ucrt	2022-12-16 00:56:37.000000000 +0100
+++ patched/imager/src/Makevars.ucrt	2023-03-20 09:29:47.021720100 +0100
@@ -1,3 +1,3 @@
 PKG_CPPFLAGS += -I../inst/include -DSTRICT_R_HEADERS -Dcimg_use_r -DCIMG_COMPILING -Dcimg_use_fftw3_singlethread -Dcimg_use_tiff  -Dcimg_use_rng -Dcimg_verbosity=1 -fpermissive -I$(LIB_TIFF)/include -I$(LIB_FFTW)/include 
 PKG_CXXFLAGS += $(SHLIB_OPENMP_CXXFLAGS)
-PKG_LIBS += -L$(LIB_TIFF)/lib -ltiff  -ljpeg -lz -lzstd -lwebp -llzma -L$(LIB_FFTW)/lib -lgdi32 -lfftw3 $(RCPP_LDFLAGS) $(SHLIB_OPENMP_CXXFLAGS)
+PKG_LIBS += -L$(LIB_TIFF)/lib -ltiff  -ljpeg -lz -lzstd -lwebp -lsharpyuv -llzma -L$(LIB_FFTW)/lib -lgdi32 -lfftw3 $(RCPP_LDFLAGS) $(SHLIB_OPENMP_CXXFLAGS)
