diff -Nru orig/image.textlinedetector/src/Makevars.win patched/image.textlinedetector/src/Makevars.win
--- orig/image.textlinedetector/src/Makevars.win	2021-01-25 09:26:00.000000000 +0100
+++ patched/image.textlinedetector/src/Makevars.win	2021-02-09 16:44:27.581548000 +0100
@@ -1,34 +1,11 @@
-VERSION = 4.4.0
-RWINLIB = ../windows/opencv-$(VERSION)
-PKG_CPPFLAGS = -I$(RWINLIB)/include/opencv4
-TARGET = lib$(subst gcc,,$(COMPILED_BY))$(R_ARCH)
+PKG_CPPFLAGS = -I$(LOCAL_SOFT)/include/opencv4
 
 PKG_LIBS = \
-	-L$(RWINLIB)/$(TARGET) \
-	-L$(RWINLIB)/lib$(R_ARCH) \
-	-lopencv_ml -lopencv_objdetect -lopencv_photo -lopencv_stitching \
-	-lopencv_video -lopencv_calib3d -lopencv_features2d -lopencv_highgui -lopencv_flann \
-	-lopencv_videoio -lopencv_imgcodecs -lopencv_imgproc -lopencv_core -ltbb \
-	-ljpeg -lwebp -lpng -lz -ltiff \
-	-lcomctl32 -lgdi32 -lole32 -lsetupapi -lws2_32 -lavifil32 -lavicap32 -lwinmm -lmsvfw32 \
-	-lopengl32 -lglu32 -lcomdlg32 -lOleAut32 -luuid
-
-PKG_LIBS = \
-	-L$(RWINLIB)/$(TARGET) \
-	-L$(RWINLIB)/lib$(R_ARCH) \
-	-lopencv_highgui -lopencv_imgproc -lopencv_core -ltbb \
-	-ljpeg -lwebp -lpng -lz -ltiff \
-	-lcomctl32 -lgdi32 -lole32 -lsetupapi -lws2_32 -lavifil32 -lavicap32 -lwinmm -lmsvfw32 \
-	-lopengl32 -lglu32 -lcomdlg32 -lOleAut32 -luuid
+-lamstrmid -ldmoguids -ldxguid -lksguid -lksuser -lopencv_highgui451 -lopencv_objdetect451 -lopencv_video451 -lopencv_videoio451 -lstrmiids -luuid -lwbemuuid -lopencv_imgcodecs451 -lswscale -lavformat -lass -lgnutls -lbluray -lavcodec -lopencv_imgproc451 -lhogweed -ltasn1 -lxvidcore -lx264 -lvpx -lvorbisenc -ltheoraenc -lswresample -lspeex -lopus -lmp3lame -lvo-amrwbenc -lopencore-amrnb -lopencore-amrwb -lopencv_core451 -lnettle -lgmp -lvorbis -ltheoradec -ltheora -lavutil -logg -lncrypt -lbcrypt -lxml2 -lopenjp2 -lksecdd -lfreetype_too -lharfbuzz_too -lfreetype -lpng16 -lpng -lharfbuzz -lbz2 -lidn2 -lunistring -liconv -lwsock32 -lws2_32 -ltiff -llzma -ljpeg -lz -lzstd -lwebpdecoder -lwebp -lsbml-static -lSDL2 -lgdi32 -lole32 -lopengl32 -lcomdlg32 -lcrypt32 -lwldap32 -loleaut32 -fstack-protector
 
 CXX_STD=CXX11
 
-all: clean winlibs
+all: clean 
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" ${VERSION}
-	mkdir -p ../inst
-	cp -r $(RWINLIB)/share ../inst/
