diff -Nru orig/imager/src/Makevars.win patched/imager/src/Makevars.win
--- orig/imager/src/Makevars.win	2020-05-10 00:40:43.000000000 +0200
+++ patched/imager/src/Makevars.win	2020-07-02 18:06:04.952214900 +0200
@@ -1,3 +1,3 @@
 PKG_CPPFLAGS += -I../inst/include -DSTRICT_R_HEADERS -Dcimg_use_r -DCIMG_COMPILING -Dcimg_use_fftw3_singlethread -Dcimg_use_tiff  -Dcimg_use_rng -Dcimg_verbosity=1 -fpermissive -I$(LIB_TIFF)/include -I$(LIB_FFTW)/include 
 PKG_CXXFLAGS += $(SHLIB_OPENMP_CXXFLAGS)
-PKG_LIBS += -L$(LIB_TIFF)/lib -ltiff  -ljpeg -lz -L$(LIB_FFTW)/lib -lgdi32 -lfftw3-3 $(RCPP_LDFLAGS) $(SHLIB_OPENMP_CXXFLAGS)
+PKG_LIBS += -L$(LIB_TIFF)/lib -ltiff  -ljpeg -lz -lzstd -lwebp -llzma -L$(LIB_FFTW)/lib -lgdi32 -lfftw3 $(RCPP_LDFLAGS) $(SHLIB_OPENMP_CXXFLAGS)
