diff -Nru orig/IP/src/Makevars.win patched/IP/src/Makevars.win
--- orig/IP/src/Makevars.win	2022-10-25 17:29:50.000000000 +0200
+++ patched/IP/src/Makevars.win	2024-02-12 17:11:58.057827900 +0100
@@ -1,13 +1,20 @@
-PKG_CFLAGS= \
-	-I../windows/libidn-1.30/include -D__RIP_IDN__
+ifeq (,$(shell pkg-config --version 2>/dev/null))
+  PKG_LIBS = \
+    -lidn2 -lunistring -lRiconv -lcharset -lws2_32
+else
+  RIP_IDN_LIBS = $(shell pkg-config --libs libidn 2>/dev/null)
+  ifeq (,$(RIP_IDN_LIBS))
+    PKG_LIBS = $(shell pkg-config --libs libidn2)
+  else
+    PKG_LIBS = $(shell pkg-config --libs libidn2 libidn)
+    PKG_CFLAGS = -D__RIP_IDN__
+  endif
+endif
 
-PKG_LIBS= \
-  -L../windows/libidn-1.30/lib${R_ARCH} -lws2_32 -lidn -lRiconv
+PKG_LIBS += -lws2_32
 
-all: clean winlibs
+all: clean 
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
 
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R"
diff -Nru orig/IP/src/Rip-host-info.c patched/IP/src/Rip-host-info.c
--- orig/IP/src/Rip-host-info.c	2024-01-10 19:42:58.000000000 +0100
+++ patched/IP/src/Rip-host-info.c	2024-02-12 17:15:50.478110000 +0100
@@ -849,7 +849,7 @@
   return Rip;
 #else
   sprintf(errmsg, "unavailable '%s' function at line %d in file '%s'.", __func__, __LINE__, __FILE__);
-  error(errmsg);
+  error("%s", errmsg);
   return ScalarLogical(0);
 #endif
 }
