diff -ruN orig/ipaddress/inst/include/ipaddress/IpAddress.h patched/ipaddress/inst/include/ipaddress/IpAddress.h
--- orig/ipaddress/inst/include/ipaddress/IpAddress.h	2020-09-11 22:58:01.000000000 -0700
+++ patched/ipaddress/inst/include/ipaddress/IpAddress.h	2021-01-07 09:02:51.519742600 -0800
@@ -3,6 +3,7 @@
 
 #include <array>
 #include <cstring>
+#include <cstdint>
 #include <ipaddress/endian.h>
 
 
diff -ruN orig/ipaddress/src/Makevars.win patched/ipaddress/src/Makevars.win
--- orig/ipaddress/src/Makevars.win	2020-08-10 17:22:52.000000000 -0700
+++ patched/ipaddress/src/Makevars.win	2021-01-07 08:59:42.489803300 -0800
@@ -2,5 +2,6 @@
 
 # Windows target version: https://docs.microsoft.com/en-us/cpp/porting/modifying-winver-and-win32-winnt
 # set value corresponds to Windows 7
-PKG_CXXFLAGS = -DASIO_STANDALONE -DASIO_NO_DEPRECATED -D_WIN32_WINNT=0x0601
+PKG_CXXFLAGS = -DASIO_STANDALONE -DASIO_NO_DEPRECATED
+#-D_WIN32_WINNT=0x0601
 PKG_LIBS = -lws2_32 -lwsock32
