diff -Nru orig/KSgeneral/src/Makevars.win patched/KSgeneral/src/Makevars.win
--- orig/KSgeneral/src/Makevars.win	2018-05-14 18:57:00.000000000 +0200
+++ patched/KSgeneral/src/Makevars.win	2020-07-02 18:05:51.559683000 +0200
@@ -6,10 +6,9 @@
 ###PKG_CPPFLAGS = -I$(FFTW3_32)
 ###endif
 
-PKG_CPPFLAGS = -I$(LIB_FFTW)/include -I$(LIB_FFTW)
 CXX11 = g++
 PKG_CXXFLAGS = -std=c++11
-PKG_LIBS = -L$(LIB_FFTW)/lib${R_ARCH} -L$(LIB_FFTW)/lib -lfftw3 -lm
+PKG_LIBS = -lfftw3 -lm
 #PKG_LIBS = -L$(LIB_FFTW)/lib -lfftw3
 
 
