diff -Nru orig/KSgeneral/src/Makevars.ucrt patched/KSgeneral/src/Makevars.ucrt
--- orig/KSgeneral/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/KSgeneral/src/Makevars.ucrt	2021-02-17 19:45:26.128432000 +0100
@@ -0,0 +1,15 @@
+###ifeq “$(WIN)” “64”
+###PKG_LIBS = -L$(FFTW3_64) -lfftw3
+###PKG_CPPFLAGS = -I$(FFTW3_64)
+###else
+###PKG_LIBS = -L$(FFTW3_32) -lfftw3
+###PKG_CPPFLAGS = -I$(FFTW3_32)
+###endif
+
+CXX11 = g++
+PKG_CXXFLAGS = -std=c++11
+PKG_LIBS = -lfftw3 -lm
+#PKG_LIBS = -L$(LIB_FFTW)/lib -lfftw3
+
+
+
