diff -Nru orig/leidenbase/src/Makevars.ucrt patched/leidenbase/src/Makevars.ucrt
--- orig/leidenbase/src/Makevars.ucrt	2023-07-24 17:01:49.000000000 +0200
+++ patched/leidenbase/src/Makevars.ucrt	2024-02-13 09:58:50.556124800 +0100
@@ -10,7 +10,7 @@
   -I${LIB_GMP}/include -Ileidenalg -Ileidenbase
 
 PKG_LIBS = -L${LIB_XML}/lib -lxml2 -liconv -lz -lws2_32 -lstdc++ -L${GLPK_HOME}/lib \
-  -lglpk -lgmp -lgfortran -L$(LIB_GMP)/lib $(BLAS_LIBS) $(LAPACK_LIBS) -llzma
+  -lglpk -lgmp -L$(LIB_GMP)/lib $(BLAS_LIBS) $(LAPACK_LIBS) $(FLIBS) -llzma
 
 #
 # Debian i386 fix.
