diff -Nru orig/libimath/src/Makevars.win.in patched/libimath/src/Makevars.win.in
--- orig/libimath/src/Makevars.win.in	2025-06-29 05:12:55.000000000 +0200
+++ patched/libimath/src/Makevars.win.in	2025-10-14 16:40:23.870777700 +0200
@@ -37,8 +37,8 @@
 	src/Imath/build/config/Makefile \
 	src/Imath/build/src/Imath/Makefile \
 	src/Makevars \
-	src/Imath/build/CMakeFiles/3.30.4/CompilerIdC/CMakeCCompilerId.c \
-	src/Imath/build/CMakeFiles/3.30.4/CompilerIdCXX/CMakeCXXCompilerId.cpp \
+	src/Imath/build/CMakeFiles/*/CompilerIdC/CMakeCCompilerId.c \
+	src/Imath/build/CMakeFiles/*/CompilerIdCXX/CMakeCXXCompilerId.cpp \
 	src/Imath/build/config/ImathConfig.h \
 	inst/include/Imath/ImathConfig.h
 
@@ -50,4 +50,4 @@
 	rm -rf @PACKAGE_BASE_DIR@/config.status && \
 	rm -rf @PACKAGE_BASE_DIR@/autom4te.cache && \
 	rm -rf @PACKAGE_BASE_DIR@/configure~ && \
-	rm -rf @PACKAGE_BASE_DIR@/tools
\ No newline at end of file
+	rm -rf @PACKAGE_BASE_DIR@/tools
