diff -Nru orig/libsoc/src/Makevars.win patched/libsoc/src/Makevars.win
--- orig/libsoc/src/Makevars.win	2019-05-15 09:18:52.000000000 +0200
+++ patched/libsoc/src/Makevars.win	2020-07-02 18:05:30.275393100 +0200
@@ -1,17 +1,9 @@
-LIB_XML ?= ../windows/libxml2-2.9.4
+PKG_CFLAGS=-Iinclude -I$(LOCAL_SOFT)/include/libxml2  -DLIBXML_STATIC
+PKG_LIBS=-lxml2 -lz -liconv -llzma -lws2_32
 
-LIBSOC_INC_PATH=${LIB_XML}/include/libxml2
-LIBSOC_LIB_PATH=${LIB_XML}/lib/${R_ARCH}
-
-PKG_CFLAGS=-Iinclude -I${LIBSOC_INC_PATH}
-PKG_LIBS=-L${LIBSOC_LIB_PATH} -lxml2 -lz -liconv -llzma -lws2_32
-
-all: winlibs
+all: 
 
 clean:
 	rm -f $(OBJECTS) xml2.dll
 
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R"
-
-.PHONY: all winlibs clean
+.PHONY: all clean
