diff -Nru orig/lwgeom/src/Makevars.win patched/lwgeom/src/Makevars.win
--- orig/lwgeom/src/Makevars.win	2020-05-04 14:03:28.000000000 +0200
+++ patched/lwgeom/src/Makevars.win	2021-02-09 16:06:58.889270500 +0100
@@ -83,30 +83,19 @@
 
 OBJECTS= $(OBJECTS_LIBLWGEOM) $(OBJECTS_RCPP)
 
-VERSION = 3.0.4
-RWINLIB = ../windows/gdal3-$(VERSION)
 TARGET = lib$(subst gcc,,$(COMPILED_BY))$(R_ARCH)
 
 PKG_CPPFLAGS = \
 	-I./liblwgeom \
-	-I$(RWINLIB)/include/geos-3.8.0 \
-	-I$(RWINLIB)/include/proj-6.3.1 \
 	-DUSE_PROJ_H
 
 PKG_LIBS = \
-	-L$(RWINLIB)/$(TARGET) \
-	-L$(RWINLIB)/lib$(R_ARCH) \
-	-lproj -lgeos_c -lgeos -ljson-c -lexpat -lxml2 -liconv -lsqlite3
+-lgeos_c -lgeos -lproj -lcurl -ltiff -lsqlite3 -lssh2 -lidn2 -lunistring -lintl -lcharset -liconv -lgcrypt -lgpg-error -lwsock32 -lws2_32 -lzstd -lz -lwebpdecoder -lwebp -llzma -ljpeg -lcfitsio -lwldap32 -lcrypt32
 
-all: clean winlibs
+all: clean
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
 
-winlibs:
-	cp postgis_config.win postgis_config.h
-	mkdir -p ../inst
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" --vanilla "../tools/winlibs.R" $(VERSION)
-	# cp -r $(RWINLIB)/share/proj ../inst/
 
-.PHONY: all winlibs clean
+.PHONY: all clean
