diff -Nru orig/magick/src/Makevars.ucrt patched/magick/src/Makevars.ucrt
--- orig/magick/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/magick/src/Makevars.ucrt	2021-02-17 19:45:41.414999600 +0100
@@ -0,0 +1,15 @@
+
+PKG_CPPFLAGS = -I${LOCAL_SOFT}/include/ImageMagick-7
+PKG_CXXFLAGS = -DMAGICKCORE_HDRI_ENABLE=0 -DMAGICKCORE_QUANTUM_DEPTH=16 -D_LIB
+
+CXX_STD = CXX11
+
+PKG_LIBS = -fopenmp \
+-lMagick++-7.Q16HDRI -lsbml-static -lMagickWand-7.Q16HDRI -lMagickCore-7.Q16HDRI -llqr-1 -lraw_r -lraw -lfftw3 -lwebpmux -lIlmImf -lpangocairo-1.0 -ljasper -llcms2 -lHalf -lIlmThread -lpangoft2-1.0 -lcairo -lpangowin32-1.0 -lIex -lpixman-1 -lfontconfig -lpango-1.0 -lusp10 -lgdi32 -lgobject-2.0 -lffi -lxml2 -lopenjp2 -lexpat -lfreetype_too -lharfbuzz_too -lfreetype -lpng16 -lpng -lharfbuzz -lglib-2.0 -lpcre -lbz2 -lintl -liconv -lwsock32 -lws2_32 -ltiff -llzma -ljpeg -lz -lcfitsio -lzstd -lwebpdecoder -lwebp -lole32
+
+all: clean 
+
+clean:
+	rm -f $(OBJECTS) $(SHLIB)
+
+.PHONY: all clean
