diff -Nru orig/mlpack/src/Makevars.win patched/mlpack/src/Makevars.win
--- orig/mlpack/src/Makevars.win	2023-11-27 19:46:30.000000000 +0100
+++ patched/mlpack/src/Makevars.win	2024-02-01 19:43:33.052654800 +0100
@@ -1,3 +1,3 @@
-PKG_CXXFLAGS = -I. -I../inst/include $(SHLIB_OPENMP_CXXFLAGS) -ftrack-macro-expansion=0 -pipe --param ggc-min-expand=10 --param ggc-min-heapsize=8192
+PKG_CXXFLAGS = -I. -I../inst/include $(SHLIB_OPENMP_CXXFLAGS) -pipe --param ggc-min-expand=10 --param ggc-min-heapsize=8192
 PKG_LIBS = $(SHLIB_OPENMP_CXXFLAGS) $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)
 CXX_STD = CXX17
