diff -Nru orig/mongolite/src/Makevars.win patched/mongolite/src/Makevars.win
--- orig/mongolite/src/Makevars.win	2020-03-17 17:01:22.000000000 +0100
+++ patched/mongolite/src/Makevars.win	2020-07-02 18:05:57.998104300 +0200
@@ -1,5 +1,4 @@
 PKG_CPPFLAGS = \
-	-I../windows/openssl-1.1.1/include \
 	-I. -Icommon \
 	-DBSON_COMPILATION -DMONGOC_COMPILATION \
 	-DMONGOC_HAVE_SASL_CLIENT_DONE -D__USE_MINGW_ANSI_STDIO
@@ -8,7 +7,7 @@
 	-Wno-deprecated-declarations
 
 PKG_LIBS = \
-	-L. -L../windows/openssl-1.1.1/lib${R_ARCH} \
+	-L. \
 	-Lmongoc -lstatmongoc -lssl -lcrypto \
 	-lcrypto -lgdi32 -lcrypt32 -lws2_32 -lsecur32 -ldnsapi -lz
 
@@ -61,14 +60,10 @@
 
 $(SHLIB): $(STATLIB)
 
-$(OBJECTS): winlibs
-
 $(STATLIB): $(LIBBSON) $(LIBMONGOC)
 
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R"
 
 clean:
 	rm -f $(SHLIB) $(STATLIB) $(LIBBSON) $(LIBMONGOC) $(OBJECTS)
 
-.PHONY: all clean winlibs
+.PHONY: all clean 
