diff -ruN orig/nanotime/src/Makevars.win patched/nanotime/src/Makevars.win
--- orig/nanotime/src/Makevars.win	2020-07-29 14:17:53.000000000 -0700
+++ patched/nanotime/src/Makevars.win	2021-01-12 13:39:29.834888700 -0800
@@ -2,4 +2,4 @@
 CXX_STD = CXX11
 
 ## We need headers from our package, the directory is not automatically included
-PKG_CXXFLAGS = -I../inst/include -mno-ms-bitfields
+PKG_CXXFLAGS = -I../inst/include -mno-ms-bitfields -D_POSIX_THREAD_SAFE_FUNCTIONS
