diff -Nru orig/nbfar/src/Makevars.win patched/nbfar/src/Makevars.win
--- orig/nbfar/src/Makevars.win	2022-02-18 17:41:37.000000000 +0100
+++ patched/nbfar/src/Makevars.win	2025-01-29 12:10:02.107182400 +0100
@@ -1,5 +1,5 @@
 PKG_LIBS = $(LAPACK_LIBS) $(BLAS_LIBS)
-PKG_LIBS += `$(R_HOME)/bin${R_ARCH_BIN}/Rscript.exe -e "RcppParallel::RcppParallelLibs()"`
+PKG_LIBS += $(shell "$(R_HOME)/bin${R_ARCH_BIN}/Rscript.exe" -e "RcppParallel::RcppParallelLibs()")
 PKG_CXXFLAGS = -DRCPP_PARALLEL_USE_TBB=1 -DARMA_64BIT_WORD=1
 PKG_CPPFLAGS = -DARMA_DONT_PRINT_OPENMP_WARNING=1
 CXX_STD = CXX11
