diff -Nru orig/nloptr/src/Makevars.win patched/nloptr/src/Makevars.win
--- orig/nloptr/src/Makevars.win	2018-04-21 22:42:20.000000000 +0200
+++ patched/nloptr/src/Makevars.win	2020-07-02 18:06:26.705319300 +0200
@@ -19,5 +19,5 @@
 # 2 May 2014: Added quotes around include paths to allow for spaces in NLOPT_HOME.
 
 # C Compiler options
-PKG_CFLAGS = -I"$(NLOPT_HOME)$(R_ARCH)/include"
-PKG_LIBS = -L"$(NLOPT_HOME)$(R_ARCH)/lib" -lnlopt_cxx
+PKG_CFLAGS = 
+PKG_LIBS = -lnlopt
