diff -Nru orig/opencv/src/Makevars.ucrt patched/opencv/src/Makevars.ucrt
--- orig/opencv/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/opencv/src/Makevars.ucrt	2021-02-17 19:45:41.883754200 +0100
@@ -0,0 +1,13 @@
+PKG_CPPFLAGS = -I$(LOCAL_SOFT)/include/opencv4 
+#-DHAVE_XFEATURES2D
+
+PKG_LIBS = \
+-lamstrmid -ldmoguids -ldxguid -lksguid -lksuser -lopencv_highgui451 -lopencv_objdetect451 -lopencv_video451 -lopencv_videoio451 -lstrmiids -luuid -lwbemuuid -lopencv_imgcodecs451 -lswscale -lavformat -lass -lgnutls -lbluray -lavcodec -lopencv_imgproc451 -lhogweed -ltasn1 -lxvidcore -lx264 -lvpx -lvorbisenc -ltheoraenc -lswresample -lspeex -lopus -lmp3lame -lvo-amrwbenc -lopencore-amrnb -lopencore-amrwb -lopencv_core451 -lnettle -lgmp -lvorbis -ltheoradec -ltheora -lavutil -logg -lncrypt -lbcrypt -lxml2 -lopenjp2 -lksecdd -lfreetype_too -lharfbuzz_too -lfreetype -lpng16 -lpng -lharfbuzz -lbz2 -lidn2 -lunistring -liconv -lwsock32 -lws2_32 -ltiff -llzma -ljpeg -lz -lzstd -lwebpdecoder -lwebp -lsbml-static -lSDL2 -lgdi32 -lole32 -lopengl32 -lcomdlg32 -lcrypt32 -lwldap32 -loleaut32 -fstack-protector
+
+CXX_STD=CXX11
+
+all: clean 
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
