diff -Nru orig/openssl/src/Makevars.win patched/openssl/src/Makevars.win
--- orig/openssl/src/Makevars.win	2019-03-21 21:31:26.000000000 +0100
+++ patched/openssl/src/Makevars.win	2020-07-02 18:05:48.668602900 +0200
@@ -1,25 +1,11 @@
-VERSION = 1.1.1
-PKG_CPPFLAGS = -I../windows/openssl-$(VERSION)/include
-PKG_LIBS = $(WIN32FIX) -Lbcrypt -lstatbcrypt -L../windows/openssl-$(VERSION)/lib${R_ARCH} \
-	-lssl -lcrypto -lws2_32 -lgdi32 -lcrypt32
-STATLIB = bcrypt/libstatbcrypt.a
-LIBBCRYPT = bcrypt/bcrypt_pbkdf.o bcrypt/blowfish.o
-WIN32FIX = win32/ipv6.o
+PKG_LIBS = -lbcrypt -lssh2 -lgcrypt -lgpg-error -lssl -lcrypto -lws2_32 -lgdi32 -lcrypt32 -lz
 
 #all: clean
 all: $(SHLIB)
 
-$(OBJECTS): winlibs
-
-$(SHLIB): $(WIN32FIX) $(OBJECTS) $(STATLIB)
-
-$(STATLIB): $(LIBBCRYPT)
-	$(AR) rcs $(STATLIB) $(LIBBCRYPT)
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
+$(SHLIB): $(OBJECTS)
 
 clean:
-	rm -f $(SHLIB) $(OBJECTS) $(STATLIB) $(LIBBCRYPT)
+	rm -f $(SHLIB) $(OBJECTS) 
 
-.PHONY: all clean winlibs
+.PHONY: all clean 
