diff -Nru orig/opencv/src/Makevars.win patched/opencv/src/Makevars.win
--- orig/opencv/src/Makevars.win	2019-03-30 17:18:22.000000000 +0100
+++ patched/opencv/src/Makevars.win	2020-07-02 18:06:10.609276400 +0200
@@ -1,26 +1,18 @@
-VERSION = 4.0.1
-RWINLIB = ../windows/opencv-$(VERSION)
-PKG_CPPFLAGS = -I$(RWINLIB)/include/opencv4
-TARGET = lib$(subst gcc,,$(COMPILED_BY))$(R_ARCH)
+PKG_CPPFLAGS = -I$(LOCAL_SOFT)/include/opencv2
 
 PKG_LIBS = \
-	-L$(RWINLIB)/$(TARGET) \
-	-L$(RWINLIB)/lib$(R_ARCH) \
-	-lopencv_gapi -lopencv_ml -lopencv_objdetect -lopencv_photo -lopencv_stitching \
+	-lopencv_ml -lopencv_objdetect -lopencv_photo -lopencv_stitching \
 	-lopencv_video -lopencv_calib3d -lopencv_features2d -lopencv_highgui -lopencv_flann \
-	-lopencv_videoio -lopencv_imgcodecs -lopencv_imgproc -lopencv_core -lquirc -ltbb \
-	-lade -lcomctl32 -lgdi32 -lole32 -lsetupapi -lws2_32 -ljpeg -lwebp -lpng -lz -ltiff \
+	-lopencv_videoio -lopencv_imgcodecs -ljasper -lIlmImf -lIlmImfUtil -lIlmThread -lIex -lHalf -lopencv_imgproc -lopencv_core \
+	-lcomctl32 -lgdi32 -lole32 -lsetupapi -lws2_32 -ltiff -ljpeg -lpng -lz \
 	-lcomctl32 -lgdi32 -lole32 -lsetupapi -lws2_32 -lavifil32 -lavicap32 -lwinmm -lmsvfw32 \
-	-lopengl32 -lglu32 -lcomdlg32 -lOleAut32 -luuid
+	-lopengl32 -lglu32 -lcomdlg32 -lOleAut32 -luuid \
+	-lzstd -lwebp -llzma
 
 CXX_STD=CXX11
 
-all: clean winlibs
+all: clean 
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
 
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" ${VERSION}
-	mkdir -p ../inst
-	cp -r $(RWINLIB)/share ../inst/
