diff -ruN orig/paropt/src/Makevars.win patched/paropt/src/Makevars.win
--- orig/paropt/src/Makevars.win	2020-06-29 23:09:18.000000000 -0700
+++ patched/paropt/src/Makevars.win	2021-01-12 04:02:03.743758000 -0800
@@ -2,7 +2,7 @@
 $(info The name of the shared library to be created is: $(SHLIB))
 
 CXX_STD=CXX11
-PKG_CPPFLAGS=-I../inst/include -I../inst
+PKG_CPPFLAGS=-I../inst/include -I../inst -fcommon
 PKG_CXXFLAGS= $(SHLIB_OPENMP_CXXFLAGS) -I../inst/include -I../inst
 PKG_LIBS = $(SHLIB_OPENMP_CXXFLAGS) $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS) -L../inst -lsundials
 PKG_LIB = -std = gnu++11
