diff -ruN orig/pcaL1/src/Makevars.win patched/pcaL1/src/Makevars.win
--- orig/pcaL1/src/Makevars.win	2019-02-07 04:54:40.000000000 -0800
+++ patched/pcaL1/src/Makevars.win	2021-01-12 04:08:54.180017100 -0800
@@ -1,2 +1,2 @@
-PKG_CPPFLAGS=-D_R_=1 -DUSE_R=1 -I${CLP_PATH}/include/coin -I${CLP_PATH}/include
-PKG_LIBS=-L${CLP_PATH}/lib/coin -L${CLP_PATH}/lib -lClp -lCoinUtils -lstdc++ $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS) -lz
+PKG_CPPFLAGS=-D_R_=1 -DUSE_R=1 -I${LOCAL_SOFT}/include/coin
+PKG_LIBS=-lClp -lCoinUtils -lstdc++ $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS) -lz -lbz2
