diff -Nru orig/pdftools/src/Makevars.ucrt patched/pdftools/src/Makevars.ucrt
--- orig/pdftools/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/pdftools/src/Makevars.ucrt	2021-02-17 19:45:42.211836200 +0100
@@ -0,0 +1,15 @@
+
+CXX_STD=CXX11
+PKG_CXXFLAGS=-Dpoppler_cpp_EXPORTS -DBUNDLE_POPPLER_DATA
+PKG_CPPFLAGS=-I$(LOCAL_SOFT)/include/poppler/cpp -I$(LOCAL_SOFT)/include/poppler
+
+PKG_LIBS= \
+	-lpoppler-cpp -lpoppler -llcms2 -ljpeg -lpng16 -ltiff -lopenjp2 \
+	-lfreetype -lharfbuzz -lfreetype -lbz2 -liconv -lz -lzstd -lwebp -llzma
+
+all: clean 
+
+clean:
+	rm -f $(OBJECTS) $(SHLIB)
+
+.PHONY: all clean
