diff -Nru orig/PearsonDS/configure.win patched/PearsonDS/configure.win
--- orig/PearsonDS/configure.win	2021-10-06 12:14:42.000000000 +0200
+++ patched/PearsonDS/configure.win	2024-02-07 18:06:17.869439600 +0100
@@ -5,4 +5,6 @@
 
 echo "  configure.win32..."
 echo "  creating src/config.h..."
-echo "#define HAVE_FPU_INLINE_ASM_X86 1" > src/config.h
+echo "#if defined(__i386) || defined(__x86_64)" > src/config.h
+echo "# define HAVE_FPU_INLINE_ASM_X86 1" >> src/config.h
+echo "#endif" >> src/config.h
