diff -Nru orig/proj4/src/Makevars.ucrt patched/proj4/src/Makevars.ucrt
--- orig/proj4/src/Makevars.ucrt	2022-11-30 01:11:36.000000000 +0100
+++ patched/proj4/src/Makevars.ucrt	2023-03-17 09:36:25.385635800 +0100
@@ -1,3 +1,3 @@
 ## from Tomas, uses P6 API so verify on toolchain updates
-PKG_LIBS = -lproj -lsqlite3 -lcurl -ltiff -ljpeg -lrtmp -lssl -lssh2 -lgcrypt -lcrypto -lgdi32 -lz -lzstd -lwebp -llzma -lgdi32 -lcrypt32 -lidn2 -lunistring -liconv -lgpg-error -lws2_32 -lwldap32 -lwinmm -lstdc++
+PKG_LIBS = -lproj -lsqlite3 -lcurl -lbcrypt -ltiff -ljpeg -lrtmp -lssl -lssh2 -lgcrypt -lcrypto -lgdi32 -lz -lzstd -lwebp -lsharpyuv -llzma -lgdi32 -lcrypt32 -lidn2 -lunistring -liconv -lgpg-error -lws2_32 -lwldap32 -lwinmm -lstdc++
 PKG_CPPFLAGS = -DUSE_PROJ6_API=1
