diff -Nru orig/proj4/configure.win patched/proj4/configure.win
--- orig/proj4/configure.win	2020-03-02 21:03:42.000000000 +0100
+++ patched/proj4/configure.win	2021-02-17 19:45:42.274384900 +0100
@@ -1,51 +0,0 @@
-#!/bin/sh
-
-echo "  checking PROJ4 headers and libraries"
-allok=yes
-
-if [ -e "${LOCAL_SOFT}/include/proj_api.h" ]; then
-    echo " --- Using local version of PROJ4 found in LOCAL_SOFT: ${LOCAL_SOFT}"
-    echo " --- Proceed at your own risk, if in doubt unset LOCAL_SOFT to use our binary."
-    exit 0
-fi
-
-if [ ! -e src/win32/libproj.a ]; then
-    if [ ! -e src/proj4-current-win.tar.gz ]; then
-	echo "  cannot find current PROJ4 files"
-	echo "  attempting to download them"
-	echo 'download.file("http://www.rforge.net/proj4/files/proj4-current-win.tar.gz","src/proj4-current-win.tar.gz",mode="wb",quiet=TRUE)'|${R_HOME}/bin/R --vanilla --slave
-    fi
-    if [ ! -e src/proj4-current-win.tar.gz ]; then
-	allok=no
-    else
-	echo "  unpacking current PROJ4"
-	tar fxz src/proj4-current-win.tar.gz -C src
-        if [ ! -e src/win32/libproj.a ]; then
-	    allok=no
-	fi
-    fi
-fi
-
-if [ ! -e src/win32/libproj.a ]; then
-    allok=no
-fi
-
-if [ ${allok} != yes ]; then
-    echo ""
-    echo " *** ERROR: unable to find PROJ4 files"
-    echo ""
-    echo " They must be either in src/win32 or in a tar-ball"
-    echo " src/proj4-current-win.tar.gz"
-    echo ""
-    echo " You can get the latest binary tar ball from"
-    echo " http://www.rforge.net/proj4/files/"
-    echo ""
-    echo " Alternatively you can set LOCAL_SOFT environment variable"
-    echo " to point to the prefix (like /usr/local) containing PROJ4 binaries."
-    echo ""
-    exit 1
-fi
-
-echo "  seems ok, ready to go"
-
-exit 0
diff -Nru orig/proj4/src/Makevars.ucrt patched/proj4/src/Makevars.ucrt
--- orig/proj4/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/proj4/src/Makevars.ucrt	2021-02-17 19:45:42.305633400 +0100
@@ -0,0 +1,2 @@
+PKG_LIBS = -lproj -lsqlite3 -lcurl -ltiff -ljpeg -lrtmp -lssl -lssh2 -lgcrypt -lcrypto -lgdi32 -lz -lzstd -lwebp -llzma -lgdi32 -lcrypt32 -lidn2 -lunistring -liconv -lgpg-error -lws2_32 -lwldap32 -lwinmm -lstdc++
+PKG_CPPFLAGS = -DACCEPT_USE_OF_DEPRECATED_PROJ_API_H
