diff -Nru orig/protolite/src/Makevars.win patched/protolite/src/Makevars.win
--- orig/protolite/src/Makevars.win	2020-01-11 13:40:38.000000000 +0100
+++ patched/protolite/src/Makevars.win	2020-07-02 18:05:40.667501100 +0200
@@ -1,11 +1,4 @@
-# ProtoBuf Version
-VERSION = 3.8.0
-RWINLIB = ../windows/protobuf-${VERSION}
-
-PKG_CPPFLAGS= -I$(RWINLIB)/include
-PKG_LIBS= -L$(RWINLIB)/lib${subst gcc,,${COMPILED_BY}}${R_ARCH} \
-	-L$(RWINLIB)/lib${R_ARCH} \
-	-lprotobuf
+PKG_LIBS= -lprotobuf
 
 CXX_STD=CXX11
 PKG_CXXFLAGS = -Wno-long-long
@@ -13,8 +6,7 @@
 all: clean winlibs
 
 winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
-	"${RWINLIB}/bin$(WIN)/protoc" *.proto --cpp_out=.
+	protoc *.proto --cpp_out=.
 
 clean:
 	rm -f $(OBJECTS) $(SHLIB) *.pb.cc *.pb.h
