diff -ruN orig/PythonInR/src/Makevars.in patched/PythonInR/src/Makevars.in
--- orig/PythonInR/src/Makevars.in	2020-06-21 12:49:59.000000000 -0700
+++ patched/PythonInR/src/Makevars.in	2021-01-07 05:59:18.078047200 -0800
@@ -2,7 +2,7 @@
 #   ‘-Wdate-time’ ‘-Werror=format-security’ ‘-Wformat’
 
 PKG_LIBS=@PKG_LIBS@
-PKG_CFLAGS=@PKG_CFLAGS@ -D@PYXY@ -DPYTHON_IN_R_NO_EXPLICIT_LINKING
+PKG_CFLAGS=@PKG_CFLAGS@ -D@PYXY@ -DPYTHON_IN_R_NO_EXPLICIT_LINKING 
 
 OBJECTS = CToR.o PythonInR.o CastPyObjects.o CastRObjects.o GetPyObjects.o init.o PyCall.o PyRunString.o Python_Explicit_Linking.o SetPyObjects.o
 
diff -ruN orig/PythonInR/src/Makevars.win patched/PythonInR/src/Makevars.win
--- orig/PythonInR/src/Makevars.win	1969-12-31 16:00:00.000000000 -0800
+++ patched/PythonInR/src/Makevars.win	2021-01-07 06:21:48.957420500 -0800
@@ -0,0 +1,23 @@
+# Compilation used the following non-portable flag(s):
+#   ‘-Wdate-time’ ‘-Werror=format-security’ ‘-Wformat’
+
+PKG_CFLAGS = -fcommon
+OBJECTS = CToR.o PythonInR.o CastPyObjects.o CastRObjects.o GetPyObjects.o init.o PyCall.o PyRunString.o Python_Explicit_Linking.o SetPyObjects.o
+
+all: $(SHLIB)
+
+$(SHLIB): $(OBJECTS)
+
+#ALL_CFLAGS = $(R_XTRA_CFLAGS) $(PKG_CFLAGS) $(CPICFLAGS) $(SHLIB_CFLAGS) $(CFLAGS)
+#ALL_CPPFLAGS = $(R_XTRA_CPPFLAGS) $(PKG_CPPFLAGS) $(CLINK_CPPFLAGS) $(CPPFLAGS)
+
+ALL_CFLAGS = $(R_XTRA_CFLAGS) $(PKG_CFLAGS) 
+ALL_CPPFLAGS = $(R_XTRA_CPPFLAGS) $(PKG_CPPFLAGS)
+
+
+%.o: %.c
+	$(CC) $(R_XTRA_CPPFLAGS) $(R_XTRA_CFLAGS) $(PKG_CFLAGS) $(CPICFLAGS) $(SHLIB_CFLAGS) -c $< -o $@
+
+# original from /etc/R/Makeconf
+# %.o: %.c
+# $(CC) $(ALL_CPPFLAGS) $(ALL_CFLAGS) -c $< -o $@
