diff -Nru orig/R2SWF/src/Makevars.win patched/R2SWF/src/Makevars.win
--- orig/R2SWF/src/Makevars.win	2020-05-09 01:51:36.000000000 +0200
+++ patched/R2SWF/src/Makevars.win	2020-07-02 18:06:53.912330100 +0200
@@ -1,5 +1,5 @@
-PKG_CPPFLAGS = -I./include -I./include/freetype -I./libming
-PKG_LIBS = -L./libming -lming -L.${R_ARCH} -lfreetype -lpng -lz
+PKG_CPPFLAGS = -I./include -I$(LOCAL_SOFT)/include/freetype2 -I./libming
+PKG_LIBS = -L./libming -lming -lfreetype -lharfbuzz -lfreetype -lpng -lz -lbz2
 
 
 .PHONY: all deps clean
@@ -9,13 +9,7 @@
 $(SHLIB): libming/libming.a
 
 libming/libming.a:
-	cd libming && $(MAKE) all CC="$(CC)" AR="$(AR)" CPPFLAGS="$(R_XTRA_CPPFLAGS)" CFLAGS="$(ALL_CFLAGS)"
-
-deps: sysfonts-dep-win-biarch.tar.gz
-	tar xzf sysfonts-dep-win-biarch.tar.gz
-
-sysfonts-dep-win-biarch.tar.gz:
-	${R_HOME}/bin$(R_ARCH_BIN)/Rscript -e "u = 'https://raw.githubusercontent.com/yixuan/sysfonts-dep/master/sysfonts-dep-win-biarch.tar.gz'; utils::download.file(u, basename(u), mode = 'wb')"
+	cd libming && $(MAKE) all CC="$(CC)" AR="$(AR)" CPPFLAGS="-I$(LOCAL_SOFT)/include/freetype2 $(R_XTRA_CPPFLAGS)" CFLAGS="$(ALL_CFLAGS)"
 
 clean:
 	$(RM) *.o
