diff -Nru orig/ragg/src/Makevars.ucrt patched/ragg/src/Makevars.ucrt
--- orig/ragg/src/Makevars.ucrt	1969-12-31 16:00:00.000000000 -0800
+++ patched/ragg/src/Makevars.ucrt	2021-02-18 02:28:13.834852100 -0800
@@ -0,0 +1,25 @@
+
+AGG_OBJECTS = agg/src/agg_curves.o agg/src/agg_font_freetype.o \
+	agg/src/agg_image_filters.o agg/src/agg_trans_affine.o \
+	agg/src/agg_vcgen_dash.o agg/src/agg_vcgen_stroke.o
+
+PKG_CPPFLAGS = -I${LOCAL_SOFT}/include/cairo \
+	-I${LOCAL_SOFT}/include/freetype2 -I./agg/include \
+	-DSTRICT_R_HEADERS
+
+PKG_LIBS = -Lagg -lstatagg \
+	-lcairo -lfontconfig -lintl -liconv -lexpat -lharfbuzz -lfreetype \
+	-lharfbuzz -lfreetype -lpixman-1 -lpng -ltiff -ljpeg -lbz2 -lz \
+	-lzstd -lgdi32 -lmsimg32 -lwebp -llzma -lws2_32 $(GRAPHAPP_LIB)
+	
+STATLIB = agg/libstatagg.a
+
+all: clean 
+
+$(SHLIB): $(STATLIB)
+
+$(STATLIB): $(AGG_OBJECTS)
+
+
+clean:
+	rm -f $(OBJECTS) $(STATLIB) $(SHLIB) $(AGG_OBJECTS)
