diff -Nru orig/RcppGSL/src/Makevars.win patched/RcppGSL/src/Makevars.win
--- orig/RcppGSL/src/Makevars.win	2019-10-20 15:40:09.000000000 +0200
+++ patched/RcppGSL/src/Makevars.win	2020-07-02 18:05:29.072056400 +0200
@@ -1,10 +1,7 @@
-PKG_CPPFLAGS= -I../windows/gsl-2.4/include -I../inst/include
-PKG_LIBS= -L../windows/gsl-2.4/lib${R_ARCH} -lgsl -lgslcblas
+PKG_CPPFLAGS= -I../inst/include
+PKG_LIBS= -lgsl -lgslcblas
 
-all: clean winlibs
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R"
+all: clean
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
diff -Nru orig/RcppGSL/tools/winlibs.R patched/RcppGSL/tools/winlibs.R
--- orig/RcppGSL/tools/winlibs.R	2017-12-23 14:48:43.000000000 +0100
+++ patched/RcppGSL/tools/winlibs.R	1970-01-01 01:00:00.000000000 +0100
@@ -1,7 +0,0 @@
-# Download GSL 2.4 Rtools build
-if(!file.exists("../windows/gsl-2.4/include/gsl/gsl_blas.h")){
-  download.file("https://github.com/rwinlib/gsl/archive/v2.4.zip", "lib.zip", quiet = TRUE)
-  dir.create("../windows", showWarnings = FALSE)
-  unzip("lib.zip", exdir = "../windows")
-  unlink("lib.zip")
-}
