diff -Nru orig/RcppZiggurat/src/Makevars.ucrt patched/RcppZiggurat/src/Makevars.ucrt
--- orig/RcppZiggurat/src/Makevars.ucrt	2021-07-12 22:43:42.000000000 +0200
+++ patched/RcppZiggurat/src/Makevars.ucrt	2025-03-26 17:39:43.626019500 +0100
@@ -1,2 +1,9 @@
-CRT=-ucrt
-include Makevars.win
+ifeq (,$(shell pkg-config --version 2>/dev/null))
+  PKG_LIBS = -lgsl -lgslcblas -lm
+else
+  PKG_CPPFLAGS = $(shell pkg-config --cflags gsl)
+  PKG_LIBS = $(shell pkg-config --libs gsl)
+endif
+
+PKG_CPPFLAGS += -I../inst/include
+
