diff -Nru orig/redland/src/Makevars.win patched/redland/src/Makevars.win
--- orig/redland/src/Makevars.win	2020-10-12 19:18:54.000000000 +0200
+++ patched/redland/src/Makevars.win	2021-01-22 18:00:29.406837000 +0100
@@ -1,24 +1,13 @@
-PKG_CPPFLAGS= -I../windows/redland-1.0.17/include \
-	-I../windows/redland-1.0.17/include/rasqal \
-	-I../windows/redland-1.0.17/include/raptor2
-	
-PKG_LIBS=-L../windows/redland-1.0.17/lib${R_ARCH} -lrdf \
-  -pipe -lrasqal -lpcre -lraptor2 -lxml2 -lpcre -lws2_32
 
-## Workaround for missing _mkgmtime on Win32, See also
-## https://sourceforge.net/p/mingw-w64/bugs/473/
-ifeq "${R_ARCH}" "/i386"
-ifeq "${R_COMPILED_BY}" "gcc 4.9.3"
-PKG_LIBS += -lmsvcr100
-endif
-endif
+PKG_CPPFLAGS= -I${LOCAL_SOFT}/include/raptor2 \
+              -I${LOCAL_SOFT}/include/rasqal	
 
-PKG_CFLAGS=-DRASQAL_STATIC -DRAPTOR_STATIC -DREDLAND_STATIC
+PKG_LIBS= \
+-lrdf -lltdl -lrasqal -lsox -lraptor2 -lraptor2 -ldl -lxslt -lgmp -lxml2 -lpcre -lcurl -lssh2 -lidn2 -lunistring -liconv -lgcrypt -lgpg-error -lwsock32 -lws2_32 -llzma -lz -lcfitsio -lzstd -lsbml-static -lcrypt32 -lwldap32
 
-all: clean winlibs
+PKG_CFLAGS=-DRASQAL_STATIC -DRAPTOR_STATIC -DREDLAND_STATIC
 
-winlibs: clean
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R"
+all: clean 
 
 clean:
 	rm -f $(OBJECTS) redland.dll
diff -Nru orig/redland/tools/winlibs.R patched/redland/tools/winlibs.R
--- orig/redland/tools/winlibs.R	2020-10-12 19:18:54.000000000 +0200
+++ patched/redland/tools/winlibs.R	1970-01-01 01:00:00.000000000 +0100
@@ -1,8 +0,0 @@
-# Build against mingw-w64 build of redland
-if(!file.exists("../windows/redland-1.0.17/include/redland.h")){
-  if(getRversion() < "3.3.0") setInternet2()
-  download.file("https://github.com/rwinlib/redland/archive/v1.0.17.zip", "lib.zip", quiet = TRUE)
-  dir.create("../windows", showWarnings = FALSE)
-  unzip("lib.zip", exdir = "../windows")
-  unlink("lib.zip")
-}
