diff -Nru orig/redland/src/Makevars.ucrt patched/redland/src/Makevars.ucrt
--- orig/redland/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/redland/src/Makevars.ucrt	2021-02-17 19:45:42.758761100 +0100
@@ -0,0 +1,13 @@
+
+PKG_CPPFLAGS= -I${LOCAL_SOFT}/include/raptor2 \
+              -I${LOCAL_SOFT}/include/rasqal	
+
+PKG_LIBS= \
+-lrdf -lltdl -lrasqal -lsox -lraptor2 -lraptor2 -ldl -lxslt -lgmp -lxml2 -lpcre -lcurl -lssh2 -lidn2 -lunistring -liconv -lgcrypt -lgpg-error -lwsock32 -lws2_32 -llzma -lz -lcfitsio -lzstd -lsbml-static -lcrypt32 -lwldap32
+
+PKG_CFLAGS=-DRASQAL_STATIC -DRAPTOR_STATIC -DREDLAND_STATIC
+
+all: clean 
+
+clean:
+	rm -f $(OBJECTS) redland.dll
