diff -Nru orig/rego/src/Makevars.win patched/rego/src/Makevars.win
--- orig/rego/src/Makevars.win	2023-08-09 12:02:30.000000000 +0200
+++ patched/rego/src/Makevars.win	2024-02-01 20:15:31.332335900 +0100
@@ -1,5 +1,4 @@
 ## This assume that we can call Rscript to ask Rcpp about its locations
 ## Use the R_HOME indirection to support installations of multiple R version
 PKG_CPPFLAGS = -I../inst/include/armadillo/include -I../inst/include/optim-master/header_only_version
-#PKG_LIBS = `$(R_HOME)/bin/Rscript -e "Rcpp:::LdFlags()"` $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS) -ftrack-macro-expansion=0
-PKG_LIBS = `$(R_HOME)/bin/Rscript -e "Rcpp:::LdFlags()"` $(FLIBS) -ftrack-macro-expansion=0
+PKG_LIBS = `$(R_HOME)/bin/Rscript -e "Rcpp:::LdFlags()"` $(FLIBS) 
