diff -Nru orig/Rfast/src/Makevars.win patched/Rfast/src/Makevars.win
--- orig/Rfast/src/Makevars.win	2025-01-14 19:12:09.000000000 +0100
+++ patched/Rfast/src/Makevars.win	2025-01-29 12:32:35.282972700 +0100
@@ -1,4 +1,4 @@
-PKG_LIBS += $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS) $(SHLIB_OPENMP_CXXFLAGS) `"${R_HOME}/bin/Rscript" -e "RcppParallel::LdFlags()"`
+PKG_LIBS += $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS) $(SHLIB_OPENMP_CXXFLAGS) $(shell "${R_HOME}/bin$(R_ARCH_BIN)/Rscript" -e "RcppParallel::LdFlags()")
 
 PKG_CXXFLAGS= $(SHLIB_OPENMP_CXXFLAGS)
 
