diff -Nru orig/rgdal/src/Makevars.ucrt patched/rgdal/src/Makevars.ucrt
--- orig/rgdal/src/Makevars.ucrt	2023-01-30 09:47:39.000000000 +0100
+++ patched/rgdal/src/Makevars.ucrt	2023-03-17 09:44:53.215004200 +0100
@@ -6,7 +6,7 @@
         -fopenmp -lgdal -larmadillo -lopenblas -lgfortran -lquadmath -lblosc -lkea -lhdf5_cpp -lhdf5 -lgta -lsqlite3 -lmysqlclient -lspatialite -lminizip -lproj -lgeos_c -lgeos  \
         -ljson-c -lnetcdf -lpq -lpgcommon -lpgport \
         -lpoppler -llcms2 -lfreetype -lharfbuzz -lfreetype -llz4 -lpcre2-8 \
-        -lwebp -lcurl -lbcrypt -lidn2 -lunistring -lssh2 -lgcrypt -lgpg-error -lssl \
+        -lwebp -lsharpyuv -lcurl -lbcrypt -lidn2 -lunistring -lssh2 -lgcrypt -lgpg-error -lssl \
         -lhdf5_hl -lhdf5 -lsz -lexpat -lfreexl \
         -lmfhdf -ldf -lportablexdr  -lpcre \
         -lopenjp2 -ljasper -lpng -ljpeg -ltiff -lwebp -lgeotiff -lgif -lxml2 -llzma \
