diff -ruN orig/rGEDI/src/Makevars.win patched/rGEDI/src/Makevars.win
--- orig/rGEDI/src/Makevars.win	2020-07-13 19:52:55.000000000 -0700
+++ patched/rGEDI/src/Makevars.win	2021-01-12 05:52:19.067251100 -0800
@@ -4,15 +4,6 @@
 LIBCLIDAR_ROOT = libclidar
 CMPFIT_ROOT = cmpfit-1.2
 HDF5_VERSION = 1.8.16
-HDF5_LIB = ../windows/hdf5-${HDF5_VERSION}
-ifdef LIB_GSL
-	GSL_INC = ${LIB_GSL}/include
-	GSL_LIB = ${LIB_GSL}/lib
-else
-	LIB_GSL = ../windows/gsl-2.4
-	GSL_INC = ${LIB_GSL}/include
-	GSL_LIB = ${LIB_GSL}/lib${R_ARCH}
-endif
 
 
 PKG_CFLAGS += -DDLLEXPORT -D_USE_MATH_DEFINES -D_WIN32 \
@@ -22,12 +13,9 @@
 ARCH=$(PROCESSOR_ARCHITECTURE)
 
 PKG_CPPFLAGS += -Igedisimulator -I$(HANCOCKTOOLS_ROOT) -I$(CMPFIT_ROOT) \
--I${LIBCLIDAR_ROOT} -I. -I${HDF5_LIB}/include \
--I${GSL_INC}
+-I${LIBCLIDAR_ROOT} -I. 
 
 PKG_LIBS += \
-	-L${HDF5_LIB}/lib-4.9.3${R_ARCH} \
-	-L${GSL_LIB} \
 	-lhdf5 -lhdf5_hl -lz -lm -lgsl
 
 src = $(LIBCLIDAR_ROOT)/libLasProcess.c $(LIBCLIDAR_ROOT)/libLasRead.c \
@@ -45,12 +33,5 @@
 all: $(SHLIB)
 $(SHLIB): $(OBJECTS)
 $(OBJECTS): $(src)
-$(src): winlibs deps
 
-winlibs:
-	"$(R_HOME)/bin/Rscript" "../tools/winlibs.R" $(HDF5_VERSION)
-
-deps:
-	"$(R_HOME)/bin/Rscript" "../tools/deps.R"
-
-.PHONY: all winlibs deps
+.PHONY: all 
