diff -Nru orig/rgeos/src/Makevars.win patched/rgeos/src/Makevars.win
--- orig/rgeos/src/Makevars.win	2020-05-08 14:53:31.000000000 +0200
+++ patched/rgeos/src/Makevars.win	2020-07-02 18:06:10.015443200 +0200
@@ -24,28 +24,17 @@
 
 #.PHONY: all winlibs clean
 
-VERSION = 3.0.4
-RWINLIB = ../windows/gdal3-$(VERSION)
 TARGET = lib$(subst gcc,,$(COMPILED_BY))$(R_ARCH)
 
-PKG_CPPFLAGS =\
-	-I$(RWINLIB)/include/geos-3.8.0
-
 PKG_LIBS = \
-	-L$(RWINLIB)/$(TARGET) \
-	-L$(RWINLIB)/lib$(R_ARCH) \
 	-lgeos_c -lgeos
 
-all: clean winlibs
+all: clean 
 
 CXX_STD = CXX11
 
-winlibs:
-	mkdir -p ../inst
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" --vanilla "../tools/winlibs.R" $(VERSION)
-
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
 
-.PHONY: all winlibs clean
+.PHONY: all clean
 
diff -Nru orig/rgeos/tools/winlibs.R patched/rgeos/tools/winlibs.R
--- orig/rgeos/tools/winlibs.R	2020-05-08 14:49:53.000000000 +0200
+++ patched/rgeos/tools/winlibs.R	1970-01-01 01:00:00.000000000 +0100
@@ -1,16 +0,0 @@
-if(getRversion() < "3.3.0") {
-  stop("Your version of R is too old. This package requires R-3.3.0 or newer on Windows.")
-}
-
-# Download gdal-2.2.* from rwinlib
-VERSION <- commandArgs(TRUE)
-#if(!file.exists(sprintf("../windows/gdal2-%s/include/gdal/gdal.h", VERSION))){
-testfile <- sprintf("../windows/gdal3-%s/include/gdal-%s/gdal.h", VERSION, VERSION)
-if(!file.exists(testfile)){
-  if(getRversion() < "3.3.0") setInternet2()
-#  download.file(sprintf("https://github.com/rwinlib/gdal2/archive/v%s.zip", VERSION), "lib.zip", quiet = TRUE)
-  download.file(sprintf("https://github.com/rwinlib/gdal3/archive/v%s.zip", VERSION), "lib.zip", quiet = TRUE)
-  dir.create("../windows", showWarnings = FALSE)
-  unzip("lib.zip", exdir = "../windows")
-  unlink("lib.zip")
-}
