diff -Nru orig/rgeos/src/Makevars.ucrt patched/rgeos/src/Makevars.ucrt
--- orig/rgeos/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/rgeos/src/Makevars.ucrt	2021-02-17 19:45:42.930644800 +0100
@@ -0,0 +1,40 @@
+#VERSION = 2.2.3
+#COMPILED_BY ?= gcc-4.6.3
+#RWINLIB = ../windows/gdal2-$(VERSION)
+#TARGET = lib$(subst gcc,,$(COMPILED_BY))$(R_ARCH)
+
+#PKG_CPPFLAGS =\
+#	-I$(RWINLIB)/include/geos
+
+#PKG_LIBS = \
+#	-L$(RWINLIB)/$(TARGET) \
+#        -L$(RWINLIB)/lib$(R_ARCH) \
+#	-lgeos_c -lgeos
+
+#all: clean winlibs
+
+#CXX_STD = CXX11
+
+#winlibs:
+#	mkdir -p ../inst
+#	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" --vanilla "../tools/winlibs.R" $(VERSION)
+
+#clean:
+#	rm -f $(SHLIB) $(OBJECTS)
+
+#.PHONY: all winlibs clean
+
+TARGET = lib$(subst gcc,,$(COMPILED_BY))$(R_ARCH)
+
+PKG_LIBS = \
+	-lgeos_c -lgeos
+
+all: clean 
+
+CXX_STD = CXX11
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
+.PHONY: all clean
+
