diff -Nru orig/rgeolocate/src/IP2Location.c patched/rgeolocate/src/IP2Location.c
--- orig/rgeolocate/src/IP2Location.c	2020-05-10 22:22:44.000000000 +0200
+++ patched/rgeolocate/src/IP2Location.c	2021-08-05 12:13:22.249860000 +0200
@@ -34,6 +34,11 @@
 #include "IP2Location.h"
 #include "IP2Loc_DBInterface.h"
 
+
+// inet_pton is available in Windows since Windows 2008
+// the replacement conflicts with UCRT version of inet_pton (the last argument is void *)
+#ifndef _UCRT
+
 #ifdef _WIN64
 #define IS_WINDOWS
 #elif defined _WIN32
@@ -173,6 +178,7 @@
   }
 }
 #endif
+#endif
 
 typedef struct ipv_t
 {
