diff -Nru orig/rgl/src/init.cpp patched/rgl/src/init.cpp
--- orig/rgl/src/init.cpp	2021-11-18 15:51:07.000000000 +0100
+++ patched/rgl/src/init.cpp	2022-05-06 11:45:16.375199800 +0200
@@ -79,7 +79,7 @@
   int stderr_copy = STDERR_FILENO; /* suppress "maybe undefined" warning */
   if (!rglDebug) {
      int devNull = 
-#ifdef windows
+#ifdef _WIN32
         open("nul", O_WRONLY);
 #else
         open("/dev/null", O_WRONLY);
