diff -ruN orig/RGtk2/src/Makevars.win patched/RGtk2/src/Makevars.win
--- orig/RGtk2/src/Makevars.win	2019-02-23 15:36:13.000000000 -0800
+++ patched/RGtk2/src/Makevars.win	2021-01-11 10:50:03.718734500 -0800
@@ -1,3 +1,5 @@
-PKG_CPPFLAGS=-D_R_=1 -DUSE_R=1 -mms-bitfields -I$(GTK_PATH)/include/gtk-2.0 -I$(GTK_PATH)/lib/gtk-2.0/include -I$(GTK_PATH)/include/atk-1.0 -I$(GTK_PATH)/include/cairo -I$(GTK_PATH)/include/pango-1.0 -I$(GTK_PATH)/include/glib-2.0 -I$(GTK_PATH)/lib/glib-2.0/include -I$(GTK_PATH)/include/libxml2  -I$(GTK_PATH)/include/gdk-pixbuf-2.0 -I$(GTK_PATH)/include -I.
+PKG_CPPFLAGS=-D_R_=1 -DUSE_R=1 -mms-bitfields -I$(LOCAL_SOFT)/include/gtk-2.0 -I$(LOCAL_SOFT)/lib/gtk-2.0/include -I$(LOCAL_SOFT)/include/atk-1.0 -I$(LOCAL_SOFT)/include/cairo -I$(LOCAL_SOFT)/include/pango-1.0 -I$(LOCAL_SOFT)/include/glib-2.0 -I$(LOCAL_SOFT)/lib/glib-2.0/include -I$(LOCAL_SOFT)/include/libxml2  -I$(LOCAL_SOFT)/include/gdk-pixbuf-2.0 -I$(LOCAL_SOFT)/include -I.
+
+PKG_LIBS=-ldmoguids -ldxguid -lksguid -lksuser -lstrmiids -luuid -lwbemuuid -lgtk-win32-2.0 -lass -lgdk-win32-2.0 -latk-1.0 -lwinspool -lspoolss -lavcodec -lfribidi -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lpangoft2-1.0 -lcairo -lpangowin32-1.0 -lgio-2.0 -lpixman-1 -lfontconfig -lpango-1.0 -lusp10 -lgdi32 -lgmodule-2.0 -liphlpapi -ldnsapi -lgobject-2.0 -lffi -lexpat -lfreetype_too -lharfbuzz_too -lfreetype -lpng16 -lpng -lharfbuzz -lglib-2.0 -lpcre -lintl -liconv -lws2_32 -ltiff -llzma -ljpeg -lz -lbz2 -lcfitsio -lzstd -lwebpdecoder -lwebp -lsbml-static -lSDL2 -limm32 -lmincore -lcomctl32 -lcomdlg32 -lmsimg32 -lole32 -luuid
+
 
-PKG_LIBS=-L$(GTK_PATH)/lib -lgtk-win32-2.0 -lgdk-win32-2.0 -latk-1.0 -lgdk_pixbuf-2.0 -lpangocairo-1.0 -lpango-1.0 -lcairo -lgobject-2.0 -lglib-2.0 -lgio-2.0 -lintl
diff -ruN orig/RGtk2/src/atkManuals.c patched/RGtk2/src/atkManuals.c
--- orig/RGtk2/src/atkManuals.c	2019-03-03 11:22:47.000000000 -0800
+++ patched/RGtk2/src/atkManuals.c	2021-01-11 10:40:45.720247800 -0800
@@ -22,6 +22,7 @@
 }
 
 /* This function is not provided in the win32 build of the library */
+/*
 #ifdef WIN32
 void
 atk_text_free_ranges (AtkTextRange **ranges)
@@ -43,3 +44,4 @@
     }
 }
 #endif
+*/
diff -ruN orig/RGtk2/src/gobject.c patched/RGtk2/src/gobject.c
--- orig/RGtk2/src/gobject.c	2019-03-03 11:40:29.000000000 -0800
+++ patched/RGtk2/src/gobject.c	2021-01-11 10:40:03.536459000 -0800
@@ -1798,7 +1798,7 @@
   return etype;
 }
 
-
+/*
 GType
 g_io_condition_get_type (void)
 {
@@ -1820,6 +1820,7 @@
 
   return etype;
 }
+*/
 
 /* Pointer comparison for ==.RGtkObject */
 
