diff -Nru orig/rjags/R/windows/zzz.R patched/rjags/R/windows/zzz.R
--- orig/rjags/R/windows/zzz.R	2018-10-09 14:01:47.000000000 +0200
+++ patched/rjags/R/windows/zzz.R	2021-07-28 10:29:10.598805000 +0200
@@ -27,7 +27,7 @@
     ## JAGS-major.x.y, or NULL if no release is found
   
     regkey <- try(readRegistry("SOFTWARE\\JAGS", hive = hive, maxdepth = 2,
-                               view="32-bit"), silent = TRUE)
+                               view="64-bit"), silent = TRUE)
     if (inherits(regkey, "try-error")) {
         return(NULL)
     }
diff -Nru orig/rjags/rjags/src/Makevars.ucrt patched/rjags/rjags/src/Makevars.ucrt
--- orig/rjags/rjags/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/rjags/rjags/src/Makevars.ucrt	2021-07-28 10:28:40.467291600 +0200
@@ -0,0 +1,3 @@
+JAGS_ROOT ?= "/c/progra~1/JAGS/JAGS-4.3.0"
+PKG_CPPFLAGS=-I"$(JAGS_ROOT)/include" 
+PKG_LIBS=-L"$(JAGS_ROOT)/${R_ARCH}/bin" -ljags-4
