diff -Nru orig/RMariaDB/src/Makevars.ucrt patched/RMariaDB/src/Makevars.ucrt
--- orig/RMariaDB/src/Makevars.ucrt	1969-12-31 16:00:00.000000000 -0800
+++ patched/RMariaDB/src/Makevars.ucrt	2021-03-19 04:35:26.386761200 -0700
@@ -0,0 +1,11 @@
+
+CXX_STD=CXX11
+PKG_CPPFLAGS = -I$(R_TOOLS_SOFT)/include/mariadb -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR
+PKG_LIBS= \
+	-lmariadbclient -lbcrypt -lssl -lcrypto -lz -lsecur32 -lshlwapi -lcrypt32 -lgdi32 -lws2_32 -pthread
+
+all: clean
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
