diff -Nru orig/RMySQL/src/Makevars.ucrt patched/RMySQL/src/Makevars.ucrt
--- orig/RMySQL/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/RMySQL/src/Makevars.ucrt	2021-02-17 19:45:26.472244000 +0100
@@ -0,0 +1,10 @@
+
+CXX_STD=CXX11
+PKG_LIBS= \
+	-lmysqlclient -lssl -lcrypto -lz -lsecur32 -lshlwapi \
+        -lcrypt32 -lgdi32 -lws2_32 -pthread
+
+all: clean 
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
diff -Nru orig/RMySQL/src/db-apply.c patched/RMySQL/src/db-apply.c
--- orig/RMySQL/src/db-apply.c	2020-02-03 14:55:17.000000000 +0100
+++ patched/RMySQL/src/db-apply.c	2021-02-17 19:45:26.503490300 +0100
@@ -423,9 +423,7 @@
         return (END_GROUP|BEGIN_GROUP);
       break;
     default:
-      PROBLEM
-      "un-regongnized R/S data type %d", fld_Sclass[jcol]
-    ERROR;
+      error("un-regongnized R/S data type %d", fld_Sclass[jcol]);
     break;
     }
 
