diff -ruN orig/RNetCDF/src/Makevars.win patched/RNetCDF/src/Makevars.win
--- orig/RNetCDF/src/Makevars.win	2020-09-11 18:54:55.000000000 -0700
+++ patched/RNetCDF/src/Makevars.win	2021-01-06 09:06:30.734422000 -0800
@@ -1,4 +1,3 @@
-VERSION=4.4.1.1-dap
 
 ifeq "$(WIN)" "64"
 PKG_SIZE_T = 8
@@ -6,7 +5,7 @@
 PKG_SIZE_T = 4
 endif
 
-PKG_CPPFLAGS = -I../windows/netcdf-${VERSION}/include \
+PKG_CPPFLAGS = \
         -DSIZEOF_INT=4 -DSIZEOF_LONG_LONG=8 -DSIZEOF_SIZE_T=$(PKG_SIZE_T) \
 	-DHAVE_LIBUDUNITS2 -DHAVE_UDUNITS2_H \
 	-DHAVE_NC_RENAME_GRP \
@@ -14,22 +13,16 @@
 	-DHAVE_NC_INQ_VAR_SZIP \
 	-DHAVE_NC_INQ_VAR_ENDIAN \
 
-WINLIBS = ../windows/netcdf-${VERSION}/lib${R_ARCH}
+PKG_LIBS = -lnetcdf -lcurl -lrtmp -lssl -lssh2 -lgcrypt -lcrypto \
+  -lidn2 -lunistring -liconv -lhdf5_hl -lhdf5 -lmfhdf -ldf \
+  -lportablexdr -ljpeg \
+  -ludunits2 -lexpat \
+  -lzstd -lz -lgpg-error -lws2_32 -lcrypt32 -lwldap32
 
-PKG_LIBS = $(WINLIBS)/libnetcdf.a $(WINLIBS)/libcurl.a \
-  $(WINLIBS)/libhdf5_hl.a $(WINLIBS)/libhdf5.a $(WINLIBS)/libszip.a \
-  $(WINLIBS)/libudunits2.a $(WINLIBS)/libexpat.a \
-  -lz -lws2_32 -lcrypt32 -lwldap32
-
-all: clean winlibs
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" ${VERSION}
-	mkdir -p ../inst
-	rm -Rf ../inst/share
-	cp -r ../windows/netcdf-${VERSION}/share ../inst/
+all: clean 
 
 clean:
 	rm -Rf $(SHLIB) $(OBJECTS)
 
-.PHONY: all winlibs clean
+.PHONY: all clean
+
diff -ruN orig/RNetCDF/tools/winlibs.R patched/RNetCDF/tools/winlibs.R
--- orig/RNetCDF/tools/winlibs.R	2020-09-11 18:54:49.000000000 -0700
+++ patched/RNetCDF/tools/winlibs.R	1969-12-31 16:00:00.000000000 -0800
@@ -1,9 +0,0 @@
-# Download netcdf library and headers from rwinlib project
-VERSION <- commandArgs(TRUE)
-if(!file.exists(sprintf("../windows/netcdf-%s/include/netcdf.h", VERSION))){
-  if(getRversion() < "3.3.0") setInternet2()
-  download.file(sprintf("https://github.com/rwinlib/netcdf/archive/v%s.zip", VERSION), "lib.zip", quiet = TRUE)
-  dir.create("../windows", showWarnings = FALSE)
-  unzip("lib.zip", exdir = "../windows")
-  unlink("lib.zip")
-}
