diff -Nru orig/RoBMA/src/Makevars.ucrt patched/RoBMA/src/Makevars.ucrt
--- orig/RoBMA/src/Makevars.ucrt	1969-12-31 16:00:00.000000000 -0800
+++ patched/RoBMA/src/Makevars.ucrt	2021-02-22 06:15:01.105005200 -0800
@@ -0,0 +1,9 @@
+#Set a default value for JAGS_ROOT in case the user hasn't set it, this location was specified by the CRAN-team (according to runjags, where I copied this from)
+JAGS_ROOT = c:/progra~1/JAGS/JAGS-4.3.0
+
+PKG_CPPFLAGS=-I"$(JAGS_ROOT)/include"
+PKG_LIBS=-L"$(JAGS_ROOT)/${R_ARCH}/bin" -ljags-4 -ljrmath-0
+
+# Actual sources and objects for RoBMA
+SOURCES = distributions/DWT1.cc distributions/DWT2.cc distributions/DWN1.cc distributions/DWN2.cc functions/LogWeightedtFun.cc weightedt.cc
+OBJECTS = $(SOURCES:.cc=.o)
