diff -Nru orig/RPostgres/src/Makevars.win patched/RPostgres/src/Makevars.win
--- orig/RPostgres/src/Makevars.win	2019-12-17 22:17:47.000000000 +0100
+++ patched/RPostgres/src/Makevars.win	2020-07-02 18:05:28.650121100 +0200
@@ -1,16 +1,9 @@
-VERSION = 11.1.0
-RWINLIB = ../windows/libpq-$(VERSION)
-PKG_CPPFLAGS = -I$(RWINLIB)/include -DSTRICT_R_HEADERS -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR
-PKG_LIBS = -L$(RWINLIB)/lib${R_ARCH} \
-	-lpq -lssl -lcrypto -lwsock32 -lsecur32 -lws2_32 -lgdi32 -lcrypt32 -lwldap32
+PKG_CPPFLAGS = -DSTRICT_R_HEADERS -DRCPP_DEFAULT_INCLUDE_CALL=false -DRCPP_USING_UTF8_ERROR_STRING -DBOOST_NO_AUTO_PTR
+PKG_LIBS =  \
+	-lpq -lssl -lcrypto -lz -lwsock32 -lsecur32 -lws2_32 -lgdi32 -lcrypt32 -lwldap32
 CXX_STD = CXX11
 
 $(SHLIB):
 
-$(OBJECTS): winlibs
-
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
diff -Nru orig/RPostgres/tools/winlibs.R patched/RPostgres/tools/winlibs.R
--- orig/RPostgres/tools/winlibs.R	2019-12-17 22:17:47.000000000 +0100
+++ patched/RPostgres/tools/winlibs.R	1970-01-01 01:00:00.000000000 +0100
@@ -1,9 +0,0 @@
-# Link against libpq static libraries
-VERSION <- commandArgs(TRUE)
-if(!file.exists(sprintf("../windows/libpq-%s/include/libpq-fe.h", VERSION))){
-  if(getRversion() < "3.3.0") setInternet2()
-  download.file(sprintf("https://github.com/rwinlib/libpq/archive/v%s.zip", VERSION), "lib.zip", quiet = TRUE)
-  dir.create("../windows", showWarnings = FALSE)
-  unzip("lib.zip", exdir = "../windows")
-  unlink("lib.zip")
-}
