diff -Nru orig/RProtoBuf/src/Makevars.win patched/RProtoBuf/src/Makevars.win
--- orig/RProtoBuf/src/Makevars.win	2019-06-28 13:08:39.000000000 +0200
+++ patched/RProtoBuf/src/Makevars.win	2020-07-02 18:05:26.649836400 +0200
@@ -1,17 +1,10 @@
-# ProtoBuf Version
-VERSION = 3.8.0
-RWINLIB = ../windows/protobuf-${VERSION}
-PKG_CPPFLAGS= -I$(RWINLIB)/include
-PKG_LIBS= -L$(RWINLIB)/lib${subst gcc,,${COMPILED_BY}}${R_ARCH} \
-	-L$(RWINLIB)/lib${R_ARCH} -lprotobuf
+PKG_LIBS= -lprotobuf
 
 ## Make this C++11 so that we get better int64 support and more
 CXX_STD=CXX11
 
-all: clean winlibs
+all: clean 
 
 clean:
 	rm -f $(OBJECTS) $(SHLIB)
 
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
