diff -Nru orig/RProtoBuf/src/Makevars.ucrt patched/RProtoBuf/src/Makevars.ucrt
--- orig/RProtoBuf/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/RProtoBuf/src/Makevars.ucrt	2021-02-17 19:45:26.690991100 +0100
@@ -0,0 +1,10 @@
+PKG_LIBS= -lprotobuf
+
+## Make this C++11 so that we get better int64 support and more
+CXX_STD=CXX11
+
+all: clean 
+
+clean:
+	rm -f $(OBJECTS) $(SHLIB)
+
