diff -Nru orig/Rserve/src/Makevars.win patched/Rserve/src/Makevars.win
--- orig/Rserve/src/Makevars.win	2022-11-28 06:36:21.000000000 +0100
+++ patched/Rserve/src/Makevars.win	2023-10-10 09:21:08.987346900 +0200
@@ -1,5 +1,9 @@
 PKG_CPPFLAGS=-DRSERVE_PKG -DWin32 -I. -Iinclude -Iinclude/Win32
-PKG_LIBS=-lssl -lcrypto -lws2_32 -lcrypt32 -lz
+ifeq (,$(shell pkg-config --version 2>/dev/null))
+  PKG_LIBS=-lssl -lcrypto -lws2_32 -lcrypt32 -lz
+else
+  PKG_LIBS=$(shell pkg-config --libs openssl)
+endif
 
 all: $(SHLIB) server
 #	$(MAKE) client
