diff -Nru orig/rstap/src/stan_files/stap_continuous.stan patched/rstap/src/stan_files/stap_continuous.stan
--- orig/rstap/src/stan_files/stap_continuous.stan	2018-12-22 16:13:06.000000000 +0100
+++ patched/rstap/src/stan_files/stap_continuous.stan	2021-02-17 19:45:43.149395000 +0100
@@ -69,7 +69,7 @@
 model {
 #include /model/make_eta.stan
   if( t > 0){
-  #include /model/eta_add_Wb.stan
+#include /model/eta_add_Wb.stan
   }
   if (has_intercept == 1) {
     if ((family == 1 || link == 2) || (family == 4 && link != 5)) eta = eta + gamma[1];
