diff -Nru orig/rsvg/src/Makevars.ucrt patched/rsvg/src/Makevars.ucrt
--- orig/rsvg/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/rsvg/src/Makevars.ucrt	2021-02-17 19:45:43.211891100 +0100
@@ -0,0 +1,21 @@
+
+all: clean 
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
+PKG_CPPFLAGS= \
+        -I${LOCAL_SOFT}/include/librsvg-2.0 \
+        -I${LOCAL_SOFT}/include/glib-2.0  \
+        -I${LOCAL_SOFT}/lib/glib-2.0/include \
+        -I${LOCAL_SOFT}/include/gdk-pixbuf-2.0 \
+        -I${LOCAL_SOFT}/include/cairo  
+
+PKG_LIBS= \
+	-lrsvg-2 -lcroco-0.6 -lxml2 \
+	-lpangocairo-1.0 -lpangoft2-1.0 -lpango-1.0 -lpangowin32-1.0 \
+	-lcairo -lfontconfig -lfreetype -lharfbuzz -lpixman-1 -lgdk_pixbuf-2.0 \
+	-lcairo-gobject -lgio-2.0 -lgobject-2.0 -lgmodule-2.0 -lglib-2.0 \
+	-ltiff -ljpeg -lffi -lpcre -lpng16 -lintl -lbz2 -lz -lzstd -lwebp \
+        -llzma -liconv -lexpat -lfribidi \
+	-lusp10 -liphlpapi -lgdiplus -lgdi32 -lole32 -ldnsapi -lws2_32 -luserenv
