diff -ruN orig/rsvg/src/Makevars.win patched/rsvg/src/Makevars.win
--- orig/rsvg/src/Makevars.win	2020-05-11 07:43:27.000000000 -0700
+++ patched/rsvg/src/Makevars.win	2021-01-07 03:38:50.078775900 -0800
@@ -1,26 +1,21 @@
-VERSION=2.48.4
-RWINLIB=../windows/rsvg-$(VERSION)
 
-all: clean winlibs
-
-winlibs: clean
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
+all: clean 
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
 
 PKG_CPPFLAGS= \
-	-I$(RWINLIB)/include/librsvg-2.0 \
-	-I$(RWINLIB)/include/glib-2.0 \
-	-I$(RWINLIB)/lib${R_ARCH}/glib-2.0/include \
-	-I$(RWINLIB)/include/gdk-pixbuf-2.0 \
-	-I$(RWINLIB)/include/cairo
+        -I${LOCAL_SOFT}/include/librsvg-2.0 \
+        -I${LOCAL_SOFT}/include/glib-2.0  \
+        -I${LOCAL_SOFT}/lib/glib-2.0/include \
+        -I${LOCAL_SOFT}/include/gdk-pixbuf-2.0 \
+        -I${LOCAL_SOFT}/include/cairo  
 
 PKG_LIBS= \
-	-L$(RWINLIB)/lib${R_ARCH} \
-	-lrsvg-2 -lxml2 \
-	-lpangocairo-1.0 -lpango-1.0 -lpangowin32-1.0 \
-	-lcairo -lfreetype -lpixman-1 -lgdk_pixbuf-2.0 \
+	-lrsvg-2 -lcroco-0.6 -lxml2 \
+	-lpangocairo-1.0 -lpangoft2-1.0 -lpango-1.0 -lpangowin32-1.0 \
+	-lcairo -lfontconfig -lfreetype -lharfbuzz -lpixman-1 -lgdk_pixbuf-2.0 \
 	-lcairo-gobject -lgio-2.0 -lgobject-2.0 -lgmodule-2.0 -lglib-2.0 \
-	-lffi -lpcre -lpng16 -lintl -lz -liconv -lfribidi \
+	-ltiff -ljpeg -lffi -lpcre -lpng16 -lintl -lbz2 -lz -lzstd -lwebp \
+        -llzma -liconv -lexpat -lfribidi \
 	-lusp10 -liphlpapi -lgdiplus -lgdi32 -lole32 -ldnsapi -lws2_32 -luserenv
diff -ruN orig/rsvg/tools/winlibs.R patched/rsvg/tools/winlibs.R
--- orig/rsvg/tools/winlibs.R	2020-05-11 07:43:27.000000000 -0700
+++ patched/rsvg/tools/winlibs.R	1969-12-31 16:00:00.000000000 -0800
@@ -1,9 +0,0 @@
-# Build against static libraries from rwinlib
-VERSION <- commandArgs(TRUE)
-if(!file.exists(sprintf("../windows/rsvg-%s/include/librsvg-2.0/librsvg/rsvg.h", VERSION))){
-  if(getRversion() < "3.3.0") setInternet2()
-  download.file(sprintf("https://github.com/rwinlib/rsvg/archive/v%s.zip", VERSION), "lib.zip", quiet = TRUE)
-  dir.create("../windows", showWarnings = FALSE)
-  unzip("lib.zip", exdir = "../windows")
-  unlink("lib.zip")
-}
