diff -ruN orig/rsvg/src/Makevars.win patched/rsvg/src/Makevars.win
--- orig/rsvg/src/Makevars.win	2020-05-11 07:43:27.000000000 -0700
+++ patched/rsvg/src/Makevars.win	2021-01-06 08:45:22.886300600 -0800
@@ -1,23 +1,10 @@
-VERSION=2.48.4
-RWINLIB=../windows/rsvg-$(VERSION)
 
-all: clean winlibs
-
-winlibs: clean
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
+all: clean 
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
 
-PKG_CPPFLAGS= \
-	-I$(RWINLIB)/include/librsvg-2.0 \
-	-I$(RWINLIB)/include/glib-2.0 \
-	-I$(RWINLIB)/lib${R_ARCH}/glib-2.0/include \
-	-I$(RWINLIB)/include/gdk-pixbuf-2.0 \
-	-I$(RWINLIB)/include/cairo
-
 PKG_LIBS= \
-	-L$(RWINLIB)/lib${R_ARCH} \
 	-lrsvg-2 -lxml2 \
 	-lpangocairo-1.0 -lpango-1.0 -lpangowin32-1.0 \
 	-lcairo -lfreetype -lpixman-1 -lgdk_pixbuf-2.0 \
diff -ruN orig/rsvg/tools/winlibs.R patched/rsvg/tools/winlibs.R
--- orig/rsvg/tools/winlibs.R	2020-05-11 07:43:27.000000000 -0700
+++ patched/rsvg/tools/winlibs.R	1969-12-31 16:00:00.000000000 -0800
@@ -1,9 +0,0 @@
-# Build against static libraries from rwinlib
-VERSION <- commandArgs(TRUE)
-if(!file.exists(sprintf("../windows/rsvg-%s/include/librsvg-2.0/librsvg/rsvg.h", VERSION))){
-  if(getRversion() < "3.3.0") setInternet2()
-  download.file(sprintf("https://github.com/rwinlib/rsvg/archive/v%s.zip", VERSION), "lib.zip", quiet = TRUE)
-  dir.create("../windows", showWarnings = FALSE)
-  unzip("lib.zip", exdir = "../windows")
-  unlink("lib.zip")
-}
