diff -ruN orig/Rsymphony/src/Makevars.win patched/Rsymphony/src/Makevars.win
--- orig/Rsymphony/src/Makevars.win	2016-04-12 04:37:35.000000000 -0700
+++ patched/Rsymphony/src/Makevars.win	2021-01-11 09:47:23.797652500 -0800
@@ -1,3 +1,2 @@
-SYMPHONY_LIBS = -lSym -lCgl -lOsiClp -lClp -lOsi -lCoinUtils -lz -fopenmp
-PKG_CPPFLAGS = -I$(SYMPHONY_HOME)/include
-PKG_LIBS = -L$(SYMPHONY_HOME)/lib $(SYMPHONY_LIBS)
+SYMPHONY_LIBS = -lSym -lCgl -lOsiClp -lClp -lOsi -lCoinUtils -lz -lbz2 -fopenmp
+PKG_LIBS = $(SYMPHONY_LIBS)
