diff -ruN orig/runjags/src/Makevars.win patched/runjags/src/Makevars.win
--- orig/runjags/src/Makevars.win	2019-12-16 08:10:04.000000000 -0800
+++ patched/runjags/src/Makevars.win	2020-12-20 12:01:25.189598600 -0800
@@ -16,6 +16,8 @@
 ###
 ######################################################################################################
 
+JAGS_MAJOR_VERSION := 4
+
 ## Conditionally set the JAGS_ROOT variable (if it is not already set) to a location specified by the CRAN team:
 JAGS_ROOT ?= /c/progra~1/JAGS/JAGS-4.3.0
 
