diff -Nru orig/rvg/src/Makevars.ucrt patched/rvg/src/Makevars.ucrt
--- orig/rvg/src/Makevars.ucrt	2022-12-07 23:21:28.000000000 +0100
+++ patched/rvg/src/Makevars.ucrt	2025-03-26 19:14:20.122526600 +0100
@@ -1,2 +1,8 @@
-CRT=-ucrt
-include Makevars.win
+ifeq (,$(shell pkg-config --version 2>/dev/null))
+  PKG_LIBS = -lpng -lz
+else
+  PKG_CPPFLAGS = $(shell pkg-config --cflags libpng)
+  PKG_LIBS = $(shell pkg-config --libs libpng)
+endif
+
+PKG_CPPFLAGS += -DSTRICT_R_HEADERS
