diff -ruN orig/rzmq/src/Makevars.win patched/rzmq/src/Makevars.win
--- orig/rzmq/src/Makevars.win	2020-01-31 02:42:44.000000000 -0800
+++ patched/rzmq/src/Makevars.win	2021-01-11 09:43:12.757682700 -0800
@@ -1,19 +1,13 @@
 CXX_STD = CXX11
-GCC_VERSION = ${subst gcc,,${COMPILED_BY}}
-RWINLIB = ../windows/zeromq-4.2.1
 
 PKG_CPPFLAGS = -DZMQ_STATIC \
-	-I${RWINLIB}/include -I../inst
+	-I../inst
 	
 PKG_LIBS = \
-	-L${RWINLIB}/lib${GCC_VERSION}${R_ARCH} \
-	-L${RWINLIB}/lib${R_ARCH} \
-	-lzmq -lsodium -liphlpapi -lws2_32
+	-lzmq -lsodium -liphlpapi -lws2_32 -fstack-protector
 
-all: clean winlibs
+all: clean 
 
 clean:
 	rm -f $(OBJECTS) $(SHLIB)
 
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R"
diff -ruN orig/rzmq/tools/winlibs.R patched/rzmq/tools/winlibs.R
--- orig/rzmq/tools/winlibs.R	2020-01-31 02:42:44.000000000 -0800
+++ patched/rzmq/tools/winlibs.R	1969-12-31 16:00:00.000000000 -0800
@@ -1,8 +0,0 @@
-# Build against precompiled zeromq libs.
-if(!file.exists("../windows/zeromq-4.2.1/include/zmq.h")){
-  if(getRversion() < "3.3.0") setInternet2()
-  download.file("https://github.com/rwinlib/zeromq/archive/v4.2.1.zip", "lib.zip", quiet = TRUE)
-  dir.create("../windows", showWarnings = FALSE)
-  unzip("lib.zip", exdir = "../windows")
-  unlink("lib.zip")
-}
