diff -Nru orig/rzmq/src/Makevars.ucrt patched/rzmq/src/Makevars.ucrt
--- orig/rzmq/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/rzmq/src/Makevars.ucrt	2021-02-17 19:45:43.385248300 +0100
@@ -0,0 +1,13 @@
+CXX_STD = CXX11
+
+PKG_CPPFLAGS = -DZMQ_STATIC \
+	-I../inst
+	
+PKG_LIBS = \
+	-lzmq -lsodium -liphlpapi -lws2_32 -fstack-protector
+
+all: clean 
+
+clean:
+	rm -f $(OBJECTS) $(SHLIB)
+
