diff -ruN orig/s2/src/Makevars.win patched/s2/src/Makevars.win
--- orig/s2/src/Makevars.win	2020-10-13 09:27:40.000000000 -0700
+++ patched/s2/src/Makevars.win	2020-12-20 11:22:07.323192300 -0800
@@ -1,5 +1,5 @@
-PKG_CPPFLAGS = -DS2_USE_EXACTFLOAT -D_USE_MATH_DEFINES -DNDEBUG -DIS_LITTLE_ENDIAN -I../windows/openssl-1.1.1/include -I../inst/include
-PKG_LIBS = -L../windows/openssl-1.1.1/lib${R_ARCH} -lssl -lcrypto -lcrypt32 -lws2_32
+PKG_CPPFLAGS = -DS2_USE_EXACTFLOAT -D_USE_MATH_DEFINES -DNDEBUG -DIS_LITTLE_ENDIAN -I../inst/include
+PKG_LIBS = -lssl -lcrypto -lws2_32 -lgdi32 -lcrypt32 -lz
 
 CXX_STD = CXX11
 
@@ -127,13 +127,9 @@
      s2/util/math/mathutil.o \
      s2/util/units/length-units.o
 
-all: clean winlibs
-
-winlibs:
-	mkdir -p ../inst
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" --vanilla "../tools/winlibs.R"
+all: clean 
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
 
-.PHONY: all winlibs clean
+.PHONY: all clean
diff -ruN orig/s2/tools/winlibs.R patched/s2/tools/winlibs.R
--- orig/s2/tools/winlibs.R	2020-10-13 09:27:40.000000000 -0700
+++ patched/s2/tools/winlibs.R	1969-12-31 16:00:00.000000000 -0800
@@ -1,8 +0,0 @@
-# Build against openssl libraries that were compiled with the Rtools gcc toolchain.
-if(!file.exists("../windows/openssl-1.1.1/include/openssl/ssl.h")){
-  if(getRversion() < "3.3.0") setInternet2()
-  download.file("https://github.com/rwinlib/openssl/archive/v1.1.1.zip", "lib.zip", quiet = TRUE)
-  dir.create("../windows", showWarnings = FALSE)
-  unzip("lib.zip", exdir = "../windows")
-  unlink("lib.zip")
-}
