diff -Nru orig/sf/src/Makevars.win patched/sf/src/Makevars.win
--- orig/sf/src/Makevars.win	2020-06-17 11:13:58.000000000 +0200
+++ patched/sf/src/Makevars.win	2021-01-22 17:51:12.853333900 +0100
@@ -1,36 +1,15 @@
-VERSION = 3.0.4
-RWINLIB = ../windows/gdal3-$(VERSION)
-TARGET = lib$(subst gcc,,$(COMPILED_BY))$(R_ARCH)
 
 PKG_CPPFLAGS =\
-	-I$(RWINLIB)/include/gdal-3.0.4 \
-	-I$(RWINLIB)/include/geos-3.8.0 \
-	-I$(RWINLIB)/include/proj-6.3.1 \
 	-DHAVE_PROJ_H
 
-PKG_LIBS = \
-	-L$(RWINLIB)/$(TARGET) \
-	-L$(RWINLIB)/lib$(R_ARCH) \
-	-lgdal -lsqlite3 -lspatialite -lproj -lgeos_c -lgeos  \
-	-ljson-c -lnetcdf -lmariadbclient -lpq -lpgport -lpgcommon \
-	-lwebp -lcurl -lssh2 -lssl \
-	-lhdf5_hl -lhdf5 -lexpat -lfreexl -lcfitsio \
-	-lmfhdf -lhdf -lxdr -lpcre \
-	-lopenjp2 -ljasper -lpng -ljpeg -ltiff -lgeotiff -lgif -lxml2 -llzma -lz \
-	-lodbc32 -lodbccp32 -liconv -lpsapi -lwldap32 -lsecur32 -lgdi32 -lnormaliz \
-	-lcrypto -lcrypt32 -lws2_32
+PKG_LIBS =  \
+-lgeos_c -lgeos -lgdal -lxml2 -lopenjp2 -lnetcdf -lmysqlclient -ljson-c -lgta -lfreexl -lexpat -lssl -lpsapi -lgif -lmfhdf -lhdf5_hl -lcrypto -lportablexdr -ldf -lhdf5 -lpng16 -lpng -lpcre -lcurl -lssh2 -lidn2 -lunistring -liconv -lgcrypt -lgpg-error -lws2_32 -ltiff -llzma -ljpeg -lz -lcfitsio -lzstd -lwebpdecoder -lwebp -lsbml-static -lgeotiff -lproj -lsqlite3 -lbz2 -lcrypt32 -lwldap32 -lsecur32
 
-all: clean winlibs
+all: clean 
 
 CXX_STD = CXX11
 
-winlibs:
-	mkdir -p ../inst
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" --vanilla "../tools/winlibs.R" $(VERSION)
-	cp -r $(RWINLIB)/share/gdal ../inst/
-	cp -r $(RWINLIB)/share/proj ../inst/
-
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
 
-.PHONY: all winlibs clean
+.PHONY: all clean
diff -Nru orig/sf/tools/winlibs.R patched/sf/tools/winlibs.R
--- orig/sf/tools/winlibs.R	2020-06-17 11:13:58.000000000 +0200
+++ patched/sf/tools/winlibs.R	1970-01-01 01:00:00.000000000 +0100
@@ -1,15 +0,0 @@
-if(getRversion() < "3.3.0") {
-  stop("Your version of R is too old. This package requires R-3.3.0 or newer on Windows.")
-}
-
-# For details see: https://github.com/rwinlib/gdal3
-VERSION <- commandArgs(TRUE)
-# wrong path: if(!file.exists(sprintf("../windows/gdal3-%s/include/gdal/gdal.h", VERSION))){
-testfile <- sprintf("../windows/gdal3-%s/include/gdal-%s/gdal.h", VERSION, VERSION)
-if(!file.exists(testfile)){  
-  if(getRversion() < "3.3.0") setInternet2()
-  download.file(sprintf("https://github.com/rwinlib/gdal3/archive/v%s.zip", VERSION), "lib.zip", quiet = TRUE)
-  dir.create("../windows", showWarnings = FALSE)
-  unzip("lib.zip", exdir = "../windows")
-  unlink("lib.zip")
-}
