diff -Nru orig/sf/src/Makevars.ucrt patched/sf/src/Makevars.ucrt
--- orig/sf/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/sf/src/Makevars.ucrt	2021-03-15 14:24:22.755888700 +0100
@@ -0,0 +1,19 @@
+
+PKG_CPPFLAGS =\
+	-DHAVE_PROJ_H
+
+PKG_LIBS =  \
+-lgeos_c -lgeos -lgdal -lxml2 -lopenjp2 -lnetcdf -lmysqlclient -ljson-c -lgta -lfreexl -lexpat -lssl -lpsapi -lgif -lmfhdf -lhdf5_hl -lcrypto -lportablexdr -ldf -lhdf5 -lpng16 -lpng -lpcre -lcurl -lssh2 -lidn2 -lunistring -liconv -lgcrypt -lgpg-error -lws2_32 -ltiff -llzma -ljpeg -lz -lcfitsio -lzstd -lwebpdecoder -lwebp -lsbml-static -lgeotiff -lproj -lsqlite3 -lbz2 -lcrypt32 -lwldap32 -lsecur32
+
+all: clean winlibs
+
+winlibs:
+	cp -r $(LOCAL_SOFT)/share/gdal ../inst/
+	cp -r $(LOCAL_SOFT)/share/proj ../inst/
+
+CXX_STD = CXX11
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
+.PHONY: all clean
