diff -Nru orig/ssh/src/Makevars.ucrt patched/ssh/src/Makevars.ucrt
--- orig/ssh/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/ssh/src/Makevars.ucrt	2021-02-17 19:45:43.635248300 +0100
@@ -0,0 +1,13 @@
+
+PKG_CPPFLAGS = -DLIBSSH_STATIC 
+
+PKG_LIBS = \
+	-lssh -lcrypto -lz -lws2_32 -lgdi32 -lcrypt32 \
+	-lgcrypt -lgpg-error -lssl
+
+all: clean 
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
+.PHONY: all clean
