diff -Nru orig/ssh/src/Makevars.win patched/ssh/src/Makevars.win
--- orig/ssh/src/Makevars.win	2019-02-15 18:41:19.000000000 +0100
+++ patched/ssh/src/Makevars.win	2020-07-02 18:05:33.697720200 +0200
@@ -1,19 +1,13 @@
-VERSION = 0.8.6
-RWINLIB = ../windows/libssh-${VERSION}
 
-PKG_CPPFLAGS = -DLIBSSH_STATIC \
-	-I${RWINLIB}/include
+PKG_CPPFLAGS = -DLIBSSH_STATIC 
 
 PKG_LIBS = \
-	-L${RWINLIB}/lib${R_ARCH} \
-	-lssh -lcrypto -lz -lws2_32 -lgdi32 -lcrypt32
+	-lssh -lcrypto -lz -lws2_32 -lgdi32 -lcrypt32 \
+	-lgcrypt -lgpg-error -lssl
 
-all: clean winlibs
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
+all: clean 
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
 
-.PHONY: all winlibs clean
+.PHONY: all clean
