diff -ruN orig/sundialr/src/Makevars.win patched/sundialr/src/Makevars.win
--- orig/sundialr/src/Makevars.win	2020-05-25 12:06:20.000000000 -0700
+++ patched/sundialr/src/Makevars.win	2021-01-07 02:21:52.861702300 -0800
@@ -1,5 +1,5 @@
 CXX=clang++
-PKG_CPPFLAGS= -I../inst/include/ -I./inst/ -DHAVE_CONFIG_H
+PKG_CPPFLAGS= -I../inst/include/ -I./inst/ -DHAVE_CONFIG_H -fcommon
 PKG_LIBS= $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS) -L../inst/ -lsundials_all
 LIBS=-L./ -L../inst/
 
