diff -Nru orig/surveyvoi/src/Makevars.ucrt patched/surveyvoi/src/Makevars.ucrt
--- orig/surveyvoi/src/Makevars.ucrt	2022-05-31 00:46:51.000000000 +0200
+++ patched/surveyvoi/src/Makevars.ucrt	2024-02-13 14:57:55.368437300 +0100
@@ -1,2 +1,19 @@
-CRT=-ucrt
-include Makevars.win
+
+CXX_STD = CXX11
+
+# PKG_CXXFLAGS = -Wno-ignored-attributes              # uncomment for debugging
+
+ifeq (,$(shell pkg-config --version 2>/dev/null))
+  PKG_LIBS = \
+    -lmpfr -lgmp -lgmpxx \
+    $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)
+else
+  PKG_LIBS = $(shell pkg-config --libs mpfr gmp gmpxx)  
+endif
+
+all: clean 
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
+.PHONY: all clean
