diff -Nru orig/sysfonts/src/Makevars.win patched/sysfonts/src/Makevars.win
--- orig/sysfonts/src/Makevars.win	2020-05-08 16:10:07.000000000 +0200
+++ patched/sysfonts/src/Makevars.win	2020-07-02 18:06:25.017580400 +0200
@@ -1,16 +1,10 @@
-PKG_CPPFLAGS = -I./include -I./include/freetype
-PKG_LIBS = -L.${R_ARCH} -lfreetype -lpng -lz
+PKG_CPPFLAGS = -I$(LOCAL_SOFT)/include/freetype2
+PKG_LIBS = -lfreetype -lharfbuzz -lfreetype -lpng -lz -lbz2
 
 
 .PHONY: all deps clean
 
-all: deps $(SHLIB)
-
-deps: sysfonts-dep-win-biarch.tar.gz
-	tar xzf sysfonts-dep-win-biarch.tar.gz
-
-sysfonts-dep-win-biarch.tar.gz:
-	${R_HOME}/bin$(R_ARCH_BIN)/Rscript -e "u = 'https://raw.githubusercontent.com/yixuan/sysfonts-dep/master/sysfonts-dep-win-biarch.tar.gz'; utils::download.file(u, basename(u), mode = 'wb')"
+all: $(SHLIB)
 
 clean:
 	$(RM) *.o
