diff -Nru orig/systemfonts/src/Makevars.win patched/systemfonts/src/Makevars.win
--- orig/systemfonts/src/Makevars.win	2021-01-14 09:58:50.000000000 +0100
+++ patched/systemfonts/src/Makevars.win	2021-02-09 16:31:25.538219700 +0100
@@ -1,22 +1,17 @@
-VERSION = 2.7.4
-RWINLIB = ../windows/harfbuzz-$(VERSION)
 
 CXX_STD = CXX11
 
 PKG_CPPFLAGS = \
-	-I${RWINLIB}/include/freetype2
+	-I${LOCAL_SOFT}/include/freetype2
 
-PKG_LIBS = -L${RWINLIB}/lib${R_ARCH} \
-	-lfreetype -lpng -lz
+PKG_LIBS = \
+	-lfreetype -lharfbuzz -lfreetype -lpng -lbz2 -lz
 
 OBJECTS = caches.o cpp11.o dev_metrics.o font_matching.o font_registry.o \
 	ft_cache.o string_shape.o font_metrics.o font_fallback.o string_metrics.o \
 	emoji.o cache_store.o init.o win/FontManagerWindows.o
 
-all: clean winlibs
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" ${VERSION}
+all: clean
 
 clean:
 	rm -f $(OBJECTS) $(SHLIB)
