diff -ruN orig/systemfonts/src/Makevars.win patched/systemfonts/src/Makevars.win
--- orig/systemfonts/src/Makevars.win	2020-08-26 11:11:13.000000000 -0700
+++ patched/systemfonts/src/Makevars.win	2021-01-07 01:56:20.773322600 -0800
@@ -1,22 +1,16 @@
-VERSION = 2.6.2
-RWINLIB = ../windows/harfbuzz-$(VERSION)
 
 CXX_STD = CXX11
 
-PKG_CPPFLAGS = \
-	-I${RWINLIB}/include/freetype2
+PKG_CPPFLAGS = -I${LOCAL_SOFT}/include/freetype2
 
-PKG_LIBS = -L${RWINLIB}/lib${R_ARCH} \
-	-lfreetype -lpng -lz
+PKG_LIBS = -lfreetype -lharfbuzz -lfreetype -lpng -lbz2 -lz
 	
 OBJECTS = caches.o cpp11.o dev_metrics.o font_matching.o font_registry.o \
 	ft_cache.o string_shape.o font_metrics.o string_metrics.o emoji.o \
 	cache_store.o init.o win/FontManagerWindows.o
 
-all: clean winlibs
+all: clean
 
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" ${VERSION}
 
 clean:
 	rm -f $(OBJECTS) $(SHLIB)
