diff -Nru orig/systemfonts/src/Makevars.ucrt patched/systemfonts/src/Makevars.ucrt
--- orig/systemfonts/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/systemfonts/src/Makevars.ucrt	2021-02-17 19:45:43.869628100 +0100
@@ -0,0 +1,17 @@
+
+CXX_STD = CXX11
+
+PKG_CPPFLAGS = \
+	-I${LOCAL_SOFT}/include/freetype2
+
+PKG_LIBS = \
+	-lfreetype -lharfbuzz -lfreetype -lpng -lbz2 -lz
+
+OBJECTS = caches.o cpp11.o dev_metrics.o font_matching.o font_registry.o \
+	ft_cache.o string_shape.o font_metrics.o font_fallback.o string_metrics.o \
+	emoji.o cache_store.o init.o win/FontManagerWindows.o
+
+all: clean
+
+clean:
+	rm -f $(OBJECTS) $(SHLIB)
