diff -Nru orig/systemfonts/src/Makevars.win patched/systemfonts/src/Makevars.win
--- orig/systemfonts/src/Makevars.win	2019-11-04 15:10:17.000000000 +0100
+++ patched/systemfonts/src/Makevars.win	2020-07-02 18:06:24.454992300 +0200
@@ -1,20 +1,13 @@
-VERSION = 1.16.0
-RWINLIB = ../windows/cairo-${VERSION}
 
 CXX_STD = CXX11
 
-PKG_CPPFLAGS = -I${RWINLIB}/include/freetype2 \
-	-DSTRICT_R_HEADERS
+PKG_CPPFLAGS = -DSTRICT_R_HEADERS -I$(LOCAL_SOFT)/include/freetype2
 
-PKG_LIBS = -L${RWINLIB}/lib${R_ARCH} \
-	-lfreetype -lharfbuzz -lfreetype -lpng -lbz2 -lz
+PKG_LIBS = -lfreetype -lharfbuzz -lfreetype -lpng -lbz2 -lz
 
 OBJECTS = systemfonts.o ft_cache.o string_shape.o font_metrics.o string_metrics.o emoji.o init.o win/FontManagerWindows.o
 
-all: clean winlibs
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" ${VERSION}
+all: clean 
 
 clean:
 	rm -f $(OBJECTS) $(SHLIB)
