diff -Nru orig/terra/src/Makevars.ucrt patched/terra/src/Makevars.ucrt
--- orig/terra/src/Makevars.ucrt	2021-07-23 15:58:39.000000000 +0200
+++ patched/terra/src/Makevars.ucrt	2021-11-04 12:45:05.685208600 +0100
@@ -1,2 +1,24 @@
-CRT=-ucrt
-include Makevars.win
+TARGET = lib$(subst gcc,,$(COMPILED_BY))$(R_ARCH)
+
+PKG_LIBS = \
+        -lgdal -lgta -lsqlite3 -lmysqlclient -lspatialite -lproj -lgeos_c -lgeos  \
+        -ljson-c -lnetcdf -lpq \
+        -lwebp -lcurl -lidn2 -lunistring -lssh2 -lgcrypt -lgpg-error -lssl \
+        -lhdf5_hl -lhdf5 -lexpat -lfreexl \
+        -lmfhdf -ldf -lportablexdr  -lpcre \
+        -lopenjp2 -ljasper -lpng -ljpeg -ltiff -lwebp -lgeotiff -lgif -lxml2 -llzma \
+        -lodbc32 -lodbccp32 -liconv -lpsapi -lwldap32 -lsecur32 -lgdi32 -lnormaliz \
+        -lbz2 -lzstd -lz -lcrypto -lcrypt32 -lcfitsio -lws2_32
+
+CXX_STD = CXX11
+
+all: clean winlibs
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
+winlibs:
+	cp -r $(R_TOOLS_SOFT)/share/gdal ../inst/
+	cp -r $(R_TOOLS_SOFT)/share/proj ../inst/
+
+.PHONY: all winlibs clean
