diff -Nru orig/terra/src/Makevars.ucrt patched/terra/src/Makevars.ucrt
--- orig/terra/src/Makevars.ucrt	2022-08-07 09:53:08.000000000 +0200
+++ patched/terra/src/Makevars.ucrt	2022-08-15 12:36:42.088810600 +0200
@@ -1,7 +1,7 @@
 TARGET = lib$(subst gcc,,$(COMPILED_BY))$(R_ARCH)
 
 PKG_LIBS = \
-        -lgdal -lgta -lsqlite3 -lmysqlclient -lspatialite -lproj -lgeos_c -lgeos  \
+        -lgdal -lkea -lhdf5_cpp -lhdf5 -lpoppler -llcms2 -lfreetype -lharfbuzz -lfreetype -llz4 -lpcre2-8 -lgta -lsqlite3 -lmysqlclient -lspatialite -lproj -lgeos_c -lgeos  \
         -ljson-c -lnetcdf -lpq \
         -lpoppler -llcms2 -lfreetype -lharfbuzz -lfreetype -llz4 -lpcre2-8 \
         -lwebp -lcurl -lidn2 -lunistring -lssh2 -lgcrypt -lgpg-error -lssl \
