diff -Nru orig/terra/src/Makevars.ucrt patched/terra/src/Makevars.ucrt
--- orig/terra/src/Makevars.ucrt	2022-02-15 04:28:48.000000000 +0100
+++ patched/terra/src/Makevars.ucrt	2022-05-06 17:02:15.158785600 +0200
@@ -1,7 +1,7 @@
 TARGET = lib$(subst gcc,,$(COMPILED_BY))$(R_ARCH)
 
 PKG_LIBS = \
-        -lgdal -lgta -lsqlite3 -lmysqlclient -lspatialite -lproj -lgeos_c -lgeos  \
+        -lgdal -lpoppler -llcms2 -lfreetype -lharfbuzz -lfreetype -llz4 -lpcre2-8 -lgta -lsqlite3 -lmysqlclient -lspatialite -lproj -lgeos_c -lgeos  \
         -ljson-c -lnetcdf -lpq \
         -lwebp -lcurl -lidn2 -lunistring -lssh2 -lgcrypt -lgpg-error -lssl \
         -lhdf5_hl -lhdf5 -lexpat -lfreexl \
