diff -Nru orig/terra/src/Makevars.ucrt patched/terra/src/Makevars.ucrt
--- orig/terra/src/Makevars.ucrt	2022-09-10 07:50:26.000000000 -0700
+++ patched/terra/src/Makevars.ucrt	2022-09-16 00:15:48.052827400 -0700
@@ -1,7 +1,8 @@
 TARGET = lib$(subst gcc,,$(COMPILED_BY))$(R_ARCH)
 
+PKG_CXXFLAGS = -Wa,-mbig-obj 
 PKG_LIBS = \
-        -lgdal -lgta -lsqlite3 -lmysqlclient -lspatialite -lproj -lgeos_c -lgeos  \
+        -lgdal -lblosc -lkea -lhdf5_cpp -lhdf5 -lpoppler -llcms2 -lfreetype -lharfbuzz -lfreetype -llz4 -lpcre2-8 -lgta -lsqlite3 -lmysqlclient -lspatialite -lminizip -lproj -lgeos_c -lgeos  \
         -ljson-c -lnetcdf -lpq \
         -lpoppler -llcms2 -lfreetype -lharfbuzz -lfreetype -llz4 -lpcre2-8 \
         -lwebp -lcurl -lidn2 -lunistring -lssh2 -lgcrypt -lgpg-error -lssl \
