diff -Nru orig/textshaping/src/Makevars.ucrt patched/textshaping/src/Makevars.ucrt
--- orig/textshaping/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/textshaping/src/Makevars.ucrt	2021-02-17 19:56:49.371558600 +0100
@@ -0,0 +1,15 @@
+
+CXX_STD = CXX11
+
+PKG_CPPFLAGS = \
+	-I${LOCAL_SOFT}/include/harfbuzz \
+	-I${LOCAL_SOFT}/include/fribidi \
+	-I${LOCAL_SOFT}/include/freetype2
+
+PKG_LIBS = \
+	-lharfbuzz -lfreetype -lfribidi -lpng -lz -lbz2
+
+all: clean 
+
+clean:
+	rm -f $(OBJECTS) $(SHLIB)
