diff -ruN orig/textshaping/src/Makevars.win patched/textshaping/src/Makevars.win
--- orig/textshaping/src/Makevars.win	2020-08-26 12:24:31.000000000 -0700
+++ patched/textshaping/src/Makevars.win	2021-01-12 14:38:09.022516300 -0800
@@ -1,20 +1,15 @@
-VERSION = 2.6.2
-RWINLIB = ../windows/harfbuzz-$(VERSION)
 
 CXX_STD = CXX11
 
 PKG_CPPFLAGS = \
-	-I${RWINLIB}/include/harfbuzz \
-	-I${RWINLIB}/include/fribidi \
-	-I${RWINLIB}/include/freetype2
+        -I$(LOCAL_SOFT)/include/freetype2 \
+        -I$(LOCAL_SOFT)/include/harfbuzz \
+        -I$(LOCAL_SOFT)/include/fribidi
 
-PKG_LIBS = -L${RWINLIB}/lib${R_ARCH} \
-	-lharfbuzz -lfreetype -lfribidi -lpng -lz
+PKG_LIBS = \
+	-lharfbuzz -lfreetype -lfribidi -lpng -lz -lbz2
 
-all: clean winlibs
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" ${VERSION}
+all: clean
 
 clean:
 	rm -f $(OBJECTS) $(SHLIB)
