diff -ruN orig/tiff/src/Makevars.win patched/tiff/src/Makevars.win
--- orig/tiff/src/Makevars.win	2020-10-21 20:50:21.000000000 -0700
+++ patched/tiff/src/Makevars.win	2021-01-07 01:13:55.682065500 -0800
@@ -1,11 +1,6 @@
-RWINLIB = ../windows/libtiff-4.1.0/mingw$(WIN)
-PKG_CPPFLAGS = -I$(RWINLIB)/include
-PKG_LIBS = -L$(RWINLIB)/lib -ltiff -ljpeg -lz
+PKG_LIBS = -ltiff -ljpeg -lz -lzstd -lwebp -llzma 
 
-all: clean winlibs
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R"
+all: clean 
 
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
