diff -Nru orig/tiff/src/Makevars.ucrt patched/tiff/src/Makevars.ucrt
--- orig/tiff/src/Makevars.ucrt	2022-01-31 05:49:39.000000000 +0100
+++ patched/tiff/src/Makevars.ucrt	2023-10-09 13:39:02.039104400 +0200
@@ -1,4 +1,9 @@
-PKG_LIBS = -ltiff -ljpeg -lz -lzstd -lwebp -llzma 
+ifeq (,$(shell pkg-config --version 2>/dev/null))
+   LIBSHARPYUV = $(or $(and $(wildcard $(R_TOOLS_SOFT)/lib/libsharpyuv.a),-lsharpyuv),)
+   PKG_LIBS = -ltiff -ljpeg -lz -lzstd -lwebp $(LIBSHARPYUV) -llzma
+else
+   PKG_LIBS = $(shell pkg-config --libs libtiff-4)
+endif
 
 all: clean 
 
