diff -Nru orig/uFTIR/src/Makevars.ucrt patched/uFTIR/src/Makevars.ucrt
--- orig/uFTIR/src/Makevars.ucrt	1970-01-01 01:00:00.000000000 +0100
+++ patched/uFTIR/src/Makevars.ucrt	2021-02-17 19:45:44.244632300 +0100
@@ -0,0 +1,29 @@
+PKG_CPPFLAGS =\
+
+PKG_LIBS = \
+        -lgdal -lgta -lsqlite3 -lmysqlclient -lspatialite -lproj -lgeos_c -lgeos  \
+        -ljson-c -lnetcdf -lpq \
+        -lwebp -lcurl -lidn2 -lunistring -lssh2 -lgcrypt -lgpg-error -lssl \
+        -lhdf5_hl -lhdf5 -lexpat -lfreexl \
+        -lmfhdf -ldf -lportablexdr  -lpcre \
+        -lopenjp2 -ljasper -lpng -ljpeg -ltiff -lwebp -lgeotiff -lgif -lxml2 -llzma \
+        -lodbc32 -lodbccp32 -liconv -lpsapi -lwldap32 -lsecur32 -lgdi32 -lnormaliz \
+        -lbz2 -lzstd -lz -lcrypto -lcrypt32 -lcfitsio -lws2_32
+
+
+PKG_LIBS0 = \
+	-lgdal -lsqlite3 -lspatialite -lproj -lgeos_c -lgeos  \
+	-ljson-c -lnetcdf -lmariadbclient -lpq -lintl -lwebp -lcurl -lssh2 -lssl -lcrypto \
+	-lkea -lhdf5_cpp -lhdf5_hl -lhdf5 -lexpat -lfreexl -lcfitsio \
+	-lmfhdf -ldf -lxdr \
+	-lopenjp2 -ljasper -lpng16 -ljpeg -ltiff -lgeotiff -lgif -lxml2 -llzma -lszip -lz \
+	-lodbc32 -lodbccp32 -liconv -lpsapi -lws2_32 -lcrypt32 -lwldap32 -lsecur32 -lgdi32
+
+all: clean 
+
+CXX_STD = CXX11
+
+clean:
+	rm -f $(SHLIB) $(OBJECTS)
+
+.PHONY: all clean
