diff -Nru orig/units/src/Makevars.win patched/units/src/Makevars.win
--- orig/units/src/Makevars.win	2020-03-11 10:05:36.000000000 -0700
+++ patched/units/src/Makevars.win	2020-12-17 07:41:29.168782300 -0800
@@ -1,16 +1,9 @@
-PKG_CPPFLAGS = -DUDUNITS2_DIR=0 -I../windows/udunits-2.2.20/include
-PKG_LIBS = -L../windows/udunits-2.2.20/lib${R_ARCH} \
-	-ludunits2 -lexpat
+PKG_CPPFLAGS = -DUDUNITS2_DIR=0 
+PKG_LIBS = -ludunits2 -lexpat
 
-all: clean winlibs
-
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R"
-	mkdir -p ../inst
-	rm -Rf ../inst/share
-	cp -r ../windows/udunits-2.2.20/share ../inst/
+all: clean
 
 clean:
 	rm -Rf $(SHLIB) $(OBJECTS)
 
-.PHONY: all winlibs clean
+.PHONY: all clean
diff -Nru orig/units/tools/winlibs.R patched/units/tools/winlibs.R
--- orig/units/tools/winlibs.R	2020-03-11 10:05:36.000000000 -0700
+++ patched/units/tools/winlibs.R	1969-12-31 16:00:00.000000000 -0800
@@ -1,8 +0,0 @@
-# Build against mingw-w64 build of udunits
-if(!file.exists("../windows/udunits-2.2.20/include/udunits.h")){
-  if(getRversion() < "3.3.0") setInternet2()
-  download.file("https://github.com/rwinlib/udunits/archive/v2.2.20.zip", "lib.zip", quiet = TRUE)
-  dir.create("../windows", showWarnings = FALSE)
-  unzip("lib.zip", exdir = "../windows")
-  unlink("lib.zip")
-}
