diff -ruN orig/websocket/src/Makevars.win patched/websocket/src/Makevars.win
--- orig/websocket/src/Makevars.win	2020-07-03 15:29:12.000000000 -0700
+++ patched/websocket/src/Makevars.win	2021-01-07 02:53:01.739702500 -0800
@@ -1,11 +1,10 @@
-OPENSSL_VERSION=1.1.1g
-
 PKG_CFLAGS = -DSTRICT_R_HEADERS
 PKG_CXXFLAGS = -DSTRICT_R_HEADERS
 
 # Some settings from https://github.com/zaphoyd/websocketpp/issues/478
-PKG_CPPFLAGS=-I./lib -I../windows/openssl-$(OPENSSL_VERSION)/include -D_WEBSOCKETPP_CPP11_THREAD_
-PKG_LIBS= -L../windows/openssl-$(OPENSSL_VERSION)/lib${R_ARCH} -lssl -lcrypto -lws2_32 -lgdi32 -lcrypt32
+
+PKG_CPPFLAGS=-I./lib -D_WEBSOCKETPP_CPP11_THREAD_ 
+PKG_LIBS= -lssl -lcrypto -lz -lws2_32 -lgdi32
 
 CXX_STD=CXX11
 
@@ -13,16 +12,10 @@
 # Uncomment to enable thread assertions
 # PKG_CPPFLAGS += -DDEBUG_THREAD -UNDEBUG
 
-.PHONY: all winlibs
+.PHONY: all 
 
 all: $(SHLIB)
 
-# Need to make sure that openssl is fully downloaded before trying to compile
-# websocket.cpp. See https://github.com/rstudio/websocket/issues/43
-$(OBJECTS): winlibs
-
 clean:
 	rm -f $(SHLIB) $(OBJECTS)
 
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(OPENSSL_VERSION)
