diff -Nru orig/XBRL/src/Makevars.win patched/XBRL/src/Makevars.win
--- orig/XBRL/src/Makevars.win	2017-03-27 19:18:01.000000000 +0200
+++ patched/XBRL/src/Makevars.win	2020-07-02 18:06:54.678064700 +0200
@@ -1,5 +1,5 @@
 ## Use the R_HOME indirection to support installations of multiple R version
 
-PKG_CXXFLAGS=-I${LIB_XML}/include/libxml2 -I${LIB_XML}/include -DLIBXML_STATIC
+PKG_CXXFLAGS=-I${LOCAL_SOFT}/include/libxml2 -DLIBXML_STATIC
 
-PKG_LIBS = $(shell "${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" -e "Rcpp:::LdFlags()") -L${LIB_XML}/lib -lxml2 -liconv -lz -lws2_32
+PKG_LIBS = $(shell "${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" -e "Rcpp:::LdFlags()") -lxml2 -llzma -liconv -lz -lws2_32
