diff -ruN orig/xslt/src/Makevars.win patched/xslt/src/Makevars.win
--- orig/xslt/src/Makevars.win	2020-04-01 04:59:35.000000000 -0700
+++ patched/xslt/src/Makevars.win	2021-01-12 13:34:32.939095600 -0800
@@ -1,16 +1,11 @@
-VERSION=2.9.8
-RWINLIB=../windows/libxml2-$(VERSION)
-PKG_CPPFLAGS=-I$(RWINLIB)/include -I$(RWINLIB)/include/libxml2 \
+PKG_CPPFLAGS=-I$(LOCAL_SOFT)/include/libxml2 \
 	-DLIBXML_STATIC -DSTRICT_R_HEADERS
 
-PKG_LIBS=-L$(RWINLIB)/lib${R_ARCH} -lexslt -lxslt -lxml2 -llzma -liconv -lz -lws2_32
+PKG_LIBS=-lexslt -lxslt -lxml2 -llzma -liconv -lz -lws2_32
 
-all: clean winlibs
+all: clean
 
 clean:
 	rm -f $(OBJECTS) $(SHLIB)
 
-winlibs:
-	"${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "../tools/winlibs.R" $(VERSION)
-
-.PHONY: all winlibs clean
+.PHONY: all clean
