diff -Nru gcc-13.2.0-orig/libgomp/configure gcc-13.2.0-patched/libgomp/configure
--- gcc-13.2.0-orig/libgomp/configure	2023-07-27 04:17:14.000000000 -0400
+++ gcc-13.2.0-patched/libgomp/configure	2024-02-13 09:38:20.374150994 -0500
@@ -11829,6 +11829,7 @@
       FC=no
     fi ;;
 esac
+FC="$GFORTRAN"
 ac_ext=${ac_fc_srcext-f}
 ac_compile='$FC -c $FCFLAGS $ac_fcflags_srcext conftest.$ac_ext >&5'
 ac_link='$FC -o conftest$ac_exeext $FCFLAGS $LDFLAGS $ac_fcflags_srcext conftest.$ac_ext $LIBS >&5'
diff -Nru gcc-13.2.0-orig/libgomp/configure.ac gcc-13.2.0-patched/libgomp/configure.ac
--- gcc-13.2.0-orig/libgomp/configure.ac	2023-07-27 04:17:14.000000000 -0400
+++ gcc-13.2.0-patched/libgomp/configure.ac	2024-02-13 09:38:20.374150994 -0500
@@ -170,6 +170,7 @@
       FC=no
     fi ;;
 esac
+FC="$GFORTRAN"
 AC_PROG_FC(gfortran)
 FCFLAGS="$FCFLAGS -Wall -L../libgfortran"
 
