diff -Nru orig/Rhtslib/R/zzz.R patched/Rhtslib/R/zzz.R
--- orig/Rhtslib/R/zzz.R	2020-10-27 18:46:21.000000000 +0100
+++ patched/Rhtslib/R/zzz.R	2021-03-22 23:13:26.932655200 +0100
@@ -29,8 +29,7 @@
             LOCAL_SOFT <- .getRconfig("LOCAL_SOFT", default="C:/extsoft")
             ## See how PKG_LIBS is defined in Rhtslib/src/Makevars.win
             ## and make sure to produce the same value here.
-            libs <- c("curl", "rtmp", "ssl", "ssh2", "crypto",
-                      "gdi32", "z", "ws2_32", "wldap32", "winmm")
+	    libs <- c("sbml-static","winmm","wldap32","gdi32","bz2","curl","rtmp","ssl","ssh2","idn2","unistring","charset","intl","iconv","gcrypt","crypto","gpg-error", "ws2_32","lzma","z","zstd", "mincore")
             if (r_arch == "i386")
                 libs <- c(libs, "idn")
             libs <- paste(sprintf("-l%s", libs), collapse=" ")
Binary files orig/Rhtslib/src/.Makevars.win.swp and patched/Rhtslib/src/.Makevars.win.swp differ
diff -Nru orig/Rhtslib/src/Makevars.win patched/Rhtslib/src/Makevars.win
--- orig/Rhtslib/src/Makevars.win	2020-10-27 18:46:21.000000000 +0100
+++ patched/Rhtslib/src/Makevars.win	2021-03-22 23:05:25.584037100 +0100
@@ -9,10 +9,8 @@
 
 ## Linker options. Make sure the Rhtslib::pkgconfig() function (defined in
 ## R/zzz.R) produces the same PKG_LIBS value.
-PKG_LIBS+=-L${LOCAL_SOFT}/lib${R_ARCH} -lcurl -lrtmp -lssl -lssh2 -lcrypto -lgdi32 -lz -lws2_32 -lwldap32 -lwinmm
-ifeq ("${R_ARCH}","/i386")
-	PKG_LIBS+=-lidn
-endif
+
+PKG_LIBS+= -lsbml-static -lwinmm -lwldap32 -lgdi32 -lbz2 -lcurl -lrtmp -lssl -lssh2 -lidn2 -lunistring -lcharset -lintl -liconv -lgcrypt -lcrypto -lgpg-error -lws2_32 -llzma -lz -lzstd -lmincore
 
 populate-usrlib-dir: htslib mk-usrlib-dir
 	cd "${HTSLIB_SOURCE_DIR}" && cp libhts.a "${USRLIB_DIR}"
diff -Nru orig/Rhtslib/src/htslib-1.7/Makefile.Rhtslib.win patched/Rhtslib/src/htslib-1.7/Makefile.Rhtslib.win
--- orig/Rhtslib/src/htslib-1.7/Makefile.Rhtslib.win	2020-10-27 18:46:21.000000000 +0100
+++ patched/Rhtslib/src/htslib-1.7/Makefile.Rhtslib.win	2021-03-22 23:09:59.592413700 +0100
@@ -35,7 +35,7 @@
 #RANLIB = ranlib
 
 # Default libraries to link if configure is not used
-htslib_default_libs = -L${LOCAL_SOFT}/lib${R_ARCH} -lcurl -lrtmp -lssl -lssh2 -lcrypto -lgdi32 -lz -lws2_32 -lwldap32 -lwinmm
+htslib_default_libs = -lsbml-static -lwinmm -lwldap32 -lgdi32 -lbz2 -lcurl -lrtmp -lssl -lssh2 -lidn2 -lunistring -lcharset -lintl -liconv -lgcrypt -lcrypto -lgpg-error -lws2_32 -llzma -lz -lzstd -lmincore 
 ifeq ("${R_ARCH}","/i386")
 	htslib_default_libs += -lidn
 endif
diff -Nru orig/Rhtslib/src/htslib-1.7/config.h patched/Rhtslib/src/htslib-1.7/config.h
--- orig/Rhtslib/src/htslib-1.7/config.h	2020-10-27 21:14:07.000000000 +0100
+++ patched/Rhtslib/src/htslib-1.7/config.h	2021-03-22 22:57:46.246716400 +0100
@@ -3,5 +3,5 @@
 #define HAVE_LIBLZMA 1
 #define HAVE_LZMA_H 1
 #define HAVE_FSEEKO 1
-#define HAVE_DRAND48 1
+/* #define HAVE_DRAND48 1 */
 #define HAVE_LIBCURL 1
