Index: src/gnuwin32/Makefile
===================================================================
--- src/gnuwin32/Makefile	(revision 78280)
+++ src/gnuwin32/Makefile	(working copy)
@@ -68,9 +68,9 @@
 
 
 R-DLLFLAGS = -mwindows
-LIBEXTRAS = -L"$(EXT_LIBS)"/lib$(R_ARCH) -lpcre2-8 -lz -lbz2 -llzma
+LIBEXTRAS = -L"$(EXT_LIBS)"/lib -lpcre2-8 -lz -lbz2 -llzma
 ifdef USE_ICU
-LIBEXTRAS += -L"$(ICU_PATH)"/lib$(R_ARCH) $(ICU_LIBS)
+LIBEXTRAS += -L"$(ICU_PATH)"/lib $(ICU_LIBS)
 endif
 
 dllversion-RESFLAGS = -I../include
