Index: src/library/utils/src/Makefile.win
===================================================================
--- src/library/utils/src/Makefile.win	(revision 78280)
+++ src/library/utils/src/Makefile.win	(working copy)
@@ -21,7 +21,7 @@
 SHLIB = $(pkg).dll
 RESFLAGS = -I $(R_HOME)/include
 
-XZ_CPPFLAGS = -DLZMA_API_STATIC -I$(LOCAL_SOFT)/include
+XZ_CPPFLAGS = -I$(LOCAL_SOFT)/include
 
 PKG_CPPFLAGS = -I$(R_HOME)/src/include -DHAVE_CONFIG_H \
   -I$(R_HOME)/src/main -I$(R_HOME)/src/gnuwin32 -I$(R_HOME)/src/extra \
Index: src/main/Makefile.win
===================================================================
--- src/main/Makefile.win	(revision 78280)
+++ src/main/Makefile.win	(working copy)
@@ -62,11 +62,11 @@
 serialize-CPPFLAGS = -I../extra/xdr
 sysutils-CPPFLAGS = -I../gnuwin32
 
-connections-CPPFLAGS = -DLZMA_API_STATIC -I"$(EXT_LIBS)"/include
+connections-CPPFLAGS = -I"$(EXT_LIBS)"/include
 dounzip-CPPFLAGS = -I$(EXT_LIBS)/include
-grep-CPPFLAGS = -DPCRE2_STATIC -I../extra -I../gnuwin32 -I"$(EXT_LIBS)"/include
-platform-CPPFLAGS += -DPCRE2_STATIC -DLZMA_API_STATIC -I"$(EXT_LIBS)"/include
-util-CPPFLAGS = -DLZMA_API_STATIC -I"$(EXT_LIBS)"/include
+grep-CPPFLAGS = -I../extra -I../gnuwin32 -I"$(EXT_LIBS)"/include
+platform-CPPFLAGS += -I"$(EXT_LIBS)"/include
+util-CPPFLAGS = -I"$(EXT_LIBS)"/include
 
 ifdef USE_ICU
 platform-CPPFLAGS += -DUSE_ICU -I"$(ICU_PATH)"/include
Index: src/modules/internet/Makefile.win
===================================================================
--- src/modules/internet/Makefile.win	(revision 78280)
+++ src/modules/internet/Makefile.win	(working copy)
@@ -11,7 +11,7 @@
 EXTRA_LIBS = -lwininet -lws2_32
 
 ifdef USE_LIBCURL
-libcurl-CPPFLAGS += -DHAVE_CURL_CURL_H -DHAVE_LIBCURL -DCURL_STATICLIB -I"$(CURL_PATH)"/include
+libcurl-CPPFLAGS += -DHAVE_CURL_CURL_H -DHAVE_LIBCURL -I"$(CURL_PATH)"/include
 EXTRA_LIBS += -L"$(CURL_PATH)"/lib$(R_ARCH) $(CURL_LIBS)
 endif
 
