diff -Nru orig/libbacktrace/configure new/libbacktrace/configure
--- orig/libbacktrace/configure	2020-07-23 02:35:54.764796795 -0400
+++ new/libbacktrace/configure	2020-08-12 13:10:30.900715979 -0400
@@ -12992,6 +12992,7 @@
 
   fi
 fi
+have_mmap=no
 if test "$have_mmap" = "no"; then
   VIEW_FILE=read.lo
   ALLOC_FILE=alloc.lo
diff -Nru orig/libbacktrace/configure.ac new/libbacktrace/configure.ac
--- orig/libbacktrace/configure.ac	2020-07-23 02:35:18.700399491 -0400
+++ new/libbacktrace/configure.ac	2020-08-12 13:09:53.036910230 -0400
@@ -301,6 +301,8 @@
     AC_CHECK_FUNC(mmap, [have_mmap=yes], [have_mmap=no])
   fi
 fi
+# with mingw, mmap.h may be present via win32-mmap
+have_mmap=no
 if test "$have_mmap" = "no"; then
   VIEW_FILE=read.lo
   ALLOC_FILE=alloc.lo
diff -Nru orig/libsanitizer/configure new/libsanitizer/configure
--- orig/libsanitizer/configure	2020-07-23 02:35:54.756796706 -0400
+++ new/libsanitizer/configure	2020-08-12 13:09:05.245163579 -0400
@@ -16632,6 +16632,7 @@
 
   fi
 fi
+have_mmap=no
 if test "$have_mmap" = "no"; then
   VIEW_FILE=read.lo
   ALLOC_FILE=alloc.lo
diff -Nru orig/libsanitizer/configure.ac new/libsanitizer/configure.ac
--- orig/libsanitizer/configure.ac	2020-07-23 02:35:19.128404206 -0400
+++ new/libsanitizer/configure.ac	2020-08-12 13:08:06.949486176 -0400
@@ -256,6 +256,8 @@
     AC_CHECK_FUNC(mmap, [have_mmap=yes], [have_mmap=no])
   fi
 fi
+# with mingw, mmap.h may be present via win32-mmap
+have_mmap=no
 if test "$have_mmap" = "no"; then
   VIEW_FILE=read.lo
   ALLOC_FILE=alloc.lo
