diff -Nru orig/reinstallr/tests/testthat/test_03_show_missing_packages.R patched/reinstallr/tests/testthat/test_03_show_missing_packages.R
--- orig/reinstallr/tests/testthat/test_03_show_missing_packages.R	2016-12-09 12:39:45.000000000 +0100
+++ patched/reinstallr/tests/testthat/test_03_show_missing_packages.R	2021-11-26 18:06:01.708804800 +0100
@@ -16,8 +16,9 @@
   writeLines(text = test_source, con = con)
 
   # check if repo is set, otherwise test is going to fail...
+  chooseCRANmirror(graphics=FALSE, ind=1, local.only=TRUE)
   repo <- getOption("repos")
-  if (is.null(repo) | repo == '@CRAN@') {
+  if (is.null(repo)) {
     repo <- 'https://cloud.R-project.org'
   }
 
